// @storybook/test-runner@0.16.0/dist/config/jest-playwright.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var jest_playwright_exports = {};
__export(jest_playwright_exports, {
  getJestConfig: () => getJestConfig
});
module.exports = __toCommonJS(jest_playwright_exports);
var import_path = __toESM(require("path"));
var import_core_common = require("@storybook/core-common");
const TEST_RUNNER_PATH = process.env.STORYBOOK_TEST_RUNNER_PATH ?? "@storybook/test-runner";
const getJestPlaywrightConfig = /* @__PURE__ */ __name(() => {
  const presetBasePath = import_path.default.dirname(require.resolve("jest-playwright-preset", {
    paths: [
      import_path.default.join(__dirname, "../node_modules")
    ]
  }));
  const expectPlaywrightPath = import_path.default.dirname(require.resolve("expect-playwright", {
    paths: [
      import_path.default.join(__dirname, "../node_modules")
    ]
  }));
  return {
    runner: import_path.default.join(presetBasePath, "runner.js"),
    globalSetup: require.resolve(`${TEST_RUNNER_PATH}/playwright/global-setup.js`),
    globalTeardown: require.resolve(`${TEST_RUNNER_PATH}/playwright/global-teardown.js`),
    testEnvironment: require.resolve(`${TEST_RUNNER_PATH}/playwright/custom-environment.js`),
    setupFilesAfterEnv: [
      require.resolve(`${TEST_RUNNER_PATH}/playwright/jest-setup.js`),
      expectPlaywrightPath,
      import_path.default.join(presetBasePath, "lib", "extends.js")
    ]
  };
}, "getJestPlaywrightConfig");
const getJestConfig = /* @__PURE__ */ __name(() => {
  const { TEST_ROOT, TEST_MATCH, STORYBOOK_STORIES_PATTERN, TEST_BROWSERS, STORYBOOK_COLLECT_COVERAGE, STORYBOOK_JUNIT } = process.env;
  const jestJunitPath = import_path.default.dirname(require.resolve("jest-junit", {
    paths: [
      import_path.default.join(__dirname, "../node_modules")
    ]
  }));
  const jestSerializerHtmlPath = import_path.default.dirname(require.resolve("jest-serializer-html", {
    paths: [
      import_path.default.join(__dirname, "../node_modules")
    ]
  }));
  const swcJestPath = import_path.default.dirname(require.resolve("@swc/jest", {
    paths: [
      import_path.default.join(__dirname, "../node_modules")
    ]
  }));
  const reporters = STORYBOOK_JUNIT ? [
    "default",
    jestJunitPath
  ] : [
    "default"
  ];
  const testMatch = STORYBOOK_STORIES_PATTERN?.split(";") ?? [];
  const config = {
    rootDir: (0, import_core_common.getProjectRoot)(),
    roots: TEST_ROOT ? [
      TEST_ROOT
    ] : void 0,
    reporters,
    testMatch,
    transform: {
      "^.+\\.(story|stories)\\.[jt]sx?$": require.resolve(`${TEST_RUNNER_PATH}/playwright/transform`),
      "^.+\\.[jt]sx?$": swcJestPath
    },
    snapshotSerializers: [
      jestSerializerHtmlPath
    ],
    testEnvironmentOptions: {
      "jest-playwright": {
        browsers: TEST_BROWSERS?.split(",").map((p) => p.trim().toLowerCase()).filter(Boolean),
        collectCoverage: STORYBOOK_COLLECT_COVERAGE === "true",
        exitOnPageError: false
      }
    },
    watchPlugins: [
      require.resolve("jest-watch-typeahead/filename"),
      require.resolve("jest-watch-typeahead/testname")
    ],
    watchPathIgnorePatterns: [
      "coverage",
      ".nyc_output",
      ".cache"
    ],
    ...getJestPlaywrightConfig()
  };
  if (TEST_MATCH) {
    config.testMatch = [
      TEST_MATCH
    ];
  }
  return config;
}, "getJestConfig");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getJestConfig
});
