// @storybook/test-runner@0.16.0/dist/config/jest-playwright.mjs
var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
import path from "path";
import { getProjectRoot } from "@storybook/core-common";
const TEST_RUNNER_PATH = process.env.STORYBOOK_TEST_RUNNER_PATH ?? "@storybook/test-runner";
const getJestPlaywrightConfig = /* @__PURE__ */ __name(() => {
  const presetBasePath = path.dirname(require.resolve("jest-playwright-preset", {
    paths: [
      path.join(__dirname, "../node_modules")
    ]
  }));
  const expectPlaywrightPath = path.dirname(require.resolve("expect-playwright", {
    paths: [
      path.join(__dirname, "../node_modules")
    ]
  }));
  return {
    runner: path.join(presetBasePath, "runner.js"),
    globalSetup: require.resolve(`${TEST_RUNNER_PATH}/playwright/global-setup.js`),
    globalTeardown: require.resolve(`${TEST_RUNNER_PATH}/playwright/global-teardown.js`),
    testEnvironment: require.resolve(`${TEST_RUNNER_PATH}/playwright/custom-environment.js`),
    setupFilesAfterEnv: [
      require.resolve(`${TEST_RUNNER_PATH}/playwright/jest-setup.js`),
      expectPlaywrightPath,
      path.join(presetBasePath, "lib", "extends.js")
    ]
  };
}, "getJestPlaywrightConfig");
const getJestConfig = /* @__PURE__ */ __name(() => {
  const { TEST_ROOT, TEST_MATCH, STORYBOOK_STORIES_PATTERN, TEST_BROWSERS, STORYBOOK_COLLECT_COVERAGE, STORYBOOK_JUNIT } = process.env;
  const jestJunitPath = path.dirname(require.resolve("jest-junit", {
    paths: [
      path.join(__dirname, "../node_modules")
    ]
  }));
  const jestSerializerHtmlPath = path.dirname(require.resolve("jest-serializer-html", {
    paths: [
      path.join(__dirname, "../node_modules")
    ]
  }));
  const swcJestPath = path.dirname(require.resolve("@swc/jest", {
    paths: [
      path.join(__dirname, "../node_modules")
    ]
  }));
  const reporters = STORYBOOK_JUNIT ? [
    "default",
    jestJunitPath
  ] : [
    "default"
  ];
  const testMatch = STORYBOOK_STORIES_PATTERN?.split(";") ?? [];
  const config = {
    rootDir: getProjectRoot(),
    roots: TEST_ROOT ? [
      TEST_ROOT
    ] : void 0,
    reporters,
    testMatch,
    transform: {
      "^.+\\.(story|stories)\\.[jt]sx?$": require.resolve(`${TEST_RUNNER_PATH}/playwright/transform`),
      "^.+\\.[jt]sx?$": swcJestPath
    },
    snapshotSerializers: [
      jestSerializerHtmlPath
    ],
    testEnvironmentOptions: {
      "jest-playwright": {
        browsers: TEST_BROWSERS?.split(",").map((p) => p.trim().toLowerCase()).filter(Boolean),
        collectCoverage: STORYBOOK_COLLECT_COVERAGE === "true",
        exitOnPageError: false
      }
    },
    watchPlugins: [
      require.resolve("jest-watch-typeahead/filename"),
      require.resolve("jest-watch-typeahead/testname")
    ],
    watchPathIgnorePatterns: [
      "coverage",
      ".nyc_output",
      ".cache"
    ],
    ...getJestPlaywrightConfig()
  };
  if (TEST_MATCH) {
    config.testMatch = [
      TEST_MATCH
    ];
  }
  return config;
}, "getJestConfig");
export {
  getJestConfig
};
