// @spotter/ui/es/components/SptTabWithBadge/index.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptTabWithBadge/index.tsx
import { Badge, Tabs } from "antd";
import { useMemo, useState } from "react";
import SptLabelWithNumber from "../SptLabelWithNumber";
import React from "react";
import "./style/index.less";
var SptTabBadgeType = /* @__PURE__ */ ((SptTabBadgeType2) => {
  SptTabBadgeType2[SptTabBadgeType2["Inline"] = 0] = "Inline";
  SptTabBadgeType2[SptTabBadgeType2["Corner"] = 1] = "Corner";
  return SptTabBadgeType2;
})(SptTabBadgeType || {});
var SptTabWithBadge = (_a) => {
  var _b = _a, {
    items,
    onChange,
    badgeType = 0 /* Inline */,
    badgeProps = {}
  } = _b, props = __objRest(_b, [
    "items",
    "onChange",
    "badgeType",
    "badgeProps"
  ]);
  var _a2;
  const [current, setCurrent] = useState((_a2 = props.activeKey) != null ? _a2 : props.defaultActiveKey);
  const innerTabs = useMemo(
    () => items == null ? void 0 : items.map((item) => {
      var _a3;
      return {
        label: badgeType === 0 /* Inline */ ? /* @__PURE__ */ React.createElement(
          SptLabelWithNumber,
          {
            active: item.key === current,
            label: item.label,
            count: (_a3 = item.count) != null ? _a3 : 0
          }
        ) : /* @__PURE__ */ React.createElement(
          Badge,
          __spreadValues({
            count: item.count,
            offset: [6, -6],
            size: props.size === "small" ? "small" : "default",
            className: "spt-component-tab-with-badge"
          }, badgeProps),
          item.label
        ),
        key: item.key,
        children: item.children
      };
    }),
    [items, current]
  );
  return /* @__PURE__ */ React.createElement(
    Tabs,
    __spreadValues({
      items: innerTabs,
      onChange: (activeKey) => {
        setCurrent(activeKey);
        onChange == null ? void 0 : onChange(activeKey);
      }
    }, props)
  );
};
var SptTabWithBadge_default = SptTabWithBadge;
export {
  SptTabBadgeType,
  SptTabWithBadge_default as default
};
