// @spotter/ui/es/components/EmbededRegister/renderer/index.js
// src/components/EmbededRegister/renderer/index.tsx
import React from "react";
import { BiEmbedded } from "./BiEmbedded";
import { BiTableauDashboard } from "./BiTableauEmbeded";
var BI_DATA_SOURCE_TYPE = /* @__PURE__ */ ((BI_DATA_SOURCE_TYPE2) => {
  BI_DATA_SOURCE_TYPE2["METABASE"] = "metabase";
  BI_DATA_SOURCE_TYPE2["TABLEAU"] = "tableau";
  return BI_DATA_SOURCE_TYPE2;
})(BI_DATA_SOURCE_TYPE || {});
var EmbeddedRenderer = ({ type, data }) => {
  switch (type) {
    case "metabase" /* METABASE */: {
      return /* @__PURE__ */ React.createElement(BiEmbedded, { i18nKey: data == null ? void 0 : data.name, id: data == null ? void 0 : data.biDashboardId });
    }
    case "tableau" /* TABLEAU */: {
      return /* @__PURE__ */ React.createElement(
        BiTableauDashboard,
        {
          name: data == null ? void 0 : data.name,
          style: data == null ? void 0 : data.biOverrideStyle,
          url: data == null ? void 0 : data.biTableauUrl
        }
      );
    }
    default: {
      return null;
    }
  }
};
export {
  BI_DATA_SOURCE_TYPE,
  EmbeddedRenderer
};
