// @spotter/ui/es/components/EmbededRegister/renderer/BiTableauEmbeded/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/components/EmbededRegister/renderer/BiTableauEmbeded/index.tsx
import React from "react";
import { useMount } from "@spotter/app-client-toolkit";
import { useEffect, useRef, useState } from "react";
import { Dropdown } from "antd";
import { DatabaseFilled, DownOutlined, DownloadOutlined, TabletFilled } from "@ant-design/icons";
import { TableauDialogType } from "./tableauTypes";
function fetchWithTimeout(_0) {
  return __async(this, arguments, function* (input, init = {}) {
    const _a = init, { timeout = 2e4 } = _a, restInit = __objRest(_a, ["timeout"]);
    const controller = new AbortController();
    const id = setTimeout(() => {
      controller.abort();
      console.log("fetch canceled");
    }, timeout);
    const response = yield fetch(input, __spreadProps(__spreadValues({}, restInit), {
      signal: controller.signal
    }));
    clearTimeout(id);
    return response.json();
  });
}
var BiTableauDashboard = ({ url, style }) => {
  const [isLoaded, setIsLoaded] = useState(false);
  const [canDownLoadData, setCanDownLoadData] = useState(false);
  const tableauRef = useRef();
  const [tableauInfo, setTableauInfo] = useState({});
  function initTableau() {
    return __async(this, null, function* () {
      const res = yield fetchWithTimeout(`/api/v1/tableau?viewPath=${encodeURIComponent(url)}`, {
        method: "get"
      });
      setTableauInfo(res);
    });
  }
  useMount(initTableau);
  useEffect(() => {
    var _a, _b;
    const handleMarkselectionChange = () => {
      if (!canDownLoadData) {
        setCanDownLoadData(true);
      }
    };
    const handleInitialized = () => {
      setIsLoaded(true);
    };
    (_a = tableauRef.current) == null ? void 0 : _a.addEventListener("markselectionchanged", handleMarkselectionChange);
    (_b = tableauRef.current) == null ? void 0 : _b.addEventListener("firstinteractive", handleInitialized);
    return () => {
      var _a2, _b2;
      (_a2 = tableauRef.current) == null ? void 0 : _a2.removeEventListener(
        "markselectionchanged",
        handleMarkselectionChange
      );
      (_b2 = tableauRef.current) == null ? void 0 : _b2.removeEventListener("firstinteractive", handleInitialized);
    };
  }, [tableauInfo]);
  const handleDownload = (type) => {
    var _a;
    return (_a = tableauRef.current) == null ? void 0 : _a.displayDialogAsync(type);
  };
  const items = [
    {
      label: "Data",
      key: TableauDialogType.ExportData,
      disabled: !canDownLoadData,
      icon: /* @__PURE__ */ React.createElement(DatabaseFilled, { rev: void 0 })
    },
    {
      label: "CrossTab",
      key: TableauDialogType.ExportCrossTab,
      icon: /* @__PURE__ */ React.createElement(TabletFilled, { rev: void 0 })
    }
  ];
  const handleMenuItemClick = ({ key }) => {
    handleDownload(key);
  };
  return /* @__PURE__ */ React.createElement("div", { className: "h-full w-full" }, tableauInfo.token ? /* @__PURE__ */ React.createElement("div", { className: "relative h-full w-full p-b-40px", style }, /* @__PURE__ */ React.createElement(
    "tableau-viz",
    {
      src: tableauInfo.url,
      token: tableauInfo.token,
      ref: tableauRef,
      toolbar: "hidden",
      "hide-tabs": true
    }
  ), /* @__PURE__ */ React.createElement("div", { className: "absolute bottom-0 left-0 right-0 flex justify-end bg-[#fafafa] p-6px p-r-20px text-[#666666]" }, isLoaded && /* @__PURE__ */ React.createElement(
    Dropdown,
    {
      menu: { items, onClick: handleMenuItemClick },
      trigger: ["click"]
    },
    /* @__PURE__ */ React.createElement("span", null, /* @__PURE__ */ React.createElement(
      DownloadOutlined,
      {
        style: { fontSize: "24px" },
        rev: void 0
      }
    ), /* @__PURE__ */ React.createElement(DownOutlined, { rev: void 0 }))
  ))) : null);
};
export {
  BiTableauDashboard
};
