// @spotter/ui/lib/components/SptTable/Form/Field.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _omit = _interopRequireDefault(require("omit.js"));
var _proComponents = require("@ant-design/pro-components");
var _react = _interopRequireWildcard(require("react"));
var _runFunction = require("../../../utils/runFunction");
var _SptSearchSelect = _interopRequireDefault(require("../../SptSearchSelect"));
var _SptSearchText = _interopRequireDefault(require("../../SptSearchText"));
var _SptSearchDatePicker = require("../../SptSearchDatePicker");
var _SptSearchDateRangePicker = require("../../SptSearchDateRangePicker");
var _SptSearchTreeSelect = require("../../SptSearchTreeSelect");
var _omitUndefined = require("../../../utils/omitUndefined");
var _SptEntitySelect = _interopRequireDefault(require("../../SptEntitySelect"));
var _SptSearchCascader = _interopRequireDefault(require("../../SptSearchCascader"));
var _SptSearchDateTimeRangePicker = require("../../SptSearchDateTimeRangePicker");
var _antd = require("antd");
var _excluded = ["fieldProps"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
/**
 * 列表中的valueType，在Filter区域已经被组件处理了，不需要再次处理
 */
var blackList = ['company', 'companyCode', 'storage', 'svcCode', 'svcAccount', 'vendorCode', 'sptDateDay', 'sptDateMonth'];

// const getDataFormtByValueType = (valueType: string, format?: string) => {
//     if (format) return format;
//     if (valueType === 'dateMonthRange') {
//         return 'YYYY-MM';
//     }

//     return 'string';
// };

var SptField = /*#__PURE__*/(0, _react.memo)(function (_ref) {
  var _item$getFormItemProp, _item$formItemProps, _item$formItemProps2, _item$getFieldProps2, _item$proFieldProps3, _item$name;
  var item = _ref.item,
    helpers = _ref.helpers;
  var context = (0, _react.useContext)(_proComponents.ProProvider);
  var type = helpers.type,
    action = helpers.action,
    originItem = helpers.originItem,
    formRef = helpers.formRef;
  var formItemProps = item === null || item === void 0 || (_item$getFormItemProp = item.getFormItemProps) === null || _item$getFormItemProp === void 0 ? void 0 : _item$getFormItemProp.call(item);
  var formFieldProps = _objectSpread(_objectSpread({}, (0, _omit.default)(item, ['dataIndex', 'width', 'render', 'renderFormItem', 'renderText', 'title', 'valueType', 'label'])), {}, {
    label: (formItemProps === null || formItemProps === void 0 ? void 0 : formItemProps.label) || item.label,
    name: ((_item$formItemProps = item.formItemProps) === null || _item$formItemProps === void 0 ? void 0 : _item$formItemProps.name) || item.name || item.key || item.dataIndex,
    width: item.width,
    initialValue: item.initialValue || ((_item$formItemProps2 = item.formItemProps) === null || _item$formItemProps2 === void 0 ? void 0 : _item$formItemProps2.initialValue) || (formItemProps === null || formItemProps === void 0 ? void 0 : formItemProps.initialValue),
    render: item !== null && item !== void 0 && item.render ? function (dom, entity, renderIndex) {
      var _item$render, _item$key, _item$getFormItemProp2, _item$getFieldProps;
      return item === null || item === void 0 || (_item$render = item.render) === null || _item$render === void 0 ? void 0 : _item$render.call(item, dom, entity, renderIndex, action === null || action === void 0 ? void 0 : action.current, _objectSpread(_objectSpread({
        type: type
      }, item), {}, {
        key: (_item$key = item.key) === null || _item$key === void 0 ? void 0 : _item$key.toString(),
        formItemProps: (_item$getFormItemProp2 = item.getFormItemProps) === null || _item$getFormItemProp2 === void 0 ? void 0 : _item$getFormItemProp2.call(item),
        fieldProps: (_item$getFieldProps = item.getFieldProps) === null || _item$getFieldProps === void 0 ? void 0 : _item$getFieldProps.call(item)
      }));
    } : undefined,
    valueEnum: (0, _runFunction.runFunction)(item.valueEnum),
    fieldProps: item === null || item === void 0 || (_item$getFieldProps2 = item.getFieldProps) === null || _item$getFieldProps2 === void 0 ? void 0 : _item$getFieldProps2.call(item),
    formItemProps: formItemProps
  });

  // 兜底恢复使用ProComponents的render
  var defaultRender = function defaultRender() {
    return /*#__PURE__*/_react.default.createElement(_proComponents.ProFormField, _extends({}, formFieldProps, {
      ignoreFormItem: true
    }));
  };
  var renderFormItem = item !== null && item !== void 0 && item.renderFormItem ? function (_, config) {
    var _item$renderFormItem, _item$key2, _item$getFormItemProp3, _item$getFieldProps3;
    var renderConfig = (0, _omitUndefined.omitUndefined)(_objectSpread(_objectSpread({}, config), {}, {
      onChange: undefined
    }));
    return item === null || item === void 0 || (_item$renderFormItem = item.renderFormItem) === null || _item$renderFormItem === void 0 ? void 0 : _item$renderFormItem.call(item, _objectSpread(_objectSpread({
      type: type
    }, item), {}, {
      key: (_item$key2 = item.key) === null || _item$key2 === void 0 ? void 0 : _item$key2.toString(),
      formItemProps: (_item$getFormItemProp3 = item.getFormItemProps) === null || _item$getFormItemProp3 === void 0 ? void 0 : _item$getFormItemProp3.call(item),
      fieldProps: (_item$getFieldProps3 = item.getFieldProps) === null || _item$getFieldProps3 === void 0 ? void 0 : _item$getFieldProps3.call(item),
      originProps: originItem
    }), _objectSpread(_objectSpread({}, renderConfig), {}, {
      defaultRender: defaultRender,
      type: type
    }), formRef.current);
  } : undefined;
  var renderedDom = function renderedDom() {
    var _item$fieldProps, _ref2, _item$fieldProps$valu, _item$fieldProps2;
    if (item !== null && item !== void 0 && item.renderFormItem) {
      var dom = renderFormItem === null || renderFormItem === void 0 ? void 0 : renderFormItem(null, {});
      // 如果renderFormItem的返回值是null或者undefined，那么就不渲染了
      if (!dom || item.ignoreFormItem) return dom;
    }

    // 从fieldProps中读取值, 目前只有value，后续如果要重构render函数，估计得考虑从dataSource中获取dataValue
    var text = item === null || item === void 0 || (_item$fieldProps = item.fieldProps) === null || _item$fieldProps === void 0 ? void 0 : _item$fieldProps['value'];
    var dataValue = (_ref2 = (_item$fieldProps$valu = item === null || item === void 0 || (_item$fieldProps2 = item.fieldProps) === null || _item$fieldProps2 === void 0 ? void 0 : _item$fieldProps2.value) !== null && _item$fieldProps$valu !== void 0 ? _item$fieldProps$valu : text) !== null && _ref2 !== void 0 ? _ref2 : '';

    // 兼容自定义ValueType的场景
    var valueTypeMap = context.valueTypeMap || {};
    var customValueTypeConfig = valueTypeMap && valueTypeMap[item.valueType];
    if (customValueTypeConfig && !(blackList !== null && blackList !== void 0 && blackList.includes(item.valueType))) {
      var _item$proFieldProps, _customValueTypeConfi;
      var props = _objectSpread(_objectSpread(_objectSpread({
        label: item.label,
        tooltip: (_item$proFieldProps = item.proFieldProps) === null || _item$proFieldProps === void 0 ? void 0 : _item$proFieldProps.tooltip,
        transform: item.transform,
        request: item.request,
        params: item.params
      }, item.fieldProps), item.proFieldProps), formFieldProps);
      return customValueTypeConfig === null || customValueTypeConfig === void 0 || (_customValueTypeConfi = customValueTypeConfig.renderFormItem) === null || _customValueTypeConfi === void 0 ? void 0 : _customValueTypeConfi.call(customValueTypeConfig, dataValue, _objectSpread(_objectSpread({
        text: dataValue
      }, props), {}, {
        valueEnum: (0, _runFunction.runFunction)(item.valueEnum),
        viewMode: 'light',
        // 直接给自定义渲染一个viewMode light的参数
        mode: 'edit'
      }), /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, dataValue));
    }
    if (renderFormItem) {
      var _formFieldProps$field;
      // 处理自定义renderFormItem的场景
      var newDom = renderFormItem === null || renderFormItem === void 0 ? void 0 : renderFormItem(text, _objectSpread({
        mode: 'edit'
      }, (_formFieldProps$field = formFieldProps === null || formFieldProps === void 0 ? void 0 : formFieldProps.fieldProps) !== null && _formFieldProps$field !== void 0 ? _formFieldProps$field : {}));
      // renderFormItem 之后的dom可能没有props，所以需要手动加上
      if ( /*#__PURE__*/_react.default.isValidElement(newDom)) return /*#__PURE__*/_react.default.cloneElement(newDom, _objectSpread(_objectSpread({
        viewMode: 'light'
      }, formFieldProps), newDom.props || {}));
      return newDom;
    }
    if (item.valueType === 'select' || item.valueType === 'text' && (formFieldProps.valueEnum || item.request)) {
      var _item$proFieldProps2;
      return /*#__PURE__*/_react.default.createElement(_SptSearchSelect.default, _extends({
        noStyle: true,
        label: item.label,
        tooltip: (_item$proFieldProps2 = item.proFieldProps) === null || _item$proFieldProps2 === void 0 ? void 0 : _item$proFieldProps2.tooltip,
        transform: item.transform,
        request: item.request,
        params: item.params
      }, item.fieldProps, item.proFieldProps, {
        fieldProps: _objectSpread({}, item.fieldProps)
      }, formFieldProps));
    }
    if (item.valueType === 'date') {
      return /*#__PURE__*/_react.default.createElement(_SptSearchDatePicker.SptSearchDatePicker, _extends({
        noStyle: true
      }, formFieldProps));
    }
    if (['dateRange', 'sptDateDay'].includes(item.valueType)) {
      return /*#__PURE__*/_react.default.createElement(_SptSearchDateRangePicker.SptSearchDateRangePicker, _extends({
        noStyle: true
      }, formFieldProps));
    }
    if (item.valueType === 'dateTimeRange') {
      return /*#__PURE__*/_react.default.createElement(_SptSearchDateTimeRangePicker.SptSearchDateTimeRangePicker, _extends({
        noStyle: true
      }, formFieldProps));
    }
    if (['dateMonthRange', 'sptDateMonth'].includes(item.valueType)) {
      var fieldProps = formFieldProps.fieldProps,
        otherFormFieldProps = _objectWithoutProperties(formFieldProps, _excluded);
      return /*#__PURE__*/_react.default.createElement(_SptSearchDateRangePicker.SptSearchDateRangePicker, _extends({
        noStyle: true,
        dataFormat: "YYYY-MM",
        fieldProps: _objectSpread({
          picker: 'month',
          format: 'YYYY-MM'
        }, fieldProps)
      }, otherFormFieldProps));
    }
    if (item.valueType === 'treeSelect') {
      return /*#__PURE__*/_react.default.createElement(_SptSearchTreeSelect.SptSearchTreeSelect, _extends({
        noStyle: true
      }, formFieldProps));
    }
    if (item.valueType === 'company') {
      return /*#__PURE__*/_react.default.createElement(_SptEntitySelect.default.Company, _extends({
        viewMode: "light",
        noStyle: true
      }, formFieldProps.fieldProps, formFieldProps));
    }
    if (item.valueType === 'svcCode' || item.valueType === 'svcAccount') {
      return /*#__PURE__*/_react.default.createElement(_SptEntitySelect.default.SvcAccount, _extends({
        viewMode: "light",
        noStyle: true
      }, formFieldProps.fieldProps, formFieldProps));
    }
    if (item.valueType === 'store' || item.valueType === 'storage') {
      return /*#__PURE__*/_react.default.createElement(_SptEntitySelect.default.Store, _extends({
        viewMode: "light"
      }, formFieldProps.fieldProps, {
        noStyle: true
      }, formFieldProps));
    }
    if (item.valueType === 'cascader') {
      return /*#__PURE__*/_react.default.createElement(_SptSearchCascader.default, _extends({
        noStyle: true
      }, formFieldProps));
    }
    if (item.valueType === 'companyCode') {
      return /*#__PURE__*/_react.default.createElement(_SptEntitySelect.default.Company, _extends({
        noStyle: true,
        viewMode: "light",
        valueName: "businessCode"
      }, formFieldProps.fieldProps, formFieldProps));
    }
    if (item.valueType === 'vendorCode') {
      return /*#__PURE__*/_react.default.createElement(_SptEntitySelect.default.VendorCode, _extends({
        noStyle: true,
        viewMode: "light",
        valueName: "vendorCode"
      }, formFieldProps.fieldProps, formFieldProps));
    }
    return /*#__PURE__*/_react.default.createElement(_SptSearchText.default, _extends({
      noStyle: true
    }, formFieldProps));
  };
  return /*#__PURE__*/_react.default.createElement(_antd.Form.Item, _extends({
    name: formFieldProps.name,
    style: {
      marginBottom: 0
    }
    // transform={item.transform}
    // dataFormat={getDataFormtByValueType(item.valueType, item.fieldProps?.format)}
    // valueType={item.valueType}
    ,
    key: ((_item$proFieldProps3 = item.proFieldProps) === null || _item$proFieldProps3 === void 0 ? void 0 : _item$proFieldProps3.proFieldKey) || ((_item$name = item.name) === null || _item$name === void 0 ? void 0 : _item$name.toString())
  }, (0, _omit.default)(formFieldProps.formItemProps, ['label', 'initialValue'])), renderedDom());
}, function (prevProps, nextProps) {
  return (0, _proComponents.isDeepEqualReact)(nextProps, prevProps, ['onChange', 'onBlur']);
});
SptField.displayName = 'SptField';
var _default = exports.default = SptField;
