// @spotter/ui/lib/components/SptTable/Form/Submitter.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _icons = require("@ant-design/icons");
var _i18nSdk = require("@spotter/i18n-sdk");
var _antd = require("antd");
var _omit = _interopRequireDefault(require("omit.js"));
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
/** @name 用于配置操作栏 */

var Submitter = function Submitter(props) {
  var submit = props.submit,
    reset = props.reset,
    _props$searchConfig = props.searchConfig,
    searchConfig = _props$searchConfig === void 0 ? {} : _props$searchConfig,
    submitButtonProps = props.submitButtonProps,
    resetButtonProps = props.resetButtonProps;
  if (props.render === false) {
    return null;
  }
  var _useSptTranslation = (0, _i18nSdk.useSptTranslation)(),
    t = _useSptTranslation.t;
  // const form = Form.useFormInstance();

  // const submit = () => {
  //     form.submit();
  //     onSubmit?.();
  // };

  // const reset = () => {
  //     form.resetFields();
  //     onReset?.();
  // };

  var _searchConfig$submitT = searchConfig.submitText,
    submitText = _searchConfig$submitT === void 0 ? t('common.action_query') : _searchConfig$submitT,
    _searchConfig$resetTe = searchConfig.resetText,
    resetText = _searchConfig$resetTe === void 0 ? t('common.action_reset') : _searchConfig$resetTe;
  var renderDom = [];
  if (submitButtonProps !== false) {
    renderDom.push( /*#__PURE__*/_react.default.createElement(_antd.Button, _extends({
      key: "submit"
    }, (0, _omit.default)(submitButtonProps || {}, ['preventDefault']), {
      type: "primary",
      onClick: function onClick(e) {
        var _submitButtonProps$on;
        if (!(submitButtonProps !== null && submitButtonProps !== void 0 && submitButtonProps.preventDefault)) {
          submit === null || submit === void 0 || submit();
        }
        submitButtonProps === null || submitButtonProps === void 0 || (_submitButtonProps$on = submitButtonProps.onClick) === null || _submitButtonProps$on === void 0 || _submitButtonProps$on.call(submitButtonProps, e);
      }
    }), submitText));
  }
  if (resetButtonProps !== false) {
    renderDom.push( /*#__PURE__*/_react.default.createElement(_antd.Button, _extends({
      type: "link",
      icon: /*#__PURE__*/_react.default.createElement(_icons.UndoOutlined, null)
    }, (0, _omit.default)(resetButtonProps || {}, ['preventDefault']), {
      key: "reset",
      onClick: function onClick(e) {
        var _resetButtonProps$onC;
        if (!(resetButtonProps !== null && resetButtonProps !== void 0 && resetButtonProps.preventDefault)) {
          reset === null || reset === void 0 || reset();
        }
        resetButtonProps === null || resetButtonProps === void 0 || (_resetButtonProps$onC = resetButtonProps.onClick) === null || _resetButtonProps$onC === void 0 || _resetButtonProps$onC.call(resetButtonProps, e);
      }
    }), resetText));
  }
  // 避免死循环
  // const renderDom = render ? render({ ...props, form, submit, reset }, dom) : dom;
  // const renderDom = dom;

  if (Array.isArray(renderDom)) {
    if ((renderDom === null || renderDom === void 0 ? void 0 : renderDom.length) < 1) {
      return null;
    }
    if ((renderDom === null || renderDom === void 0 ? void 0 : renderDom.length) === 1) {
      return renderDom[0];
    }
    return /*#__PURE__*/_react.default.createElement("div", {
      style: {
        display: 'flex',
        alignItems: 'center',
        gap: '8px'
      }
    }, renderDom);
  }
  return renderDom;
};
Submitter.displayName = 'Submitter';
var _default = exports.default = /*#__PURE__*/(0, _react.memo)(Submitter);
