// @spotter/ui/es/components/SptSearch/components/DateMonthRange/index.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptSearch/components/DateMonthRange/index.tsx
import { ProFormDateMonthRangePicker } from "@ant-design/pro-components";
import React from "react";
import { useRef } from "react";
import "./style/index.less";
import classNames from "classnames";
var SptSearchDateMonthRangePicker = (_a) => {
  var _b = _a, {
    label,
    className,
    fieldProps
  } = _b, proFormItemProps = __objRest(_b, [
    "label",
    "className",
    "fieldProps"
  ]);
  const baseClassName = "spt-search-month-range-date";
  const ref = useRef();
  return /* @__PURE__ */ React.createElement("div", { className: classNames(baseClassName, className) }, label || (fieldProps == null ? void 0 : fieldProps.label) ? /* @__PURE__ */ React.createElement(
    "div",
    {
      className: classNames(`${baseClassName}-title`),
      onClick: () => {
        var _a2;
        (_a2 = ref.current) == null ? void 0 : _a2.focus();
      }
    },
    label || (fieldProps == null ? void 0 : fieldProps.label)
  ) : null, /* @__PURE__ */ React.createElement(
    ProFormDateMonthRangePicker,
    __spreadValues({
      noStyle: true,
      fieldProps: __spreadValues({
        ref,
        bordered: false
      }, fieldProps)
    }, proFormItemProps)
  ));
};
export {
  SptSearchDateMonthRangePicker
};
