// @spotter/ui/es/api/http-client.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/api/http-client.ts
import axios from "axios";
import JSONBigInt from "json-bigint";
var JSONBig2String = JSONBigInt({
  storeAsString: true,
  constructorAction: "preserve",
  protoAction: "preserve"
});
var defaultStatusSetMap = {
  authError: /* @__PURE__ */ new Set([401, 401e4, 4010001, 4010002, 4010003]),
  permissionError: /* @__PURE__ */ new Set([403]),
  success: /* @__PURE__ */ new Set([200, 201])
};
var HttpClient = class {
  constructor(options) {
    var _a, _b, _c, _d, _e, _f;
    this.options = options;
    this.options.res.isSuccess = (_b = (_a = options.res) == null ? void 0 : _a.isSuccess) != null ? _b : (res) => {
      var _a2;
      return defaultStatusSetMap.success.has(res.status) && defaultStatusSetMap.success.has((_a2 = res.data) == null ? void 0 : _a2.code);
    };
    this.options.res.isAuthError = (_d = (_c = options.res) == null ? void 0 : _c.isAuthError) != null ? _d : (res) => {
      var _a2;
      return defaultStatusSetMap.authError.has(res.status) || defaultStatusSetMap.authError.has((_a2 = res.data) == null ? void 0 : _a2.code);
    };
    this.options.res.isPermissionError = (_f = (_e = options.res) == null ? void 0 : _e.isPermissionError) != null ? _f : (res) => {
      var _a2;
      return defaultStatusSetMap.permissionError.has(res.status) || defaultStatusSetMap.permissionError.has((_a2 = res.data) == null ? void 0 : _a2.code);
    };
    this.httpClient = axios.create(__spreadProps(__spreadValues({}, this.options.axiosOptions), {
      baseURL: this.options.context.href,
      withCredentials: true,
      // CORS with cookies,
      transformResponse: [
        (data) => {
          try {
            return JSONBig2String.parse(data || "{}");
          } catch (error) {
            console.error(error);
            return data;
          }
        }
      ]
    }));
    this.httpClient.interceptors.request.use(
      (config) => {
        if (!config.data) {
          config.data = {};
        }
        return config;
      },
      (error) => {
        console.log("request", error);
        return Promise.reject(error);
      }
    );
    this.httpClient.interceptors.response.use(
      (response) => {
        var _a2, _b2, _c2, _d2, _e2, _f2, _g;
        if (this.options.res.isSuccess(response)) {
          return Promise.resolve(response);
        }
        if (this.options.res.isAuthError(response)) {
          console.error(response);
          (_b2 = (_a2 = this.options.res).onAuthError) == null ? void 0 : _b2.call(_a2, response);
        } else if (this.options.res.isPermissionError(response)) {
          console.error(response);
          (_d2 = (_c2 = this.options.res).onPermissionError) == null ? void 0 : _d2.call(_c2, response);
        } else {
          if ((_e2 = response == null ? void 0 : response.config) == null ? void 0 : _e2.ignoreError) {
            return Promise.resolve(response);
          }
          (_g = (_f2 = this.options.res).onOtherError) == null ? void 0 : _g.call(_f2, response);
          console.error(JSON.stringify(response.data));
        }
        return Promise.reject(response);
      },
      (error) => {
        var _a2, _b2, _c2, _d2, _e2, _f2;
        const response = (error == null ? void 0 : error.response) || {};
        if (this.options.res.isAuthError(response)) {
          console.log(error);
          (_b2 = (_a2 = this.options.res).onAuthError) == null ? void 0 : _b2.call(_a2, response);
        } else if (this.options.res.isPermissionError(response)) {
          console.error(response);
          (_d2 = (_c2 = this.options.res).onPermissionError) == null ? void 0 : _d2.call(_c2, response);
        } else {
          (_f2 = (_e2 = this.options.res).onOtherError) == null ? void 0 : _f2.call(_e2, response);
          console.error(JSON.stringify(error.toJSON()));
        }
        return Promise.reject(error);
      }
    );
  }
  request(config) {
    return __async(this, null, function* () {
      return this.httpClient.request(config);
    });
  }
  getAxios() {
    return this.httpClient;
  }
};
export {
  JSONBig2String,
  HttpClient as default
};
