// @spotter/ui/lib/components/EmbededRegister/renderer/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EmbeddedRenderer = exports.BI_DATA_SOURCE_TYPE = void 0;
var _react = _interopRequireDefault(require("react"));
var _BiEmbedded = require("./BiEmbedded");
var _BiTableauEmbeded = require("./BiTableauEmbeded");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var BI_DATA_SOURCE_TYPE = exports.BI_DATA_SOURCE_TYPE = /*#__PURE__*/function (BI_DATA_SOURCE_TYPE) {
  BI_DATA_SOURCE_TYPE["METABASE"] = "metabase";
  BI_DATA_SOURCE_TYPE["TABLEAU"] = "tableau";
  return BI_DATA_SOURCE_TYPE;
}({});
var EmbeddedRenderer = exports.EmbeddedRenderer = function EmbeddedRenderer(_ref) {
  var type = _ref.type,
    data = _ref.data;
  switch (type) {
    case BI_DATA_SOURCE_TYPE.METABASE:
      {
        return /*#__PURE__*/_react.default.createElement(_BiEmbedded.BiEmbedded, {
          i18nKey: data === null || data === void 0 ? void 0 : data.name,
          id: data === null || data === void 0 ? void 0 : data.biDashboardId
        });
      }
    case BI_DATA_SOURCE_TYPE.TABLEAU:
      {
        return /*#__PURE__*/_react.default.createElement(_BiTableauEmbeded.BiTableauDashboard, {
          name: data === null || data === void 0 ? void 0 : data.name,
          style: data === null || data === void 0 ? void 0 : data.biOverrideStyle,
          url: data === null || data === void 0 ? void 0 : data.biTableauUrl
        });
      }
    default:
      {
        return null;
      }
  }
};
