// @spotter/ui/es/components/SptPageContainer/index.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptPageContainer/index.tsx
import { Spin, Typography } from "antd";
import { Outlet, useLocation, useNavigate } from "react-router-dom";
import React, { useEffect, useMemo, useRef, useState } from "react";
import { pathJoin } from "@spotter/app-client-toolkit";
import classNames from "classnames";
import SptTabWithBadge, { SptTabBadgeType } from "../SptTabWithBadge";
import "./style/index.less";
import SptFooterToolbar from "../SptFooterToolbar";
import { SptComponentProvider } from "../Provider";
var SptPageContainerContext = React.createContext(
  null
);
var SptPageContainerProvider = SptPageContainerContext.Provider;
var { Text } = Typography;
var SptPageContainerBase = ({
  title,
  subTitle,
  tabs = [],
  tabBadgeType = SptTabBadgeType.Corner,
  basePath = "",
  topSection,
  children = /* @__PURE__ */ React.createElement(Outlet, null),
  extra,
  action,
  footerToolbar,
  loading = false,
  tabNavigationMode = true,
  autoDefaultTabKey = true,
  fitContainer = false,
  tableAction,
  onTabChange
}) => {
  var _a, _b;
  const innerTabs = useMemo(
    () => tabs == null ? void 0 : tabs.map((p) => ({
      // ...p,
      label: p.tab,
      key: tabNavigationMode ? pathJoin(basePath, p.tabKey) : p.tabKey,
      count: p == null ? void 0 : p.count
    })),
    [basePath, tabs, tabNavigationMode]
  );
  const location = useLocation();
  const navigate = useNavigate();
  const [activeKey, setActiveKey] = useState(
    tabNavigationMode ? location.pathname : (_b = (_a = innerTabs[0]) == null ? void 0 : _a.key) != null ? _b : void 0
  );
  useEffect(() => {
    console.log("刷新。。。。");
    if (tabNavigationMode && location.pathname !== pathJoin("/", basePath)) {
      setActiveKey(location.pathname);
    } else {
      autoDefaultTabKey && innerTabs.length && setActiveKey(innerTabs[0].key);
    }
  }, [location.pathname, tabNavigationMode, innerTabs]);
  useEffect(() => {
    onTabChange == null ? void 0 : onTabChange(activeKey);
  }, [activeKey]);
  const ref = useRef(null);
  const tableProvideAction = useMemo(() => {
    return {
      tableAction
    };
  }, [activeKey, innerTabs, tabNavigationMode, basePath]);
  return /* @__PURE__ */ React.createElement(SptPageContainerProvider, { value: tableProvideAction }, /* @__PURE__ */ React.createElement(
    "div",
    {
      className: classNames(
        {
          "with-tabs": tabs == null ? void 0 : tabs.length,
          "fit-container": fitContainer,
          "tab-navigation-mode": tabNavigationMode,
          "spotter-page-container-fit-pagination": footerToolbar
        },
        "spotter-page-container"
      ),
      ref
    },
    topSection,
    (title || subTitle || action || extra) && /* @__PURE__ */ React.createElement("div", { className: "spotter-page-container-title" }, (title || subTitle || action) && /* @__PURE__ */ React.createElement("div", { className: "spotter-page-container-title-inner" }, /* @__PURE__ */ React.createElement("div", { className: "spotter-page-container-title-inner-wrap" }, /* @__PURE__ */ React.createElement(
      Text,
      {
        style: { verticalAlign: "middle", overflow: "hidden" },
        className: "spotter-page-container-title-main",
        ellipsis: {
          tooltip: title
        }
      },
      title
    ), subTitle ? /* @__PURE__ */ React.createElement("div", { className: "spotter-page-container-title-sub" }, subTitle) : null), /* @__PURE__ */ React.createElement("div", { className: "spotter-page-container-title-action" }, action)), extra),
    innerTabs && (innerTabs == null ? void 0 : innerTabs.length) ? /* @__PURE__ */ React.createElement(
      SptTabWithBadge,
      {
        className: "spotter-page-container-tabs",
        activeKey,
        onChange: (key) => {
          setActiveKey(key);
          tabNavigationMode && navigate(key);
        },
        items: innerTabs.map((item, i) => ({
          label: tabs[i].tab,
          key: tabNavigationMode ? item.key : tabs[i].tabKey,
          count: item == null ? void 0 : item.count
        })),
        badgeType: tabBadgeType,
        tabBarGutter: 24
      }
    ) : null,
    /* @__PURE__ */ React.createElement(Spin, { style: { height: "" }, spinning: loading }, children),
    footerToolbar ? /* @__PURE__ */ React.createElement(SptFooterToolbar, null, " ", footerToolbar, " ") : null
  ));
};
var SptPageContainer = (_a) => {
  var props = __objRest(_a, []);
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(SptPageContainerBase, __spreadValues({}, props)));
};
var SptPageContainer_default = SptPageContainer;
export {
  SptPageContainerContext,
  SptPageContainerProvider,
  SptPageContainer_default as default
};
