// @spotter/ui/lib/components/SptLabelWithNumber/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _antd = require("antd");
var _react = _interopRequireDefault(require("react"));
require("./style/index.less");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var SptLabelWithNumber = function SptLabelWithNumber(_ref) {
  var label = _ref.label,
    _ref$count = _ref.count,
    count = _ref$count === void 0 ? 0 : _ref$count,
    _ref$showBadge = _ref.showBadge,
    showBadge = _ref$showBadge === void 0 ? true : _ref$showBadge,
    _ref$active = _ref.active,
    active = _ref$active === void 0 ? false : _ref$active,
    _ref$showZero = _ref.showZero,
    showZero = _ref$showZero === void 0 ? false : _ref$showZero;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "spt-component-label-with-number"
  }, /*#__PURE__*/_react.default.createElement("div", null, label), showBadge && count > 0 ? /*#__PURE__*/_react.default.createElement(_antd.Badge, {
    count: count,
    showZero: showZero,
    color: "#F0F5FF",
    style: {
      marginLeft: '5px',
      color: active ? 'var(--color-primary)' : 'rgba(0,0,0, 0.45)'
    }
  }) : null);
};
var _default = exports.default = SptLabelWithNumber;
