// @spotter/ui/es/components/SptTable/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/components/SptTable/index.tsx
import {
  omitUndefined,
  useMountMergeState,
  useRefFunction
} from "@ant-design/pro-components";
import React, {
  useCallback,
  useContext,
  useEffect,
  useImperativeHandle,
  useMemo,
  useRef,
  useState
} from "react";
import classnames from "classnames";
import { useResizable } from "./Resizable";
import { useFit } from "./useFit";
import { nanoid } from "nanoid";
import { SptComponentContext, SptComponentProvider } from "../Provider";
import { eventBus, useLocalCache } from "@spotter/app-client-toolkit";
import { useLocation, useSearchParams } from "react-router-dom";
import "./style/index.less";
import Filter from "./Form";
import classNames from "classnames";
import Table from "./Table";
import { Spin } from "antd";
import omit from "omit.js";
import Pagination from "./Pagination";
import transformKey from "./Form/transformKey";
import { SptPageContainerContext } from "../SptPageContainer";
var getValidColumnWidth = (column, cacheColumn, globalColumnResizableMinWidth) => {
  var _a, _b, _c;
  return column.dataIndex && cacheColumn && cacheColumn.originWidth === column.width ? cacheColumn.width : (
    // 当没有设定列宽时，默认使用全局的最小列宽
    (_c = (_b = column.width) != null ? _b : (_a = column.resizable) == null ? void 0 : _a.minWidth) != null ? _c : globalColumnResizableMinWidth
  );
};
var transformOriginColumns = ({
  originColumns,
  getCache,
  columnResizable,
  drawerTrigger,
  resizedColumns
}) => originColumns.map((column, index) => {
  var _a, _b;
  let transformdColumn = __spreadValues({}, column);
  const resizeData = column.dataIndex && resizedColumns[column.dataIndex.toString()] || resizedColumns[index.toString()];
  if (resizeData && !column.hideInTable) {
    transformdColumn = __spreadProps(__spreadValues({}, column), {
      width: column.width + ((_a = resizeData == null ? void 0 : resizeData.width) != null ? _a : 0)
    });
  }
  const c = !transformdColumn.hideInTable ? __spreadProps(__spreadValues({}, transformdColumn), {
    width: getValidColumnWidth(
      transformdColumn,
      getCache()[transformdColumn.dataIndex],
      columnResizable == null ? void 0 : columnResizable.minWidth
    )
  }) : transformdColumn;
  if (!((_b = transformdColumn.drawerTrigger) != null ? _b : drawerTrigger)) {
    return __spreadProps(__spreadValues({}, c), {
      className: classnames(transformdColumn.className, "disable-drawer-trigger"),
      render: (node, ...rp) => {
        var _a2;
        return /* @__PURE__ */ React.createElement(
          "div",
          {
            className: classnames("disabled-drawer-trigger-table-cell-wrapper", {
              "items-start": !transformdColumn.align || transformdColumn.align === "left",
              "items-center": transformdColumn.align === "center",
              "items-end": transformdColumn.align === "right"
            }),
            onClick: (e) => {
              e.stopPropagation();
            }
          },
          ((_a2 = transformdColumn.render) == null ? void 0 : _a2.call(transformdColumn, node, ...rp)) || node
        );
      }
    });
  }
  return c;
});
var SpotterTableBase = (_a) => {
  var _b = _a, {
    className,
    width = 0,
    fit = "content",
    sticky = true,
    onRowDrawerChange: _onRowDrawerChange,
    onRowSelection,
    drawerTrigger = true,
    columnResizable: _columnResizable,
    cacheKey,
    params: propsParams,
    columns: _columns,
    components: _components,
    onRow,
    request,
    search,
    formRef: formRef,
    manualRequest,
    onReset: onPropsReset,
    actionRef: propsActionRef,
    form: searchForm,
    pagination: propsPagination,
    dataSource: propsDataSource,
    loading: propsLoading,
    inheritAction,
    toolbar: propsToolbar
  } = _b, restProps = __objRest(_b, [
    "className",
    "width",
    "fit",
    "sticky",
    "onRowDrawerChange",
    "onRowSelection",
    "drawerTrigger",
    "columnResizable",
    "cacheKey",
    "params",
    "columns",
    "components",
    "onRow",
    "request",
    "search",
    // pagination,
    "formRef",
    "manualRequest",
    "onReset",
    "actionRef",
    "form",
    "pagination",
    "dataSource",
    "loading",
    "inheritAction",
    "toolbar"
  ]);
  var _a2;
  const onRowDrawerChange = useMemo(() => _onRowDrawerChange, []);
  const columnResizable = useMemo(
    () => _columnResizable != null ? _columnResizable : {
      minWidth: 88,
      preferResizeWidth: false
    },
    [_columnResizable]
  );
  const [formSearch, setFormSearch] = useMountMergeState(() => {
    if (manualRequest || search !== false) {
      return void 0;
    }
    return {};
  });
  const manualRequestRef = useRef(formSearch === void 0);
  const initFormRef = useRef();
  const baseFormRef = formRef != null ? formRef : initFormRef;
  const params = useMemo(() => {
    return __spreadValues(__spreadValues({}, propsParams || {}), formSearch || {});
  }, [propsParams, formSearch]);
  const [loading, setLoading] = useMountMergeState(false, {
    value: typeof propsLoading === "object" ? propsLoading == null ? void 0 : propsLoading.spinning : propsLoading,
    onChange: restProps.onLoadingChange
  });
  const originComponents = useMemo(() => _components, []);
  const _cacheKey = useMemo(() => `${cacheKey ? `${cacheKey}_` : ""}spotter_table`, []);
  const _componentId = useMemo(() => nanoid(), []);
  const { routes } = useContext(SptComponentContext);
  const location = useLocation();
  const {
    getCache,
    setCache,
    cacheKey: cacheKeyWithRoute
  } = useLocalCache(_cacheKey, {}, routes != null ? routes : [], location);
  const { scrollY, fitClass } = useFit(fit);
  const { ResizingHeaderCell, ResizingIndicator, updateResizableColumns } = useResizable({
    key: _componentId,
    headerCell: (_a2 = originComponents == null ? void 0 : originComponents.header) == null ? void 0 : _a2.cell
  });
  const [resizedColumns, setResizedColumns] = useState({});
  useEffect(() => {
    eventBus.on("resizeColumn", ({ columnKey, width: width2 }) => {
      setResizedColumns((prev) => {
        var _a3;
        const item = prev[columnKey];
        return __spreadProps(__spreadValues({}, prev), {
          [columnKey]: {
            // 保存旧的宽度，用于计算增量
            width: ((_a3 = item == null ? void 0 : item.width) != null ? _a3 : 0) + width2
          }
        });
      });
    });
    return () => {
      eventBus.off("resizeColumn");
    };
  }, []);
  const columns = useMemo(() => {
    return updateResizableColumns(
      transformOriginColumns({
        originColumns: _columns,
        getCache,
        columnResizable,
        drawerTrigger,
        resizedColumns
      })
    );
  }, [_columns, resizedColumns]);
  const [searchParams] = useSearchParams();
  useEffect(() => {
    var _a3, _b2, _c;
    try {
      const params2 = searchParams.get("table_search_params");
      if (params2) {
        const table_search_params = JSON.parse(decodeURIComponent(params2));
        if (table_search_params.resetFields) {
          (_a3 = baseFormRef.current) == null ? void 0 : _a3.resetFields();
        }
        (_b2 = baseFormRef.current) == null ? void 0 : _b2.setFieldsValue(table_search_params);
        (_c = baseFormRef.current) == null ? void 0 : _c.submit();
      }
    } catch (error) {
      console.log(error);
    }
  }, [searchParams]);
  useEffect(() => {
    if (columns.length) {
      setCache(
        columns.reduce((p, c, i) => {
          var _a3, _b2;
          if (c.dataIndex && c.width) {
            p[c.dataIndex] = {
              /**
               * 现在存在一个问题，改了代码里的 table 列宽以后，如果客户端有缓存，那么改动后的列宽并不会优先生效.
               * 这在一些情况下会导致显示问题，所以补充了一个自动根据原始列宽和当前的代码列宽做判断后再增量覆盖缓存的方案，来让改动后的列宽优先生效，同时不会影响其他未改动的列宽的缓存
               * */
              originWidth: (
                /**
                 * 这里优先使用数组索引去查找，提高效率。但是由于可能动态的变更 columns 的顺序和数量，所以使用数组索引进行查找并不严谨.
                 * 故我们再增加一步对比 dataIndex，如果 dataIndex 不相同，那么说明错位了，则使用 find 来查找 dataIndex 相同的原始列
                 */
                ((_a3 = _columns[i]) == null ? void 0 : _a3.dataIndex) === c.dataIndex ? _columns[i].width : (_b2 = _columns.find(({ dataIndex }) => dataIndex === c.dataIndex)) == null ? void 0 : _b2.width
              ),
              width: c.width
            };
          }
          return p;
        }, {})
      );
    }
  }, [columns]);
  const [pagination, setPagination] = useState({
    current: propsPagination && (propsPagination == null ? void 0 : propsPagination.current) || 1,
    pageSize: propsPagination && (propsPagination == null ? void 0 : propsPagination.pageSize) || 20
  });
  const [paginationInfo, setPaginationInfo] = useState({
    total: propsPagination && propsPagination.total || 0,
    // 是否已经加载过数据
    loaded: false
  });
  const sorter = useRef();
  const [dataSource, setDataSource] = useState(
    propsDataSource ? propsDataSource == null ? void 0 : propsDataSource.slice(
      (pagination.current - 1) * pagination.pageSize,
      pagination.current * pagination.pageSize
    ) : []
  );
  useEffect(() => {
    if (propsDataSource) {
      setDataSource(
        propsDataSource == null ? void 0 : propsDataSource.slice(
          (pagination.current - 1) * pagination.pageSize,
          pagination.current * pagination.pageSize
        )
      );
      setPaginationInfo(__spreadProps(__spreadValues({}, paginationInfo), {
        loaded: true,
        total: propsDataSource.length
      }));
    }
  }, [propsDataSource]);
  const queryData = (params2) => __async(void 0, null, function* () {
    var _a3;
    if (!request)
      return;
    if (manualRequestRef.current) {
      manualRequestRef.current = false;
      return;
    }
    setLoading(true);
    try {
      const { data, total } = (_a3 = yield request == null ? void 0 : request(
        params2,
        sorter.current,
        // sort
        {}
      )) != null ? _a3 : {};
      setDataSource(data || []);
      setPaginationInfo(__spreadProps(__spreadValues({}, paginationInfo), {
        loaded: true,
        total: total || 0
      }));
    } catch (error) {
      console.error(error);
    }
    setLoading(false);
  });
  const reload = () => __async(void 0, null, function* () {
    yield queryData(__spreadValues({
      current: pagination.current,
      pageSize: pagination.pageSize
    }, omit(params, ["_timestamp"])));
  });
  const reloadWithPagination = (nextPagination) => __async(void 0, null, function* () {
    yield queryData(__spreadValues({
      current: nextPagination.current,
      pageSize: nextPagination.pageSize
    }, omit(params, ["_timestamp"])));
  });
  useEffect(() => {
    request && reload();
  }, [params]);
  const onFormSearchSubmit = useCallback((values) => {
    setPagination(__spreadProps(__spreadValues({}, pagination), {
      current: 1
    }));
    setFormSearch(values);
  }, []);
  const searchKeyMap = useMemo(() => {
    var _a3;
    if (!columns) {
      return {};
    }
    const searchKeyMap2 = {};
    for (const item of columns) {
      searchKeyMap2[`${item.key || item.dataIndex}`] = (_a3 = item == null ? void 0 : item.search) == null ? void 0 : _a3.transform;
    }
    return searchKeyMap2;
  }, [columns]);
  const actionRef = useRef();
  useImperativeHandle(propsActionRef, () => __spreadProps(__spreadValues({}, actionRef.current || {}), {
    reset: () => {
      var _a3, _b2, _c;
      (_a3 = baseFormRef == null ? void 0 : baseFormRef.current) == null ? void 0 : _a3.resetFields();
      const values = omitUndefined(((_c = (_b2 = baseFormRef == null ? void 0 : baseFormRef.current) == null ? void 0 : _b2.getFieldsValue) == null ? void 0 : _c.call(_b2)) || {});
      setFormSearch(transformKey(values, searchKeyMap));
    },
    reload: (resetPageIndex) => __async(void 0, null, function* () {
      if (resetPageIndex) {
        const nextPagination = __spreadProps(__spreadValues({}, pagination), {
          current: 1
        });
        setPagination(nextPagination);
        reloadWithPagination(nextPagination);
      } else {
        reload();
      }
    }),
    reloadAndRest: () => {
      const nextPagination = __spreadProps(__spreadValues({}, pagination), {
        current: 1
      });
      setPagination(nextPagination);
      request && reloadWithPagination(nextPagination);
    }
  }));
  const onPaginationChange = useRefFunction((page, pageSize) => {
    const nextPagination = __spreadProps(__spreadValues({}, pagination), {
      current: page,
      pageSize
    });
    setPagination(nextPagination);
    if (propsDataSource) {
      setDataSource(
        propsDataSource == null ? void 0 : propsDataSource.slice(
          (nextPagination.current - 1) * nextPagination.pageSize,
          nextPagination.current * nextPagination.pageSize
        )
      );
    }
    request && reloadWithPagination(nextPagination);
    if (propsPagination !== false && (propsPagination == null ? void 0 : propsPagination.onChange)) {
      propsPagination.onChange(page, pageSize);
    }
  });
  const onTableChange = useRefFunction((_pagination, _filter, _sorter) => {
    var _a3;
    const nextSorter = _sorter && Object.keys(_sorter).length > 0 && _sorter.order ? {
      [(_sorter == null ? void 0 : _sorter.columnKey) || (_sorter == null ? void 0 : _sorter.field) || ((_a3 = _sorter == null ? void 0 : _sorter.column) == null ? void 0 : _a3.dataIndex)]: _sorter.order
    } : void 0;
    sorter.current = nextSorter;
    request && reload();
  });
  const tableAction = useContext(SptPageContainerContext);
  const toolbar = useMemo(() => {
    var _b2, _c, _d;
    if (!inheritAction)
      return propsToolbar;
    const _a3 = propsToolbar != null ? propsToolbar : {}, { actions = [], settings = [] } = _a3, restToolbarProps = __objRest(_a3, ["actions", "settings"]);
    const tableActions = [
      (actions == null ? void 0 : actions.length) > 0 || (settings == null ? void 0 : settings.length) > 0 ? /* @__PURE__ */ React.createElement("div", { key: "spt-table-global-divider", className: "spt-table-global-divider" }) : void 0,
      ...(_b2 = tableAction == null ? void 0 : tableAction.tableAction) != null ? _b2 : []
    ];
    return __spreadProps(__spreadValues({}, restToolbarProps), {
      actions: (_d = [
        ...actions,
        ...(tableAction == null ? void 0 : tableAction.tableAction) && ((_c = tableAction == null ? void 0 : tableAction.tableAction) == null ? void 0 : _c.length) > 0 ? tableActions : []
      ]) == null ? void 0 : _d.filter((item) => item),
      settings
    });
  }, [propsToolbar]);
  return /* @__PURE__ */ React.createElement("div", { className: classNames("spt-table-wrap", fitClass) }, search === false ? null : /* @__PURE__ */ React.createElement(
    Filter,
    {
      action: actionRef,
      loading: !!loading,
      columns,
      formRef: baseFormRef,
      form: searchForm,
      search,
      manualRequest,
      type: restProps.type || "table",
      onSubmit: restProps.onSubmit,
      onReset: onPropsReset,
      onFormSearchSubmit,
      searchKeyMap
    }
  ), /* @__PURE__ */ React.createElement(Spin, { spinning: loading }, /* @__PURE__ */ React.createElement(
    Table,
    __spreadValues({
      columns,
      width,
      sticky,
      fit,
      fitClass,
      scrollY,
      onRow,
      className,
      componentId: _componentId,
      cacheKeyWithRoute,
      ResizingIndicator,
      onRowSelection,
      onRowDrawerChange,
      ResizingHeaderCell,
      dataSource,
      onChange: onTableChange,
      pagination: propsPagination,
      toolbar
    }, restProps)
  )), paginationInfo.loaded && propsPagination !== false ? /* @__PURE__ */ React.createElement(
    Pagination,
    __spreadProps(__spreadValues(__spreadProps(__spreadValues({}, pagination), {
      total: paginationInfo.total,
      pageSize: pagination.pageSize
    }), propsPagination != null ? propsPagination : {}), {
      onChange: onPaginationChange
    })
  ) : null);
};
var SptTable = (_a) => {
  var props = __objRest(_a, []);
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(SpotterTableBase, __spreadValues({}, props)));
};
var SptTable_default = SptTable;
export {
  SptTable_default as default
};
