// @spotter/ui/es/components/SptTable/Resizable.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptTable/Resizable.tsx
import React, { useCallback, useEffect, useMemo, useRef } from "react";
import { eventBus } from "@spotter/app-client-toolkit";
import { clearSelection, toFloat } from "./utils";
var getResizingEvent = (id) => `${id}_resizing`;
var ResizingHeaderCell = (props) => {
  const _a = props, {
    onResizeStart,
    width,
    onResizeStop,
    OriginHeaderCell,
    children: overrideChildren
  } = _a, restProps = __objRest(_a, [
    "onResizeStart",
    "width",
    "onResizeStop",
    "OriginHeaderCell",
    "children"
  ]);
  const nodeRef = useRef();
  const thNodeRef = useRef();
  useEffect(() => {
    for (const elem of document.querySelectorAll(
      ".spotter-table-header-column-resize-trigger"
    )) {
      if (elem.onclick) {
        return;
      }
      elem.addEventListener("click", (event) => {
        event.stopPropagation();
        return false;
      });
    }
  }, []);
  if (!width) {
    return /* @__PURE__ */ React.createElement("th", __spreadValues({}, props), OriginHeaderCell ? /* @__PURE__ */ React.createElement(OriginHeaderCell, __spreadValues({}, props), overrideChildren) : overrideChildren);
  }
  return /* @__PURE__ */ React.createElement(
    "th",
    __spreadProps(__spreadValues({}, restProps), {
      ref: (dom) => {
        dom && (thNodeRef.current = dom);
      }
    }),
    /* @__PURE__ */ React.createElement(
      "span",
      {
        className: "inline-block whitespace-pre-line",
        style: { wordBreak: "break-word" },
        ref: (dom) => {
          dom && (nodeRef.current = dom);
        }
      },
      overrideChildren
    ),
    /* @__PURE__ */ React.createElement(
      "div",
      {
        onMouseDown: (triggerMouseDownEvent) => {
          clearSelection();
          const triggerRect = triggerMouseDownEvent.target.getBoundingClientRect();
          const mouseOffsetX = toFloat(
            triggerMouseDownEvent.clientX - (triggerRect.x + triggerRect.width * 0.5)
          );
          window.addEventListener(
            "mouseup",
            (e) => {
              e.stopPropagation();
              const _triggerRect = triggerMouseDownEvent.target.getBoundingClientRect();
              onResizeStop(
                // 鼠标的 x 轴坐标 - 鼠标相对指示器的偏移量 = 指示器的 x 轴坐标
                toFloat(e.clientX - mouseOffsetX) - toFloat(_triggerRect.x + 0.5 * _triggerRect.width),
                toFloat(_triggerRect.x + 0.5 * _triggerRect.width)
              );
            },
            { once: true }
          );
          onResizeStart(
            thNodeRef.current.getBoundingClientRect(),
            nodeRef.current.getBoundingClientRect(),
            triggerMouseDownEvent.target.getBoundingClientRect()
          );
        },
        className: "spotter-table-header-column-resize-trigger"
      }
    )
  );
};
var ResizingIndicator = ({ id }) => {
  const indicatorRef = useRef();
  const resizingRef = useRef({
    inResizing: false,
    boundary: 0,
    startX: 0,
    startMouseOffsetX: 0,
    offsetX: 0
  });
  const onDrag = useCallback(
    (e) => {
      var _a;
      if (resizingRef.current.inResizing) {
        !resizingRef.current.startMouseOffsetX && (resizingRef.current.startMouseOffsetX = e.clientX - resizingRef.current.startX);
        (_a = indicatorRef.current) == null ? void 0 : _a.setAttribute(
          "style",
          `transform:translateX(${// 鼠标 x 轴坐标 - 鼠标相对指示器的偏移量 = 指示器的坐标
          Math.max(
            e.clientX - resizingRef.current.startMouseOffsetX,
            resizingRef.current.boundary
          ) - resizingRef.current.offsetX}px);display:block;`
        );
      }
    },
    [id]
  );
  const resizingHandler = useCallback(() => {
    if (indicatorRef.current) {
      if (resizingRef.current.inResizing) {
        indicatorRef.current.setAttribute(
          "style",
          `transform:translateX(${resizingRef.current.startX - resizingRef.current.offsetX}px);display:block;`
        );
      } else {
        indicatorRef.current.setAttribute("style", "display:none;");
        resizingRef.current.startMouseOffsetX = 0;
      }
    }
  }, [id]);
  useEffect(() => {
    if (indicatorRef.current) {
      resizingRef.current.offsetX = indicatorRef.current.parentNode.getBoundingClientRect().x;
      const { ownerDocument } = indicatorRef.current;
      resizingHandler();
      eventBus.on(getResizingEvent(id), (resizingPayload) => {
        resizingRef.current = __spreadProps(__spreadValues(__spreadValues({}, resizingRef.current), resizingPayload), {
          // 在每次开始拖动前重新校正一遍 offsetX，因为可能随着视窗 resizing 后导致 offsetX 已经变化
          offsetX: indicatorRef.current.parentNode.getBoundingClientRect().x
        });
        resizingHandler();
      });
      ownerDocument.addEventListener("mousemove", onDrag);
    }
    return () => {
      eventBus.off(getResizingEvent(id));
      if (indicatorRef.current) {
        const { ownerDocument } = indicatorRef.current;
        ownerDocument.removeEventListener("mousemove", onDrag);
      }
    };
  }, [id]);
  return /* @__PURE__ */ React.createElement(
    "div",
    {
      ref: (dom) => {
        dom && (indicatorRef.current = dom);
      },
      className: "spotter-table-column-resizable-marker"
    }
  );
};
var useResizable = ({
  headerCell: OriginHeaderCell,
  key
}) => {
  const resizingSettingRef = useRef({ inResizing: false, boundary: 0, startX: 0 });
  const updateResizableColumns = useCallback((originColumns) => {
    const cols = originColumns.map((originCol) => __spreadProps(__spreadValues({}, originCol), {
      onHeaderCell: (col) => {
        var _a;
        return __spreadProps(__spreadValues({}, (_a = originCol.onHeaderCell) == null ? void 0 : _a.call(originCol, col)), {
          width: col.width,
          OriginHeaderCell,
          onResizeStart: (thRect, thChildRect, triggerRect) => {
            var _a2, _b;
            resizingSettingRef.current = {
              inResizing: true,
              // 必须用 thRect 的 x 坐标因为 thChild 可能是右对齐.
              boundary: toFloat(
                thRect.x + Math.min(thChildRect.width, (_b = (_a2 = col.resizable) == null ? void 0 : _a2.boundary) != null ? _b : 71) + // 16 for <th> padding block 8px
                17
              ),
              startX: toFloat(triggerRect.x + triggerRect.width * 0.5)
            };
            eventBus.emit(getResizingEvent(key), resizingSettingRef.current);
          },
          onResizeStop: (diffWidthByIndicator, triggerX) => {
            const { boundary } = resizingSettingRef.current;
            resizingSettingRef.current = {
              inResizing: false,
              boundary: 0,
              startX: 0
            };
            eventBus.emit(getResizingEvent(key), resizingSettingRef.current);
            eventBus.emit("resizeColumn", {
              columnKey: col.key,
              width: Math.max((boundary != null ? boundary : 0) - triggerX, diffWidthByIndicator)
            });
          }
        });
      }
    }));
    return !cols.some((i) => i.key === "resize-placeholder") ? [
      ...cols,
      // 自动注入空白列来吸收多余的宽度，否则 resizing 后的总列宽小于容器宽度时会因为 antd 会强制其他列按比例吸收被调低的列宽出现抖动问题
      {
        key: "resize-placeholder",
        search: false,
        editable: false,
        className: "resize-placeholder"
      }
    ] : cols;
  }, []);
  return useMemo(
    () => ({
      ResizingHeaderCell,
      ResizingIndicator,
      updateResizableColumns
    }),
    []
  );
};
export {
  useResizable
};
