// @spotter/ui/es/components/SptTable/StickyScrollbar.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptTable/StickyScrollbar.tsx
import React, { useCallback, useEffect, useRef } from "react";
import { eventBus } from "@spotter/app-client-toolkit";
import { createPortal } from "react-dom";
import { clearSelection, toFloat } from "./utils";
var getScrollbarEvent = (id, event) => `${id}_SCROLL_BAR_${event}`;
var StickyScrollbar = ({
  id,
  getContainer
}) => {
  const tableSizeRef = useRef({
    fullWidth: 0,
    scrollbarWidth: 0,
    scrollbarScale: 0,
    scrollbarThumbOffset: 0,
    scrollbarClientX: 0
  });
  const scrollbarThumbRef = useRef();
  const scrollbarRef = useRef();
  const moveScrollbar = useCallback(
    (tableBodyScrollLeft) => {
      tableSizeRef.current.scrollbarThumbOffset = tableBodyScrollLeft * tableSizeRef.current.scrollbarScale;
      if (scrollbarThumbRef.current) {
        scrollbarThumbRef.current.style.setProperty(
          "transform",
          `translateX(${toFloat(tableSizeRef.current.scrollbarThumbOffset)}px)`
        );
      }
    },
    [id]
  );
  useEffect(() => {
    eventBus.on(
      getScrollbarEvent(id, 0 /* FULL_WIDTH_UPDATE */),
      ({ tableFullWidth, tableBodyScrollLeft }) => {
        const { width, x } = scrollbarRef.current.getBoundingClientRect();
        const scrollbarScale = Math.max(Math.min(width / tableFullWidth, 1), 0);
        tableSizeRef.current = __spreadProps(__spreadValues({}, tableSizeRef.current), {
          scrollbarWidth: width,
          scrollbarClientX: x,
          fullWidth: tableFullWidth,
          scrollbarThumbOffset: tableBodyScrollLeft * scrollbarScale,
          scrollbarScale
        });
        if (tableSizeRef.current.scrollbarScale === 1) {
          scrollbarThumbRef.current.style.setProperty("visibility", "hidden");
        } else {
          scrollbarThumbRef.current.setAttribute(
            "style",
            `visibility:visible;width:${toFloat(
              tableSizeRef.current.scrollbarWidth * tableSizeRef.current.scrollbarScale
            )}px;transform:translateX(${toFloat(
              tableSizeRef.current.scrollbarThumbOffset
            )}px)`
          );
        }
      }
    );
    eventBus.on(getScrollbarEvent(id, 1 /* TABLE_DODY_SCROLL */), moveScrollbar);
    return () => {
      eventBus.off(getScrollbarEvent(id, 0 /* FULL_WIDTH_UPDATE */));
      eventBus.off(getScrollbarEvent(id, 1 /* TABLE_DODY_SCROLL */));
    };
  }, [id]);
  return createPortal(
    /* @__PURE__ */ React.createElement(
      "div",
      {
        ref: (dom) => {
          if (dom) {
            scrollbarRef.current = dom;
            if (scrollbarRef.current) {
              const { width, x } = scrollbarRef.current.getBoundingClientRect();
              tableSizeRef.current = __spreadProps(__spreadValues({}, tableSizeRef.current), {
                scrollbarWidth: width,
                scrollbarClientX: x,
                scrollbarScale: Math.max(
                  Math.min(width / tableSizeRef.current.fullWidth, 1),
                  0
                )
              });
            }
          }
        },
        className: "spotter-table-sticky-scroll-bar"
      },
      /* @__PURE__ */ React.createElement(
        "div",
        {
          onClick: (e) => {
            e.stopPropagation();
            e.preventDefault();
          },
          onMouseDown: (mouseDownEvent) => {
            mouseDownEvent.stopPropagation();
            clearSelection();
            const { ownerDocument } = scrollbarThumbRef.current;
            const mouseOffsetXWithScrollBarThumb = mouseDownEvent.clientX - tableSizeRef.current.scrollbarClientX - tableSizeRef.current.scrollbarThumbOffset;
            scrollbarRef.current.classList.add(
              "spotter-table-sticky-scroll-bar-thumb-dragging"
            );
            const scrollbarDrag = (mouseMoveEvent) => {
              tableSizeRef.current.scrollbarThumbOffset = Math.min(
                Math.max(
                  mouseMoveEvent.clientX - mouseOffsetXWithScrollBarThumb - tableSizeRef.current.scrollbarClientX,
                  0
                ),
                tableSizeRef.current.scrollbarWidth * (1 - tableSizeRef.current.scrollbarScale)
              );
              eventBus.emit(
                getScrollbarEvent(id, 2 /* SCROLLBAR_MOVE */),
                // 将滚动条的 scrollLeft 通过比例转为实际 table body 的 scrollLeft
                tableSizeRef.current.scrollbarThumbOffset / tableSizeRef.current.scrollbarScale
              );
            };
            ownerDocument.addEventListener("mousemove", scrollbarDrag);
            ownerDocument.addEventListener(
              "mouseup",
              (mouseUpEvent) => {
                mouseUpEvent.stopPropagation();
                ownerDocument.removeEventListener("mousemove", scrollbarDrag);
                scrollbarRef.current.classList.remove(
                  "spotter-table-sticky-scroll-bar-thumb-dragging"
                );
              },
              { once: true }
            );
          },
          className: "spotter-table-sticky-scroll-bar-thumb",
          ref: (dom) => {
            dom && (scrollbarThumbRef.current = dom);
          }
        }
      ),
      /* @__PURE__ */ React.createElement(
        "div",
        {
          className: "spotter-table-sticky-scroll-bar-track",
          onClick: (e) => {
            eventBus.emit(
              getScrollbarEvent(id, 2 /* SCROLLBAR_MOVE */),
              toFloat(e.clientX - tableSizeRef.current.scrollbarClientX)
            );
          }
        }
      )
    ),
    getContainer()
  );
};
var ScrollBodyWrapper = (_a) => {
  var _b = _a, { id } = _b, restProps = __objRest(_b, ["id"]);
  const tbodyRef = useRef();
  const scrollContainerRef = useRef();
  const scrollHandler = useCallback(
    (e) => {
      eventBus.emit(
        getScrollbarEvent(id, 1 /* TABLE_DODY_SCROLL */),
        e.target.scrollLeft
      );
    },
    [id]
  );
  const scrollbarDragHandler = useCallback(
    (scrollLeft) => {
      scrollContainerRef.current.scrollLeft = scrollLeft;
    },
    [id]
  );
  const resizeHandler = useCallback(() => {
    eventBus.emit(getScrollbarEvent(id, 0 /* FULL_WIDTH_UPDATE */), {
      tableFullWidth: tbodyRef.current.getBoundingClientRect().width,
      tableBodyScrollLeft: tbodyRef.current.parentElement.parentElement.scrollLeft
    });
  }, [id]);
  useEffect(() => {
    if (tbodyRef.current) {
      tbodyRef.current.addEventListener("scroll", scrollHandler);
      window.addEventListener("resize", resizeHandler);
    }
    return () => {
      if (tbodyRef.current) {
        window.removeEventListener("resize", resizeHandler);
        eventBus.off(
          getScrollbarEvent(id, 2 /* SCROLLBAR_MOVE */),
          scrollbarDragHandler
        );
      }
    };
  }, [id]);
  return /* @__PURE__ */ React.createElement(
    "tbody",
    __spreadValues({
      ref: (dom) => {
        if (dom) {
          tbodyRef.current = dom;
          scrollContainerRef.current = dom.parentElement.parentElement;
          resizeHandler();
          scrollContainerRef.current.removeEventListener("scroll", scrollHandler);
          scrollContainerRef.current.addEventListener("scroll", scrollHandler);
          eventBus.off(
            getScrollbarEvent(id, 2 /* SCROLLBAR_MOVE */),
            scrollbarDragHandler
          );
          eventBus.on(
            getScrollbarEvent(id, 2 /* SCROLLBAR_MOVE */),
            scrollbarDragHandler
          );
        }
      }
    }, restProps)
  );
};
export {
  ScrollBodyWrapper,
  StickyScrollbar
};
