// @spotter/ui/es/components/SptTable/Table.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptTable/Table.tsx
import { ProTable } from "@ant-design/pro-components";
import classnames from "classnames";
import React, { memo, useCallback, useMemo, useRef } from "react";
import { ScrollBodyWrapper, StickyScrollbar } from "./StickyScrollbar";
var NODE_NAME_BLACK_LIST_DRAWER_TRIGGER = /* @__PURE__ */ new Set(["A"]);
function Table(_a) {
  var _b = _a, {
    className,
    width,
    fit,
    fitClass,
    sticky,
    columns,
    cacheKeyWithRoute,
    onRow,
    onRowSelection,
    onRowDrawerChange,
    ResizingHeaderCell,
    components: originComponents,
    ResizingIndicator,
    scrollY,
    pagination,
    componentId: _componentId
  } = _b, restProps = __objRest(_b, [
    "className",
    "width",
    "fit",
    "fitClass",
    "sticky",
    "columns",
    "cacheKeyWithRoute",
    "onRow",
    "onRowSelection",
    "onRowDrawerChange",
    "ResizingHeaderCell",
    "components",
    "ResizingIndicator",
    "scrollY",
    "pagination",
    "componentId"
  ]);
  const spotterTableWrapperRef = useRef();
  const tableViewRender = useCallback(
    (_, defaultDom) => /* @__PURE__ */ React.createElement(
      "div",
      {
        id: _componentId,
        className: classnames("spotter-table-wrapper", {
          "with-pagination": pagination === void 0 ? true : pagination
          // 'with-pagination': true,
        }),
        ref: (dom) => {
          spotterTableWrapperRef.current = dom;
        }
      },
      defaultDom,
      /* @__PURE__ */ React.createElement(ResizingIndicator, { id: _componentId }),
      /* @__PURE__ */ React.createElement(
        StickyScrollbar,
        {
          id: _componentId,
          getContainer: () => {
            var _a2;
            return ((_a2 = spotterTableWrapperRef.current) == null ? void 0 : _a2.querySelector(
              ".ant-table-wrapper >.ant-spin-nested-loading >.ant-spin-container >.ant-table"
            )) || document.body;
          }
        }
      )
    ),
    [ResizingIndicator, _componentId]
  );
  const spotterTablePresetProps = useMemo(
    () => ({
      className: classnames(className, fitClass, "spotter-table"),
      sticky,
      scroll: {
        y: scrollY,
        x: width,
        scrollToFirstRowOnChange: true
      },
      options: false,
      size: "small",
      onRow: (record) => {
        const eventHandlers = onRow == null ? void 0 : onRow(record);
        return __spreadProps(__spreadValues({}, eventHandlers), {
          onClick: (e) => {
            var _a2, _b2;
            if ((_a2 = document.getSelection()) == null ? void 0 : _a2.toString()) {
              onRowSelection == null ? void 0 : onRowSelection(document.getSelection(), e);
            } else {
              !NODE_NAME_BLACK_LIST_DRAWER_TRIGGER.has(
                e.target.nodeName
              ) && (onRowDrawerChange == null ? void 0 : onRowDrawerChange(true, record));
            }
            (_b2 = eventHandlers == null ? void 0 : eventHandlers.onClick) == null ? void 0 : _b2.call(eventHandlers, e);
          }
        });
      },
      // eslint-disable-next-line react/no-unstable-nested-components
      tableViewRender
    }),
    [
      className,
      width,
      sticky,
      fitClass,
      fit,
      onRowDrawerChange,
      onRowSelection,
      cacheKeyWithRoute,
      scrollY
    ]
  );
  const componentHeader = useMemo(
    () => __spreadProps(__spreadValues({}, originComponents == null ? void 0 : originComponents.header), {
      cell: ResizingHeaderCell
    }),
    [originComponents, ResizingHeaderCell]
  );
  const componentBody = useMemo(
    () => __spreadProps(__spreadValues({}, originComponents == null ? void 0 : originComponents.body), {
      // eslint-disable-next-line react/no-unstable-nested-components
      wrapper: (props) => /* @__PURE__ */ React.createElement(ScrollBodyWrapper, __spreadValues({ id: _componentId }, props))
    }),
    [originComponents, _componentId]
  );
  return /* @__PURE__ */ React.createElement(
    ProTable,
    __spreadProps(__spreadValues(__spreadValues({}, restProps), spotterTablePresetProps), {
      search: false,
      pagination: false,
      loading: false,
      columns,
      components: {
        table: originComponents == null ? void 0 : originComponents.table,
        header: componentHeader,
        body: componentBody
      }
    })
  );
}
var Table_default = memo(Table);
export {
  Table_default as default
};
