// @spotter/ui/es/components/SptSearchText/index.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptSearchText/index.tsx
import { SearchOutlined } from "@ant-design/icons";
import { ProFormText } from "@ant-design/pro-components";
import React, { useRef, useState } from "react";
import { useMemo } from "react";
import { SptComponentProvider } from "../Provider";
import { Tooltip } from "antd";
import "./style/index.less";
var SptSearchText = (_a) => {
  var _b = _a, {
    autoSearch = false,
    label,
    fieldProps,
    placeholder
  } = _b, proFormTextProps = __objRest(_b, [
    "autoSearch",
    "label",
    "fieldProps",
    "placeholder"
  ]);
  const ref = useRef(null);
  const [showTip, setShowTip] = useState(false);
  const fix = useMemo(
    () => ({
      prefix: autoSearch ? /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(
        SearchOutlined,
        {
          className: "cursor-pointer py-4px",
          style: { color: "var(--color-text-tertiary)" },
          rev: void 0
        }
      ), /* @__PURE__ */ React.createElement("div", { style: { color: "var(--color-text-tertiary)", paddingRight: "4px" } }, (fieldProps == null ? void 0 : fieldProps.label) || label)) : /* @__PURE__ */ React.createElement("div", { style: { color: "var(--color-text-tertiary)", paddingRight: "4px" } }, (fieldProps == null ? void 0 : fieldProps.label) || label)
    }),
    [autoSearch, fieldProps == null ? void 0 : fieldProps.label, label]
  );
  const handleMouseOver = () => {
    var _a2;
    const inputElement = (_a2 = ref == null ? void 0 : ref.current) == null ? void 0 : _a2.querySelector("input");
    const inputElementWidth = inputElement == null ? void 0 : inputElement.clientWidth;
    const placeholderStr = inputElement == null ? void 0 : inputElement.getAttribute("placeholder");
    const placeholderElement = document.createElement("div");
    placeholderElement.style.position = "absolute";
    placeholderElement.style.width = `${inputElementWidth}px`;
    placeholderElement.style.visibility = "hidden";
    placeholderElement.style.whiteSpace = "nowrap";
    placeholderElement.innerHTML = placeholderStr || "";
    document.body.appendChild(placeholderElement);
    setShowTip(placeholderElement.scrollWidth > placeholderElement.clientWidth);
    document.body.removeChild(placeholderElement);
  };
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement("div", { ref, onMouseOver: handleMouseOver }, /* @__PURE__ */ React.createElement(
    Tooltip,
    {
      overlayClassName: "spt-search-text-tooltip",
      mouseLeaveDelay: 0,
      title: showTip ? (fieldProps == null ? void 0 : fieldProps.placeholder) || placeholder : ""
    },
    /* @__PURE__ */ React.createElement(
      ProFormText,
      __spreadValues({
        noStyle: true,
        fieldProps: __spreadValues(__spreadValues({}, fix), fieldProps)
      }, proFormTextProps)
    ),
    /* @__PURE__ */ React.createElement("div", { style: { display: "none" } }, (fieldProps == null ? void 0 : fieldProps.placeholder) || placeholder)
  )));
};
var SptSearchText_default = SptSearchText;
export {
  SptSearchText,
  SptSearchText_default as default
};
