// @spotter/ui/es/components/Downgrade/ClientError.js
// src/components/Downgrade/ClientError.tsx
import { useEffect, useState } from "react";
import { Button, Result } from "antd";
import { useSptTranslation } from "@spotter/i18n-sdk";
import React from "react";
import { domBreakingAutoReporting } from "@spotter/app-client-toolkit";
var ClientError = ({ error }) => {
  const [openDetail, setOpenDetail] = useState(false);
  const { t } = useSptTranslation();
  useEffect(() => {
    domBreakingAutoReporting(new Error(error == null ? void 0 : error.message));
  }, []);
  return /* @__PURE__ */ React.createElement("div", { className: "h-full" }, /* @__PURE__ */ React.createElement("div", { className: "overflow-auto h-full flex flex-col box-border justify-between bg-white" }, /* @__PURE__ */ React.createElement(
    Result,
    {
      title: t("infra.downgrade_client_crashed_title"),
      subTitle: openDetail ? /* @__PURE__ */ React.createElement("div", null, /* @__PURE__ */ React.createElement("span", { className: "mr-8px" }, " ", error == null ? void 0 : error.message)) : /* @__PURE__ */ React.createElement(
        "a",
        {
          onClick: () => {
            setOpenDetail(true);
          }
        },
        t("common.detail")
      ),
      status: "500",
      extra: /* @__PURE__ */ React.createElement(Button, { type: "primary", href: "/" }, t("infra.back_home"))
    }
  )));
};
export {
  ClientError
};
