// @spotter/eslint-config-atlas@1.0.6/index.js
const OFF = 0;
const WARNING = 1;
const ERROR = 2;

module.exports = {
  env: {
    es6: true,
    node: true,
  },
  reportUnusedDisableDirectives: true,
  plugins: [
    "unicorn",
    "promise",
    "@typescript-eslint",
    "jsonc",
    "yml",
    "import",
  ],
  extends: [
    "airbnb-base",
    "plugin:eslint-comments/recommended",
    "plugin:unicorn/recommended",
    "plugin:promise/recommended",
    "plugin:jsonc/recommended-with-jsonc",
    "plugin:yml/standard",
    "plugin:@typescript-eslint/recommended",
    "plugin:import/typescript",
    "prettier",
  ],
  parser: "@typescript-eslint/parser",
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 2020,
    sourceType: "module",
  },
  settings: {
    "import/resolver": {
      node: {
        // import 模块时，不写后缀将尝试导入的后缀，出现频率高的文件类型放前面
        extensions: [
          ".js",
          ".jsx",
          ".mjs",
          ".ts",
          ".tsx",
          ".d.ts",
          ".json",
          "scss",
        ],
      },
      typescript: {
        project: ["tsconfig.json"],
      },
    },
  },
  ignorePatterns: [
    "*.min.*",
    "*.d.ts",
    "CHANGELOG.md",
    "dist",
    "LICENSE*",
    "output",
    "coverage",
    "public",
    "temp",
    "package-lock.json",
    "pnpm-lock.yaml",
    "pnpm-workspace.yaml",
    "yarn.lock",
    "__snapshots__",
    "!.github",
    "!.vitepress",
    "!.vscode",
  ],
  overrides: [
    {
      files: ["*.json", "*.json5"],
      parser: "jsonc-eslint-parser",
      rules: {
        "jsonc/array-bracket-spacing": ["error", "never"],
        "jsonc/comma-dangle": ["error", "never"],
        "jsonc/comma-style": ["error", "last"],
        "jsonc/key-spacing": [
          "error",
          { beforeColon: false, afterColon: true },
        ],
        "jsonc/no-octal-escape": "error",
        "jsonc/object-curly-newline": [
          "error",
          { multiline: true, consistent: true },
        ],
        "jsonc/object-curly-spacing": ["error", "always"],
        "jsonc/object-property-newline": [
          "error",
          { allowMultiplePropertiesPerLine: true },
        ],
      },
    },
    {
      files: ["*.yaml", "*.yml"],
      parser: "yaml-eslint-parser",
      rules: {
        "spaced-comment": "off",
      },
    },
    {
      files: ["package.json"],
      parser: "jsonc-eslint-parser",
      rules: {
        "jsonc/sort-keys": [
          "error",
          {
            pathPattern: "^$",
            order: [
              "publisher",
              "name",
              "displayName",
              "type",
              "version",
              "private",
              "packageManager",
              "description",
              "author",
              "license",
              "funding",
              "homepage",
              "repository",
              "bugs",
              "keywords",
              "categories",
              "sideEffects",
              "exports",
              "main",
              "module",
              "unpkg",
              "jsdelivr",
              "types",
              "typesVersions",
              "bin",
              "icon",
              "files",
              "engines",
              "activationEvents",
              "contributes",
              "scripts",
              "peerDependencies",
              "peerDependenciesMeta",
              "dependencies",
              "optionalDependencies",
              "devDependencies",
              "pnpm",
              "overrides",
              "resolutions",
              "husky",
              "simple-git-hooks",
              "lint-staged",
              "eslintConfig",
            ],
          },
          {
            pathPattern: "^(?:dev|peer|optional|bundled)?[Dd]ependencies$",
            order: { type: "asc" },
          },
          {
            pathPattern: "^exports.*$",
            order: ["types", "require", "import"],
          },
        ],
      },
    },

    {
      files: ["**/*.d.ts"],
      rules: {
        "import/no-duplicates": OFF,
      },
    },
    {
      files: [
        "scripts/**/*.*s",
        "Gulpfile.ts",
        "**/test/**/*.ts",
        "**/*.test.js",
        "**/*.spec.js",
        "client/**/*.ts",
      ],
      rules: {
        "import/no-extraneous-dependencies": OFF,
      },
    },
    {
      files: ["*.config.ts", "*.config.js"],
      rules: {
        "import/no-extraneous-dependencies": OFF,
        "unicorn/prefer-module": OFF,
        "@typescript-eslint/no-var-requires": OFF,
      },
    },
  ],
  rules: {
    "eslint-comments/disable-enable-pair": [ERROR, { allowWholeFile: true }],
    "unicorn/prevent-abbreviations": OFF,
    "unicorn/filename-case": [
      WARNING,
      {
        cases: {
          // 中划线
          kebabCase: true,
          // 小驼峰
          camelCase: true,
          // 下划线
          snakeCase: true,
          // 大驼峰
          pascalCase: true,
        },
      },
    ],
    "unicorn/no-process-exit": OFF,
    "unicorn/explicit-length-check": OFF,
    "unicorn/no-null": OFF,
    "unicorn/no-array-reduce": OFF,
    "unicorn/expiring-todo-comments": OFF,
    "unicorn/numeric-separators-style": [WARNING],
    "promise/always-return": OFF,
    "func-names": OFF,
    "lines-between-class-members": OFF,
    "max-classes-per-file": OFF,
    "class-methods-use-this": OFF,
    "no-console": OFF,
    "no-empty": OFF,
    "no-param-reassign": OFF,
    "no-plusplus": OFF,
    "no-underscore-dangle": OFF,
    "no-unused-expressions": OFF,
    "no-useless-constructor": OFF,
    "unicorn/no-useless-undefined": OFF,
    "no-bitwise": OFF,
    "consistent-return": OFF,
    "no-unsafe-finally": OFF,
    "no-restricted-syntax": OFF,
    "no-use-before-define": OFF,
    "no-return-assign": OFF,
    "no-template-curly-in-string": OFF,
    "no-shadow": OFF,
    "promise/catch-or-return": OFF,
    "arrow-body-style": OFF,
    camelcase: OFF,
    "@typescript-eslint/explicit-function-return-type": OFF,
    "@typescript-eslint/no-explicit-any": OFF,
    "@typescript-eslint/no-non-null-assertion": OFF,
    "@typescript-eslint/no-useless-constructor": ERROR,
    "@typescript-eslint/camelCase": OFF,
    "@typescript-eslint/no-use-before-define": [ERROR],
    "@typescript-eslint/no-shadow": [ERROR],
    "@typescript-eslint/ban-ts-comment": OFF,
    "@typescript-eslint/no-restricted-imports": [
      ERROR,
      {
        paths: [
          {
            name: "xlsx",
            message: '请使用异步导入的方式使用xlsx，例如：import("xlsx")',
            allowTypeImports: true,
          },
          {
            name: "exceljs",
            message: '请使用异步导入的方式使用exceljs，例如：import("xlsx")',
            allowTypeImports: true,
          },
        ],
      },
    ],
    "import/extensions": [
      ERROR,
      "ignorePackages",
      {
        ts: "never",
        tsx: "never",
        // json: 'never',
        js: "never",
      },
    ],
    "import/no-cycle": OFF,
    "import/prefer-default-export": OFF,
    "import/no-extraneous-dependencies": [WARNING],
    "unicorn/prefer-node-protocol": OFF,
    "unicorn/import-style": OFF,
    "unicorn/prefer-module": OFF,
    "@typescript-eslint/no-var-requires": OFF,
  },
};
