// @spotter/i18n-sdk@3.1.20/lib/api-sdk/domain/i18nApiSdk/i18n.sdk.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterI18nApiSdk = void 0;
const base_api_sdk_1 = __importDefault(require("../../utils/base-api-sdk"));
const const_1 = require("../../../const");
/**
 * SpotterI18nApiSdk
 * i18n SDK
 */
class SpotterI18nApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
     * 获取国际化数据
     * @param data
     */
    getLanguageFile(data) {
        return __awaiter(this, void 0, void 0, function* () {
            const { project, component, language, domainUrl } = data;
            return this.httpClient.request({
                method: 'get',
                // 如果 domainUrl 不存在，回退到常量
                baseURL: domainUrl || const_1.DOMAIN_URL,
                // 如何 language 不存在，回退到 zh_Hans
                url: `/api/translations/${project}/${component}/${
                // zh_Hans to zh映射
                language === 'zh' ? 'zh_Hans' : (language || 'zh_Hans')}/file/`,
                headers: {
                    'Content-Type': 'application/json',
                    // token这里直接写死
                    Authorization: 'Token wlu_xRmTEHTCLVlpGhYunVeyFm4kO7AORbHUcZfv',
                    'Cache-Control': 'no-cache'
                },
            });
        });
    }
}
exports.SpotterI18nApiSdk = SpotterI18nApiSdk;
