// @spotter/ui/lib/types/enum.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WAREHOUSE_TYPE = exports.ValidateStorageName = exports.UserStatus = exports.UserAccountAvailableStatus = exports.RoleCode = exports.PermissionType = exports.MenuTypeMap = exports.MenuType = exports.Actions = exports.AccountType = void 0;
var _MenuTypeMap;
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
var AccountType = exports.AccountType = /*#__PURE__*/function (AccountType) {
  AccountType[AccountType["EMAIL"] = 1] = "EMAIL";
  AccountType[AccountType["PHONE"] = 2] = "PHONE";
  return AccountType;
}({});
var UserAccountAvailableStatus = exports.UserAccountAvailableStatus = /*#__PURE__*/function (UserAccountAvailableStatus) {
  UserAccountAvailableStatus[UserAccountAvailableStatus["DISABLE"] = -1] = "DISABLE";
  UserAccountAvailableStatus[UserAccountAvailableStatus["ENABLE"] = 1] = "ENABLE";
  UserAccountAvailableStatus[UserAccountAvailableStatus["PENDING"] = 2] = "PENDING";
  return UserAccountAvailableStatus;
}({}); // 权限分类
var PermissionType = exports.PermissionType = /*#__PURE__*/function (PermissionType) {
  PermissionType[PermissionType["MENU"] = 1] = "MENU";
  PermissionType[PermissionType["API"] = 2] = "API";
  PermissionType[PermissionType["DATA"] = 3] = "DATA";
  return PermissionType;
}({}); // 菜单分类
var MenuType = exports.MenuType = /*#__PURE__*/function (MenuType) {
  MenuType[MenuType["GMESH"] = 1] = "GMESH";
  MenuType[MenuType["ROMERX"] = 2] = "ROMERX";
  return MenuType;
}({});
var MenuTypeMap = exports.MenuTypeMap = (_MenuTypeMap = {}, _defineProperty(_MenuTypeMap, MenuType.GMESH, 'gmesh'), _defineProperty(_MenuTypeMap, MenuType.ROMERX, 'sevc'), _MenuTypeMap);
var Actions = exports.Actions = /*#__PURE__*/function (Actions) {
  Actions["EDIT"] = "edit";
  Actions["ADD"] = "add";
  Actions["DELETE"] = "delete";
  return Actions;
}({});
var RoleCode = exports.RoleCode = /*#__PURE__*/function (RoleCode) {
  RoleCode["accountManager"] = "account_manager";
  RoleCode["businessManager"] = "business_manager_leader";
  return RoleCode;
}({});
var UserStatus = exports.UserStatus = /*#__PURE__*/function (UserStatus) {
  UserStatus[UserStatus["FREEZE"] = -1] = "FREEZE";
  UserStatus[UserStatus["ACTIVE"] = 1] = "ACTIVE";
  UserStatus[UserStatus["INACTIVE"] = 2] = "INACTIVE";
  return UserStatus;
}({}); // 仓库类型
var WAREHOUSE_TYPE = exports.WAREHOUSE_TYPE = /*#__PURE__*/function (WAREHOUSE_TYPE) {
  WAREHOUSE_TYPE[WAREHOUSE_TYPE["Third"] = 1] = "Third";
  WAREHOUSE_TYPE[WAREHOUSE_TYPE["Cloud"] = 2] = "Cloud";
  return WAREHOUSE_TYPE;
}({});
var ValidateStorageName = exports.ValidateStorageName = /*#__PURE__*/function (ValidateStorageName) {
  ValidateStorageName["PERMIT_DATA"] = "PERMIT_DATA";
  ValidateStorageName["PERMIT_TAG_REMOTE"] = "permitEtag";
  ValidateStorageName["PERMIT_TAG_LOCAL"] = "PERMIT_ETAG_LOCAL";
  ValidateStorageName["MENU_DATA"] = "MENU_DATA";
  ValidateStorageName["MENU_TAG_REMOTE"] = "menuEtag";
  ValidateStorageName["MENU_TAG_LOCAL"] = "MEN_ETAG_LOCAL";
  ValidateStorageName["ROLE_DATA"] = "ROLE_DATA";
  ValidateStorageName["ROLE_TAG_REMOTE"] = "roleEtag";
  ValidateStorageName["ROLE_TAG_LOCAL"] = "ROLE_ETAG_LOCAL";
  return ValidateStorageName;
}({});
