// @spotter/ui/es/components/SptSearchSelect/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptSearchSelect/index.tsx
import { ProFormSelect } from "@ant-design/pro-components";
import classNames from "classnames";
import { nanoid } from "nanoid";
import React, { useMemo } from "react";
import { useEffect, useState } from "react";
import { SptComponentProvider } from "../Provider";
import "./style/index.less";
var SptSearchSelect = (_a) => {
  var _b = _a, {
    label,
    className,
    fieldProps
  } = _b, proFormSelectProps = __objRest(_b, [
    "label",
    "className",
    "fieldProps"
  ]);
  const baseClassName = "spt-search-select-with-inner-label";
  const { mode: fieldPropsModel } = fieldProps || {};
  const [searchText, setSearchText] = useState("");
  const [select, setSelect] = useState([]);
  const mode = (proFormSelectProps == null ? void 0 : proFormSelectProps.mode) || fieldPropsModel;
  const id = `${baseClassName}-${nanoid(8).toLocaleLowerCase()}`;
  const onChangeGuard = (_value, _label) => {
    setSelect(_value);
    (fieldProps == null ? void 0 : fieldProps.onChange) && fieldProps.onChange(_value, _label);
  };
  const onSearchGuard = (value) => {
    value !== searchText && setSearchText(value);
    (fieldProps == null ? void 0 : fieldProps.onSearch) && fieldProps.onSearch(value);
  };
  const onBlurGuard = (e) => {
    setSearchText(searchText === "" ? "-" : "");
    (fieldProps == null ? void 0 : fieldProps.onBlur) && fieldProps.onBlur(e);
  };
  const onClearGuard = () => {
    setSearchText(searchText === "" ? "-" : "");
    (fieldProps == null ? void 0 : fieldProps.onClear) && fieldProps.onClear();
  };
  useEffect(() => {
    var _a2, _b2, _c, _d, _e;
    (_a2 = document.querySelector(`.${id} .ant-select-selector`)) == null ? void 0 : _a2.setAttribute("data-spt-search-select-inner-label", label);
    (_b2 = document.querySelector(`.${id} .ant-select-selector .ant-select-selection-placeholder`)) == null ? void 0 : _b2.setAttribute("data-spt-search-select-inner-label", label);
    mode !== "multiple" && mode !== "tags" && ((_c = document.querySelector(`.${id} .ant-select-selector .ant-select-selection-search`)) == null ? void 0 : _c.setAttribute("data-spt-search-select-inner-label", label));
    if (mode === "multiple" && select.length === 0) {
      (_d = document.querySelector(`.${id} .ant-select-selector .ant-select-selection-placeholder`)) == null ? void 0 : _d.setAttribute("data-spt-search-select-inner-label", label);
    }
    if (mode === "tags") {
      (_e = document.querySelector(`.${id} .ant-select-selector .ant-select-selection-placeholder`)) == null ? void 0 : _e.setAttribute("data-spt-search-select-inner-label", label);
    }
  });
  const showSearchProperty = useMemo(() => {
    if ((fieldProps == null ? void 0 : fieldProps.showSearch) !== void 0) {
      return {
        showSearch: fieldProps == null ? void 0 : fieldProps.showSearch
      };
    }
    return mode === "multiple" || mode === "tags" ? {
      showSearch: true
    } : {};
  }, []);
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(
    ProFormSelect,
    __spreadValues({
      noStyle: true,
      fieldProps: __spreadProps(__spreadValues(__spreadValues({}, showSearchProperty), fieldProps), {
        className: classNames(
          baseClassName,
          mode,
          id,
          className,
          fieldProps == null ? void 0 : fieldProps.className
        ),
        onClear: onClearGuard,
        onSearch: onSearchGuard,
        onChange: onChangeGuard,
        onBlur: onBlurGuard,
        getPopupContainer: (triggerNode) => triggerNode.parentNode
      })
    }, proFormSelectProps)
  ));
};
var SptSearchSelect_default = SptSearchSelect;
export {
  SptSearchSelect,
  SptSearchSelect_default as default
};
