// @spotter/i18n-sdk@3.1.21/lib/hooks.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sptI18n = exports.useSptTranslation = void 0;
const react_1 = require("react");
const react_i18next_1 = require("react-i18next");
const i18next_1 = __importDefault(require("i18next"));
function useSptTranslation(monitor = true) {
    const { t, i18n } = (0, react_i18next_1.useTranslation)();
    return {
        i18n,
        t: (0, react_1.useCallback)((key, options = {}) => {
            var _a, _b;
            if (!key)
                return '';
            if (monitor && !((_a = options === null || options === void 0 ? void 0 : options.defaultValue) === null || _a === void 0 ? void 0 : _a.trim())) {
                console.error(`i18nKey 加载错误：${key} `);
                const error = (options === null || options === void 0 ? void 0 : options.defaultValue) === undefined
                    ? new Error(`i18nKey 丢失：${key} `)
                    : new Error(`i18nKey 兜底文案丢失：${key} `);
                (_b = window === null || window === void 0 ? void 0 : window.__OSS_MONITOR_SDK_) === null || _b === void 0 ? void 0 : _b.error({
                    message: error.message,
                    stack: error.stack
                });
            }
            if (typeof key === 'string') {
                return t(key, options);
            }
            if (typeof key === 'object') {
                let i18Vo = key;
                return t(i18Vo.i18nKey, Object.assign(Object.assign(Object.assign({}, i18Vo.params), options), { 
                    // 先使用后端给的，最后使用option的，option为应用本地生成的
                    defaultValue: i18Vo.defaultValue || options.defaultValue }));
            }
            return '国际化格式错误，请检查!';
        }, [t])
    };
}
exports.useSptTranslation = useSptTranslation;
/**
 * 用于对后端接口返回进行翻译，多用于catch
 * @param key
 * @param options
 * @returns
 */
const sptI18n = (key, options = {}) => {
    if (!key)
        return '';
    if (typeof key === 'string') {
        return i18next_1.default.t(key, options);
    }
    if (typeof key === 'object') {
        let i18Vo = key;
        return i18next_1.default.t(i18Vo.i18nKey, Object.assign({ defaultValue: i18Vo.defaultValue }, i18Vo.params));
    }
    return '国际化格式错误，请检查!';
};
exports.sptI18n = sptI18n;
