// @spotter/i18n-sdk@3.1.21/lib/utils.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.asyncUpdateI18n = exports.i18nInit = exports.setLanguageCache = exports.getLanguageCache = exports.getProjectLanguage = exports.getLanguage = void 0;
const services_1 = require("./services");
const _1 = __importStar(require("."));
const i18next_icu_1 = __importDefault(require("i18next-icu"));
/**
 * @param options
 * @returns 返回指定部件指定语言json数据
 */
function getLanguage(options) {
    return new Promise((resolve, reject) => {
        services_1.i18nSdk
            .getLanguageFile(options)
            .then((res) => {
            // 304命中缓存去拿数据
            if (res.status === 200) {
                // TODO更新缓存
                resolve(res.data);
            }
            else {
                reject('拉取翻译失败');
            }
        })
            .catch((err) => reject(err));
    });
}
exports.getLanguage = getLanguage;
/**
 * @param options
 * @param isMain 默认为false 为true会进行加载common 和后端common数据
 * @returns 返回应用必需翻译数据
 */
const getProjectLanguage = (options) => new Promise((resolve, reject) => {
    var _a;
    // 获取指定语言翻译
    Promise.all((_a = options === null || options === void 0 ? void 0 : options.componentList) === null || _a === void 0 ? void 0 : _a.map((component) => getLanguage(Object.assign(Object.assign({}, options), { component }))))
        .then((resList) => {
        var _a, _b;
        const source = Object.assign({}, ...resList);
        // 拿取common部件的index
        const commonIndex = (_a = options === null || options === void 0 ? void 0 : options.componentList) === null || _a === void 0 ? void 0 : _a.findIndex((component) => component === 'common');
        // 如果找到了位置，设置localStorage缓存
        if (~commonIndex) {
            // 获取现有的缓存key
            // TODO 兼容 dev 后续需要删掉
            const cacheData = (0, exports.getLanguageCache)(Object.assign(Object.assign({}, options), { component: 'common' }));
            (0, exports.setLanguageCache)(Object.assign(Object.assign({}, options), { component: (_b = options === null || options === void 0 ? void 0 : options.componentList) === null || _b === void 0 ? void 0 : _b[commonIndex] }), Object.assign(Object.assign({}, cacheData), resList[commonIndex]));
        }
        resolve(source);
    })
        .catch((error) => {
        console.log(reject(error));
    });
});
exports.getProjectLanguage = getProjectLanguage;
/**
 *
 * @param options 语言配置项
 * 包含dev的项目设置dev缓存
 * @returns
 */
const getLanguageCache = (options) => {
    var _a, _b;
    try {
        // 返回空对象
        return JSON.parse(localStorage.getItem(`i18n_cache_common_${options.language}_${
        // TODO 兼容 dev 后续需要删掉
        ((_a = options === null || options === void 0 ? void 0 : options.project) === null || _a === void 0 ? void 0 : _a.includes('local')) || ((_b = options === null || options === void 0 ? void 0 : options.project) === null || _b === void 0 ? void 0 : _b.includes('dev'))
            ? 'dev'
            : 'prd'}`) || '{}');
    }
    catch (error) {
        console.log(error);
        // 返回空对象source
        return {};
    }
};
exports.getLanguageCache = getLanguageCache;
/**
 *
 * @param options 语言配置项
 * @param value 数据源
 * @returns
 */
const setLanguageCache = (options, value) => {
    var _a, _b;
    try {
        localStorage.setItem(`i18n_cache_common_${options.language}_${
        // TODO 兼容 dev 后续需要删掉
        ((_a = options === null || options === void 0 ? void 0 : options.project) === null || _a === void 0 ? void 0 : _a.includes('local')) || ((_b = options === null || options === void 0 ? void 0 : options.project) === null || _b === void 0 ? void 0 : _b.includes('dev'))
            ? 'dev'
            : 'prd'}`, JSON.stringify(value || ''));
    }
    catch (error) {
        console.log(error);
    }
};
exports.setLanguageCache = setLanguageCache;
/**
 * 调用 i18n init 初始化
 * @param options
 * @param data
 */
const i18nInit = (options, data) => __awaiter(void 0, void 0, void 0, function* () {
    yield _1.default
        .use(i18next_icu_1.default)
        .use(_1.initReactI18next)
        .init({
        lng: options.language,
        resources: {
            [options === null || options === void 0 ? void 0 : options.language]: {
                // 如果存在source文件去灌入数据
                translation: Object.assign({}, data)
            }
        },
        saveMissing: false,
        interpolation: {
            escapeValue: false // not needed for react as it escapes by default,
        }
    });
});
exports.i18nInit = i18nInit;
/** 异步更新翻译资源
 *
 * @param options
 * @param component // 需要异步请求的部件
 * @param allData
 * @param originalSource
 */
const asyncUpdateI18n = (options, component) => __awaiter(void 0, void 0, void 0, function* () {
    try {
        const newOptions = Object.assign({}, options);
        // backend-common 单独访问 spotter 项目
        if (component === 'backend-common') {
            if (options.project.includes('local') || options.project.includes('dev')) {
                newOptions.project = 'spotter-dev';
            }
            else {
                newOptions.project = 'spotter';
            }
        }
        const res = yield getLanguage(Object.assign(Object.assign({}, newOptions), { component }));
        // 添加异步请求的资源
        _1.default.addResourceBundle(newOptions.language, 'translation', Object.assign({}, res), true, true);
    }
    catch (initError) {
        console.error('异步添加失败：', initError);
    }
});
exports.asyncUpdateI18n = asyncUpdateI18n;
