// @spotter/ui/es/components/SptBrand/BrandCopyright.js
// src/components/SptBrand/BrandCopyright.tsx
import React from "react";
import { BrandFull } from "./BrandFull";
import { Divider } from "antd";
import "./style/brand-copyright.less";
var BrandCopyright = () => /* @__PURE__ */ React.createElement("div", { className: "brand-copyright-container" }, /* @__PURE__ */ React.createElement("div", { className: "brand-copyright-container-logo" }, /* @__PURE__ */ React.createElement("span", { className: "brand-copyright-container-logo-pow" }, "Powered By"), /* @__PURE__ */ React.createElement(
  BrandFull,
  {
    logoProps: { className: "brand-copyright-container-logo-icon" },
    nameProps: {
      className: "brand-copyright-container-logo-name"
    }
  }
)), /* @__PURE__ */ React.createElement("span", { className: "brand-copyright-container-copyright" }, "Copyright © Since 2022 Spotterio.com, Inc. and its affiliates. All Rights Reserved.", /* @__PURE__ */ React.createElement(Divider, { type: "vertical" }), /* @__PURE__ */ React.createElement(
  "a",
  {
    className: "brand-copyright-container-copyright-link",
    target: "_blank",
    rel: "noopener noreferrer",
    href: "https://beian.miit.gov.cn"
  },
  "粤ICP备2022005977号"
)));
export {
  BrandCopyright
};
