// @spotter/ui/es/components/SptBrand/BrandLogo.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptBrand/BrandLogo.tsx
import React from "react";
var BrandLogo = (_a) => {
  var _b = _a, { color } = _b, props = __objRest(_b, ["color"]);
  return /* @__PURE__ */ React.createElement("div", __spreadValues({}, props), /* @__PURE__ */ React.createElement(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      xmlnsXlink: "http://www.w3.org/1999/xlink",
      x: "0px",
      y: "0px",
      height: "100%",
      viewBox: "0 0 632 840.8",
      xmlSpace: "preserve",
      version: "1.1"
    },
    /* @__PURE__ */ React.createElement("g", { fill: color != null ? color : "var(--color-primary)" }, /* @__PURE__ */ React.createElement(
      "path",
      {
        d: "M309.3,1.9L98.1,60.7v13.7c0,15-44.7,41.8-84.8,56.4L0,135.8v321.1c0,106.6,53.2,206.3,154,288.2\n					c75.3,61.3,149.9,90.9,153,92.2l9,3.5l9-3.5c3.1-1.2,77.7-30.9,153-92.2c100.7-82,154-181.7,154-288.2V135.8l-13.3-4.9\n					c-40.1-14.7-84.7-41.5-84.7-56.4V60.7L316,0L309.3,1.9z M316,39.7l177.4,49.4c13,33.8,67.2,59.9,95,71.4v296.4\n					c0,96.9-46.8,184.4-139.2,259.9C393.5,762.2,337.3,789.5,316,799c-21.3-9.5-77.5-36.8-133.1-82.2\n					c-92.4-75.5-139.2-163-139.2-259.9V160.5c27.8-11.5,82-37.7,95-71.4L316,39.7z"
      }
    ))
  ));
};
export {
  BrandLogo
};
