// @spotter/ui/lib/components/SptTable/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _proComponents = require("@ant-design/pro-components");
var _react = _interopRequireWildcard(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _Resizable = require("./Resizable");
var _useFit2 = require("./useFit");
var _nanoid = require("nanoid");
var _Provider = require("../Provider");
var _appClientToolkit = require("@spotter/app-client-toolkit");
var _reactRouterDom = require("react-router-dom");
require("./style/index.less");
var _Form = _interopRequireDefault(require("./Form"));
var _Table = _interopRequireDefault(require("./Table"));
var _antd = require("antd");
var _omit = _interopRequireDefault(require("omit.js"));
var _Pagination = _interopRequireDefault(require("./Pagination"));
var _transformKey = _interopRequireDefault(require("./Form/transformKey"));
var _SptPageContainer = require("../SptPageContainer");
var _excluded = ["className", "width", "fit", "sticky", "onRowDrawerChange", "onRowSelection", "drawerTrigger", "columnResizable", "cacheKey", "params", "columns", "components", "onRow", "request", "search", "formRef", "manualRequest", "onReset", "actionRef", "form", "pagination", "dataSource", "loading", "inheritAction", "toolbar"],
  _excluded2 = ["actions", "settings"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure " + obj); }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }
function _regeneratorRuntime() { "use strict"; /*! regenerator-runtime -- Copyright (c) 2014-present, Facebook, Inc. -- license (MIT): https://github.com/facebook/regenerator/blob/main/LICENSE */ _regeneratorRuntime = function _regeneratorRuntime() { return e; }; var t, e = {}, r = Object.prototype, n = r.hasOwnProperty, o = Object.defineProperty || function (t, e, r) { t[e] = r.value; }, i = "function" == typeof Symbol ? Symbol : {}, a = i.iterator || "@@iterator", c = i.asyncIterator || "@@asyncIterator", u = i.toStringTag || "@@toStringTag"; function define(t, e, r) { return Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }), t[e]; } try { define({}, ""); } catch (t) { define = function define(t, e, r) { return t[e] = r; }; } function wrap(t, e, r, n) { var i = e && e.prototype instanceof Generator ? e : Generator, a = Object.create(i.prototype), c = new Context(n || []); return o(a, "_invoke", { value: makeInvokeMethod(t, r, c) }), a; } function tryCatch(t, e, r) { try { return { type: "normal", arg: t.call(e, r) }; } catch (t) { return { type: "throw", arg: t }; } } e.wrap = wrap; var h = "suspendedStart", l = "suspendedYield", f = "executing", s = "completed", y = {}; function Generator() {} function GeneratorFunction() {} function GeneratorFunctionPrototype() {} var p = {}; define(p, a, function () { return this; }); var d = Object.getPrototypeOf, v = d && d(d(values([]))); v && v !== r && n.call(v, a) && (p = v); var g = GeneratorFunctionPrototype.prototype = Generator.prototype = Object.create(p); function defineIteratorMethods(t) { ["next", "throw", "return"].forEach(function (e) { define(t, e, function (t) { return this._invoke(e, t); }); }); } function AsyncIterator(t, e) { function invoke(r, o, i, a) { var c = tryCatch(t[r], t, o); if ("throw" !== c.type) { var u = c.arg, h = u.value; return h && "object" == _typeof(h) && n.call(h, "__await") ? e.resolve(h.__await).then(function (t) { invoke("next", t, i, a); }, function (t) { invoke("throw", t, i, a); }) : e.resolve(h).then(function (t) { u.value = t, i(u); }, function (t) { return invoke("throw", t, i, a); }); } a(c.arg); } var r; o(this, "_invoke", { value: function value(t, n) { function callInvokeWithMethodAndArg() { return new e(function (e, r) { invoke(t, n, e, r); }); } return r = r ? r.then(callInvokeWithMethodAndArg, callInvokeWithMethodAndArg) : callInvokeWithMethodAndArg(); } }); } function makeInvokeMethod(e, r, n) { var o = h; return function (i, a) { if (o === f) throw new Error("Generator is already running"); if (o === s) { if ("throw" === i) throw a; return { value: t, done: !0 }; } for (n.method = i, n.arg = a;;) { var c = n.delegate; if (c) { var u = maybeInvokeDelegate(c, n); if (u) { if (u === y) continue; return u; } } if ("next" === n.method) n.sent = n._sent = n.arg;else if ("throw" === n.method) { if (o === h) throw o = s, n.arg; n.dispatchException(n.arg); } else "return" === n.method && n.abrupt("return", n.arg); o = f; var p = tryCatch(e, r, n); if ("normal" === p.type) { if (o = n.done ? s : l, p.arg === y) continue; return { value: p.arg, done: n.done }; } "throw" === p.type && (o = s, n.method = "throw", n.arg = p.arg); } }; } function maybeInvokeDelegate(e, r) { var n = r.method, o = e.iterator[n]; if (o === t) return r.delegate = null, "throw" === n && e.iterator.return && (r.method = "return", r.arg = t, maybeInvokeDelegate(e, r), "throw" === r.method) || "return" !== n && (r.method = "throw", r.arg = new TypeError("The iterator does not provide a '" + n + "' method")), y; var i = tryCatch(o, e.iterator, r.arg); if ("throw" === i.type) return r.method = "throw", r.arg = i.arg, r.delegate = null, y; var a = i.arg; return a ? a.done ? (r[e.resultName] = a.value, r.next = e.nextLoc, "return" !== r.method && (r.method = "next", r.arg = t), r.delegate = null, y) : a : (r.method = "throw", r.arg = new TypeError("iterator result is not an object"), r.delegate = null, y); } function pushTryEntry(t) { var e = { tryLoc: t[0] }; 1 in t && (e.catchLoc = t[1]), 2 in t && (e.finallyLoc = t[2], e.afterLoc = t[3]), this.tryEntries.push(e); } function resetTryEntry(t) { var e = t.completion || {}; e.type = "normal", delete e.arg, t.completion = e; } function Context(t) { this.tryEntries = [{ tryLoc: "root" }], t.forEach(pushTryEntry, this), this.reset(!0); } function values(e) { if (e || "" === e) { var r = e[a]; if (r) return r.call(e); if ("function" == typeof e.next) return e; if (!isNaN(e.length)) { var o = -1, i = function next() { for (; ++o < e.length;) if (n.call(e, o)) return next.value = e[o], next.done = !1, next; return next.value = t, next.done = !0, next; }; return i.next = i; } } throw new TypeError(_typeof(e) + " is not iterable"); } return GeneratorFunction.prototype = GeneratorFunctionPrototype, o(g, "constructor", { value: GeneratorFunctionPrototype, configurable: !0 }), o(GeneratorFunctionPrototype, "constructor", { value: GeneratorFunction, configurable: !0 }), GeneratorFunction.displayName = define(GeneratorFunctionPrototype, u, "GeneratorFunction"), e.isGeneratorFunction = function (t) { var e = "function" == typeof t && t.constructor; return !!e && (e === GeneratorFunction || "GeneratorFunction" === (e.displayName || e.name)); }, e.mark = function (t) { return Object.setPrototypeOf ? Object.setPrototypeOf(t, GeneratorFunctionPrototype) : (t.__proto__ = GeneratorFunctionPrototype, define(t, u, "GeneratorFunction")), t.prototype = Object.create(g), t; }, e.awrap = function (t) { return { __await: t }; }, defineIteratorMethods(AsyncIterator.prototype), define(AsyncIterator.prototype, c, function () { return this; }), e.AsyncIterator = AsyncIterator, e.async = function (t, r, n, o, i) { void 0 === i && (i = Promise); var a = new AsyncIterator(wrap(t, r, n, o), i); return e.isGeneratorFunction(r) ? a : a.next().then(function (t) { return t.done ? t.value : a.next(); }); }, defineIteratorMethods(g), define(g, u, "Generator"), define(g, a, function () { return this; }), define(g, "toString", function () { return "[object Generator]"; }), e.keys = function (t) { var e = Object(t), r = []; for (var n in e) r.push(n); return r.reverse(), function next() { for (; r.length;) { var t = r.pop(); if (t in e) return next.value = t, next.done = !1, next; } return next.done = !0, next; }; }, e.values = values, Context.prototype = { constructor: Context, reset: function reset(e) { if (this.prev = 0, this.next = 0, this.sent = this._sent = t, this.done = !1, this.delegate = null, this.method = "next", this.arg = t, this.tryEntries.forEach(resetTryEntry), !e) for (var r in this) "t" === r.charAt(0) && n.call(this, r) && !isNaN(+r.slice(1)) && (this[r] = t); }, stop: function stop() { this.done = !0; var t = this.tryEntries[0].completion; if ("throw" === t.type) throw t.arg; return this.rval; }, dispatchException: function dispatchException(e) { if (this.done) throw e; var r = this; function handle(n, o) { return a.type = "throw", a.arg = e, r.next = n, o && (r.method = "next", r.arg = t), !!o; } for (var o = this.tryEntries.length - 1; o >= 0; --o) { var i = this.tryEntries[o], a = i.completion; if ("root" === i.tryLoc) return handle("end"); if (i.tryLoc <= this.prev) { var c = n.call(i, "catchLoc"), u = n.call(i, "finallyLoc"); if (c && u) { if (this.prev < i.catchLoc) return handle(i.catchLoc, !0); if (this.prev < i.finallyLoc) return handle(i.finallyLoc); } else if (c) { if (this.prev < i.catchLoc) return handle(i.catchLoc, !0); } else { if (!u) throw new Error("try statement without catch or finally"); if (this.prev < i.finallyLoc) return handle(i.finallyLoc); } } } }, abrupt: function abrupt(t, e) { for (var r = this.tryEntries.length - 1; r >= 0; --r) { var o = this.tryEntries[r]; if (o.tryLoc <= this.prev && n.call(o, "finallyLoc") && this.prev < o.finallyLoc) { var i = o; break; } } i && ("break" === t || "continue" === t) && i.tryLoc <= e && e <= i.finallyLoc && (i = null); var a = i ? i.completion : {}; return a.type = t, a.arg = e, i ? (this.method = "next", this.next = i.finallyLoc, y) : this.complete(a); }, complete: function complete(t, e) { if ("throw" === t.type) throw t.arg; return "break" === t.type || "continue" === t.type ? this.next = t.arg : "return" === t.type ? (this.rval = this.arg = t.arg, this.method = "return", this.next = "end") : "normal" === t.type && e && (this.next = e), y; }, finish: function finish(t) { for (var e = this.tryEntries.length - 1; e >= 0; --e) { var r = this.tryEntries[e]; if (r.finallyLoc === t) return this.complete(r.completion, r.afterLoc), resetTryEntry(r), y; } }, catch: function _catch(t) { for (var e = this.tryEntries.length - 1; e >= 0; --e) { var r = this.tryEntries[e]; if (r.tryLoc === t) { var n = r.completion; if ("throw" === n.type) { var o = n.arg; resetTryEntry(r); } return o; } } throw new Error("illegal catch attempt"); }, delegateYield: function delegateYield(e, r, n) { return this.delegate = { iterator: values(e), resultName: r, nextLoc: n }, "next" === this.method && (this.arg = t), y; } }, e; }
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }
function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
// TODO: 兼容searchConfig，移除不需要的search配置

var getValidColumnWidth = function getValidColumnWidth(column, cacheColumn, globalColumnResizableMinWidth) {
  var _ref, _column$width, _column$resizable;
  return column.dataIndex && cacheColumn && cacheColumn.originWidth === column.width ? cacheColumn.width : // 当没有设定列宽时，默认使用全局的最小列宽
  (_ref = (_column$width = column.width) !== null && _column$width !== void 0 ? _column$width : (_column$resizable = column.resizable) === null || _column$resizable === void 0 ? void 0 : _column$resizable.minWidth) !== null && _ref !== void 0 ? _ref : globalColumnResizableMinWidth;
};
var transformOriginColumns = function transformOriginColumns(_ref2) {
  var originColumns = _ref2.originColumns,
    getCache = _ref2.getCache,
    columnResizable = _ref2.columnResizable,
    drawerTrigger = _ref2.drawerTrigger,
    resizedColumns = _ref2.resizedColumns;
  return originColumns.map(function (column, index) {
    var _transformdColumn$dra;
    var transformdColumn = _objectSpread({}, column);
    var resizeData = column.dataIndex && resizedColumns[column.dataIndex.toString()] || resizedColumns[index.toString()];
    if (resizeData && !column.hideInTable) {
      var _resizeData$width;
      transformdColumn = _objectSpread(_objectSpread({}, column), {}, {
        width: column.width + ((_resizeData$width = resizeData === null || resizeData === void 0 ? void 0 : resizeData.width) !== null && _resizeData$width !== void 0 ? _resizeData$width : 0)
      });
    }
    var c = !transformdColumn.hideInTable ? _objectSpread(_objectSpread({}, transformdColumn), {}, {
      width: getValidColumnWidth(transformdColumn, getCache()[transformdColumn.dataIndex], columnResizable === null || columnResizable === void 0 ? void 0 : columnResizable.minWidth)
    }) : transformdColumn;
    if (!((_transformdColumn$dra = transformdColumn.drawerTrigger) !== null && _transformdColumn$dra !== void 0 ? _transformdColumn$dra : drawerTrigger)) {
      return _objectSpread(_objectSpread({}, c), {}, {
        className: (0, _classnames.default)(transformdColumn.className, 'disable-drawer-trigger'),
        render: function render(node) {
          var _transformdColumn$ren, _transformdColumn;
          for (var _len = arguments.length, rp = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
            rp[_key - 1] = arguments[_key];
          }
          return /*#__PURE__*/_react.default.createElement("div", {
            className: (0, _classnames.default)('disabled-drawer-trigger-table-cell-wrapper', {
              'items-start': !transformdColumn.align || transformdColumn.align === 'left',
              'items-center': transformdColumn.align === 'center',
              'items-end': transformdColumn.align === 'right'
            }),
            onClick: function onClick(e) {
              e.stopPropagation();
            }
          }, ((_transformdColumn$ren = (_transformdColumn = transformdColumn).render) === null || _transformdColumn$ren === void 0 ? void 0 : _transformdColumn$ren.call.apply(_transformdColumn$ren, [_transformdColumn, node].concat(rp))) || node);
        }
      });
    }
    return c;
  });
};
var SpotterTableBase = function SpotterTableBase(_ref3) {
  var _originComponents$hea;
  var className = _ref3.className,
    _ref3$width = _ref3.width,
    width = _ref3$width === void 0 ? 0 : _ref3$width,
    _ref3$fit = _ref3.fit,
    fit = _ref3$fit === void 0 ? 'content' : _ref3$fit,
    _ref3$sticky = _ref3.sticky,
    sticky = _ref3$sticky === void 0 ? true : _ref3$sticky,
    _onRowDrawerChange = _ref3.onRowDrawerChange,
    onRowSelection = _ref3.onRowSelection,
    _ref3$drawerTrigger = _ref3.drawerTrigger,
    drawerTrigger = _ref3$drawerTrigger === void 0 ? true : _ref3$drawerTrigger,
    _columnResizable = _ref3.columnResizable,
    cacheKey = _ref3.cacheKey,
    propsParams = _ref3.params,
    _columns = _ref3.columns,
    _components = _ref3.components,
    onRow = _ref3.onRow,
    request = _ref3.request,
    search = _ref3.search,
    formRef = _ref3.formRef,
    manualRequest = _ref3.manualRequest,
    onPropsReset = _ref3.onReset,
    propsActionRef = _ref3.actionRef,
    searchForm = _ref3.form,
    propsPagination = _ref3.pagination,
    propsDataSource = _ref3.dataSource,
    propsLoading = _ref3.loading,
    inheritAction = _ref3.inheritAction,
    propsToolbar = _ref3.toolbar,
    restProps = _objectWithoutProperties(_ref3, _excluded);
  var onRowDrawerChange = (0, _react.useMemo)(function () {
    return _onRowDrawerChange;
  }, []);
  var columnResizable = (0, _react.useMemo)(function () {
    return _columnResizable !== null && _columnResizable !== void 0 ? _columnResizable : {
      minWidth: 88,
      preferResizeWidth: false
    };
  }, [_columnResizable]);
  var _useMountMergeState = (0, _proComponents.useMountMergeState)(function () {
      if (manualRequest || search !== false) {
        return undefined;
      }
      return {};
    }),
    _useMountMergeState2 = _slicedToArray(_useMountMergeState, 2),
    formSearch = _useMountMergeState2[0],
    setFormSearch = _useMountMergeState2[1];
  var manualRequestRef = (0, _react.useRef)(formSearch === undefined);
  var initFormRef = (0, _react.useRef)();
  var baseFormRef = formRef !== null && formRef !== void 0 ? formRef : initFormRef;
  var params = (0, _react.useMemo)(function () {
    return _objectSpread(_objectSpread({}, propsParams || {}), formSearch || {});
  }, [propsParams, formSearch]);

  // 获取loading
  var _useMountMergeState3 = (0, _proComponents.useMountMergeState)(false, {
      value: _typeof(propsLoading) === 'object' ? propsLoading === null || propsLoading === void 0 ? void 0 : propsLoading.spinning : propsLoading,
      onChange: restProps.onLoadingChange
    }),
    _useMountMergeState4 = _slicedToArray(_useMountMergeState3, 2),
    loading = _useMountMergeState4[0],
    setLoading = _useMountMergeState4[1];
  var originComponents = (0, _react.useMemo)(function () {
    return _components;
  }, []);
  var _cacheKey = (0, _react.useMemo)(function () {
    return "".concat(cacheKey ? "".concat(cacheKey, "_") : '', "spotter_table");
  }, []);
  // spotter table 组件在一次完整的生命周期中的唯一 id
  var _componentId = (0, _react.useMemo)(function () {
    return (0, _nanoid.nanoid)();
  }, []);
  var _useContext = (0, _react.useContext)(_Provider.SptComponentContext),
    routes = _useContext.routes;
  var location = (0, _reactRouterDom.useLocation)();
  var _useLocalCache = (0, _appClientToolkit.useLocalCache)(_cacheKey, {}, routes !== null && routes !== void 0 ? routes : [], location),
    getCache = _useLocalCache.getCache,
    setCache = _useLocalCache.setCache,
    cacheKeyWithRoute = _useLocalCache.cacheKey;
  // y 轴滚动设置
  var _useFit = (0, _useFit2.useFit)(fit),
    scrollY = _useFit.scrollY,
    fitClass = _useFit.fitClass;
  var _useResizable = (0, _Resizable.useResizable)({
      key: _componentId,
      headerCell: originComponents === null || originComponents === void 0 || (_originComponents$hea = originComponents.header) === null || _originComponents$hea === void 0 ? void 0 : _originComponents$hea.cell
    }),
    ResizingHeaderCell = _useResizable.ResizingHeaderCell,
    ResizingIndicator = _useResizable.ResizingIndicator,
    updateResizableColumns = _useResizable.updateResizableColumns;
  var _useState = (0, _react.useState)({}),
    _useState2 = _slicedToArray(_useState, 2),
    resizedColumns = _useState2[0],
    setResizedColumns = _useState2[1];
  (0, _react.useEffect)(function () {
    _appClientToolkit.eventBus.on('resizeColumn', function (_ref4) {
      var columnKey = _ref4.columnKey,
        width = _ref4.width;
      setResizedColumns(function (prev) {
        var _item$width;
        var item = prev[columnKey];
        return _objectSpread(_objectSpread({}, prev), {}, _defineProperty({}, columnKey, {
          // 保存旧的宽度，用于计算增量
          width: ((_item$width = item === null || item === void 0 ? void 0 : item.width) !== null && _item$width !== void 0 ? _item$width : 0) + width
        }));
      });
    });
    return function () {
      _appClientToolkit.eventBus.off('resizeColumn');
    };
  }, []);
  var columns = (0, _react.useMemo)(function () {
    return updateResizableColumns(transformOriginColumns({
      originColumns: _columns,
      getCache: getCache,
      columnResizable: columnResizable,
      drawerTrigger: drawerTrigger,
      resizedColumns: resizedColumns
    }));
  }, [_columns, resizedColumns]);

  // 识别searchParams并插入
  var _useSearchParams = (0, _reactRouterDom.useSearchParams)(),
    _useSearchParams2 = _slicedToArray(_useSearchParams, 1),
    searchParams = _useSearchParams2[0];
  (0, _react.useEffect)(function () {
    try {
      var _params = searchParams.get('table_search_params');
      if (_params) {
        var _baseFormRef$current2, _baseFormRef$current3;
        var table_search_params = JSON.parse(decodeURIComponent(_params));
        if (table_search_params.resetFields) {
          var _baseFormRef$current;
          (_baseFormRef$current = baseFormRef.current) === null || _baseFormRef$current === void 0 || _baseFormRef$current.resetFields();
        }
        (_baseFormRef$current2 = baseFormRef.current) === null || _baseFormRef$current2 === void 0 || _baseFormRef$current2.setFieldsValue(table_search_params);
        (_baseFormRef$current3 = baseFormRef.current) === null || _baseFormRef$current3 === void 0 || _baseFormRef$current3.submit();
      }
    } catch (error) {
      console.log(error);
    }
  }, [searchParams]);
  (0, _react.useEffect)(function () {
    if (columns.length) {
      setCache(columns.reduce(function (p, c, i) {
        // 只缓存有 dataIndex 的列宽度，因为在没有 dataIndex 会使用 index 作为 key，而我们后面修改顺序或者 column 数量发生变更时会存在问题
        if (c.dataIndex && c.width) {
          var _columns$i, _columns$find;
          p[c.dataIndex] = {
            /**
             * 现在存在一个问题，改了代码里的 table 列宽以后，如果客户端有缓存，那么改动后的列宽并不会优先生效.
             * 这在一些情况下会导致显示问题，所以补充了一个自动根据原始列宽和当前的代码列宽做判断后再增量覆盖缓存的方案，来让改动后的列宽优先生效，同时不会影响其他未改动的列宽的缓存
             * */
            originWidth:
            /**
             * 这里优先使用数组索引去查找，提高效率。但是由于可能动态的变更 columns 的顺序和数量，所以使用数组索引进行查找并不严谨.
             * 故我们再增加一步对比 dataIndex，如果 dataIndex 不相同，那么说明错位了，则使用 find 来查找 dataIndex 相同的原始列
             */
            ((_columns$i = _columns[i]) === null || _columns$i === void 0 ? void 0 : _columns$i.dataIndex) === c.dataIndex ? _columns[i].width : (_columns$find = _columns.find(function (_ref5) {
              var dataIndex = _ref5.dataIndex;
              return dataIndex === c.dataIndex;
            })) === null || _columns$find === void 0 ? void 0 : _columns$find.width,
            width: c.width
          };
        }
        return p;
      }, {}));
    }
  }, [columns]);
  var _useState3 = (0, _react.useState)({
      current: propsPagination && (propsPagination === null || propsPagination === void 0 ? void 0 : propsPagination.current) || 1,
      pageSize: propsPagination && (propsPagination === null || propsPagination === void 0 ? void 0 : propsPagination.pageSize) || 20
    }),
    _useState4 = _slicedToArray(_useState3, 2),
    pagination = _useState4[0],
    setPagination = _useState4[1];
  var _useState5 = (0, _react.useState)({
      total: propsPagination && propsPagination.total || 0,
      // 是否已经加载过数据
      loaded: false
    }),
    _useState6 = _slicedToArray(_useState5, 2),
    paginationInfo = _useState6[0],
    setPaginationInfo = _useState6[1];
  var sorter = (0, _react.useRef)();
  var _useState7 = (0, _react.useState)(propsDataSource ? propsDataSource === null || propsDataSource === void 0 ? void 0 : propsDataSource.slice((pagination.current - 1) * pagination.pageSize, pagination.current * pagination.pageSize) : []),
    _useState8 = _slicedToArray(_useState7, 2),
    dataSource = _useState8[0],
    setDataSource = _useState8[1];
  (0, _react.useEffect)(function () {
    if (propsDataSource) {
      setDataSource(propsDataSource === null || propsDataSource === void 0 ? void 0 : propsDataSource.slice((pagination.current - 1) * pagination.pageSize, pagination.current * pagination.pageSize));
      setPaginationInfo(_objectSpread(_objectSpread({}, paginationInfo), {}, {
        loaded: true,
        total: propsDataSource.length
      }));
    }
  }, [propsDataSource]);
  var queryData = /*#__PURE__*/function () {
    var _ref6 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee(params) {
      var _yield$request, _ref7, data, total;
      return _regeneratorRuntime().wrap(function _callee$(_context) {
        while (1) switch (_context.prev = _context.next) {
          case 0:
            if (request) {
              _context.next = 2;
              break;
            }
            return _context.abrupt("return");
          case 2:
            if (!manualRequestRef.current) {
              _context.next = 5;
              break;
            }
            manualRequestRef.current = false;
            return _context.abrupt("return");
          case 5:
            setLoading(true);
            _context.prev = 6;
            _context.next = 9;
            return request === null || request === void 0 ? void 0 : request(params, sorter.current,
            // sort
            {} // filter
            );
          case 9:
            _context.t1 = _yield$request = _context.sent;
            _context.t0 = _context.t1 !== null;
            if (!_context.t0) {
              _context.next = 13;
              break;
            }
            _context.t0 = _yield$request !== void 0;
          case 13:
            if (!_context.t0) {
              _context.next = 17;
              break;
            }
            _context.t2 = _yield$request;
            _context.next = 18;
            break;
          case 17:
            _context.t2 = {};
          case 18:
            _ref7 = _context.t2;
            data = _ref7.data;
            total = _ref7.total;
            setDataSource(data || []);
            setPaginationInfo(_objectSpread(_objectSpread({}, paginationInfo), {}, {
              loaded: true,
              total: total || 0
            }));
            _context.next = 28;
            break;
          case 25:
            _context.prev = 25;
            _context.t3 = _context["catch"](6);
            console.error(_context.t3);
          case 28:
            setLoading(false);
          case 29:
          case "end":
            return _context.stop();
        }
      }, _callee, null, [[6, 25]]);
    }));
    return function queryData(_x) {
      return _ref6.apply(this, arguments);
    };
  }();
  var _reload = /*#__PURE__*/function () {
    var _ref8 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee2() {
      return _regeneratorRuntime().wrap(function _callee2$(_context2) {
        while (1) switch (_context2.prev = _context2.next) {
          case 0:
            _context2.next = 2;
            return queryData(_objectSpread({
              current: pagination.current,
              pageSize: pagination.pageSize
            }, (0, _omit.default)(params, ['_timestamp'])));
          case 2:
          case "end":
            return _context2.stop();
        }
      }, _callee2);
    }));
    return function reload() {
      return _ref8.apply(this, arguments);
    };
  }();
  var reloadWithPagination = /*#__PURE__*/function () {
    var _ref9 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee3(nextPagination) {
      return _regeneratorRuntime().wrap(function _callee3$(_context3) {
        while (1) switch (_context3.prev = _context3.next) {
          case 0:
            _context3.next = 2;
            return queryData(_objectSpread({
              current: nextPagination.current,
              pageSize: nextPagination.pageSize
            }, (0, _omit.default)(params, ['_timestamp'])));
          case 2:
          case "end":
            return _context3.stop();
        }
      }, _callee3);
    }));
    return function reloadWithPagination(_x2) {
      return _ref9.apply(this, arguments);
    };
  }();
  (0, _react.useEffect)(function () {
    // TODO: 清理request副作用
    request && _reload();
  }, [params]);
  var onFormSearchSubmit = (0, _react.useCallback)(function (values) {
    setPagination(_objectSpread(_objectSpread({}, pagination), {}, {
      current: 1
    }));
    setFormSearch(values);
  }, []);
  var searchKeyMap = (0, _react.useMemo)(function () {
    if (!columns) {
      return {};
    }
    var searchKeyMap = {};
    var _iterator = _createForOfIteratorHelper(columns),
      _step;
    try {
      for (_iterator.s(); !(_step = _iterator.n()).done;) {
        var _item$search;
        var item = _step.value;
        searchKeyMap["".concat(item.key || item.dataIndex)] = item === null || item === void 0 || (_item$search = item.search) === null || _item$search === void 0 ? void 0 : _item$search.transform;
      }
    } catch (err) {
      _iterator.e(err);
    } finally {
      _iterator.f();
    }
    return searchKeyMap;
  }, [columns]);

  // 拦截ActionRef
  var actionRef = (0, _react.useRef)();
  (0, _react.useImperativeHandle)(propsActionRef, function () {
    return _objectSpread(_objectSpread({}, actionRef.current || {}), {}, {
      reset: function reset() {
        var _baseFormRef$current4, _baseFormRef$current5, _baseFormRef$current6;
        baseFormRef === null || baseFormRef === void 0 || (_baseFormRef$current4 = baseFormRef.current) === null || _baseFormRef$current4 === void 0 || _baseFormRef$current4.resetFields();
        var values = (0, _proComponents.omitUndefined)((baseFormRef === null || baseFormRef === void 0 || (_baseFormRef$current5 = baseFormRef.current) === null || _baseFormRef$current5 === void 0 || (_baseFormRef$current6 = _baseFormRef$current5.getFieldsValue) === null || _baseFormRef$current6 === void 0 ? void 0 : _baseFormRef$current6.call(_baseFormRef$current5)) || {});
        setFormSearch((0, _transformKey.default)(values, searchKeyMap));
      },
      reload: function () {
        var _reload2 = _asyncToGenerator( /*#__PURE__*/_regeneratorRuntime().mark(function _callee4(resetPageIndex) {
          var nextPagination;
          return _regeneratorRuntime().wrap(function _callee4$(_context4) {
            while (1) switch (_context4.prev = _context4.next) {
              case 0:
                if (resetPageIndex) {
                  nextPagination = _objectSpread(_objectSpread({}, pagination), {}, {
                    current: 1
                  });
                  setPagination(nextPagination);
                  reloadWithPagination(nextPagination);
                } else {
                  _reload();
                }
              case 1:
              case "end":
                return _context4.stop();
            }
          }, _callee4);
        }));
        function reload(_x3) {
          return _reload2.apply(this, arguments);
        }
        return reload;
      }(),
      reloadAndRest: function reloadAndRest() {
        var nextPagination = _objectSpread(_objectSpread({}, pagination), {}, {
          current: 1
        });
        setPagination(nextPagination);
        request && reloadWithPagination(nextPagination);
      }
    });
  });
  var onPaginationChange = (0, _proComponents.useRefFunction)(function (page, pageSize) {
    var nextPagination = _objectSpread(_objectSpread({}, pagination), {}, {
      current: page,
      pageSize: pageSize
    });
    setPagination(nextPagination);
    if (propsDataSource) {
      setDataSource(propsDataSource === null || propsDataSource === void 0 ? void 0 : propsDataSource.slice((nextPagination.current - 1) * nextPagination.pageSize, nextPagination.current * nextPagination.pageSize));
    }
    request && reloadWithPagination(nextPagination);
    if (propsPagination !== false && propsPagination !== null && propsPagination !== void 0 && propsPagination.onChange) {
      propsPagination.onChange(page, pageSize);
    }
  });
  var onTableChange = (0, _proComponents.useRefFunction)(function (_pagination, _filter, _sorter) {
    var _sorter$column;
    var nextSorter = _sorter && Object.keys(_sorter).length > 0 && _sorter.order ? _defineProperty({}, (_sorter === null || _sorter === void 0 ? void 0 : _sorter.columnKey) || (_sorter === null || _sorter === void 0 ? void 0 : _sorter.field) || (_sorter === null || _sorter === void 0 || (_sorter$column = _sorter.column) === null || _sorter$column === void 0 ? void 0 : _sorter$column.dataIndex), _sorter.order) : undefined;
    sorter.current = nextSorter;
    request && _reload();
  });
  var tableAction = (0, _react.useContext)(_SptPageContainer.SptPageContainerContext);
  var toolbar = (0, _react.useMemo)(function () {
    var _tableAction$tableAct, _ref12, _tableAction$tableAct2;
    if (!inheritAction) return propsToolbar;
    var _ref11 = propsToolbar !== null && propsToolbar !== void 0 ? propsToolbar : {},
      _ref11$actions = _ref11.actions,
      actions = _ref11$actions === void 0 ? [] : _ref11$actions,
      _ref11$settings = _ref11.settings,
      settings = _ref11$settings === void 0 ? [] : _ref11$settings,
      restToolbarProps = _objectWithoutProperties(_ref11, _excluded2);
    var tableActions = [(actions === null || actions === void 0 ? void 0 : actions.length) > 0 || (settings === null || settings === void 0 ? void 0 : settings.length) > 0 ? /*#__PURE__*/_react.default.createElement("div", {
      key: "spt-table-global-divider",
      className: "spt-table-global-divider"
    }) : undefined].concat(_toConsumableArray((_tableAction$tableAct = tableAction === null || tableAction === void 0 ? void 0 : tableAction.tableAction) !== null && _tableAction$tableAct !== void 0 ? _tableAction$tableAct : []));
    return _objectSpread(_objectSpread({}, restToolbarProps), {}, {
      actions: (_ref12 = [].concat(_toConsumableArray(actions), _toConsumableArray(tableAction !== null && tableAction !== void 0 && tableAction.tableAction && (tableAction === null || tableAction === void 0 || (_tableAction$tableAct2 = tableAction.tableAction) === null || _tableAction$tableAct2 === void 0 ? void 0 : _tableAction$tableAct2.length) > 0 ? tableActions : []))) === null || _ref12 === void 0 ? void 0 : _ref12.filter(function (item) {
        return item;
      }),
      settings: settings
    });
  }, [propsToolbar]);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)('spt-table-wrap', fitClass)
  }, search === false ? null : /*#__PURE__*/_react.default.createElement(_Form.default, {
    action: actionRef,
    loading: !!loading,
    columns: columns,
    formRef: baseFormRef,
    form: searchForm,
    search: search,
    manualRequest: manualRequest,
    type: restProps.type || 'table',
    onSubmit: restProps.onSubmit,
    onReset: onPropsReset,
    onFormSearchSubmit: onFormSearchSubmit,
    searchKeyMap: searchKeyMap
  }), /*#__PURE__*/_react.default.createElement(_antd.Spin, {
    spinning: loading
  }, /*#__PURE__*/_react.default.createElement(_Table.default, _extends({
    columns: columns,
    width: width,
    sticky: sticky,
    fit: fit,
    fitClass: fitClass,
    scrollY: scrollY,
    onRow: onRow,
    className: className,
    componentId: _componentId,
    cacheKeyWithRoute: cacheKeyWithRoute,
    ResizingIndicator: ResizingIndicator,
    onRowSelection: onRowSelection,
    onRowDrawerChange: onRowDrawerChange,
    ResizingHeaderCell: ResizingHeaderCell,
    dataSource: dataSource,
    onChange: onTableChange,
    pagination: propsPagination,
    toolbar: toolbar
    // TODO: DON'T WRITE LIKE THIS. IT'S NOT A GOOD PRACTICE
  }, restProps))), paginationInfo.loaded && propsPagination !== false ? /*#__PURE__*/_react.default.createElement(_Pagination.default, _extends({}, pagination, {
    total: paginationInfo.total,
    pageSize: pagination.pageSize
  }, propsPagination !== null && propsPagination !== void 0 ? propsPagination : {}, {
    onChange: onPaginationChange
  })) : null);
};
var SptTable = function SptTable(_ref13) {
  var props = Object.assign({}, (_objectDestructuringEmpty(_ref13), _ref13));
  return /*#__PURE__*/_react.default.createElement(_Provider.SptComponentProvider, null, /*#__PURE__*/_react.default.createElement(SpotterTableBase, props));
};
var _default = exports.default = SptTable;
