// @spotter/ui/lib/components/SptTable/Resizable.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useResizable = void 0;
var _react = _interopRequireWildcard(require("react"));
var _appClientToolkit = require("@spotter/app-client-toolkit");
var _utils = require("./utils");
var _excluded = ["onResizeStart", "width", "onResizeStop", "OriginHeaderCell", "children"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var getResizingEvent = function getResizingEvent(id) {
  return "".concat(id, "_resizing");
};
var ResizingHeaderCell = function ResizingHeaderCell(props) {
  var onResizeStart = props.onResizeStart,
    width = props.width,
    onResizeStop = props.onResizeStop,
    OriginHeaderCell = props.OriginHeaderCell,
    overrideChildren = props.children,
    restProps = _objectWithoutProperties(props, _excluded);
  var nodeRef = (0, _react.useRef)();
  var thNodeRef = (0, _react.useRef)();
  (0, _react.useEffect)(function () {
    // 使用选择器，获取所有拖动按钮的 Dom 对象
    var _iterator = _createForOfIteratorHelper(document.querySelectorAll('.spotter-table-header-column-resize-trigger')),
      _step;
    try {
      for (_iterator.s(); !(_step = _iterator.n()).done;) {
        var elem = _step.value;
        if (elem.onclick) {
          return;
        }
        elem.addEventListener('click', function (event) {
          // 阻止冒泡
          event.stopPropagation();
          return false;
        });
      }
    } catch (err) {
      _iterator.e(err);
    } finally {
      _iterator.f();
    }
  }, []);
  // 没有原始宽度的列，不支持伸缩；会出现从自适应宽度一下子跳到拖动位置；也可以自行增加参数，如 disableResize
  if (!width) {
    return /*#__PURE__*/_react.default.createElement("th", props, OriginHeaderCell ? /*#__PURE__*/_react.default.createElement(OriginHeaderCell, props, overrideChildren) : overrideChildren);
  }
  return /*#__PURE__*/_react.default.createElement("th", _extends({}, restProps, {
    ref: function ref(dom) {
      dom && (thNodeRef.current = dom);
    }
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: "inline-block whitespace-pre-line",
    style: {
      wordBreak: 'break-word'
    },
    ref: function ref(dom) {
      dom && (nodeRef.current = dom);
    }
  }, overrideChildren), /*#__PURE__*/_react.default.createElement("div", {
    onMouseDown: function onMouseDown(triggerMouseDownEvent) {
      (0, _utils.clearSelection)();
      var triggerRect = triggerMouseDownEvent.target.getBoundingClientRect();
      // 鼠标相对指示器的偏移量
      var mouseOffsetX = (0, _utils.toFloat)(triggerMouseDownEvent.clientX - (triggerRect.x + triggerRect.width * 0.5));
      window.addEventListener('mouseup', function (e) {
        e.stopPropagation();
        // +1 for ::before 1px width
        var _triggerRect = triggerMouseDownEvent.target.getBoundingClientRect();
        onResizeStop(
        // 鼠标的 x 轴坐标 - 鼠标相对指示器的偏移量 = 指示器的 x 轴坐标
        (0, _utils.toFloat)(e.clientX - mouseOffsetX) - (0, _utils.toFloat)(_triggerRect.x + 0.5 * _triggerRect.width), (0, _utils.toFloat)(_triggerRect.x + 0.5 * _triggerRect.width));
      }, {
        once: true
      });
      onResizeStart(thNodeRef.current.getBoundingClientRect(), nodeRef.current.getBoundingClientRect(), triggerMouseDownEvent.target.getBoundingClientRect());
    },
    className: "spotter-table-header-column-resize-trigger"
  }));
};
var ResizingIndicator = function ResizingIndicator(_ref) {
  var id = _ref.id;
  // 指示器 dom 节点
  var indicatorRef = (0, _react.useRef)();
  var resizingRef = (0, _react.useRef)({
    inResizing: false,
    boundary: 0,
    startX: 0,
    startMouseOffsetX: 0,
    offsetX: 0
  });
  var onDrag = (0, _react.useCallback)(function (e) {
    if (resizingRef.current.inResizing) {
      var _indicatorRef$current;
      !resizingRef.current.startMouseOffsetX && (resizingRef.current.startMouseOffsetX = e.clientX - resizingRef.current.startX);
      (_indicatorRef$current = indicatorRef.current) === null || _indicatorRef$current === void 0 || _indicatorRef$current.setAttribute('style', "transform:translateX(".concat(
      // 鼠标 x 轴坐标 - 鼠标相对指示器的偏移量 = 指示器的坐标
      Math.max(e.clientX - resizingRef.current.startMouseOffsetX, resizingRef.current.boundary) - resizingRef.current.offsetX, "px);display:block;"));
    }
  }, [id]);
  var resizingHandler = (0, _react.useCallback)(function () {
    if (indicatorRef.current) {
      if (resizingRef.current.inResizing) {
        // 显示指示器并将指示器定位到 resizing dragger 的中线
        indicatorRef.current.setAttribute('style', "transform:translateX(".concat(resizingRef.current.startX - resizingRef.current.offsetX, "px);display:block;"));
      } else {
        // 隐藏指示器
        indicatorRef.current.setAttribute('style', 'display:none;');
        // 重置初始指示器和鼠标点击的偏移量
        resizingRef.current.startMouseOffsetX = 0;
      }
    }
  }, [id]);
  (0, _react.useEffect)(function () {
    if (indicatorRef.current) {
      resizingRef.current.offsetX = indicatorRef.current.parentNode.getBoundingClientRect().x;
      var _ref2 = indicatorRef.current,
        ownerDocument = _ref2.ownerDocument;
      resizingHandler();
      // 原生的方式替代 react state 驱动，用来解决性能问题
      _appClientToolkit.eventBus.on(getResizingEvent(id), function (resizingPayload) {
        resizingRef.current = _objectSpread(_objectSpread(_objectSpread({}, resizingRef.current), resizingPayload), {}, {
          // 在每次开始拖动前重新校正一遍 offsetX，因为可能随着视窗 resizing 后导致 offsetX 已经变化
          offsetX: indicatorRef.current.parentNode.getBoundingClientRect().x
        });
        resizingHandler();
      });
      ownerDocument.addEventListener('mousemove', onDrag);
    }
    return function () {
      _appClientToolkit.eventBus.off(getResizingEvent(id));
      if (indicatorRef.current) {
        var _ref3 = indicatorRef.current,
          _ownerDocument = _ref3.ownerDocument;
        _ownerDocument.removeEventListener('mousemove', onDrag);
      }
    };
  }, [id]);
  return /*#__PURE__*/_react.default.createElement("div", {
    ref: function ref(dom) {
      dom && (indicatorRef.current = dom);
    },
    className: "spotter-table-column-resizable-marker"
  });
};
var useResizable = exports.useResizable = function useResizable(_ref4) {
  var OriginHeaderCell = _ref4.headerCell,
    key = _ref4.key;
  var resizingSettingRef = (0, _react.useRef)({
    inResizing: false,
    boundary: 0,
    startX: 0
  });
  var updateResizableColumns = (0, _react.useCallback)(function (originColumns) {
    var cols = originColumns.map(function (originCol) {
      return _objectSpread(_objectSpread({}, originCol), {}, {
        onHeaderCell: function onHeaderCell(col) {
          var _originCol$onHeaderCe;
          return _objectSpread(_objectSpread({}, (_originCol$onHeaderCe = originCol.onHeaderCell) === null || _originCol$onHeaderCe === void 0 ? void 0 : _originCol$onHeaderCe.call(originCol, col)), {}, {
            width: col.width,
            OriginHeaderCell: OriginHeaderCell,
            onResizeStart: function onResizeStart(thRect, thChildRect, triggerRect) {
              var _col$resizable$bounda, _col$resizable;
              resizingSettingRef.current = {
                inResizing: true,
                // 必须用 thRect 的 x 坐标因为 thChild 可能是右对齐.
                boundary: (0, _utils.toFloat)(thRect.x + Math.min(thChildRect.width, (_col$resizable$bounda = (_col$resizable = col.resizable) === null || _col$resizable === void 0 ? void 0 : _col$resizable.boundary) !== null && _col$resizable$bounda !== void 0 ? _col$resizable$bounda : 71) +
                // 16 for <th> padding block 8px
                17),
                startX: (0, _utils.toFloat)(triggerRect.x + triggerRect.width * 0.5)
              };
              _appClientToolkit.eventBus.emit(getResizingEvent(key), resizingSettingRef.current);
            },
            onResizeStop: function onResizeStop(diffWidthByIndicator, triggerX) {
              // 因为要提前清空来尽可能早的隐藏指示器，清空后还会临时用一次计算，所以在这里缓存一下
              var boundary = resizingSettingRef.current.boundary;
              // 重置列 resizing 配置
              resizingSettingRef.current = {
                inResizing: false,
                boundary: 0,
                startX: 0
              };
              _appClientToolkit.eventBus.emit(getResizingEvent(key), resizingSettingRef.current);
              _appClientToolkit.eventBus.emit('resizeColumn', {
                columnKey: col.key,
                width: Math.max((boundary !== null && boundary !== void 0 ? boundary : 0) - triggerX, diffWidthByIndicator)
              });
            }
          });
        }
      });
    });
    return !cols.some(function (i) {
      return i.key === 'resize-placeholder';
    }) ? [].concat(_toConsumableArray(cols), [
    // 自动注入空白列来吸收多余的宽度，否则 resizing 后的总列宽小于容器宽度时会因为 antd 会强制其他列按比例吸收被调低的列宽出现抖动问题
    {
      key: 'resize-placeholder',
      search: false,
      editable: false,
      className: 'resize-placeholder'
    }]) : cols;
  }, []);
  return (0, _react.useMemo)(function () {
    return {
      ResizingHeaderCell: ResizingHeaderCell,
      ResizingIndicator: ResizingIndicator,
      updateResizableColumns: updateResizableColumns
    };
  }, []);
};
