// @spotter/ui/lib/components/SptTable/StickyScrollbar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StickyScrollbar = exports.ScrollBodyWrapper = void 0;
var _react = _interopRequireWildcard(require("react"));
var _appClientToolkit = require("@spotter/app-client-toolkit");
var _reactDom = require("react-dom");
var _utils = require("./utils");
var _excluded = ["id"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
var SCROLLBAR_EVENT = /*#__PURE__*/function (SCROLLBAR_EVENT) {
  SCROLLBAR_EVENT[SCROLLBAR_EVENT["FULL_WIDTH_UPDATE"] = 0] = "FULL_WIDTH_UPDATE";
  SCROLLBAR_EVENT[SCROLLBAR_EVENT["TABLE_DODY_SCROLL"] = 1] = "TABLE_DODY_SCROLL";
  SCROLLBAR_EVENT[SCROLLBAR_EVENT["SCROLLBAR_MOVE"] = 2] = "SCROLLBAR_MOVE";
  return SCROLLBAR_EVENT;
}(SCROLLBAR_EVENT || {});
var getScrollbarEvent = function getScrollbarEvent(id, event) {
  return "".concat(id, "_SCROLL_BAR_").concat(event);
};
var StickyScrollbar = exports.StickyScrollbar = function StickyScrollbar(_ref) {
  var id = _ref.id,
    getContainer = _ref.getContainer;
  var tableSizeRef = (0, _react.useRef)({
    fullWidth: 0,
    scrollbarWidth: 0,
    scrollbarScale: 0,
    scrollbarThumbOffset: 0,
    scrollbarClientX: 0
  });
  var scrollbarThumbRef = (0, _react.useRef)();
  var scrollbarRef = (0, _react.useRef)();
  var moveScrollbar = (0, _react.useCallback)(function (tableBodyScrollLeft) {
    tableSizeRef.current.scrollbarThumbOffset = tableBodyScrollLeft * tableSizeRef.current.scrollbarScale;
    if (scrollbarThumbRef.current) {
      scrollbarThumbRef.current.style.setProperty('transform', "translateX(".concat((0, _utils.toFloat)(tableSizeRef.current.scrollbarThumbOffset), "px)"));
    }
  }, [id]);
  (0, _react.useEffect)(function () {
    // 当视窗尺寸更新时触发重新计算 scrollbar 宽度
    _appClientToolkit.eventBus.on(getScrollbarEvent(id, SCROLLBAR_EVENT.FULL_WIDTH_UPDATE), function (_ref2) {
      var tableFullWidth = _ref2.tableFullWidth,
        tableBodyScrollLeft = _ref2.tableBodyScrollLeft;
      var _scrollbarRef$current = scrollbarRef.current.getBoundingClientRect(),
        width = _scrollbarRef$current.width,
        x = _scrollbarRef$current.x;
      var scrollbarScale = Math.max(Math.min(width / tableFullWidth, 1), 0);
      tableSizeRef.current = _objectSpread(_objectSpread({}, tableSizeRef.current), {}, {
        scrollbarWidth: width,
        scrollbarClientX: x,
        fullWidth: tableFullWidth,
        scrollbarThumbOffset: tableBodyScrollLeft * scrollbarScale,
        scrollbarScale: scrollbarScale
      });
      // 如果比例等于 1 说明无法滚动
      if (tableSizeRef.current.scrollbarScale === 1) {
        scrollbarThumbRef.current.style.setProperty('visibility', 'hidden');
      } else {
        scrollbarThumbRef.current.setAttribute('style', "visibility:visible;width:".concat((0, _utils.toFloat)(tableSizeRef.current.scrollbarWidth * tableSizeRef.current.scrollbarScale), "px;transform:translateX(").concat((0, _utils.toFloat)(tableSizeRef.current.scrollbarThumbOffset), "px)"));
      }
    });
    _appClientToolkit.eventBus.on(getScrollbarEvent(id, SCROLLBAR_EVENT.TABLE_DODY_SCROLL), moveScrollbar);
    return function () {
      _appClientToolkit.eventBus.off(getScrollbarEvent(id, SCROLLBAR_EVENT.FULL_WIDTH_UPDATE));
      _appClientToolkit.eventBus.off(getScrollbarEvent(id, SCROLLBAR_EVENT.TABLE_DODY_SCROLL));
    };
  }, [id]);
  return /*#__PURE__*/(0, _reactDom.createPortal)( /*#__PURE__*/_react.default.createElement("div", {
    ref: function ref(dom) {
      if (dom) {
        scrollbarRef.current = dom;
        if (scrollbarRef.current) {
          var _ref3;
          var _scrollbarRef$current2 = scrollbarRef.current.getBoundingClientRect(),
            width = _scrollbarRef$current2.width,
            x = _scrollbarRef$current2.x;
          tableSizeRef.current = _objectSpread(_objectSpread({}, tableSizeRef.current), {}, {
            scrollbarWidth: width,
            scrollbarClientX: x,
            scrollbarScale: Math.max(Math.min((_ref3 = width / tableSizeRef.current.fullWidth) !== null && _ref3 !== void 0 ? _ref3 : width, 1), 0)
          });
        }
      }
    },
    className: "spotter-table-sticky-scroll-bar"
  }, /*#__PURE__*/_react.default.createElement("div", {
    onClick: function onClick(e) {
      e.stopPropagation();
      e.preventDefault();
    },
    onMouseDown: function onMouseDown(mouseDownEvent) {
      // 清空选项
      mouseDownEvent.stopPropagation();
      (0, _utils.clearSelection)();
      var _ref4 = scrollbarThumbRef.current,
        ownerDocument = _ref4.ownerDocument;
      // 鼠标单击时与滚动条的 x 轴相对偏移量，用来保持拖动过程中维持相对位置
      var mouseOffsetXWithScrollBarThumb = mouseDownEvent.clientX - tableSizeRef.current.scrollbarClientX - tableSizeRef.current.scrollbarThumbOffset;
      scrollbarRef.current.classList.add('spotter-table-sticky-scroll-bar-thumb-dragging');
      var scrollbarDrag = function scrollbarDrag(mouseMoveEvent) {
        // 通过该类名来实现 table 不可选中的效果，避免拖动时候选中文本
        tableSizeRef.current.scrollbarThumbOffset = Math.min(Math.max(mouseMoveEvent.clientX - mouseOffsetXWithScrollBarThumb - tableSizeRef.current.scrollbarClientX, 0), tableSizeRef.current.scrollbarWidth * (1 - tableSizeRef.current.scrollbarScale));
        _appClientToolkit.eventBus.emit(getScrollbarEvent(id, SCROLLBAR_EVENT.SCROLLBAR_MOVE),
        // 将滚动条的 scrollLeft 通过比例转为实际 table body 的 scrollLeft
        tableSizeRef.current.scrollbarThumbOffset / tableSizeRef.current.scrollbarScale);
      };
      ownerDocument.addEventListener('mousemove', scrollbarDrag);
      ownerDocument.addEventListener('mouseup', function (mouseUpEvent) {
        mouseUpEvent.stopPropagation();
        ownerDocument.removeEventListener('mousemove', scrollbarDrag);
        scrollbarRef.current.classList.remove('spotter-table-sticky-scroll-bar-thumb-dragging');
      }, {
        once: true
      });
    },
    className: "spotter-table-sticky-scroll-bar-thumb",
    ref: function ref(dom) {
      dom && (scrollbarThumbRef.current = dom);
    }
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-table-sticky-scroll-bar-track",
    onClick: function onClick(e) {
      _appClientToolkit.eventBus.emit(getScrollbarEvent(id, SCROLLBAR_EVENT.SCROLLBAR_MOVE), (0, _utils.toFloat)(e.clientX - tableSizeRef.current.scrollbarClientX));
    }
  })), getContainer());
};
var ScrollBodyWrapper = exports.ScrollBodyWrapper = function ScrollBodyWrapper(_ref5) {
  var id = _ref5.id,
    restProps = _objectWithoutProperties(_ref5, _excluded);
  var tbodyRef = (0, _react.useRef)();
  var scrollContainerRef = (0, _react.useRef)();
  var scrollHandler = (0, _react.useCallback)(function (e) {
    _appClientToolkit.eventBus.emit(getScrollbarEvent(id, SCROLLBAR_EVENT.TABLE_DODY_SCROLL), e.target.scrollLeft);
  }, [id]);
  var scrollbarDragHandler = (0, _react.useCallback)(function (scrollLeft) {
    scrollContainerRef.current.scrollLeft = scrollLeft;
  }, [id]);
  var resizeHandler = (0, _react.useCallback)(function () {
    _appClientToolkit.eventBus.emit(getScrollbarEvent(id, SCROLLBAR_EVENT.FULL_WIDTH_UPDATE), {
      tableFullWidth: tbodyRef.current.getBoundingClientRect().width,
      tableBodyScrollLeft: tbodyRef.current.parentElement.parentElement.scrollLeft
    });
  }, [id]);
  (0, _react.useEffect)(function () {
    if (tbodyRef.current) {
      tbodyRef.current.addEventListener('scroll', scrollHandler);
      // 当视窗尺寸更新时触发重新计算容器宽度
      window.addEventListener('resize', resizeHandler);
    }
    return function () {
      if (tbodyRef.current) {
        window.removeEventListener('resize', resizeHandler);
        _appClientToolkit.eventBus.off(getScrollbarEvent(id, SCROLLBAR_EVENT.SCROLLBAR_MOVE), scrollbarDragHandler);
      }
    };
  }, [id]);
  return /*#__PURE__*/_react.default.createElement("tbody", _extends({
    ref: function ref(dom) {
      if (dom) {
        tbodyRef.current = dom;
        // 向上找两级找到 div.ant-table-body，作为可滚动的元素容器
        scrollContainerRef.current = dom.parentElement.parentElement;
        resizeHandler();
        // 每次检测到更新 dom 都重新进行 scroll 事件订阅
        scrollContainerRef.current.removeEventListener('scroll', scrollHandler);
        scrollContainerRef.current.addEventListener('scroll', scrollHandler);
        // 每次检测到更新 dom 都重新进行 scroll-bar 事件订阅
        _appClientToolkit.eventBus.off(getScrollbarEvent(id, SCROLLBAR_EVENT.SCROLLBAR_MOVE), scrollbarDragHandler);
        _appClientToolkit.eventBus.on(getScrollbarEvent(id, SCROLLBAR_EVENT.SCROLLBAR_MOVE), scrollbarDragHandler);
      }
    }
  }, restProps));
};

/**
 * 这里必须要自己实现一个 scrollbar 因为 antd 原生自带的 scrollbar 存在一个 bug，在滚动到底部时他手动点击一次就会消除掉 scrollbar, 不然可以通过 css 来 hack
 * @param id
 */
