// @spotter/ui/lib/components/SptTable/Table.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _proComponents = require("@ant-design/pro-components");
var _classnames = _interopRequireDefault(require("classnames"));
var _react = _interopRequireWildcard(require("react"));
var _StickyScrollbar = require("./StickyScrollbar");
var _excluded = ["className", "width", "fit", "fitClass", "sticky", "columns", "cacheKeyWithRoute", "onRow", "onRowSelection", "onRowDrawerChange", "ResizingHeaderCell", "components", "ResizingIndicator", "scrollY", "pagination", "componentId"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
// 元素名称抽屉触发黑名单，点击在此列表中的元素时不会触发抽屉
var NODE_NAME_BLACK_LIST_DRAWER_TRIGGER = new Set(['A']);
function Table(_ref) {
  var className = _ref.className,
    width = _ref.width,
    fit = _ref.fit,
    fitClass = _ref.fitClass,
    sticky = _ref.sticky,
    columns = _ref.columns,
    cacheKeyWithRoute = _ref.cacheKeyWithRoute,
    _onRow = _ref.onRow,
    onRowSelection = _ref.onRowSelection,
    onRowDrawerChange = _ref.onRowDrawerChange,
    ResizingHeaderCell = _ref.ResizingHeaderCell,
    originComponents = _ref.components,
    ResizingIndicator = _ref.ResizingIndicator,
    scrollY = _ref.scrollY,
    pagination = _ref.pagination,
    _componentId = _ref.componentId,
    restProps = _objectWithoutProperties(_ref, _excluded);
  var spotterTableWrapperRef = (0, _react.useRef)();

  // eslint-disable-next-line react/no-unstable-nested-components
  var tableViewRender = (0, _react.useCallback)(function (_, defaultDom) {
    return /*#__PURE__*/_react.default.createElement("div", {
      id: _componentId,
      className: (0, _classnames.default)('spotter-table-wrapper', {
        'with-pagination': pagination === undefined ? true : pagination
        // 'with-pagination': true,
      }),

      ref: function ref(dom) {
        spotterTableWrapperRef.current = dom;
      }
    }, defaultDom, /*#__PURE__*/_react.default.createElement(ResizingIndicator, {
      id: _componentId
    }), /*#__PURE__*/_react.default.createElement(_StickyScrollbar.StickyScrollbar, {
      id: _componentId,
      getContainer: function getContainer() {
        var _spotterTableWrapperR;
        return ((_spotterTableWrapperR = spotterTableWrapperRef.current) === null || _spotterTableWrapperR === void 0 ? void 0 : _spotterTableWrapperR.querySelector('.ant-table-wrapper >.ant-spin-nested-loading >.ant-spin-container >.ant-table')) || document.body;
      }
    }));
  }, [ResizingIndicator, _componentId]);

  // 预置参数.
  var spotterTablePresetProps = (0, _react.useMemo)(function () {
    return {
      className: (0, _classnames.default)(className, fitClass, 'spotter-table'),
      sticky: sticky,
      scroll: {
        y: scrollY,
        x: width,
        scrollToFirstRowOnChange: true
      },
      options: false,
      size: 'small',
      onRow: function onRow(record) {
        var eventHandlers = _onRow === null || _onRow === void 0 ? void 0 : _onRow(record);
        return _objectSpread(_objectSpread({}, eventHandlers), {}, {
          onClick: function onClick(e) {
            var _document$getSelectio, _eventHandlers$onClic;
            // 选中时不触发展开，并取消开放状态
            if ((_document$getSelectio = document.getSelection()) !== null && _document$getSelectio !== void 0 && _document$getSelectio.toString()) {
              onRowSelection === null || onRowSelection === void 0 || onRowSelection(document.getSelection(), e);
            } else {
              !NODE_NAME_BLACK_LIST_DRAWER_TRIGGER.has(e.target.nodeName) && (onRowDrawerChange === null || onRowDrawerChange === void 0 ? void 0 : onRowDrawerChange(true, record));
            }
            eventHandlers === null || eventHandlers === void 0 || (_eventHandlers$onClic = eventHandlers.onClick) === null || _eventHandlers$onClic === void 0 || _eventHandlers$onClic.call(eventHandlers, e);
          }
        });
      },
      // eslint-disable-next-line react/no-unstable-nested-components
      tableViewRender: tableViewRender
    };
  }, [className, width, sticky, fitClass, fit, onRowDrawerChange, onRowSelection, cacheKeyWithRoute, scrollY]);
  var componentHeader = (0, _react.useMemo)(function () {
    return _objectSpread(_objectSpread({}, originComponents === null || originComponents === void 0 ? void 0 : originComponents.header), {}, {
      cell: ResizingHeaderCell
    });
  }, [originComponents, ResizingHeaderCell]);
  var componentBody = (0, _react.useMemo)(function () {
    return _objectSpread(_objectSpread({}, originComponents === null || originComponents === void 0 ? void 0 : originComponents.body), {}, {
      // eslint-disable-next-line react/no-unstable-nested-components
      wrapper: function wrapper(props) {
        return /*#__PURE__*/_react.default.createElement(_StickyScrollbar.ScrollBodyWrapper, _extends({
          id: _componentId
        }, props));
      }
    });
  }, [originComponents, _componentId]);
  return /*#__PURE__*/_react.default.createElement(_proComponents.ProTable
  // TODO: 处理类型报错的问题
  , _extends({}, restProps, spotterTablePresetProps, {
    // 关闭默认的搜索、分页功能，解决搜索和分页功能导致延迟的问题,避免不必要的刷新率
    search: false,
    pagination: false,
    loading: false,
    columns: columns,
    components: {
      table: originComponents === null || originComponents === void 0 ? void 0 : originComponents.table,
      header: componentHeader,
      body: componentBody
    }
  }));
}
var _default = exports.default = /*#__PURE__*/(0, _react.memo)(Table);
