// @spotter/ui/lib/components/SptTable/useFit.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useFit = void 0;
var _react = require("react");
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var SPOTTER_TABLE_FIT_CLASS = /*#__PURE__*/function (SPOTTER_TABLE_FIT_CLASS) {
  SPOTTER_TABLE_FIT_CLASS["CONTAINER"] = "spotter-table-fit-container";
  SPOTTER_TABLE_FIT_CLASS["CONTENT"] = "spotter-table-fit-content";
  SPOTTER_TABLE_FIT_CLASS["NONE"] = "spotter-table-fit-none";
  return SPOTTER_TABLE_FIT_CLASS;
}(SPOTTER_TABLE_FIT_CLASS || {});
var getScrollY = function getScrollY(fitMode) {
  return fitMode === 'auto' && window.innerHeight > 1080 || fitMode === 'container' ? '100%' : undefined;
};
var useFit = exports.useFit = function useFit(fitMode) {
  // y 轴滚动设置
  var _useState = (0, _react.useState)(
    // 初始化时检测高度
    getScrollY(fitMode)),
    _useState2 = _slicedToArray(_useState, 2),
    scrollY = _useState2[0],
    setScrollY = _useState2[1];
  var fitClass = (0, _react.useMemo)(function () {
    switch (fitMode) {
      case 'auto':
        {
          return window.innerHeight > 1080 ? SPOTTER_TABLE_FIT_CLASS.CONTAINER : SPOTTER_TABLE_FIT_CLASS.CONTENT;
        }
      case 'content':
        {
          return SPOTTER_TABLE_FIT_CLASS.CONTENT;
        }
      case 'container':
        {
          return SPOTTER_TABLE_FIT_CLASS.CONTAINER;
        }
      case 'none':
        {
          return SPOTTER_TABLE_FIT_CLASS.NONE;
        }
      default:
        {
          return SPOTTER_TABLE_FIT_CLASS.CONTENT;
        }
    }
  }, [scrollY, fitMode]);
  (0, _react.useEffect)(function () {
    var resizeHandler = function resizeHandler() {
      setScrollY(getScrollY(fitMode));
    };
    window.addEventListener('resize', resizeHandler);
    return function () {
      return window.removeEventListener('resize', resizeHandler);
    };
  }, []);
  return {
    scrollY: scrollY,
    fitClass: fitClass
  };
};
