// @spotter/ui/es/components/SptTransfer/index.js
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/components/SptTransfer/index.tsx
import { Button, Checkbox, Input, Spin, Tooltip } from "antd";
import React, { useEffect, useMemo, useState } from "react";
import classnames from "classnames";
import { CloseOutlined } from "@ant-design/icons";
import "./style/index.less";
import { SptComponentProvider } from "../Provider";
var SptTransfer = ({
  className,
  style,
  options,
  defaultValue = [],
  value,
  request,
  searchPlaceholder = "请输入",
  loading = false,
  onChange
}) => {
  const [checkedList, _setCheckedList] = useState(defaultValue);
  const [allOptions, setAllOptions] = useState([]);
  const [searchValue, setSearchValue] = useState("");
  const [_loading, setLoading] = useState(loading);
  const finalOptions = useMemo(() => options != null ? options : allOptions, [options, allOptions]);
  const finalValue = useMemo(() => value != null ? value : checkedList, [value, checkedList]);
  const checkedAll = useMemo(
    () => !!finalOptions.length && finalValue.length === finalOptions.length,
    [finalOptions, finalValue]
  );
  const setCheckedList = (checkedOptions2) => {
    _setCheckedList(checkedOptions2);
    onChange && onChange(checkedOptions2);
  };
  const onCheckedOptionChange = (e) => {
    const isChecked = !e.target.checked;
    const checkValue = e.target.value;
    if (isChecked) {
      const _index = finalValue.indexOf(checkValue);
      if (_index > -1) {
        const _list = [...finalValue];
        _list.splice(_index, 1);
        setCheckedList(_list);
      }
    } else {
      setCheckedList([...finalValue, checkValue]);
    }
  };
  const filteredOptions = finalOptions.filter(
    (o) => {
      var _a;
      return (_a = o.label) == null ? void 0 : _a.toLowerCase().includes(searchValue == null ? void 0 : searchValue.toLowerCase());
    }
  );
  const checkedOptions = finalOptions.filter((option) => finalValue.includes(option.value));
  useEffect(() => {
    setLoading(loading);
  }, [loading]);
  useEffect(() => {
    (() => __async(void 0, null, function* () {
      if (request) {
        try {
          setLoading(true);
          const o = yield request();
          setAllOptions(o);
        } finally {
          setLoading(false);
        }
      }
    }))();
  }, [request]);
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement("div", { style, className: classnames("spotter-transfer", className) }, /* @__PURE__ */ React.createElement(Spin, { spinning: _loading }, /* @__PURE__ */ React.createElement("div", { className: "spotter-transfer-loading-container" }, /* @__PURE__ */ React.createElement("div", { className: "spotter-transfer-all-options-container" }, /* @__PURE__ */ React.createElement(
    Input,
    {
      placeholder: searchPlaceholder,
      onChange: (e) => {
        setSearchValue(e.target.value);
      }
    }
  ), /* @__PURE__ */ React.createElement("div", { className: "spotter-transfer-all-options-title" }, /* @__PURE__ */ React.createElement("span", { className: "spotter-transfer-all-options-title-summary" }, `总共：${finalOptions.length} 项`), /* @__PURE__ */ React.createElement(
    Button,
    {
      onClick: () => {
        setCheckedList(
          !checkedAll ? finalOptions.map(({ value: v }) => v) : []
        );
      },
      type: "link"
    },
    checkedAll ? "取消全选" : "全选"
  )), /* @__PURE__ */ React.createElement("div", { className: "spotter-transfer-all-options-list" }, filteredOptions == null ? void 0 : filteredOptions.map((item) => /* @__PURE__ */ React.createElement(
    "div",
    {
      key: item.value,
      className: "spotter-transfer-all-options-list-item"
    },
    /* @__PURE__ */ React.createElement(
      Checkbox,
      {
        value: item.value,
        checked: finalValue.includes(item.value),
        onChange: onCheckedOptionChange
      },
      item.label
    ),
    /* @__PURE__ */ React.createElement("br", null)
  )))), /* @__PURE__ */ React.createElement("div", { className: "spotter-transfer-checked-options-container" }, /* @__PURE__ */ React.createElement("div", { className: "spotter-transfer-checked-options-title" }, /* @__PURE__ */ React.createElement("span", { className: "spotter-transfer-checked-options-title-summary" }, `已选：${finalValue.length} 项`), /* @__PURE__ */ React.createElement(
    Button,
    {
      onClick: () => {
        setCheckedList([]);
      },
      type: "link"
    },
    "清空"
  )), /* @__PURE__ */ React.createElement("div", { className: "spotter-transfer-checked-options-list" }, checkedOptions.map(({ label, value: v }) => /* @__PURE__ */ React.createElement(
    "div",
    {
      className: "spotter-transfer-checked-options-list-item",
      key: label
    },
    /* @__PURE__ */ React.createElement("span", { className: "spotter-transfer-checked-options-list-item-label" }, label),
    /* @__PURE__ */ React.createElement(Tooltip, { title: "移除" }, /* @__PURE__ */ React.createElement(
      "span",
      {
        className: "spotter-transfer-checked-options-list-item-action remove mr-6px",
        onClick: () => {
          setCheckedList(
            checkedList.filter((i) => i !== v)
          );
        }
      },
      /* @__PURE__ */ React.createElement(CloseOutlined, null)
    ))
  ))))))));
};
var SptTransfer_default = SptTransfer;
export {
  SptTransfer_default as default
};
