// @spotter/ui/es/components/SptRestModal/index.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};

// src/components/SptRestModal/index.tsx
import { SearchOutlined } from "@ant-design/icons";
import { ModalForm } from "@ant-design/pro-components";
import { Empty, Input } from "antd";
import React from "react";
import { useMemo, useState } from "react";
import { SptComponentProvider } from "../Provider";
var SptRestModal = ({
  style,
  title,
  className,
  trigger,
  options,
  placeholder = "请输入",
  open,
  modalProps
}) => {
  const [search, setSearch] = useState();
  const list = useMemo(() => {
    if (search) {
      return options == null ? void 0 : options.filter((item) => {
        var _a;
        return ((_a = `${item}`) == null ? void 0 : _a.indexOf(search)) > -1;
      });
    }
    return options;
  }, [options, search]);
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(
    ModalForm,
    __spreadValues({
      style,
      title,
      width: 572,
      open,
      className,
      trigger,
      submitter: false
    }, modalProps),
    /* @__PURE__ */ React.createElement(
      Input,
      {
        prefix: /* @__PURE__ */ React.createElement(SearchOutlined, null),
        allowClear: true,
        value: search,
        style: {
          width: 240
        },
        placeholder,
        onChange: (e) => {
          setSearch(e.target.value);
        }
      }
    ),
    /* @__PURE__ */ React.createElement(
      "div",
      {
        className: "receive-accounts-wrap",
        style: {
          background: "#F7F9FB",
          marginTop: "16px",
          padding: "16px",
          height: "324px",
          overflow: "auto"
        }
      },
      list == null ? void 0 : list.map((item, index) => (
        // eslint-disable-next-line react/no-array-index-key
        /* @__PURE__ */ React.createElement("div", { key: index, className: "mb-8px" }, item)
      )),
      !list || list.length === 0 ? /* @__PURE__ */ React.createElement("div", { className: "h-full flex items-center justify-center" }, /* @__PURE__ */ React.createElement(Empty, null)) : null
    )
  ));
};
var SptRestModal_default = SptRestModal;
export {
  SptRestModal_default as default
};
