// @spotter/ui/es/components/SptUserAvatar/index.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptUserAvatar/index.tsx
import { Avatar, Spin } from "antd";
import React from "react";
import { LoadingOutlined } from "@ant-design/icons";
import { SptComponentProvider } from "../Provider";
var UserAvatarBase = ({
  maxNameLength = 1,
  size = 32,
  fontSize,
  name,
  loading,
  style
}) => {
  var _a;
  return /* @__PURE__ */ React.createElement(
    Avatar,
    {
      className: "bg-primary-500",
      style: __spreadValues({
        display: "flex",
        alignItems: "center",
        backgroundColor: "var(--color-primary)",
        width: typeof size === "number" ? size : size.width,
        height: typeof size === "number" ? size : size.height,
        fontSize: fontSize != null ? fontSize : Math.max(14, (typeof size === "number" ? size : size.height) / 3)
      }, style),
      gap: 16
    },
    loading ? /* @__PURE__ */ React.createElement(
      Spin,
      {
        indicator: /* @__PURE__ */ React.createElement(LoadingOutlined, null),
        spinning: true,
        style: { color: "#fff", display: "flex", alignItems: "center" },
        size: "small"
      }
    ) : name == null ? void 0 : name.slice(0, Math.min((_a = name == null ? void 0 : name.length) != null ? _a : Number.MAX_SAFE_INTEGER, maxNameLength))
  );
};
var SptUserAvatar = (_a) => {
  var props = __objRest(_a, []);
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(UserAvatarBase, __spreadValues({}, props)));
};
var SptUserAvatar_default = SptUserAvatar;
export {
  SptUserAvatar_default as default
};
