// @spotter/app-client-toolkit@1.0.42-nic.8/lib/infra/mainApp/network/hijack/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unproxy = exports.proxy = exports.EnableScope = void 0;
const ajax_hook_1 = require("ajax-hook");
const fetch_1 = require("./fetch");
var EnableScope;
(function (EnableScope) {
    EnableScope["FETCH"] = "fetch";
    EnableScope["XHR"] = "xhr";
})(EnableScope = exports.EnableScope || (exports.EnableScope = {}));
const _enabledScope = new Set([]);
function proxy({ enableScope = Object.values(EnableScope), beforeRequest, afterResponse, }) {
    if (enableScope.includes(EnableScope.XHR)) {
        (0, ajax_hook_1.proxy)({
            onRequest(config, handler) {
                beforeRequest &&
                    beforeRequest(config).then((overrideConfig) => {
                        handler.next({
                            ...config,
                            ...overrideConfig,
                        });
                    });
            },
            onResponse(config, handler) {
                afterResponse &&
                    afterResponse(config).then((overrideConfig) => {
                        handler.next({
                            ...config,
                            ...overrideConfig,
                        });
                    });
            },
        });
    }
    if (enableScope.includes(EnableScope.FETCH)) {
        (0, fetch_1.proxyFetch)({
            async onRequest(info, init) {
                if (beforeRequest) {
                    const r = new Request(info, init);
                    const config = await beforeRequest({
                        url: r.url,
                        method: r.method,
                        // @ts-ignore
                        headers: r.headers ? Object.fromEntries(r.headers) : undefined,
                    });
                    return [new Request(r, config)];
                }
                return [info, init];
            },
            async onResponse(response) {
                if (afterResponse) {
                    afterResponse({
                        status: response.status,
                        statusText: response.statusText,
                        response,
                        config: {
                            url: response.url,
                        },
                    });
                }
                return response;
            },
        });
    }
}
exports.proxy = proxy;
function unproxy() {
    if (_enabledScope.has(EnableScope.XHR)) {
        (0, ajax_hook_1.unProxy)();
    }
    if (_enabledScope.has(EnableScope.FETCH)) {
        (0, fetch_1.unproxyFetch)();
    }
}
exports.unproxy = unproxy;
