// @spotter/app-client-toolkit/lib/tool/formatter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.formatStorageName = exports.formatLength = exports.formatWeight = exports.formatVolume = exports.sortQueryFormatter = exports.formatNumber = exports.CurrencyFormatter = exports.getCurrencySymbol = exports.formatEmpty = exports.formatUnit = exports.amountFormatter = exports.dataSizeFormatter = void 0;
const const_1 = require("../const");
/**
 * 格式化工具
 * - 数据大小
 * - 金额
 * - 数字
 */
/**
 * 数据大小格式化工具
 */
exports.dataSizeFormatter = function (size) {
    if (size > const_1.TB) {
        return `${(size / const_1.TB).toFixed(2)} TB`;
    }
    if (size > const_1.GB) {
        return `${(size / const_1.GB).toFixed(2)} GB`;
    }
    if (size > const_1.MB) {
        return `${(size / const_1.MB).toFixed(2)} MB`;
    }
    return `${(size / const_1.KB).toFixed(2)} KB`;
};
/**
 * 金额格式化工具
 * @deprecated 【请勿使用，已废弃，未来即将移除】仅兼容旧代码，新的金额/货币格式化工具请使用 formatCurrency
 * @param amount
 * @param decimal
 */
function amountFormatter(amount, decimal = 2) {
    return amount || String(amount) === '0'
        ? Number.parseFloat(String(amount)).toLocaleString('en-US', {
            minimumFractionDigits: decimal,
            maximumFractionDigits: decimal
        })
        : const_1.DEFAULT_EMPTY_VALUE;
}
exports.amountFormatter = amountFormatter;
/**
 * 单位格式化工具 【测试版本】
 * @param digit
 * @param opt 可以直接传string作为单位；也可传`UnitOption`作为配置
 */
function formatUnit(digit, opt) {
    if (const_1.EMPTY_SET.has(digit))
        return const_1.DEFAULT_EMPTY_VALUE;
    const { decimal = 0, unit = '', pattern = 'DU', transform } = typeof opt === 'string' ? { unit: opt } : opt ?? {};
    const temp = {
        U: unit,
        D: amountFormatter(transform ? transform(digit) : digit, decimal)
    };
    return pattern.replace(/[DU]/g, (res) => temp[res]);
}
exports.formatUnit = formatUnit;
/** 空数据格式化工具 */
function formatEmpty(value, empty = const_1.DEFAULT_EMPTY_VALUE) {
    return const_1.EMPTY_SET.has(value) ? empty : value;
}
exports.formatEmpty = formatEmpty;
/**
 * @description: 获取货币符号
 * @param {string} code
 */
function getCurrencySymbol(code) {
    return const_1.currencySymbolMap[code];
}
exports.getCurrencySymbol = getCurrencySymbol;
/**
 * 货币格式化工具
 */
class CurrencyFormatter {
    /**
     * @param opt
     */
    constructor(opt) {
        /**
         * @description: 获取货币符号
         * @param code
         */
        this.symbol = (code = this.code) => this.symbolMap[code];
        /**
         * 数字格式化
         * @param amount
         * @param decimal
         */
        this.amount = (amount, decimal = this.decimal) => !this.empty.has(amount)
            ? Number.parseFloat(String(amount)).toLocaleString('en-US', {
                minimumFractionDigits: decimal,
                maximumFractionDigits: decimal
            })
            : const_1.DEFAULT_EMPTY_VALUE;
        /**
         * 货币格式化
         * @param amount
         * @param code
         * @param opt [opt]
         */
        this.format = (amount, opt) => {
            if (this.empty.has(amount))
                return const_1.DEFAULT_EMPTY_VALUE;
            const { decimal = this.decimal, pattern = this.pattern, code = opt?.code ?? this.code } = { ...opt };
            const temp = {
                C: code,
                S: this.symbol(code),
                D: this.amount(amount, decimal)
            };
            return pattern.replace(/[CDS]/g, (res) => temp[res]);
        };
        this.decimal = opt?.decimal ?? 2;
        this.pattern = opt?.pattern ?? 'SD';
        this.code = opt?.code ?? 'USD';
        this.empty = opt?.emptySet ?? const_1.EMPTY_SET;
        this.symbolMap = opt?.symbolMap ?? const_1.currencySymbolMap;
    }
}
exports.CurrencyFormatter = CurrencyFormatter;
/**
 * 数字格式化工具
 * @param number
 * @param decimal
 */
function formatNumber(number, decimal = 2) {
    return number !== null && !Number.isNaN(+number)
        ? Number.parseFloat(String(number)).toFixed(decimal)
        : const_1.DEFAULT_EMPTY_VALUE;
}
exports.formatNumber = formatNumber;
/**
 * 排序参数格式化（与后端保持一致）
 * @param {Record} sort
 */
const sortQueryFormatter = (sort) => {
    if (sort) {
        const [column] = Object.keys(sort);
        return { column, sortType: sort[column] };
    }
    return {};
};
exports.sortQueryFormatter = sortQueryFormatter;
/**
 * 体积格式化
 * @param {number} volume
 * @param {number} [decimal=2]
 * @return {*}
 */
const formatVolume = (volume, decimal = 6) => formatUnit(volume, { pattern: 'D U', unit: 'cbm', decimal });
exports.formatVolume = formatVolume;
/**
 * 重量格式化
 * @param {number} weight
 * @param {number} [decimal=2]
 * @return {*}
 */
const formatWeight = (weight, decimal = 2) => formatUnit(weight, { pattern: 'D U', unit: 'lb', decimal });
exports.formatWeight = formatWeight;
/**
 * 长度格式化
 * @param {number} weight
 * @param {number} [decimal=2]
 */
const formatLength = (weight, decimal = 2) => formatUnit(weight, { pattern: 'D U', unit: 'inch', decimal });
exports.formatLength = formatLength;
/**
 * 仓库名称格式化
 * @param {Record<string, any>} [r]
 */
const formatStorageName = (r) => !!r?.storageName || !!r?.storageCode
    ? `${r.storageName}(${r.storageCode})`
    : const_1.DEFAULT_EMPTY_VALUE;
exports.formatStorageName = formatStorageName;
