// @spotter/app-client-toolkit/lib/tool/sso.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SSOSDKClient = exports.SSOSDK = exports.PermissionErrorCode = exports.AuthErrorCode = exports.SessionKeys = void 0;
var ENV;
(function (ENV) {
    ENV["local"] = "local";
    ENV["development"] = "development";
    ENV["test"] = "test";
    ENV["production"] = "production";
})(ENV || (ENV = {}));
const SSO_API_URL_MAP = {
    [ENV.local]: 'https://sso.dev.spotterio.com',
    [ENV.development]: 'https://sso.dev.spotterio.com',
    [ENV.test]: 'https://sso.tst.spotterio.com',
    [ENV.production]: 'https://sso.spotterio.com'
};
const SSO_TOKEN = 'Authentication-Token';
exports.SessionKeys = 'spotter_token';
exports.AuthErrorCode = new Set([401,
    4010000,
    4010001,
    4010002,
    4010003,
    4010004,
    -1,
    -2,
    -3,
    -4,
    -5,
    -6,
    -7,
]);
exports.PermissionErrorCode = new Set([403, 403001]);
/**
 * 纯前端使用
 */
class SSOSDK {
    constructor({ env, ssoUrl, appCode, loginPath }) {
        this.env = 'production';
        this.appCode = '';
        this.SSO_TICKET = 'spotter_sso_ticket';
        this.SPOTTER_SSO_REDIRECT_COUNT = 'spotter_sso_redirect_count';
        this.SPOTTER_SSO_START_TIME = 'spotter_sso_start_time';
        this.loginPath = ''; // 登录后指定跳转页面地址
        this.ssoUrl = '';
        this.env = env;
        this.appCode = appCode;
        this.loginPath = loginPath;
        this.ssoUrl = ssoUrl || SSO_API_URL_MAP[env];
    }
    async login() {
        console.log('ssoUrl', this.ssoUrl);
        const ticket = this.getLastSpotterSsoTicket(window.location.search);
        const noTicket = this.removeQueryParam(this.getRedirectUrl(), this.SSO_TICKET);
        if (ticket) {
            try {
                const { code, msg, data } = await this.doLoginByTicket(ticket);
                if (code === 200) {
                    localStorage.setItem(SSO_TOKEN, data?.token);
                    window.history.replaceState('', '', noTicket);
                }
                else {
                    throw new Error(msg);
                }
            }
            catch (error) {
                throw new Error(error?.msg || 'sso login error');
            }
        }
        const ticketToken = localStorage.getItem(SSO_TOKEN);
        if (ticketToken) {
            try {
                const { code, msg } = await this.checkLogin(ticketToken);
                if (code === 200)
                    window.history.replaceState('', '', noTicket);
                else
                    throw new Error(msg);
            }
            catch (error) {
                throw new Error(error?.msg || 'sso check login error');
            }
        }
    }
    async doLoginByTicket(ticket) {
        const res = (await (await fetch(`${this.ssoUrl}/sso/client/doLoginByTicket`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                ticket,
                appCode: this.appCode,
            }),
        })).json());
        return res;
    }
    async checkLogin(token) {
        const res = (await (await fetch(`${this.ssoUrl}/sso/client/isLogin`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                token,
            }),
        })).json());
        return res;
    }
    getLastSpotterSsoTicket(url) {
        const params = new URLSearchParams(url);
        let lastValue = null;
        params.getAll(this.SSO_TICKET).forEach((value) => {
            lastValue = value;
        });
        return lastValue;
    }
    removeQueryParam(url, param) {
        const urlObj = new URL(url);
        const searchParams = new URLSearchParams(urlObj.search);
        searchParams.delete(param);
        urlObj.search = searchParams.toString();
        return urlObj.toString();
    }
    redirectToSSO() {
        window.location.href = `${this.ssoUrl}/sso/auth?redirect=${encodeURIComponent(this.getRedirectUrl())}&appCode=${this.appCode}`;
    }
    redirectToSSOLogout() {
        localStorage.removeItem(SSO_TOKEN);
        localStorage.removeItem(this.SPOTTER_SSO_REDIRECT_COUNT);
        window.location.href = `${this.ssoUrl}/sso/client/signout?back=${encodeURIComponent(this.getRedirectUrl())}&appCode=${this.appCode}`;
    }
    redirectWithLimit(limit = 10) {
        const redirectCount = Number(sessionStorage.getItem(this.SPOTTER_SSO_REDIRECT_COUNT)) || 0;
        const lastTime = Number(sessionStorage.getItem(this.SPOTTER_SSO_START_TIME)) || 0;
        if (redirectCount >= limit && Date.now() - lastTime < 2000) {
            sessionStorage.setItem(this.SPOTTER_SSO_START_TIME, String(Date.now()));
            throw new Error('sso 重定向过多，请退出重新登录');
        }
        if (Date.now() - lastTime < 1000)
            sessionStorage.setItem(this.SPOTTER_SSO_REDIRECT_COUNT, String(redirectCount + 1));
        sessionStorage.setItem(this.SPOTTER_SSO_START_TIME, String(Date.now()));
        this.redirectToSSO();
    }
    getSSOToken() {
        return SSO_TOKEN;
    }
    getRedirectUrl() {
        if (this.loginPath) {
            const url = `${window.location.protocol}//${window.location.hostname}:${window.location.port}${this.loginPath}`;
            return url;
        }
        else {
            return window.location.href;
        }
    }
}
exports.SSOSDK = SSOSDK;
/**
 * 存在前后端使用
 */
class SSOSDKClient {
    constructor({ env, appCode }) {
        this.env = 'production';
        this.SSO_TICKET = 'spotter_sso_ticket';
        this.appCode = '';
        this.removeQueryParam = (url, param) => {
            const urlObj = new URL(url);
            const searchParams = new URLSearchParams(urlObj.search);
            searchParams.delete(param);
            urlObj.search = searchParams.toString();
            return urlObj.toString();
        };
        this.env = env;
        this.appCode = appCode;
    }
    login() {
        window.location.href = `${SSO_API_URL_MAP[this.env]}/sso/auth?redirect=${encodeURIComponent(this.removeQueryParam(window.location.href, this.SSO_TICKET))}&appCode=${this.appCode}`;
    }
    logout() {
        const cookieKey = exports.SessionKeys;
        console.log('清除cookie前', document.cookie);
        this.clearCookie(cookieKey);
        console.log('清除cookie后', document.cookie);
        window.location.replace(`${window.location.origin}/api/v1/sso/logout?client=${encodeURIComponent(this.removeQueryParam(window.location.href, this.SSO_TICKET))}`);
    }
    clearCookie(name) {
        document.cookie = `${name}=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;`;
    }
    getCookie() {
        const cookieKey = exports.SessionKeys;
        const arr = document.cookie.split('; ');
        for (const element of arr) {
            const arr2 = element.split('=');
            if (arr2[0].trim() === cookieKey) {
                return arr2[1];
            }
        }
        return '';
    }
}
exports.SSOSDKClient = SSOSDKClient;
