// @spotter/ui/es/components/SptStoreSelect/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/components/SptStoreSelect/index.tsx
import { WarehouseSdk } from "../../services/WarehouseSdk";
import { ProFormSelect } from "@ant-design/pro-components";
import React from "react";
import { SptComponentProvider } from "../Provider";
var WAREHOUSE_TYPE = /* @__PURE__ */ ((WAREHOUSE_TYPE2) => {
  WAREHOUSE_TYPE2[WAREHOUSE_TYPE2["Third"] = 1] = "Third";
  WAREHOUSE_TYPE2[WAREHOUSE_TYPE2["Cloud"] = 2] = "Cloud";
  return WAREHOUSE_TYPE2;
})(WAREHOUSE_TYPE || {});
var SptStoreSelect = (_a) => {
  var _b = _a, {
    storageType,
    fieldProps,
    debounceTime,
    codeInLabel = true,
    labelName = "storageName",
    valueName = "storageCode",
    requestParams
  } = _b, props = __objRest(_b, [
    "storageType",
    "fieldProps",
    "debounceTime",
    "codeInLabel",
    "labelName",
    "valueName",
    "requestParams"
  ]);
  var _a2, _b2, _c;
  if (codeInLabel) {
    labelName = (record) => `${record == null ? void 0 : record.storageName} (${record == null ? void 0 : record.storageCode})`;
  }
  const formatLabel = (record) => {
    return typeof labelName === "function" ? labelName(record) : record[labelName];
  };
  const formatValue = (record) => {
    return typeof valueName === "function" ? valueName(record) : record[valueName];
  };
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(
    ProFormSelect,
    __spreadValues({
      fieldProps: __spreadProps(__spreadValues({}, fieldProps), {
        placeholder: (fieldProps == null ? void 0 : fieldProps.placeholder) || "请输入"
      }),
      request: (_0) => __async(void 0, [_0], function* ({ keyWords }) {
        const { data: res } = yield WarehouseSdk.storageList(__spreadValues({
          storageType,
          keyword: keyWords
        }, requestParams));
        return res.data.map((item) => ({
          label: formatLabel(item),
          value: formatValue(item)
        }));
      }),
      debounceTime: debounceTime != null ? debounceTime : 500,
      noStyle: (_a2 = props.noStyle) != null ? _a2 : true,
      showSearch: (_b2 = props.showSearch) != null ? _b2 : true,
      allowClear: (_c = props.allowClear) != null ? _c : true
    }, props)
  ));
};
var SptStoreSelect_default = SptStoreSelect;
export {
  WAREHOUSE_TYPE,
  SptStoreSelect_default as default
};
