// @spotter/app-client-toolkit/lib/infra/mainApp/app/appRegistry/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.loadApp = exports.registerApps = exports.getDefaultAppLifeCycles = void 0;
const qiankun_1 = require("qiankun");
const lodash_1 = require("lodash");
const logger_1 = require("../../logger");
function getDefaultAppLifeCycles() {
    return {
        beforeLoad: [
            (app, window) => {
                (0, logger_1.proxyConsole)(window, app);
                console.log('[LifeCycle] before load %c%s', 'color: green;', app.name);
                return Promise.resolve();
            },
        ],
        beforeMount: [
            (app) => {
                console.log('[LifeCycle] before mount %c%s', 'color: green;', app.name);
                return Promise.resolve();
            },
        ],
        afterUnmount: [
            (app) => {
                console.log('[LifeCycle] after unmount %c%s', 'color: green;', app.name);
                return Promise.resolve();
            },
        ],
    };
}
exports.getDefaultAppLifeCycles = getDefaultAppLifeCycles;
function registerApps(microApps, { afterRegister, lifeCycles, }) {
    const uniqMircoApps = (0, lodash_1.uniqBy)(microApps, 'name');
    (0, qiankun_1.registerMicroApps)(uniqMircoApps, {
        ...getDefaultAppLifeCycles(),
        ...lifeCycles,
    });
    afterRegister && afterRegister(uniqMircoApps);
}
exports.registerApps = registerApps;
function loadApp(app, configuration, lifeCycles) {
    return (0, qiankun_1.loadMicroApp)(app, configuration, {
        ...getDefaultAppLifeCycles(),
        ...lifeCycles,
    });
}
exports.loadApp = loadApp;
