// @spotter/ui/es/types/enum.js
// src/types/enum.ts
var AccountType = /* @__PURE__ */ ((AccountType2) => {
  AccountType2[AccountType2["EMAIL"] = 1] = "EMAIL";
  AccountType2[AccountType2["PHONE"] = 2] = "PHONE";
  return AccountType2;
})(AccountType || {});
var UserAccountAvailableStatus = /* @__PURE__ */ ((UserAccountAvailableStatus2) => {
  UserAccountAvailableStatus2[UserAccountAvailableStatus2["DISABLE"] = -1] = "DISABLE";
  UserAccountAvailableStatus2[UserAccountAvailableStatus2["ENABLE"] = 1] = "ENABLE";
  UserAccountAvailableStatus2[UserAccountAvailableStatus2["PENDING"] = 2] = "PENDING";
  return UserAccountAvailableStatus2;
})(UserAccountAvailableStatus || {});
var PermissionType = /* @__PURE__ */ ((PermissionType2) => {
  PermissionType2[PermissionType2["MENU"] = 1] = "MENU";
  PermissionType2[PermissionType2["API"] = 2] = "API";
  PermissionType2[PermissionType2["DATA"] = 3] = "DATA";
  return PermissionType2;
})(PermissionType || {});
var MenuType = /* @__PURE__ */ ((MenuType2) => {
  MenuType2[MenuType2["GMESH"] = 1] = "GMESH";
  MenuType2[MenuType2["ROMERX"] = 2] = "ROMERX";
  return MenuType2;
})(MenuType || {});
var MenuTypeMap = {
  [1 /* GMESH */]: "gmesh",
  [2 /* ROMERX */]: "sevc"
};
var Actions = /* @__PURE__ */ ((Actions2) => {
  Actions2["EDIT"] = "edit";
  Actions2["ADD"] = "add";
  Actions2["DELETE"] = "delete";
  return Actions2;
})(Actions || {});
var RoleCode = /* @__PURE__ */ ((RoleCode2) => {
  RoleCode2["accountManager"] = "account_manager";
  RoleCode2["businessManager"] = "business_manager_leader";
  return RoleCode2;
})(RoleCode || {});
var UserStatus = /* @__PURE__ */ ((UserStatus2) => {
  UserStatus2[UserStatus2["FREEZE"] = -1] = "FREEZE";
  UserStatus2[UserStatus2["ACTIVE"] = 1] = "ACTIVE";
  UserStatus2[UserStatus2["INACTIVE"] = 2] = "INACTIVE";
  return UserStatus2;
})(UserStatus || {});
var WAREHOUSE_TYPE = /* @__PURE__ */ ((WAREHOUSE_TYPE2) => {
  WAREHOUSE_TYPE2[WAREHOUSE_TYPE2["Third"] = 1] = "Third";
  WAREHOUSE_TYPE2[WAREHOUSE_TYPE2["Cloud"] = 2] = "Cloud";
  return WAREHOUSE_TYPE2;
})(WAREHOUSE_TYPE || {});
var ValidateStorageName = /* @__PURE__ */ ((ValidateStorageName2) => {
  ValidateStorageName2["PERMIT_DATA"] = "PERMIT_DATA";
  ValidateStorageName2["PERMIT_TAG_REMOTE"] = "permitEtag";
  ValidateStorageName2["PERMIT_TAG_LOCAL"] = "PERMIT_ETAG_LOCAL";
  ValidateStorageName2["MENU_DATA"] = "MENU_DATA";
  ValidateStorageName2["MENU_TAG_REMOTE"] = "menuEtag";
  ValidateStorageName2["MENU_TAG_LOCAL"] = "MEN_ETAG_LOCAL";
  ValidateStorageName2["ROLE_DATA"] = "ROLE_DATA";
  ValidateStorageName2["ROLE_TAG_REMOTE"] = "roleEtag";
  ValidateStorageName2["ROLE_TAG_LOCAL"] = "ROLE_ETAG_LOCAL";
  return ValidateStorageName2;
})(ValidateStorageName || {});
export {
  AccountType,
  Actions,
  MenuType,
  MenuTypeMap,
  PermissionType,
  RoleCode,
  UserAccountAvailableStatus,
  UserStatus,
  ValidateStorageName,
  WAREHOUSE_TYPE
};
