// eslint-plugin-try-catch-failsafe@0.1.4/lib/rules/new-url.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
var findTryStatementRange = function(node) {
    if (!node.parent) {
        return null;
    }
    if (node.parent.type === "TryStatement") {
        var _node_parent, _node_parent1;
        /**
     * The `try` statement should not be the only condition to determine
     * whether the `new URL()` call is wrapped in `try-catch` block,
     * because the following code will not throw an error in JavaScript:
     *
     * ```js
     * try {
     *   // some code that throws an error
     * } catch {
     *   new URL('');
     * } finally {
     *   new URL('');
     * }
     *
     * So if node is wrapped in `catch` or `finally` block, we shoud
     * still report the error.
     */ if (((_node_parent = node.parent) === null || _node_parent === void 0 ? void 0 : _node_parent.handler) === node || ((_node_parent1 = node.parent) === null || _node_parent1 === void 0 ? void 0 : _node_parent1.finalizer) === node) {
            return null;
        }
        return node.range;
    }
    return findTryStatementRange(node.parent);
};
module.exports = {
    meta: {
        docs: {
            description: "Use try-catch to wrap new URL()"
        },
        fixable: null,
        messages: {
            shouldWrap: "new URL() should be wrapped in try-catch"
        }
    },
    create: function(context) {
        return {
            NewExpression: function(node) {
                if (node.callee.type === "Identifier" && node.callee.name === "URL") {
                    var range = findTryStatementRange(node);
                    if (!range) {
                        context.report({
                            node: node,
                            messageId: "shouldWrap"
                        });
                        return;
                    }
                    // range should be bigger than node.range
                    if (range[0] < node.range[0] && node.range[1] < range[1]) {
                        return;
                    }
                    context.report({
                        node: node,
                        messageId: "shouldWrap"
                    });
                }
            }
        };
    }
};
