// @spotter/app-client-toolkit/lib/const/enum.js
"use strict";
/********************* 枚举常量 ****************************/
Object.defineProperty(exports, "__esModule", { value: true });
exports.SUPPORT_CURRENCY = exports.NUMBER_FORMAT_STYLE = exports.SUPPORT_LOCALES = exports.BOOLEAN = void 0;
var BOOLEAN;
(function (BOOLEAN) {
    BOOLEAN[BOOLEAN["FALSE"] = 0] = "FALSE";
    BOOLEAN[BOOLEAN["TRUE"] = 1] = "TRUE";
})(BOOLEAN = exports.BOOLEAN || (exports.BOOLEAN = {}));
/**
 * 已支持的地区
 */
var SUPPORT_LOCALES;
(function (SUPPORT_LOCALES) {
    SUPPORT_LOCALES["EN_US"] = "en-US";
})(SUPPORT_LOCALES = exports.SUPPORT_LOCALES || (exports.SUPPORT_LOCALES = {}));
/**
 * intl.NumberFormat 的 style 枚举值
 */
var NUMBER_FORMAT_STYLE;
(function (NUMBER_FORMAT_STYLE) {
    /**
     * 货币
      */
    NUMBER_FORMAT_STYLE["CURRENCY"] = "currency";
    /**
     * 纯数字
     */
    NUMBER_FORMAT_STYLE["DECIMAL"] = "decimal";
    /**
     * 百分比
     */
    NUMBER_FORMAT_STYLE["PERCENT"] = "percent";
    /**
     * 单位格式
     */
    NUMBER_FORMAT_STYLE["UNIT"] = "unit";
})(NUMBER_FORMAT_STYLE = exports.NUMBER_FORMAT_STYLE || (exports.NUMBER_FORMAT_STYLE = {}));
/**
 * 已支持的货币
 */
var SUPPORT_CURRENCY;
(function (SUPPORT_CURRENCY) {
    SUPPORT_CURRENCY["USD"] = "USD";
})(SUPPORT_CURRENCY = exports.SUPPORT_CURRENCY || (exports.SUPPORT_CURRENCY = {}));
