// @spotter/app-client-toolkit/lib/const/expression.js
"use strict";
/********************* 表达式常量 ****************************/
Object.defineProperty(exports, "__esModule", { value: true });
exports.ONLY_TIME_MINUTE_FORMAT = exports.TIME_SECOND_FORMAT = exports.TIME_MINUTE_FORMAT = exports.TIME_DAY_WITH_WEEK_FORMAT = exports.TIME_DAY_FORMAT = exports.TIME_MONTH_FORMAT = exports.ORDER_CODE_REGEXP = exports.SIGNUP_PASSWORD_REGEXP = exports.LOGIN_PASSWORD_REGEXP = exports.PHONE_REGEXP = exports.EMAIL_REGEXP = void 0;
/**
 *  正则表达式
 *  - 邮箱
 *  - 手机号
 **/
/**
 * 邮箱正则表达式
 */
exports.EMAIL_REGEXP = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+\.)+[a-zA-Z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]{2,}))$/;
/**
 * 手机号的正则表达式
 */
exports.PHONE_REGEXP = /^1\d{10}$/;
/**
 * 登录时有效密码的正则表达式，兼容过去的密码格式，相对宽松
 */
exports.LOGIN_PASSWORD_REGEXP = /^\w{6,12}$/g;
/**
 * 注册时有效密码的正则表达式，相对严格
 */
exports.SIGNUP_PASSWORD_REGEXP = /^(?!\d+$)(?![A-Za-z]+$)[\dA-Za-z]{6,12}$/g;
/**
 * 销售订单号正则表达式
 * @description 数位至少5位，字符除字母和数字外只能包含 # 或者 -，最多不大于64位
 */
exports.ORDER_CODE_REGEXP = /^[\dA-Za-z#-]{5,64}$/g;
/**
 * 时间格式表达式
 * - 精度到月，举例：2022-08
 * - 精度到天，举例：2022-08-29
 * - 精度到天带周，举例：2022-08-29 (周一)
 * - 精确到分钟，举例：20222-08-29 00:00
 * - 精确到秒，举例：20222-08-29 00:00:00
 **/
/**
 * 精确到月的格式
 */
exports.TIME_MONTH_FORMAT = 'YYYY-MM';
/**
 * 精确到天的格式
 */
exports.TIME_DAY_FORMAT = 'YYYY-MM-DD';
/**
 * 精确到天带周的格式
 */
exports.TIME_DAY_WITH_WEEK_FORMAT = 'YYYY-MM-DD (ddd)';
/**
 * 精确到分钟的格式
 */
exports.TIME_MINUTE_FORMAT = 'YYYY-MM-DD HH:mm';
/**
 * 精确到秒的格式
 */
exports.TIME_SECOND_FORMAT = 'YYYY-MM-DD HH:mm:ss';
/**
 * 只有时分的格式
 */
exports.ONLY_TIME_MINUTE_FORMAT = 'HH:mm';
