// @spotter/app-client-toolkit/lib/const/platform.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SPOTTER_ROLES_CODE = exports.DEFAULT_TIMEZONE = exports.eventBus = exports.DEFAULT_TABLE_PAGE_SIZE = exports.CAPTCHA_DIGIT = exports.SUPPORT_FILE_SUFFIX_SET = exports.SUPPORT_IMAGE_SUFFIX_SET = exports.PLATFORM_FIELD_REGEXP = exports.EMPTY_SET = exports.DEFAULT_EMPTY_VALUE = void 0;
const mitt_1 = __importDefault(require("mitt"));
/********************* 平台常量 ****************************/
/**
 * 当数据为空时默认展示的值
 */
exports.DEFAULT_EMPTY_VALUE = '-';
/**
 * 空值的集合
 */
exports.EMPTY_SET = new Set(['', null, undefined]);
/**
 * 平台字段规格常量，数据属于该表的子集： https://d2vpiinh02.feishu.cn/wiki/wikcngt0VUxeaCOzVoZBb7CBGDh?table=tblP1RSZQsj6z434&view=vew2IwHmj4 【目前需要手动同步维护】
 */
exports.PLATFORM_FIELD_REGEXP = {
    // 示例：123456789101
    upc: {
        // 推荐宽度
        recommendedWidth: 108,
        // 合规正则表达式
        validRegexp: /^[\d]{12}$/g
    },
    // 示例： 810082445343
    ean: {
        // 推荐宽度
        recommendedWidth: 120,
        // 合规正则表达式
        validRegexp: /^[\d]{13}$/g
    },
    // 示例： B09JNQBX8Q
    asin: {
        recommendedWidth: 136
    },
    // 示例： 00005-0001-0002
    ssku: {
        recommendedWidth: 128
    },
    /**
     * VCPO物流单号
     */
    vcpoShipmentCode: {
        recommendedWidth: 188
    },
    /**
     * SPT物流单号
     */
    sptShipmentCode: {
        recommendedWidth: 108
    },
    // 示例： ZQUWK1
    companyCode: {
        recommendedWidth: 80
    },
    // 示例： 35141230452
    asn: {
        recommendedWidth: 100
    },
    // 示例： ZZ12001200/PS12001200
    outboundCode: {
        recommendedWidth: 96
    },
    // 示例： SPTAK01091101
    sptOrderCode: {
        recommendedWidth: 128
    },
    /**
     * SA订单号
     * @description ASIN+Ordered On+装箱数量+订单类型
     */
    saOrderCode: {
        recommendedWidth: 108
    },
    /**
     * 入库单号用SPT物流ID
     */
    inboundNo: { recommendedWidth: 100 },
    /**
     * 出库单号用ARN
     */
    outboundNo: { recommendedWidth: 100 },
    /**
     * 亚马逊采购订单号
     */
    amzOrderCode: { recommendedWidth: 116 },
    /**
     * 一期款款项编号
     */
    firstPaymentCode: { recommendedWidth: 148 },
    /**
     * 二期款款项编号
     */
    secondPaymentCode: { recommendedWidth: 148 }
};
/**
 * 平台默认支持的工单图片后缀
 */
exports.SUPPORT_IMAGE_SUFFIX_SET = new Set(['jpg', 'jpeg', 'bmp', 'png', 'gif']);
/**
 * 平台默认支持的工单文件后缀
 */
exports.SUPPORT_FILE_SUFFIX_SET = new Set([
    ...exports.SUPPORT_IMAGE_SUFFIX_SET,
    'txt',
    'zip',
    'rar',
    '7z',
    'doc',
    'docx',
    'pdf',
    'xlsx',
    'xls'
]);
/**
 * 平台验证码默认位数
 */
exports.CAPTCHA_DIGIT = 6;
/**
 * 平台默认的 table 单页行数
 */
exports.DEFAULT_TABLE_PAGE_SIZE = 20;
/**
 * 平台默认的事件总线对象
 */
exports.eventBus = (0, mitt_1.default)();
/**
 * 平台默认时区 和亚马逊保持一致
 */
exports.DEFAULT_TIMEZONE = 'America/Los_Angeles';
/**
 * spotter角色代码
 * @FYI https://d2vpiinh02.feishu.cn/wiki/wikcnD6kJOcUQrRwiZqklet8Bue
 */
var SPOTTER_ROLES_CODE;
(function (SPOTTER_ROLES_CODE) {
    /** 物流经理 */
    SPOTTER_ROLES_CODE["SPOTTER_SHIPMENT_MANAGER_CODE"] = "SPT-SHIPMENT";
    /**
     * 物流经理主管
     */
    SPOTTER_ROLES_CODE["SPOTTER_SHIPMENT_LOGISTICS_MANAGER"] = "logistics_manager";
    /**
     * 物流经理部门负责人
     */
    SPOTTER_ROLES_CODE["SPOTTER_SHIPMENT_LOGISTICS_LINE_LEADER"] = "logistics_lineleader";
    /** 运营经理 */
    SPOTTER_ROLES_CODE["BUSINESS_MANAGER_LEADER"] = "business_manager_leader";
    /** 运营专员 */
    SPOTTER_ROLES_CODE["BUSINESS_MANAGER"] = "business_manager";
    /** 账号经理 */
    SPOTTER_ROLES_CODE["ACCOUNT_MANAGER"] = "account_manager";
    /** 商务经理 */
    SPOTTER_ROLES_CODE["ACCOUNT_MANAGER_LEADER"] = "account_manager_leader";
    /** 超级管理员 */
    SPOTTER_ROLES_CODE["ADMIN"] = "admin";
    /** 财务经理 */
    SPOTTER_ROLES_CODE["FINANCIAL_MANAGER"] = "financial_assistant";
    /** 财务助理 */
    SPOTTER_ROLES_CODE["FINANCIAL_ASSISTANT"] = "finance_assistant_sub";
    /** 财务会计 */
    SPOTTER_ROLES_CODE["FINANCIAL_ACCOUNTING"] = "finance_accounting";
    /** SEVC财务 */
    SPOTTER_ROLES_CODE["FINANCIAL_SEVC"] = "sevc_finance";
    /** 出纳 */
    SPOTTER_ROLES_CODE["CASHIER"] = "cashier";
    // 站内信发送人
    SPOTTER_ROLES_CODE["MAIL_SENDER"] = "inbox_send_role";
    // 站内信审核人
    SPOTTER_ROLES_CODE["INBOX_TEMPLATE_AUDIT_ROLE_TEMP"] = "inbox_template_audit_role_temp";
    /**
     * 运营业务线负责人
     */
    SPOTTER_ROLES_CODE["BUSINESS_OWNER"] = "business_owner";
    /**
     * 财务物流处理人
     */
    SPOTTER_ROLES_CODE["FINANCE_WAREHOUSE_ASSISTANT"] = "financial_warehouse_assistant";
})(SPOTTER_ROLES_CODE = exports.SPOTTER_ROLES_CODE || (exports.SPOTTER_ROLES_CODE = {}));
