// @spotter/ui/es/components/EmbededRegister/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));

// src/components/EmbededRegister/index.tsx
import React from "react";
import { useEffect } from "react";
import { EmbeddedRenderer } from "./renderer";
import { BiEmbedded } from "./renderer/BiEmbedded";
var SUPPORT_EMBEDDED_TYPE = /* @__PURE__ */ new Set(["bi-embedded"]);
var parseData = (data) => {
  try {
    return __spreadProps(__spreadValues({}, data), {
      jsonData: JSON.parse(data == null ? void 0 : data.data)
    });
  } catch (error) {
    console.error(error);
    return null;
  }
};
var processMenus = (menus) => {
  const result = [];
  for (const menu of menus) {
    const parsedMenu = parseData(menu);
    if (parsedMenu) {
      result.push(parsedMenu);
    }
    if (menu == null ? void 0 : menu.children) {
      const parsedChildren = processMenus(menu.children);
      result.push(...parsedChildren);
    }
  }
  return result;
};
var useEmbeddedRoutes = (menus, spotterRouter) => {
  useEffect(() => {
    const processMenusData = processMenus(menus == null ? void 0 : menus.data);
    spotterRouter.addLayoutRoute({
      children: processMenusData.filter(
        (i) => SUPPORT_EMBEDDED_TYPE.has(i.jsonData.renderType)
      ).map((i) => {
        var _a;
        return {
          path: i.jsonData.url,
          i18nKey: i.jsonData.name,
          element: /* @__PURE__ */ React.createElement(
            EmbeddedRenderer,
            {
              key: i.jsonData.url,
              data: (_a = i.jsonData) != null ? _a : {},
              type: i.jsonData.biDatasource
            }
          )
        };
      })
    });
  }, [menus == null ? void 0 : menus.data]);
};
var BiEmbeddedRenderer = EmbeddedRenderer;
export {
  BiEmbedded,
  BiEmbeddedRenderer,
  useEmbeddedRoutes
};
