// @spotter/gmesh-api-sdk/lib/api/accounting-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAccountingApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AccountingApi - object-oriented interface
* @export
* @class AccountingApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAccountingApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 入库核算-金额合计
    * @summary 入库核算-金额合计
    * @param {AccountingApiInboundCountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    inboundCount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inbound/count', data: requestParameters }));
        });
    }
    /**
    * 入库核算-列表导出
    * @summary 入库核算-列表导出
    * @param {AccountingApiInboundExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    inboundExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inbound/export', data: requestParameters }));
        });
    }
    /**
    * 入库核算汇总-导出
    * @summary 入库核算汇总-导出
    * @param {AccountingApiInboundSumExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    inboundSumExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inboundSum/export', data: requestParameters }));
        });
    }
    /**
    * 入库核算汇总-列表查询
    * @summary 入库核算汇总-列表查询
    * @param {AccountingApiInboundSumPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    inboundSumPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inboundSum/queryPage', data: requestParameters }));
        });
    }
    /**
    * 库存核算-金额合计
    * @summary 库存核算-金额合计
    * @param {AccountingApiInventoryCountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    inventoryCount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inventory/count', data: requestParameters }));
        });
    }
    /**
    * 库存核算-列表导出
    * @summary 库存核算-列表导出
    * @param {AccountingApiInventoryExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    inventoryExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inventory/export', data: requestParameters }));
        });
    }
    /**
    * 库存价值变动-导出
    * @summary 库存价值变动-导出
    * @param {AccountingApiInventoryValueChangeExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    inventoryValueChangeExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inventory_value_change/export', data: requestParameters }));
        });
    }
    /**
    * 库存价值变动-分页
    * @summary 库存价值变动-分页
    * @param {AccountingApiInventoryValueChangePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    inventoryValueChangePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inventory_value_change/page', data: requestParameters }));
        });
    }
    /**
    * 出库核算-金额合计
    * @summary 出库核算-金额合计
    * @param {AccountingApiOutboundCountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    outboundCount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/outbound/count', data: requestParameters }));
        });
    }
    /**
    * 出库核算-导出
    * @summary 出库核算-导出
    * @param {AccountingApiOutboundExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    outboundExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/outbound/export', data: requestParameters }));
        });
    }
    /**
    * 出库核算-分页
    * @summary 出库核算-分页
    * @param {AccountingApiOutboundPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    outboundPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/outbound/page', data: requestParameters }));
        });
    }
    /**
    * 出库核算汇总-导出
    * @summary 出库核算汇总-导出
    * @param {AccountingApiOutboundSumExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    outboundSumExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/outboundSum/export', data: requestParameters }));
        });
    }
    /**
    * 出库核算汇总-列表查询
    * @summary 出库核算汇总-列表查询
    * @param {AccountingApiOutboundSumPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    outboundSumPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/outboundSum/queryPage', data: requestParameters }));
        });
    }
    /**
    * 盈亏核算-金额合计
    * @summary 盈亏核算-金额合计
    * @param {AccountingApiProfitLossCountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    profitLossCount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/profit_loss/count', data: requestParameters }));
        });
    }
    /**
    * 盈亏核算-导出
    * @summary 盈亏核算-导出
    * @param {AccountingApiProfitLossExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    profitLossExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/profit_loss/export', data: requestParameters }));
        });
    }
    /**
    * 盈亏核算-分页
    * @summary 盈亏核算-分页
    * @param {AccountingApiProfitLossPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    profitLossPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/profit_loss/page', data: requestParameters }));
        });
    }
    /**
    * 盈亏核算汇总-导出
    * @summary 盈亏核算汇总-导出
    * @param {AccountingApiProfitLossSumExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    profitLossSumExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/profit_loss_sum/export', data: requestParameters }));
        });
    }
    /**
    * 盈亏核算汇总-列表查询
    * @summary 盈亏核算汇总-列表查询
    * @param {AccountingApiProfitLossSumPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    profitLossSumPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/profit_loss_sum/queryPage', data: requestParameters }));
        });
    }
    /**
    * 进销存-导出
    * @summary 进销存-导出
    * @param {AccountingApiPurchaseSaleInventoryExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    purchaseSaleInventoryExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/purchase_sale_inventory/export', data: requestParameters }));
        });
    }
    /**
    * 进销存-分页
    * @summary 进销存-分页
    * @param {AccountingApiPurchaseSaleInventoryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    purchaseSaleInventoryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/purchase_sale_inventory/page', data: requestParameters }));
        });
    }
    /**
    * 入库核算-列表查询
    * @summary 入库核算-列表查询
    * @param {AccountingApiQueryInboundPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    queryInboundPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inbound/queryPage', data: requestParameters }));
        });
    }
    /**
    * 库存核算-列表查询
    * @summary 库存核算-列表查询
    * @param {AccountingApiQueryInventoryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingApi
    */
    queryInventoryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/accounting/inventory/queryPage', data: requestParameters }));
        });
    }
}
exports.SpotterAccountingApiSdk = SpotterAccountingApiSdk;
