// @spotter/gmesh-api-sdk/lib/api/accounting-voucher-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAccountingVoucherEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AccountingVoucherEndpointApi - object-oriented interface
* @export
* @class AccountingVoucherEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAccountingVoucherEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导出凭证
    * @summary 导出凭证
    * @param {AccountingVoucherEndpointApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingVoucherEndpointApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/accounting/voucher/export', data: requestParameters }));
        });
    }
    /**
    * 审核凭证
    * @summary 审核凭证
    * @param {AccountingVoucherEndpointApiAuditRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingVoucherEndpointApi
    */
    audit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/accounting/voucher/audit', data: requestParameters }));
        });
    }
    /**
    * 编辑凭证
    * @summary 编辑凭证
    * @param {AccountingVoucherEndpointApiEditRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingVoucherEndpointApi
    */
    edit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/accounting/voucher/item/edit', data: requestParameters }));
        });
    }
    /**
    * 编辑凭证日期
    * @summary 编辑凭证日期
    * @param {AccountingVoucherEndpointApiEditVoucherDateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingVoucherEndpointApi
    */
    editVoucherDate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/accounting/voucher/editVoucherDate', data: requestParameters }));
        });
    }
    /**
    * 导出金蝶凭证
    * @summary 导出金蝶凭证
    * @param {AccountingVoucherEndpointApiExportKingdeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingVoucherEndpointApi
    */
    exportKingdee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/accounting/voucher/kingdee/export', data: requestParameters }));
        });
    }
    /**
    * 获取借贷不平衡的凭证
    * @summary 获取借贷不平衡的凭证
    * @param {*} [options] Override http request option.
    * @memberof AccountingVoucherEndpointApi
    */
    getNotBalance(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-evidence-web/accounting/voucher/notBalance' }));
        });
    }
    /**
    * 获取会计科目
    * @summary 获取会计科目
    * @param {*} [options] Override http request option.
    * @memberof AccountingVoucherEndpointApi
    */
    getSubject(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-evidence-web/accounting/voucher/getSubject' }));
        });
    }
    /**
    * 分页查询凭证
    * @summary 分页查询凭证
    * @param {AccountingVoucherEndpointApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountingVoucherEndpointApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/accounting/voucher/queryPage', data: requestParameters }));
        });
    }
}
exports.SpotterAccountingVoucherEndpointApiSdk = SpotterAccountingVoucherEndpointApiSdk;
