// @spotter/gmesh-api-sdk/lib/api/account-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAccountApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AccountApi - object-oriented interface
* @export
* @class AccountApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAccountApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 获取所有运营账号信息
    * @summary 获取所有运营账号信息
    * @param {*} [options] Override http request option.
    * @memberof AccountApi
    */
    all(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/amazon/account/all' }));
        });
    }
    /**
    * 模糊查询当前account code
    * @summary 模糊查询当前account code
    * @param {AccountApiListAccountCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountApi
    */
    listAccountCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/list/code', data: requestParameters }));
        });
    }
    /**
    * 模糊查询registeredBusinessName
    * @summary 模糊查询registeredBusinessName
    * @param {AccountApiListRegisteredBusinessNameRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountApi
    */
    listRegisteredBusinessName(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/list/registered/business/name', data: requestParameters }));
        });
    }
    /**
    * 根据id更新账号字段
    * @summary 根据id更新账号字段
    * @param {AccountApiModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountApi
    */
    modify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/modify', data: requestParameters }));
        });
    }
    /**
    * 根据条件查询亚马逊运营账号（分页）
    * @summary 根据条件查询亚马逊运营账号（分页）
    * @param {AccountApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/page', data: requestParameters }));
        });
    }
    /**
    * 插入一个亚马逊账号
    * @summary 插入一个亚马逊账号
    * @param {AccountApiSaveRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AccountApi
    */
    save(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/save', data: requestParameters }));
        });
    }
}
exports.SpotterAccountApiSdk = SpotterAccountApiSdk;
