// @spotter/gmesh-api-sdk/lib/api/address-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAddressApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AddressApi - object-oriented interface
* @export
* @class AddressApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAddressApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 从谷歌获取地址预测信息
    * @summary 从谷歌获取地址预测信息
    * @param {AddressApiGetPredictionAddressRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AddressApi
    */
    getPredictionAddress(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/getPredictionAddress', data: requestParameters }));
        });
    }
    /**
    * 收货地址列表
    * @summary 收货地址列表
    * @param {AddressApiPageAddressListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AddressApi
    */
    pageAddressList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/list', data: requestParameters }));
        });
    }
    /**
    * 新增地址信息
    * @summary 新增地址信息
    * @param {AddressApiSaveAddressInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AddressApi
    */
    saveAddressInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/save', data: requestParameters }));
        });
    }
    /**
    * 更新地址信息
    * @summary 更新地址信息
    * @param {AddressApiUpdateAddressInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AddressApi
    */
    updateAddressInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/update', data: requestParameters }));
        });
    }
    /**
    * 修改地址状态，启用停用
    * @summary 修改地址状态，启用停用
    * @param {AddressApiUpdateAddressStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AddressApi
    */
    updateAddressStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/updateAddressStatus', data: requestParameters }));
        });
    }
}
exports.SpotterAddressApiSdk = SpotterAddressApiSdk;
