// @spotter/gmesh-api-sdk/lib/api/adjust-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAdjustApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AdjustApi - object-oriented interface
* @export
* @class AdjustApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAdjustApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 计算保护价格
    * @summary 计算保护价格
    * @param {AdjustApiCalProtectPriceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    calProtectPrice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/calProtectPrice', data: requestParameters }));
        });
    }
    /**
    * 取消调价
    * @summary 取消调价
    * @param {AdjustApiCancelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    cancel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/cancle', data: requestParameters }));
        });
    }
    /**
    * 审核页面权限校验
    * @summary 审核页面权限校验
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    checkPermit(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-items-web/offer/adjust/checkPermit' }));
        });
    }
    /**
    * 创建调价，取消冲突调价
    * @summary 创建新调价，取消冲突调价
    * @param {AdjustApiCoverAdjustRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    coverAdjust(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/coverAdjust', data: requestParameters }));
        });
    }
    /**
    * 创建调价
    * @summary 创建调价
    * @param {AdjustApiCreateAdjustRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    createAdjust(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/createAdjust', data: requestParameters }));
        });
    }
    /**
    * 调价详情查询
    * @summary 调价详情查询
    * @param {AdjustApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.adjustId) !== undefined) {
                params['adjustId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.adjustId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-items-web/offer/adjust/detail', params: requestParameters }));
        });
    }
    /**
    * 模糊检索ssku
    * @summary 模糊检索ssku
    * @param {AdjustApiFuzzySSKURequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    fuzzySSKU(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.fuzzySSKU) !== undefined) {
                params['fuzzySSKU'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.fuzzySSKU;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-items-web/offer/adjust/querySSKU', params: requestParameters }));
        });
    }
    /**
    * 中断调价
    * @summary 中断调价
    * @param {AdjustApiInterruptAdjustRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    interruptAdjust(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/interrupt/adjust', data: requestParameters }));
        });
    }
    /**
    * 查询部门各个业务线负责人
    * @summary 查询部门各个业务线负责人
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    listBusinessOwner(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-items-web/offer/adjust/listBusinessOwner' }));
        });
    }
    /**
    * 发起人列表
    * @summary 发起人列表
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    listSponsor(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-items-web/offer/adjust/listSponsor' }));
        });
    }
    /**
    * 调价状态列表
    * @summary 调价状态列表
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    listStatus(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-items-web/offer/adjust/listStatus' }));
        });
    }
    /**
    * 根据ssku查询vendorCode
    * @summary 根据ssku查询vendorCode
    * @param {AdjustApiListVendorRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    listVendor(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku) !== undefined) {
                params['ssku'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-items-web/offer/adjust/listVendor', params: requestParameters }));
        });
    }
    /**
    * 调价列表查询
    * @summary 调价列表分页查询
    * @param {AdjustApiPageAdjustRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    pageAdjust(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/listAdjust', data: requestParameters }));
        });
    }
    /**
    * 待审核列表查询
    * @summary 待审核列表查询
    * @param {AdjustApiPageReviewsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    pageReviews(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/listReviews', data: requestParameters }));
        });
    }
    /**
    * 添加备注
    * @summary 添加备注
    * @param {AdjustApiRemarkRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    remark(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/remark', data: requestParameters }));
        });
    }
    /**
    * 调价审核
    * @summary 调价审核
    * @param {AdjustApiReviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AdjustApi
    */
    review(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/offer/adjust/review', data: requestParameters }));
        });
    }
}
exports.SpotterAdjustApiSdk = SpotterAdjustApiSdk;
