// @spotter/gmesh-api-sdk/lib/api/agreement-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAgreementApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AgreementApi - object-oriented interface
* @export
* @class AgreementApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAgreementApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 协议概览（gmesh）
    * @summary 协议概览（gmesh）
    * @param {AgreementApiAgreementSummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    agreementSummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/gmesh/summary', data: requestParameters }));
        });
    }
    /**
    * VC协议审核-已审核报价协议列表
    * @summary VC协议审核-已审核报价协议列表
    * @param {AgreementApiAuditFinishRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    auditFinish(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/finish', data: requestParameters }));
        });
    }
    /**
    * VC协议审核-待客户确认报价协议列表
    * @summary VC协议审核-待客户确认报价协议列表
    * @param {AgreementApiAuditToBeConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    auditToBeConfirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/toBeConfirm', data: requestParameters }));
        });
    }
    /**
    * 获取待办统计
    * @summary VC协议审核-当前登录人获取待办统计
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    auditTodoStatistics(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/todo/statistics' }));
        });
    }
    /**
    * 计算保护价格（gmesh）
    * @summary 计算保护价格（gmesh）
    * @param {AgreementApiGmeshProtectPriceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    gmeshProtectPrice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/gmesh/summary/protect/price', data: requestParameters }));
        });
    }
    /**
    * VC协议审核-获取待调价VC协议列表
    * @summary VC协议审核-获取待调价VC协议列表
    * @param {AgreementApiMsrpAdjustTodosRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    msrpAdjustTodos(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/msrp/adjust/todo', data: requestParameters }));
        });
    }
    /**
    * 获取待审核vc报价列表(一二三级审核)
    * @summary VC协议审核-获取待审核VC报价列表
    * @param {AgreementApiOperatingAuditTodosRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    operatingAuditTodos(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/todo', data: requestParameters }));
        });
    }
    /**
    * VC报价审核不通过(一二三级)
    * @summary VC协议审核-VC报价审核不通过
    * @param {AgreementApiOperatingRejectRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    operatingReject(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/reject', data: requestParameters }));
        });
    }
    /**
    * 产品关联报价协议列表
    * @summary 产品关联报价协议列表
    * @param {AgreementApiProductSummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    productSummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/product/summary', data: requestParameters }));
        });
    }
    /**
    * VC协议审核-修改是否需要提供有效期
    * @summary VC协议审核-修改是否需要提供有效期
    * @param {AgreementApiUpdateNeedBatchNoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    updateNeedBatchNo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/update/needBatchNo', data: requestParameters }));
        });
    }
    /**
    * VC协议审核-亚马逊实际报价计算,不存储
    * @summary VC协议审核-亚马逊实际报价计算,不存储
    * @param {AgreementApiVcActualQuotationCalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcActualQuotationCal(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/amzon/actual/offer/cal', data: requestParameters }));
        });
    }
    /**
    * VC协议审核-亚马逊实际报价计算并更新结果
    * @summary VC协议审核-亚马逊实际报价计算并更新结果
    * @param {AgreementApiVcActualQuotationCalSaveRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcActualQuotationCalSave(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/amzon/actual/offer/cal/save', data: requestParameters }));
        });
    }
    /**
    * VC报价协议详情
    * @summary VC报价协议详情
    * @param {AgreementApiVcAgreementDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcAgreementDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementCode) !== undefined) {
                params['agreementCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/agreement/vc/detail', params: requestParameters }));
        });
    }
    /**
    * VC报价审核详情
    * @summary VC协议审核-VC报价审核详情
    * @param {AgreementApiVcAuditDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcAuditDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/agreement/vc/audit/detail', params: requestParameters }));
        });
    }
    /**
    * VC报价一级审核通过
    * @summary VC协议审核-一级审核通过
    * @param {AgreementApiVcFirstApprovalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcFirstApproval(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/first/pass', data: requestParameters }));
        });
    }
    /**
    * VC报价一级审核批量通过
    * @summary VC协议审核-一级审核批量通过
    * @param {AgreementApiVcFirstBatchApprovalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcFirstBatchApproval(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/first/batch/pass', data: requestParameters }));
        });
    }
    /**
    * VC协议审核-批量拒绝（1，2，3级审核）
    * @summary VC协议审核-批量拒绝
    * @param {AgreementApiVcFirstBatchRejectRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcFirstBatchReject(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/batch/reject', data: requestParameters }));
        });
    }
    /**
    * 重新指定任务处理人
    * @summary VC协议审核-重新指定任务处理人
    * @param {AgreementApiVcResetAssigneeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcResetAssignee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/reset/assignee', data: requestParameters }));
        });
    }
    /**
    * 重新指定任务处理人(单个)
    * @summary VC协议审核-重新指定任务处理人(单个)
    * @param {AgreementApiVcResetSingleAssigneeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcResetSingleAssignee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/reset/single/assignee', data: requestParameters }));
        });
    }
    /**
    * VC报价二级审核通过
    * @summary VC协议审核-二级审核通过
    * @param {AgreementApiVcSecondApprovalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcSecondApproval(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/second/pass', data: requestParameters }));
        });
    }
    /**
    * VC报价二级审核批量通过
    * @summary VC协议审核-二级审核批量通过
    * @param {AgreementApiVcSecondBatchApprovalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcSecondBatchApproval(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/second/batch/pass', data: requestParameters }));
        });
    }
    /**
    * VC协议审核-Spotter VC报价计算并更新结果
    * @summary VC协议审核-Spotter VC报价计算并更新结果
    * @param {AgreementApiVcSpotterOfferCalSaveRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcSpotterOfferCalSave(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/spotter/offer/cal/save', data: requestParameters }));
        });
    }
    /**
    * VC报价三级审核通过
    * @summary VC协议审核-三级审核通过
    * @param {AgreementApiVcThirdApprovalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcThirdApproval(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/third/pass', data: requestParameters }));
        });
    }
    /**
    * VC报价三级审核批量通过
    * @summary VC协议审核-三级审核批量通过
    * @param {AgreementApiVcThirdBatchApprovalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AgreementApi
    */
    vcThirdBatchApproval(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/vc/audit/third/batch/pass', data: requestParameters }));
        });
    }
}
exports.SpotterAgreementApiSdk = SpotterAgreementApiSdk;
