// @spotter/gmesh-api-sdk/lib/api/amazon-return-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAmazonReturnApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AmazonReturnApi - object-oriented interface
* @export
* @class AmazonReturnApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAmazonReturnApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 退货明细导出为Excel
    * @summary 退货明细导出为Excel
    * @param {AmazonReturnApiExportReturnDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    exportReturnDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/excel/export', data: requestParameters }));
        });
    }
    /**
    * 查询退货明细基础信息
    * @summary 查询退货明细基础信息
    * @param {AmazonReturnApiGetReturnDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    getReturnDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/details', data: requestParameters }));
        });
    }
    /**
    * 查询退货数据的金额总和
    * @summary 查询退货数据的金额总和(随条件联动)
    * @param {AmazonReturnApiGetReturnDetailTotalAmountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    getReturnDetailTotalAmount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/total_amount', data: requestParameters }));
        });
    }
    /**
    * 获取没有协议价的数量
    * @summary 获取没有协议价的数量
    * @param {AmazonReturnApiNoSptPriceCountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    noSptPriceCount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/no_spt_price/count', data: requestParameters }));
        });
    }
    /**
    * 导出没有协议价的数据excel表格
    * @summary 导出没有协议价的数据excel表格
    * @param {AmazonReturnApiNoSptPriceExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    noSptPriceExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/no_spt_price/export', data: requestParameters }));
        });
    }
    /**
    * 上传excel补充协议价数据
    * @summary 上传excel补充协议价数据
    * @param {AmazonReturnApiNoSptPriceImportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    noSptPriceImport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/no_spt_price/import', data: requestParameters }));
        });
    }
    /**
    * 查询未结算退货详情
    * @summary 查询未结算退货详情
    * @param {AmazonReturnApiNonSettlementDetailsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    nonSettlementDetails(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/non/settlement/details', data: requestParameters }));
        });
    }
    /**
    * 查询指定公司最后一次结算的退货费用拉取时间
    * @summary 查询指定公司最后一次结算的退货费用拉取时间
    * @param {AmazonReturnApiQueryLastPaymentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    queryLastPayment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/amazon/return/settlement/last_settle_info', params: requestParameters }));
        });
    }
    /**
    * 退货发起结算
    * @summary 退货发起结算
    * @param {AmazonReturnApiSettlementCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    settlementCreate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/settlement/create', data: requestParameters }));
        });
    }
    /**
    * 退货结算单撤销
    * @summary 退货结算单撤销
    * @param {AmazonReturnApiSettlementDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    settlementDelete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/settlement/delete', data: requestParameters }));
        });
    }
    /**
    * 退货结算单详情(退货明细分页)
    * @summary 退货结算单详情(退货明细分页)
    * @param {AmazonReturnApiSettlementDetailPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    settlementDetailPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/settlement/detail_page', data: requestParameters }));
        });
    }
    /**
    * 退货结算单导出
    * @summary 退货结算单导出
    * @param {AmazonReturnApiSettlementExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    settlementExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/settlement/export', data: requestParameters }));
        });
    }
    /**
    * 退货结算单分页查询
    * @summary 退货结算单分页查询
    * @param {AmazonReturnApiSettlementPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    settlementPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/settlement/page', data: requestParameters }));
        });
    }
    /**
    * 退货结算预览
    * @summary 退货结算预览
    * @param {AmazonReturnApiSettlementPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazonReturnApi
    */
    settlementPreview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/settlement/preview', data: requestParameters }));
        });
    }
    /**
    * 人工补全退货信息
    * @summary 人工补全退货信息
    * @param {AmazonReturnApiSupplementAmzReturnDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof AmazonReturnApi
    */
    supplementAmzReturnDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/detail/update', data: requestParameters }));
        });
    }
}
exports.SpotterAmazonReturnApiSdk = SpotterAmazonReturnApiSdk;
