// @spotter/gmesh-api-sdk/lib/api/amazing-auth-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAmazingAuthApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AmazingAuthApi - object-oriented interface
* @export
* @class AmazingAuthApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAmazingAuthApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 亚马逊授权回调url
    * @summary 亚马逊授权回调url
    * @param {AmazingAuthApiCallbackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazingAuthApi
    */
    callback(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.state) !== undefined) {
                params['state'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.state;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.code) !== undefined) {
                params['code'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.code;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ads-web/amazing/auth/callback', params: requestParameters }));
        });
    }
    /**
    * 亚马逊授权url
    * @summary 亚马逊授权url
    * @param {AmazingAuthApiGetAuthUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmazingAuthApi
    */
    getAuthUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountName) !== undefined) {
                params['accountName'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountName;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.country) !== undefined) {
                params['country'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.country;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ads-web/amazing/auth/url', params: requestParameters }));
        });
    }
    /**
    * 亚马逊授权url
    * @summary 亚马逊授权url
    * @param {*} [options] Override http request option.
    * @memberof AmazingAuthApi
    */
    getAuthUrlForTest(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ads-web/amazing/auth/url/test' }));
        });
    }
}
exports.SpotterAmazingAuthApiSdk = SpotterAmazingAuthApiSdk;
