// @spotter/gmesh-api-sdk/lib/api/amazon-shortage-missed-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAmazonShortageMissedApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AmazonShortageMissedApi - object-oriented interface
* @export
* @class AmazonShortageMissedApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAmazonShortageMissedApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 丢（缺）货明细导出为Excel
    * @summary 丢（缺）货明细导出为Excel
    * @param {AmazonShortageMissedApiExportShortageMissedDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof AmazonShortageMissedApi
    */
    exportShortageMissedDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/shortage/missed/excel/export', data: requestParameters }));
        });
    }
    /**
    * 查询丢缺货明细基础信息
    * @summary 查询丢缺货明细基础信息
    * @param {AmazonShortageMissedApiGetReturnDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof AmazonShortageMissedApi
    */
    getReturnDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/shortage/missed/details', data: requestParameters }));
        });
    }
    /**
    * 丢（缺）货明细Excel导入
    * @summary 丢（缺）货明细Excel导入
    * @param {AmazonShortageMissedApiImportShortageMissedDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof AmazonShortageMissedApi
    */
    importShortageMissedDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/shortage/missed/excel/import', data: requestParameters }));
        });
    }
    /**
    * 查询未结算丢缺货详情s
    * @summary 查询未结算丢缺货详情s
    * @param {AmazonShortageMissedApiNonSettlementDetailsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof AmazonShortageMissedApi
    */
    nonSettlementDetails(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/shortage/missed/non/settlement/details', data: requestParameters }));
        });
    }
    /**
    * 人工补全丢（缺）货详情
    * @summary 人工补全丢（缺）货详情
    * @param {AmazonShortageMissedApiSupplementAmzShortageMissedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof AmazonShortageMissedApi
    */
    supplementAmzShortageMissed(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/shortage/missed/detail', data: requestParameters }));
        });
    }
}
exports.SpotterAmazonShortageMissedApiSdk = SpotterAmazonShortageMissedApiSdk;
