// @spotter/gmesh-api-sdk/lib/api/amz-order-gmesh-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAmzOrderGmeshApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AmzOrderGmeshApi - object-oriented interface
* @export
* @class AmzOrderGmeshApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAmzOrderGmeshApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导出数据的下单时间区间不超过30天
    * @summary excel导出
    * @param {AmzOrderGmeshApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/export', data: requestParameters }));
        });
    }
    /**
    * 亚马逊BTR订单导出
    * @summary 亚马逊BTR订单导出
    * @param {AmzOrderGmeshApiBtrExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    btrExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/btr/export', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df订单取消
    * @summary 亚马逊df订单取消
    * @param {AmzOrderGmeshApiCancelOrderAmzDfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    cancelOrderAmzDf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/cancel/df', data: requestParameters }));
        });
    }
    /**
    * 提交所有待处理或处理中的订单
    * @summary 手动提交订单
    * @param {AmzOrderGmeshApiConfirmOrderRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    confirmOrder(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/manual/confirm', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df订单修改实际接单数量
    * @summary 亚马逊df订单修改实际接单数量
    * @param {AmzOrderGmeshApiConfirmOrderAmzDfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    confirmOrderAmzDf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/confirm/df', data: requestParameters }));
        });
    }
    /**
    * 待发货页面修改订单仓库
    * @summary 修改仓库
    * @param {AmzOrderGmeshApiEditStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof AmzOrderGmeshApi
    */
    editStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/storage/edit', data: requestParameters }));
        });
    }
    /**
    * DF订单缺货列表导出
    * @summary DF订单缺货列表导出
    * @param {AmzOrderGmeshApiExportDfLackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    exportDfLack(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/list/df/lack/export', data: requestParameters }));
        });
    }
    /**
    * 亚马逊PO订单待发货列表导出
    * @summary 待发货列表导出
    * @param {AmzOrderGmeshApiExportPoUnshippedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    exportPoUnshipped(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/export/unshipped', data: requestParameters }));
        });
    }
    /**
    * 亚马逊PO订单详情
    * @summary 亚马逊PO订单详情
    * @param {AmzOrderGmeshApiGetPoDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    getPoDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/po/detail', data: requestParameters }));
        });
    }
    /**
    * 亚马逊订单统计
    * @summary 亚马逊订单统计
    * @param {AmzOrderGmeshApiGetStatisticsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    getStatistics(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/getStatistics', data: requestParameters }));
        });
    }
    /**
    * 初始化shipment
    * @summary 初始化shipment
    * @param {AmzOrderGmeshApiInitShipmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    initShipment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/initShipment', data: requestParameters }));
        });
    }
    /**
    * 查询亚马逊订单快照列表
    * @summary 查询亚马逊订单快照列表
    * @param {AmzOrderGmeshApiListAmzItemSnapshotRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    listAmzItemSnapshot(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/itemSnapshot/list', data: requestParameters }));
        });
    }
    /**
    * DF订单缺货列表
    * @summary DF订单缺货列表
    * @param {AmzOrderGmeshApiListDfLackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    listDfLack(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/list/df/lack', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 提前提交订单列表
    * @param {AmzOrderGmeshApiListPoConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    listPoConfirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/list/po/confirm', data: requestParameters }));
        });
    }
    /**
    * 查询状态为Pending且接单数量为0的订单
    * @summary 亚马逊PO订单修改产品接单状态列表
    * @param {AmzOrderGmeshApiListPoModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    listPoModify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/list/po/modify', data: requestParameters }));
        });
    }
    /**
    * 查询已确认且接单数量大于0的PO订单
    * @summary 亚马逊PO订单待发货列表
    * @param {AmzOrderGmeshApiListPoUnshippedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    listPoUnshipped(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/list/po/unshipped', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 标记手动发货PO订单
    * @param {AmzOrderGmeshApiMarkShipRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    markShip(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/markShip', data: requestParameters }));
        });
    }
    /**
    * 修改亚马逊采购单
    * @summary 修改亚马逊采购单
    * @param {AmzOrderGmeshApiModifyAmzItemRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    modifyAmzItem(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/amzItem/modify', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 修改po订单接单状态
    * @param {AmzOrderGmeshApiModifyPoReceiveStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    modifyPoReceiveStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/modify/po/receiveStatus', data: requestParameters }));
        });
    }
    /**
    * 亚马逊BBO询盘订单分页
    * @summary 亚马逊BBO询盘订单分页
    * @param {AmzOrderGmeshApiPageBboOrderRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    pageBboOrder(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/pageOrder/bbo', data: requestParameters }));
        });
    }
    /**
    * 亚马逊po订单分页
    * @summary 亚马逊po订单分页
    * @param {AmzOrderGmeshApiPageOrderRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    pageOrder(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/pageOrder', data: requestParameters }));
        });
    }
    /**
    * 查询已确认且接单数量大于0的PO订单
    * @summary 亚马逊PO订单待发货分页
    * @param {AmzOrderGmeshApiPagePoUnshippedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    pagePoUnshipped(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/page/po/unshipped', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 手动同步bbo订单
    * @param {AmzOrderGmeshApiSyncOrderBboRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    syncOrderBbo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/sync/bbo', data: requestParameters }));
        });
    }
    /**
    * 订单库存解冻
    * @summary 订单库存解冻
    * @param {AmzOrderGmeshApiUnfreezePoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    unfreezePo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/unfreeze/po', data: requestParameters }));
        });
    }
    /**
    * 批量修改BBO装箱数量
    * @summary 修改BBO装箱数量
    * @param {AmzOrderGmeshApiUpdateBboInnerPackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    updateBboInnerPack(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/update/bbo/innerPack', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 修改frd
    * @param {AmzOrderGmeshApiUpdateFrdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    updateFrd(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/updateFrd', data: requestParameters }));
        });
    }
    /**
    * 修改装箱数量
    * @summary 修改装箱数量
    * @param {AmzOrderGmeshApiUpdateInnerPackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderGmeshApi
    */
    updateInnerPack(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/gmesh/updateInnerPack', data: requestParameters }));
        });
    }
}
exports.SpotterAmzOrderGmeshApiSdk = SpotterAmzOrderGmeshApiSdk;
