// @spotter/gmesh-api-sdk/lib/api/amz-order-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAmzOrderApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AmzOrderApi - object-oriented interface
* @export
* @class AmzOrderApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAmzOrderApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 亚马逊BBO订单统计
    * @summary 亚马逊BBO订单统计
    * @param {AmzOrderApiBboStatisticsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    bboStatistics(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/statistics/bbo', data: requestParameters }));
        });
    }
    /**
    * 修改bbo接单数量
    * @summary 修改bbo接单数量
    * @param {AmzOrderApiConfirmBboRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    confirmBbo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/bbo/confirm', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df统计
    * @summary 亚马逊df统计
    * @param {AmzOrderApiDfStatisticRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    dfStatistic(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/statistics/df', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df可售数量导出
    * @summary 亚马逊df可售数量导出
    * @param {AmzOrderApiExportDfInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    exportDfInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/export/df/inventory', data: requestParameters }));
        });
    }
    /**
    * 亚马逊DF订单列表导出
    * @summary 亚马逊DF订单列表导出
    * @param {AmzOrderApiExportDfOrderRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    exportDfOrder(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/export/df', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df待确认订单列表
    * @summary 亚马逊df待确认订单列表
    * @param {AmzOrderApiListDfNewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    listDfNew(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/list/df/new', data: requestParameters }));
        });
    }
    /**
    * 页面未处理信息提示
    * @summary 未处理信息
    * @param {AmzOrderApiListInfoTipsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    listInfoTips(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/info', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df可售数量修改
    * @summary 亚马逊df可售数量修改
    * @param {AmzOrderApiModifyDfInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    modifyDfInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/modify/df/inventory', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df订单分页
    * @summary 亚马逊df订单分页
    * @param {AmzOrderApiPageDfAllRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    pageDfAll(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/page/df', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df可售数量分页
    * @summary 亚马逊df可售数量分页
    * @param {AmzOrderApiPageDfInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    pageDfInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/page/df/inventory', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df可售数量修改分页
    * @summary 亚马逊df可售数量修改分页
    * @param {AmzOrderApiPageDfInventoryModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AmzOrderApi
    */
    pageDfInventoryModify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/page/df/inventory/modify', data: requestParameters }));
        });
    }
}
exports.SpotterAmzOrderApiSdk = SpotterAmzOrderApiSdk;
