// @spotter/gmesh-api-sdk/lib/api/auth-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAuthApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* AuthApi - object-oriented interface
* @export
* @class AuthApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterAuthApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 获取企业认证授权链接
    * @summary 获取企业认证授权链接
    * @param {*} [options] Override http request option.
    * @memberof AuthApi
    */
    getOrgAuthUrl(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getOrgAuthUrl' }));
        });
    }
    /**
    * 查询企业授权信息
    * @summary 查询企业授权信息
    * @param {*} [options] Override http request option.
    * @memberof AuthApi
    */
    getOrgAuthorizedInfo(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getOrgAuthorizedInfo' }));
        });
    }
    /**
    * 查询企业认证信息
    * @summary 查询企业认证信息
    * @param {*} [options] Override http request option.
    * @memberof AuthApi
    */
    getOrgIdentityInfo(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getOrgIdentityInfo' }));
        });
    }
    /**
    * 获取个人认证授权链接
    * @summary 获取个人认证授权链接
    * @param {AuthApiGetPsnAuthUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AuthApi
    */
    getPsnAuthUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getPsnAuthUrl', data: requestParameters }));
        });
    }
    /**
    * 查询个人授权信息
    * @summary 查询个人授权信息
    * @param {AuthApiGetPsnAuthorizedInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AuthApi
    */
    getPsnAuthorizedInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getPsnAuthorizedInfo', data: requestParameters }));
        });
    }
    /**
    * 查询个人认证信息
    * @summary 查询个人认证信息
    * @param {AuthApiGetPsnIdentityInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof AuthApi
    */
    getPsnIdentityInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getPsnIdentityInfo', data: requestParameters }));
        });
    }
}
exports.SpotterAuthApiSdk = SpotterAuthApiSdk;
