// @spotter/gmesh-api-sdk/lib/api/bind-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterBindApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* BindApi - object-oriented interface
* @export
* @class BindApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterBindApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 新建关联关系
    * @summary 新建关联关系
    * @param {BindApiCreateBindRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BindApi
    */
    createBind(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/bind', data: requestParameters }));
        });
    }
    /**
    * 删除关联关系
    * @summary 删除关联关系
    * @param {BindApiDelBindRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BindApi
    */
    delBind(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/bind', params: requestParameters }));
        });
    }
    /**
    * 关联关系日志记录
    * @summary 关联关系日志记录
    * @param {BindApiGetLogsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BindApi
    */
    getLogs(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/bind/log', data: requestParameters }));
        });
    }
    /**
    * 关联供应商分页
    * @summary 关联供应商分页
    * @param {BindApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BindApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/bind/page', data: requestParameters }));
        });
    }
}
exports.SpotterBindApiSdk = SpotterBindApiSdk;
