// @spotter/gmesh-api-sdk/lib/api/brand-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterBrandApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* BrandApi - object-oriented interface
* @export
* @class BrandApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterBrandApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 查询指定品牌（含历史销售额）
    * @summary 查询指定品牌信息（含历史销售额）
    * @param {BrandApiBrandSalesSearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BrandApi
    */
    brandSalesSearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId) !== undefined) {
                params['brandId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/brand/company/search/with/sales', params: requestParameters }));
        });
    }
    /**
    * 查询指定公司的品牌信息
    * @summary 查询指定公司的品牌信息
    * @param {BrandApiCompanySearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BrandApi
    */
    companySearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/brand/company/search', params: requestParameters }));
        });
    }
    /**
    * 查询指定公司品牌概览数据
    * @summary 查询指定公司品牌概览数据
    * @param {BrandApiCompanySummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BrandApi
    */
    companySummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/brand/company/summary', params: requestParameters }));
        });
    }
    /**
    * 条件搜索品牌列表
    * @summary 条件搜索品牌列表
    * @param {BrandApiGmeshSearchBrandListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BrandApi
    */
    gmeshSearchBrandList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/brand/gmesh/list', data: requestParameters }));
        });
    }
}
exports.SpotterBrandApiSdk = SpotterBrandApiSdk;
