// @spotter/gmesh-api-sdk/lib/api/brand-auth-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterBrandAuthApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* BrandAuthApi - object-oriented interface
* @export
* @class BrandAuthApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterBrandAuthApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 审核品牌授权书
    * @summary 审核品牌授权书
    * @param {BrandAuthApiGmeshAuditBrandAuthRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BrandAuthApi
    */
    gmeshAuditBrandAuth(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/brand/auth/gmesh/audit', data: requestParameters }));
        });
    }
    /**
    * 查询品牌授权书列表
    * @summary 查询品牌授权书列表
    * @param {BrandAuthApiGmeshFindBrandAuthPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BrandAuthApi
    */
    gmeshFindBrandAuthPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/brand/auth/gmesh/page', data: requestParameters }));
        });
    }
    /**
    * 查询超时未上传品牌授权书
    * @summary 查询超时未上传品牌授权书
    * @param {*} [options] Override http request option.
    * @memberof BrandAuthApi
    */
    gmeshFindBrandAuthUploadTimeout(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/brand/auth/gmesh/upload/timeout' }));
        });
    }
    /**
    * 查询品牌及授权书详情
    * @summary 查询品牌及授权书详情
    * @param {BrandAuthApiGmeshFindBrandByIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof BrandAuthApi
    */
    gmeshFindBrandById(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId) !== undefined) {
                params['brandId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/brand/auth/gmesh/detail', params: requestParameters }));
        });
    }
}
exports.SpotterBrandAuthApiSdk = SpotterBrandAuthApiSdk;
