// @spotter/gmesh-api-sdk/lib/api/category-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterCategoryApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* CategoryApi - object-oriented interface
* @export
* @class CategoryApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterCategoryApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * OD邮箱类型层级结构
    * @summary OD邮箱类型层级结构
    * @param {CategoryApiCategoryFindByLevelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CategoryApi
    */
    categoryFindByLevel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.name) !== undefined) {
                params['name'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.name;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/category/findByLevel', params: requestParameters }));
        });
    }
    /**
    * OD邮箱类型（树结构）
    * @summary OD邮箱类型(树结构)
    * @param {CategoryApiCategoryTreeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CategoryApi
    */
    categoryTree(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.name) !== undefined) {
                params['name'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.name;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/category/tree', params: requestParameters }));
        });
    }
    /**
    * 获取所有一级类目信息
    * @summary 获取所有一级类目信息
    * @param {*} [options] Override http request option.
    * @memberof CategoryApi
    */
    listAllPrimaryCategory(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/category/first/list/all' }));
        });
    }
    /**
    * 查询类目信息
    * @summary 查询类目信息
    * @param {CategoryApiSearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CategoryApi
    */
    search(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/category/search', data: requestParameters }));
        });
    }
}
exports.SpotterCategoryApiSdk = SpotterCategoryApiSdk;
