// @spotter/gmesh-api-sdk/lib/api/cccc-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterCcccApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* CcccApi - object-oriented interface
* @export
* @class CcccApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterCcccApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * SVC账户分页接口
    * @param {CcccApiDataverseOpenapiGmeshSvcAccountCpAmazonAccountPageGetRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CcccApi
    */
    dataverseOpenapiGmeshSvcAccountCpAmazonAccountPageGet(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.offset) !== undefined) {
                params['offset'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.offset;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.pageSize) !== undefined) {
                params['pageSize'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.pageSize;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.registeredBusinessName) !== undefined) {
                params['registeredBusinessName'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.registeredBusinessName;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountCode) !== undefined) {
                params['accountCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/dtg-dataverse-openapi/gmesh/svc-account/cp_amazon_account_page', params: requestParameters }));
        });
    }
    /**
    * SVC账户Vendor分页接口
    * @param {CcccApiDataverseOpenapiGmeshSvcAccountCpAmazonVendorPageGetRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CcccApi
    */
    dataverseOpenapiGmeshSvcAccountCpAmazonVendorPageGet(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountId) !== undefined) {
                params['accountId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.offset) !== undefined) {
                params['offset'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.offset;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.limit) !== undefined) {
                params['limit'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.limit;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/dtg-dataverse-openapi/gmesh/svc-account/cp_amazon_vendor_page', params: requestParameters }));
        });
    }
}
exports.SpotterCcccApiSdk = SpotterCcccApiSdk;
