// @spotter/gmesh-api-sdk/lib/api/common-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterCommonApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* CommonApi - object-oriented interface
* @export
* @class CommonApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterCommonApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 同类型结算单批量导出
    * @summary 同类型结算单批量导出
    * @param {CommonApiBatchExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    batchExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/common/settle/batch_export', data: requestParameters }));
        });
    }
    /**
    * 导出Pdf
    * @summary 导出Pdf
    * @param {CommonApiExportPdfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    exportPdf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/common/pdf', data: requestParameters }));
        });
    }
    /**
    * 导出Pdf
    * @summary 导出Pdf
    * @param {CommonApiExportPdf0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof CommonApi
    */
    exportPdf_1(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ess-web/ess/common/pdf', data: requestParameters }));
        });
    }
    /**
    * 导出结算单
    * @summary 导出结算单
    * @param {CommonApiExportSettleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    exportSettle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType) !== undefined) {
                params['settleType'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/common/settle/export', params: requestParameters }));
        });
    }
    /**
    * 当前时间是否存在有效合同
    * @summary 当前时间是否存在有效合同
    * @param {CommonApiGetEfficientContractNowRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    getEfficientContractNow(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/common/getEfficientContract4Now', params: requestParameters }));
        });
    }
    /**
    * 查询目前最新的结算单
    * @summary 查询目前最新的结算单
    * @param {CommonApiGetLastRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    getLast(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType) !== undefined) {
                params['settleType'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/common/settle/getLast', params: requestParameters }));
        });
    }
    /**
    * 获取财务角色用户列表
    * @summary 获取财务角色用户列表
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    getUserList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/common/getUserList' }));
        });
    }
    /**
    * 上次结算信息-结算时间（通用接口）
    * @summary 上次结算信息-结算时间（通用接口）
    * @param {CommonApiLastSettleInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof CommonApi
    */
    lastSettleInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType) !== undefined) {
                params['settleType'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/common/settle/last_settle_info', params: requestParameters }));
        });
    }
    /**
    * 所有款项撤销前检查
    * @summary 所有款项撤销前检查
    * @param {CommonApiPreRevokeCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    preRevokeCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType) !== undefined) {
                params['settleType'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/common/settle/preRevokeCheck', params: requestParameters }));
        });
    }
    /**
    * 检查用户是否有撤销该结算单的权限
    * @summary 检查用户是否有撤销该结算单的权限
    * @param {CommonApiPreRevokePermissionCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    preRevokePermissionCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType) !== undefined) {
                params['settleType'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/common/settle/preRevokePermissionCheck', params: requestParameters }));
        });
    }
    /**
    * 查询财务操作日志
    * @summary 查询财务操作日志
    * @param {CommonApiQueryLogRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    queryLog(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/common/operate/query', data: requestParameters }));
        });
    }
    /**
    * 分页查询结算单
    * @summary 分页查询结算单
    * @param {CommonApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/common/settle/queryPage', data: requestParameters }));
        });
    }
    /**
    * 撤销结算单
    * @summary 撤销结算单
    * @param {CommonApiRevokeSettleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CommonApi
    */
    revokeSettle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType) !== undefined) {
                params['settleType'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleType;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/common/settle/revoke', params: requestParameters }));
        });
    }
}
exports.SpotterCommonApiSdk = SpotterCommonApiSdk;
