// @spotter/gmesh-api-sdk/lib/api/company-bank-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterCompanyBankApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* CompanyBankApi - object-oriented interface
* @export
* @class CompanyBankApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterCompanyBankApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 新增银行信息
    * @summary 新增银行信息
    * @param {CompanyBankApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof CompanyBankApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company/bank/create', data: requestParameters }));
        });
    }
    /**
    * 查看银行信息
    * @summary 查看银行信息
    * @param {CompanyBankApiInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof CompanyBankApi
    */
    info(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/company/bank/info', params: requestParameters }));
        });
    }
    /**
    * 银行信息修改
    * @summary 银行信息修改
    * @param {CompanyBankApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof CompanyBankApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company/bank/update', data: requestParameters }));
        });
    }
}
exports.SpotterCompanyBankApiSdk = SpotterCompanyBankApiSdk;
