// @spotter/gmesh-api-sdk/lib/api/company-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterCompanyApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* CompanyApi - object-oriented interface
* @export
* @class CompanyApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterCompanyApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 添加一个公司
    * @summary 添加一个公司
    * @param {CompanyApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CompanyApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company', data: requestParameters }));
        });
    }
    /**
    * -1:冻结 1:已激活 2:待激活
    * @summary 更新公司状态
    * @param {CompanyApiEnableOrDisableCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CompanyApi
    */
    enableOrDisableCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.status) !== undefined) {
                params['status'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.status;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/company/status', params: requestParameters }));
        });
    }
    /**
    * 查询所有的国家代码枚举值
    * @summary 获取所有的国家代码
    * @param {*} [options] Override http request option.
    * @memberof CompanyApi
    */
    getCountryCode(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/company/country_code' }));
        });
    }
    /**
    * 编辑展示，公司信息
    * @summary 公司信息
    * @param {CompanyApiGetInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CompanyApi
    */
    getInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/company', params: requestParameters }));
        });
    }
    /**
    * 获取当前用户所管理的公司
    * @summary 获取当前用户所管理的公司
    * @param {*} [options] Override http request option.
    * @memberof CompanyApi
    */
    getUserManageCompany(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/company/manager/company/list' }));
        });
    }
    /**
    * Seller管理/公司管理/公司列表分页
    * @summary Seller管理/公司管理/公司列表分页
    * @param {CompanyApiListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CompanyApi
    */
    list(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company/list', data: requestParameters }));
        });
    }
    /**
    * 公司列表
    * @summary 公司列表
    * @param {CompanyApiListAllRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CompanyApi
    */
    listAll(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company/simple/list', data: requestParameters }));
        });
    }
    /**
    * 更新公司信息
    * @summary 更新公司信息
    * @param {CompanyApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CompanyApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/company', data: requestParameters }));
        });
    }
}
exports.SpotterCompanyApiSdk = SpotterCompanyApiSdk;
