// @spotter/gmesh-api-sdk/lib/api/configuration-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterConfigurationApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ConfigurationApi - object-oriented interface
* @export
* @class ConfigurationApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterConfigurationApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * OD邮箱规则-校验邮箱类型
    * @summary OD邮箱规则-校验邮箱类型
    * @param {ConfigurationApiCheckCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    checkCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/configuration/rule/checkCategory', data: requestParameters }));
        });
    }
    /**
    * OD邮箱规则创建
    * @summary OD邮箱规则创建
    * @param {ConfigurationApiCreateRuleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    createRule(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/configuration/rule/create', data: requestParameters }));
        });
    }
    /**
    * OD邮箱规则删除
    * @summary OD邮箱规则删除
    * @param {ConfigurationApiDelRuleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    delRule(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/configuration/rule/del', params: requestParameters }));
        });
    }
    /**
    * OD邮箱规则详情
    * @summary OD邮箱规则详情
    * @param {ConfigurationApiDetailsRuleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    detailsRule(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/configuration/rule/detail', params: requestParameters }));
        });
    }
    /**
    * OD邮箱模版详情
    * @summary OD邮箱模版详情
    * @param {ConfigurationApiDetailsTempRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    detailsTemp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/configuration/temp/detail', params: requestParameters }));
        });
    }
    /**
    * OD邮箱模版-通过邮箱类型查询任务说明模版
    * @summary OD邮箱模版-通过邮箱类型查询任务说明模版
    * @param {ConfigurationApiFindIllustrateTempByCategoryIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    findIllustrateTempByCategoryId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.categoryId) !== undefined) {
                params['categoryId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.categoryId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/configuration/temp/findIllustrateTemp', params: requestParameters }));
        });
    }
    /**
    * OD邮箱规则分页
    * @summary OD邮箱规则分页
    * @param {ConfigurationApiPagingRuleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    pagingRule(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/configuration/rule/page', data: requestParameters }));
        });
    }
    /**
    * OD邮箱模版分页
    * @summary OD邮箱模版分页
    * @param {ConfigurationApiPagingTempRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    pagingTemp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/configuration/temp/page', data: requestParameters }));
        });
    }
    /**
    * OD邮箱模版-通过邮箱类型查询快捷回复模版
    * @summary OD邮箱模版-通过邮箱类型查询快捷回复模版
    * @param {ConfigurationApiTempFindByCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    tempFindByCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.categoryId) !== undefined) {
                params['categoryId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.categoryId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/configuration/temp/findByCategory', params: requestParameters }));
        });
    }
    /**
    * OD邮箱规则编辑
    * @summary OD邮箱规则编辑
    * @param {ConfigurationApiUpdateRuleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    updateRule(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/configuration/rule/update', data: requestParameters }));
        });
    }
    /**
    * OD邮箱模版编辑
    * @summary OD邮箱模版编辑
    * @param {ConfigurationApiUpdateTempRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ConfigurationApi
    */
    updateTemp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/configuration/temp/update', data: requestParameters }));
        });
    }
}
exports.SpotterConfigurationApiSdk = SpotterConfigurationApiSdk;
