// @spotter/gmesh-api-sdk/lib/api/contract-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterContractApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ContractApi - object-oriented interface
* @export
* @class ContractApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterContractApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 合同创建
    * @summary 合同创建
    * @param {ContractApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ContractApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/create', data: requestParameters }));
        });
    }
    /**
    * 合同详情
    * @summary 合同详情
    * @param {ContractApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ContractApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-business-web/contract/detail', params: requestParameters }));
        });
    }
    /**
    * 合同分页查询
    * @summary 合同分页查询
    * @param {ContractApiFindPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ContractApi
    */
    findPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/list', data: requestParameters }));
        });
    }
    /**
    * 订单模式-带物流模式
    * @summary 订单模式-带物流模式
    * @param {*} [options] Override http request option.
    * @memberof ContractApi
    */
    orderTypeWithWarehouseList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/order/type/with/warehouse' }));
        });
    }
    /**
    * 付款周期-带金融服务率
    * @summary 付款周期-带金融服务率
    * @param {*} [options] Override http request option.
    * @memberof ContractApi
    */
    paymentWithRates(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/payment/cycle' }));
        });
    }
    /**
    * 合同补充协议附件增加
    * @summary 合同补充协议附件增加
    * @param {ContractApiSupplementAttachedUploadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ContractApi
    */
    supplementAttachedUpload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/supplement/attaches/upload', data: requestParameters }));
        });
    }
    /**
    * 合同剩余能上传补充协议附件数量
    * @summary 合同剩余能上传补充协议附件数量
    * @param {ContractApiSupplementResidualAttachCountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ContractApi
    */
    supplementResidualAttachCount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/supplement/attach/residual/count', params: requestParameters }));
        });
    }
    /**
    * 合同修改
    * @summary 合同修改
    * @param {ContractApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ContractApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/update', data: requestParameters }));
        });
    }
}
exports.SpotterContractApiSdk = SpotterContractApiSdk;
