// @spotter/gmesh-api-sdk/lib/api/coop-agreement-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterCoopAgreementApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* CoopAgreementApi - object-oriented interface
* @export
* @class CoopAgreementApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterCoopAgreementApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 提醒未确认的asin数据
    * @summary 提醒未确认的asin数据
    * @param {CoopAgreementApiAlertNoDutyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CoopAgreementApi
    */
    alertNoDuty(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/coopAgreement/alertNoDuty', data: requestParameters }));
        });
    }
    /**
    * 处理Coop协议批量接受
    * @summary 处理Coop协议批量接受
    * @param {CoopAgreementApiCoopAgreementBatchAcceptRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CoopAgreementApi
    */
    coopAgreementBatchAccept(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/coopAgreement/batchAccept', data: requestParameters }));
        });
    }
    /**
    * 批量接受Coop协议预校验(返回不符合的agreementId)
    * @summary 批量接受Coop协议预校验
    * @param {CoopAgreementApiCoopAgreementBatchAcceptPreCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CoopAgreementApi
    */
    coopAgreementBatchAcceptPreCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/coopAgreement/batchAcceptPreCheck', data: requestParameters }));
        });
    }
    /**
    * Coop协议明细(支持id 和 唯一键查询)
    * @summary Coop协议明细
    * @param {CoopAgreementApiCoopAgreementDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CoopAgreementApi
    */
    coopAgreementDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId) !== undefined) {
                params['agreementId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/coopAgreement/detail', params: requestParameters }));
        });
    }
    /**
    * 处理Coop协议（接受/拒绝）
    * @summary 处理Coop协议
    * @param {CoopAgreementApiCoopAgreementHandleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CoopAgreementApi
    */
    coopAgreementHandle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/coopAgreement/handle', data: requestParameters }));
        });
    }
    /**
    * 处理Coop协议预校验
    * @summary 处理Coop协议预校验
    * @param {CoopAgreementApiCoopAgreementHandlePreCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CoopAgreementApi
    */
    coopAgreementHandlePreCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/coopAgreement/handlePreCheck', data: requestParameters }));
        });
    }
    /**
    * 分页查询Coop协议
    * @summary 分页查询Coop协议
    * @param {CoopAgreementApiCoopAgreementPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CoopAgreementApi
    */
    coopAgreementPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/coopAgreement/page', data: requestParameters }));
        });
    }
    /**
    * 存在未确认责任方的agreement数量
    * @summary 存在未确认责任方的agreement数量
    * @param {*} [options] Override http request option.
    * @memberof CoopAgreementApi
    */
    noDutyCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/coopAgreement/noDutyCount' }));
        });
    }
}
exports.SpotterCoopAgreementApiSdk = SpotterCoopAgreementApiSdk;
