// @spotter/gmesh-api-sdk/lib/api/cost-income-end-point-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterCostIncomeEndPointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* CostIncomeEndPointApi - object-oriented interface
* @export
* @class CostIncomeEndPointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterCostIncomeEndPointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 收入成本明细编辑
    * @summary 收入成本明细编辑
    * @param {CostIncomeEndPointApiEditDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    editDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/detail/edit', data: requestParameters }));
        });
    }
    /**
    * 收入成本明细批量编辑
    * @summary 收入成本明细批量编辑
    * @param {CostIncomeEndPointApiEditDetailBatchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    editDetailBatch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/detail/editBatch', data: requestParameters }));
        });
    }
    /**
    * 确认收入编辑
    * @summary 确认收入编辑
    * @param {CostIncomeEndPointApiEditIncomeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    editIncome(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/income/editRemarks', data: requestParameters }));
        });
    }
    /**
    * 成本结转导出
    * @summary 成本结转导出
    * @param {CostIncomeEndPointApiExportCostRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    exportCost(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/cost/export', data: requestParameters }));
        });
    }
    /**
    * 收入成本明细导出
    * @summary 收入成本明细导出
    * @param {CostIncomeEndPointApiExportDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    exportDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/detail/export', data: requestParameters }));
        });
    }
    /**
    * 确认收入导出
    * @summary 确认收入导出
    * @param {CostIncomeEndPointApiExportIncomeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    exportIncome(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/income/export', data: requestParameters }));
        });
    }
    /**
    * 成本结转分页查询
    * @summary 成本结转分页查询
    * @param {CostIncomeEndPointApiQueryCostPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    queryCostPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/cost/queryPage', data: requestParameters }));
        });
    }
    /**
    * 收入成本明细-查询无结算日期DI列表
    * @summary 收入成本明细-查询无结算日期DI列表
    * @param {CostIncomeEndPointApiQueryDetailDiListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    queryDetailDiList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/detail/queryDiList', data: requestParameters }));
        });
    }
    /**
    * 收入成本明细分页查询
    * @summary 收入成本明细分页查询
    * @param {CostIncomeEndPointApiQueryDetailPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    queryDetailPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/detail/queryPage', data: requestParameters }));
        });
    }
    /**
    * 确认收入分页查询
    * @summary 确认收入分页查询
    * @param {CostIncomeEndPointApiQueryIncomePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CostIncomeEndPointApi
    */
    queryIncomePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-computator-web/costIncomeSettlement/income/queryPage', data: requestParameters }));
        });
    }
}
exports.SpotterCostIncomeEndPointApiSdk = SpotterCostIncomeEndPointApiSdk;
