// @spotter/gmesh-api-sdk/lib/api/coupon-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterCouponApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* CouponApi - object-oriented interface
* @export
* @class CouponApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterCouponApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 检查coupon数据完整性
    * @summary 检查coupon数据完整性
    * @param {*} [options] Override http request option.
    * @memberof CouponApi
    */
    checkCouponItemComplete(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/coupon/item/check' }));
        });
    }
    /**
    * 创建结算单
    * @summary 创建结算单
    * @param {CouponApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CouponApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/coupon/settle/create', data: requestParameters }));
        });
    }
    /**
    * 生成结算单预览
    * @summary 生成结算单预览
    * @param {CouponApiPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CouponApi
    */
    preview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/coupon/settle/preview', data: requestParameters }));
        });
    }
    /**
    * 分页获取coupon明细
    * @summary 分页获取coupon明细
    * @param {CouponApiQueryCouponRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CouponApi
    */
    queryCoupon(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/coupon/item/queryPage', data: requestParameters }));
        });
    }
    /**
    * 获取coupon结算单详情
    * @summary 获取coupon结算单详情
    * @param {CouponApiQueryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CouponApi
    */
    queryDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/coupon/settle/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 补全coupon数据
    * @summary 补全coupon数据
    * @param {CouponApiUpdateCouponItemDataRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof CouponApi
    */
    updateCouponItemData(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/coupon/item/update', data: requestParameters }));
        });
    }
}
exports.SpotterCouponApiSdk = SpotterCouponApiSdk;
