// @spotter/gmesh-api-sdk/lib/api/dashboard-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterDashboardApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* DashboardApi - object-oriented interface
* @export
* @class DashboardApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterDashboardApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 营收趋势图
    * @summary 营收趋势图
    * @param {DashboardApiFirstLineWithChartRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DashboardApi
    */
    firstLineWithChart(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/dashboard/revenue-trend-chart', data: requestParameters }));
        });
    }
    /**
    * 营收指标一级-总维度
    * @summary 营收指标一级-总维度
    * @param {DashboardApiLineRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DashboardApi
    */
    line(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/dashboard/l1', data: requestParameters }));
        });
    }
    /**
    * 营收指标二级-业务线维度
    * @summary 营收指标二级-业务线维度
    * @param {DashboardApiLineSecondRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DashboardApi
    */
    lineSecond(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/dashboard/l2', data: requestParameters }));
        });
    }
    /**
    * 营收指标三级-公司维度
    * @summary 营收指标三级-公司维度
    * @param {DashboardApiLineThirdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DashboardApi
    */
    lineThird(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/dashboard/l3', data: requestParameters }));
        });
    }
    /**
    * 公司负责人下拉框
    * @summary 公司负责人下拉框
    * @param {DashboardApiSelectAccountManagerRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DashboardApi
    */
    selectAccountManager(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year) !== undefined) {
                params['year'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.lineCode) !== undefined) {
                params['lineCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.lineCode;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/report/dashboard/select/accountManager', params: requestParameters }));
        });
    }
    /**
    * 公司下拉框
    * @summary 公司下拉框
    * @param {DashboardApiSelectCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DashboardApi
    */
    selectCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.lineCode) !== undefined) {
                params['lineCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.lineCode;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year) !== undefined) {
                params['year'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/report/dashboard/select/company', params: requestParameters }));
        });
    }
    /**
    * 业务线下拉框
    * @summary 业务线下拉框
    * @param {DashboardApiSelectLineRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DashboardApi
    */
    selectLine(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year) !== undefined) {
                params['year'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/report/dashboard/select/line', params: requestParameters }));
        });
    }
    /**
    * 年度战略目标
    * @summary 年度战略目标
    * @param {DashboardApiTargetAllRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DashboardApi
    */
    targetAll(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year) !== undefined) {
                params['year'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/report/dashboard/target-all', params: requestParameters }));
        });
    }
}
exports.SpotterDashboardApiSdk = SpotterDashboardApiSdk;
