// @spotter/gmesh-api-sdk/lib/api/data-permission-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterDataPermissionApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* DataPermissionApi - object-oriented interface
* @export
* @class DataPermissionApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterDataPermissionApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 删除数据权限规则
    * @summary 删除数据权限规则
    * @param {DataPermissionApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataPermissionApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-guard-web/data-permission', params: requestParameters }));
        });
    }
    /**
    * 添加数据权限规则
    * @summary 添加数据权限规则
    * @param {DataPermissionApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataPermissionApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/data-permission', data: requestParameters }));
        });
    }
    /**
    * 获取操作符类型
    * @summary 获取操作符类型
    * @param {*} [options] Override http request option.
    * @memberof DataPermissionApi
    */
    getOperators(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/data-permission/operators' }));
        });
    }
    /**
    * 查询数据权限规则列表
    * @summary 查询数据权限规则列表
    * @param {DataPermissionApiListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataPermissionApi
    */
    list(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/data-permission/list', data: requestParameters }));
        });
    }
    /**
    * 查询数据权限规则
    * @summary 查询数据权限规则
    * @param {DataPermissionApiOneRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataPermissionApi
    */
    one(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/data-permission', params: requestParameters }));
        });
    }
    /**
    * 更新数据权限规则
    * @summary 更新数据权限规则
    * @param {DataPermissionApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataPermissionApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/data-permission', data: requestParameters }));
        });
    }
}
exports.SpotterDataPermissionApiSdk = SpotterDataPermissionApiSdk;
