// @spotter/gmesh-api-sdk/lib/api/data-verify-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterDataVerifyApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* DataVerifyApi - object-oriented interface
* @export
* @class DataVerifyApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterDataVerifyApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 核对任务记录-列表状态数量-by款项
    * @summary 核对任务记录-列表状态数量-by款项
    * @param {DataVerifyApiCountTaskRecordByFundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    countTaskRecordByFund(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/countTaskRecordByFund', data: requestParameters }));
        });
    }
    /**
    * 核对任务记录-明细分页列表
    * @summary 核对任务记录-明细分页列表
    * @param {DataVerifyApiDetailPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    detailPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/record/detail/page', data: requestParameters }));
        });
    }
    /**
    * 核对任务记录-核对异常公司列表
    * @summary 核对任务记录-核对异常公司列表
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    listAbnormalCompany(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/listAbnormalCompanyInfo' }));
        });
    }
    /**
    * 核对任务记录-核对异常分类列表
    * @summary 核对任务记录-核对异常分类列表
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    listAbnormalTaskType(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/listAbnormalTaskType' }));
        });
    }
    /**
    * 核对任务记录-列表-by公司
    * @summary 核对任务记录-列表-by公司
    * @param {DataVerifyApiListTaskRecordByCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    listTaskRecordByCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/listTaskRecordByCompany', data: requestParameters }));
        });
    }
    /**
    * 核对任务记录-列表-by款项
    * @summary 核对任务记录-列表-by款项
    * @param {DataVerifyApiListTaskRecordByFundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    listTaskRecordByFund(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/listTaskRecordByFund', data: requestParameters }));
        });
    }
    /**
    * 对账项目下拉列表
    * @summary 对账项目下拉列表
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    listTaskType(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/listTaskType' }));
        });
    }
    /**
    * 核对任务记录-处理
    * @summary 核对任务记录-处理
    * @param {DataVerifyApiRecordOperateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    recordOperate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/record/operate', data: requestParameters }));
        });
    }
    /**
    * 核对任务记录-处理类型列表
    * @summary 核对任务记录-处理类型列表
    * @param {DataVerifyApiRecordOperateTypeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    recordOperateType(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/record/operate/type', data: requestParameters }));
        });
    }
    /**
    * 核对任务记录-预览
    * @summary 核对任务记录-预览
    * @param {DataVerifyApiRecordPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DataVerifyApi
    */
    recordPreview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-monitor-web/dataverify/record/preview', data: requestParameters }));
        });
    }
}
exports.SpotterDataVerifyApiSdk = SpotterDataVerifyApiSdk;
