// @spotter/gmesh-api-sdk/lib/api/dispute-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterDisputeApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* DisputeApi - object-oriented interface
* @export
* @class DisputeApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterDisputeApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    *
    * @summary 亚马逊dispute枚举列表
    * @param {DisputeApiGetDisputeEnumListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DisputeApi
    */
    getDisputeEnumList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.enumType) !== undefined) {
                params['enumType'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.enumType;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/dispute/enum/list', params: requestParameters }));
        });
    }
    /**
    *
    * @summary shortage申诉详情
    * @param {DisputeApiShortageDisputeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DisputeApi
    */
    shortageDisputeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber) !== undefined) {
                params['invoiceNumber'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/dispute/shortage/detail', params: requestParameters }));
        });
    }
    /**
    *
    * @summary shortage申诉编辑
    * @param {DisputeApiShortageDisputeEditRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DisputeApi
    */
    shortageDisputeEdit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/dispute/shortage/edit', data: requestParameters }));
        });
    }
    /**
    *
    * @summary shortage申诉导出
    * @param {DisputeApiShortageDisputeExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DisputeApi
    */
    shortageDisputeExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/dispute/shortage/export', data: requestParameters }));
        });
    }
    /**
    *
    * @summary shortage申诉列表
    * @param {DisputeApiShortageDisputePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof DisputeApi
    */
    shortageDisputePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/dispute/shortage/page', data: requestParameters }));
        });
    }
    /**
    *
    * @summary shortage申诉未处理计数
    * @param {*} [options] Override http request option.
    * @memberof DisputeApi
    */
    shortageUnDisputeCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/dispute/shortage/count' }));
        });
    }
}
exports.SpotterDisputeApiSdk = SpotterDisputeApiSdk;
