// @spotter/gmesh-api-sdk/lib/api/email-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterEmailApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* EmailApi - object-oriented interface
* @export
* @class EmailApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterEmailApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 获取负责人
    * @summary 获取负责人
    * @param {EmailApiCoordinatorQueryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    coordinatorQuery(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/email/coordinator/list', data: requestParameters }));
        });
    }
    /**
    * 收件箱分页查询
    * @summary 收件箱分页查询
    * @param {EmailApiGroupPageQueryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    groupPageQuery(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/email/group/page', data: requestParameters }));
        });
    }
    /**
    * 分页查询邮件基本信息
    * @summary 分页查询邮件基本信息
    * @param {EmailApiMailBasicInfoPageQueryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    mailBasicInfoPageQuery(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/email/page/basic', data: requestParameters }));
        });
    }
    /**
    * 邮件详情
    * @summary 邮件详情
    * @param {EmailApiMailDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    mailDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/email/detail', params: requestParameters }));
        });
    }
    /**
    * 邮件组详情
    * @summary 邮件组详情
    * @param {EmailApiMailGroupDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    mailGroupDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/email/group/detail', params: requestParameters }));
        });
    }
    /**
    * 发送邮件
    * @summary 发送邮件
    * @param {EmailApiSendMailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    sendMail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/email/send', data: requestParameters }));
        });
    }
    /**
    * 邮件组类型设置
    * @summary 邮件组类型设置
    * @param {EmailApiSetGroupCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    setGroupCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/email/group/category/update', data: requestParameters }));
        });
    }
    /**
    * 邮件组已读设置
    * @summary 邮件组已读设置
    * @param {EmailApiSetGroupReadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    setGroupRead(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/email/group/read', data: requestParameters }));
        });
    }
    /**
    * 邮件组标签设置
    * @summary 邮件组标签设置
    * @param {EmailApiSetGroupStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    setGroupStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/email/group/status/update', data: requestParameters }));
        });
    }
    /**
    * 邮件已读设置
    * @summary 邮件已读设置
    * @param {EmailApiSetMailReadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    setMailRead(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/email/read', data: requestParameters }));
        });
    }
    /**
    * 总未读邮件数
    * @summary 总未读邮件数
    * @param {*} [options] Override http request option.
    * @memberof EmailApi
    */
    totalUnreadCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/email/total/unreadCount' }));
        });
    }
}
exports.SpotterEmailApiSdk = SpotterEmailApiSdk;
