// @spotter/gmesh-api-sdk/lib/api/enum-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterEnumApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* EnumApi - object-oriented interface
* @export
* @class EnumApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterEnumApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 协议状态
    * @summary 枚举-协议状态
    * @param {*} [options] Override http request option.
    * @memberof EnumApi
    */
    agreementStatus(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/enum/agreement/status' }));
        });
    }
    /**
    * 协议类型
    * @summary 枚举-协议类型
    * @param {*} [options] Override http request option.
    * @memberof EnumApi
    */
    agreementType(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/enum/agreement/type' }));
        });
    }
    /**
    * 获取所有枚举
    * @summary 获取所有枚举
    * @param {*} [options] Override http request option.
    * @memberof EnumApi
    */
    allEnum(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/enum/all' }));
        });
    }
    /**
    * 获取所有枚举
    * @summary 获取所有枚举
    * @param {*} [options] Override http request option.
    * @memberof EnumApi
    */
    allEnum_1(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/invoice/enum/all' }));
        });
    }
    /**
    * 获取所有枚举
    * @summary 获取所有枚举
    * @param {*} [options] Override http request option.
    * @memberof EnumApi
    */
    allEnum_2(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-monitor-web/enum/monitor/all' }));
        });
    }
    /**
    * 免仓租期枚举
    * @summary 枚举-免仓租期
    * @param {*} [options] Override http request option.
    * @memberof EnumApi
    */
    freeLeaseDays(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/enum/free/lease/days' }));
        });
    }
    /**
    * 公司报价属性
    * @summary 枚举-公司报价属性
    * @param {*} [options] Override http request option.
    * @memberof EnumApi
    */
    quotationAttribute(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/enum/quotation/attribute' }));
        });
    }
}
exports.SpotterEnumApiSdk = SpotterEnumApiSdk;
