// @spotter/gmesh-api-sdk/lib/api/file-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFileApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FileApi - object-oriented interface
* @export
* @class FileApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFileApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * ZIP多文件打包
    * @summary ZIP多文件打包
    * @param {FileApiGetBatchZipRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof FileApi
    */
    getBatchZip(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/gmesh/zip', data: requestParameters }));
        });
    }
    /**
    * 获取下载文件的签名
    * @summary 获取文件下载的签名
    * @param {FileApiGetFileDownloadSignerRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof FileApi
    */
    getFileDownloadSigner(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/gmesh/download', data: requestParameters }));
        });
    }
    /**
    * 获取指定存储桶的临时密钥
    * @summary 获取指定存储桶的临时密钥
    * @param {FileApiGetTemporaryCredentialRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof FileApi
    */
    getTemporaryCredential(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/gmesh/credential', data: requestParameters }));
        });
    }
    /**
    * ZIP多文件打包(异步)
    * @summary ZIP多文件打包(异步)
    * @param {FileApiGmeshBatchZipAsyncRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof FileApi
    */
    gmeshBatchZipAsync(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/gmesh/zip/async', data: requestParameters }));
        });
    }
    /**
    * ZIP多文件打包(异步)
    * @summary ZIP多文件打包(异步)
    * @param {FileApiGmeshBatchZipAsyncResultRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof FileApi
    */
    gmeshBatchZipAsyncResult(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/gmesh/zip/result', data: requestParameters }));
        });
    }
}
exports.SpotterFileApiSdk = SpotterFileApiSdk;
