// @spotter/gmesh-api-sdk/lib/api/finance-warehouse-reconciliation-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceWarehouseReconciliationApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceWarehouseReconciliationApi - object-oriented interface
* @export
* @class FinanceWarehouseReconciliationApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceWarehouseReconciliationApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * FF费价格调整明细信息
    * @summary FF费价格调整明细信息
    * @param {FinanceWarehouseReconciliationApiAdjustDetailFulfillmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    adjustDetailFulfillment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust_detail/fulfillment', data: requestParameters }));
        });
    }
    /**
    * storage费价格调整明细信息
    * @summary storage费价格调整明细信息
    * @param {FinanceWarehouseReconciliationApiAdjustDetailStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    adjustDetailStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust_detail/storage', data: requestParameters }));
        });
    }
    /**
    * vas费价格调整明细信息
    * @summary vas费价格调整明细信息
    * @param {FinanceWarehouseReconciliationApiAdjustDetailVasRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    adjustDetailVas(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust_detail/vas', data: requestParameters }));
        });
    }
    /**
    * 审核海外仓费用对账单
    * @summary 审核海外仓费用对账单
    * @param {FinanceWarehouseReconciliationApiAuditReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    auditReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/audit', data: requestParameters }));
        });
    }
    /**
    * 批量审核海外仓费用对账单
    * @summary 批量审核海外仓费用对账单
    * @param {FinanceWarehouseReconciliationApiBatchAuditReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    batchAuditReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/batch_audit', data: requestParameters }));
        });
    }
    /**
    * 分类获取在某个账期时间点存在对账单和没有对账单的公司列表
    * @summary 分类获取在某个账期时间点存在对账单和没有对账单的公司列表
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    companyReconciliationGroup(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/company_group_list' }));
        });
    }
    /**
    * 查询海外仓对账单的某个状态下的数量
    * @summary 查询海外仓对账单的某个状态下的数量
    * @param {FinanceWarehouseReconciliationApiCountReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    countReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/count', data: requestParameters }));
        });
    }
    /**
    * 创建海外仓费用对账单
    * @summary 创建海外仓费用对账单
    * @param {FinanceWarehouseReconciliationApiCreateReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    createReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/create', data: requestParameters }));
        });
    }
    /**
    * 创建海外仓费用金额调整
    * @summary 创建海外仓费用金额调整
    * @param {FinanceWarehouseReconciliationApiCreateWarehouseReconciliationAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    createWarehouseReconciliationAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust/create', data: requestParameters }));
        });
    }
    /**
    * 撤销海外仓费用对账单
    * @summary 撤销海外仓费用对账单
    * @param {FinanceWarehouseReconciliationApiDeleteReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    deleteReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/delete', data: requestParameters }));
        });
    }
    /**
    * 撤销海外仓费用金额调整
    * @summary 撤销海外仓费用金额调整
    * @param {FinanceWarehouseReconciliationApiDeleteWarehouseReconciliationAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    deleteWarehouseReconciliationAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust/delete', data: requestParameters }));
        });
    }
    /**
    * 海外仓费用对账单详情
    * @summary 海外仓费用对账单详情
    * @param {FinanceWarehouseReconciliationApiDetailReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    detailReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/detail', data: requestParameters }));
        });
    }
    /**
    * 编辑海外仓费用金额调整
    * @summary 编辑海外仓费用金额调整
    * @param {FinanceWarehouseReconciliationApiEditWarehouseReconciliationAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    editWarehouseReconciliationAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust/edit', data: requestParameters }));
        });
    }
    /**
    * 导出海外仓费用对账单excel
    * @summary 导出海外仓费用对账单excel
    * @param {FinanceWarehouseReconciliationApiExportReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    exportReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/export_excel', data: requestParameters }));
        });
    }
    /**
    * 批量导出金额调整的excel
    * @summary 批量导出金额调整的excel
    * @param {FinanceWarehouseReconciliationApiExportWarehouseReconciliationAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    exportWarehouseReconciliationAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust/batch_export', data: requestParameters }));
        });
    }
    /**
    * 创建海外仓费用对账单分页查询
    * @summary 海外仓费用对账单分页查询
    * @param {FinanceWarehouseReconciliationApiPageReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    pageReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/page', data: requestParameters }));
        });
    }
    /**
    * 分页查询海外仓费用金额调整
    * @summary 分页查询海外仓费用金额调整
    * @param {FinanceWarehouseReconciliationApiPageWarehouseReconciliationAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    pageWarehouseReconciliationAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust/page', data: requestParameters }));
        });
    }
    /**
    * 创建海外仓费用对账单生成前的预览
    * @summary 海外仓费用对账单生成前的预览
    * @param {FinanceWarehouseReconciliationApiPreviewReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    previewReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/preview', data: requestParameters }));
        });
    }
    /**
    * 对账单创建者列表（财务经理+财务助理+财务会计）
    * @summary 对账单创建者列表（财务经理+财务助理+财务会计）
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    reconCreatorList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/creator_list' }));
        });
    }
    /**
    * 查询公司某款项时间下的所有海外仓费用类型的款项编号列表
    * @summary 查询公司某款项时间下的所有海外仓费用类型的款项编号列表
    * @param {FinanceWarehouseReconciliationApiSettleCodesListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    settleCodesList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/settle_codes_list', data: requestParameters }));
        });
    }
    /**
    * 详情海外仓费用金额调整
    * @summary 海外仓费用金额调整详情
    * @param {FinanceWarehouseReconciliationApiWarehouseReconciliationAdjustmentDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseReconciliationApi
    */
    warehouseReconciliationAdjustmentDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust/detail', data: requestParameters }));
        });
    }
}
exports.SpotterFinanceWarehouseReconciliationApiSdk = SpotterFinanceWarehouseReconciliationApiSdk;
