// @spotter/gmesh-api-sdk/lib/api/finance-penalty-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinancePenaltyApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinancePenaltyApi - object-oriented interface
* @export
* @class FinancePenaltyApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinancePenaltyApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * chargebacks-导出
    * @summary chargebacks-导出
    * @param {FinancePenaltyApiChargeBacksExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    chargeBacksExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/chargebacks/export', data: requestParameters }));
        });
    }
    /**
    * chargebacks-分页
    * @summary chargebacks-分页
    * @param {FinancePenaltyApiChargebacksPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    chargebacksPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/chargebacks/page', data: requestParameters }));
        });
    }
    /**
    * 确认msrp
    * @summary 确认msrp
    * @param {FinancePenaltyApiConfirmMsrpRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    confirmMsrp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/confirm_msrp', data: requestParameters }));
        });
    }
    /**
    * 确认责任方
    * @summary 确认责任方
    * @param {FinancePenaltyApiConfirmResponsibleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    confirmResponsible(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/confirm_responsible', data: requestParameters }));
        });
    }
    /**
    * 确认仓库
    * @summary 确认仓库
    * @param {FinancePenaltyApiConfirmWarehouseRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    confirmWarehouse(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/confirm_warehouse', data: requestParameters }));
        });
    }
    /**
    * 结算单-创建
    * @summary 结算单-创建
    * @param {FinancePenaltyApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/settle/create', data: requestParameters }));
        });
    }
    /**
    * 结算单-详情
    * @summary 结算单-详情
    * @param {FinancePenaltyApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/penalty/settle/detail', params: requestParameters }));
        });
    }
    /**
    * 结算单-导出
    * @summary 结算单-导出
    * @param {FinancePenaltyApiExportSettleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    exportSettle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/penalty/settle/export', params: requestParameters }));
        });
    }
    /**
    * 罚款-是否存在尚未完善的数据
    * @summary 罚款-是否存在尚未完善的数据
    * @param {FinancePenaltyApiIsExistRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    isExist(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/is_exist', data: requestParameters }));
        });
    }
    /**
    * 结算单-最近一次结算时间
    * @summary 结算单-最近一次结算时间
    * @param {FinancePenaltyApiLatestSettleDateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    latestSettleDate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/settle/preview/latest_settle_date', data: requestParameters }));
        });
    }
    /**
    * 结算单预览-最早遗漏数据
    * @summary 结算单预览-最早遗漏数据
    * @param {FinancePenaltyApiMissingDataRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    missingData(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/settle/preview/missing_data', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理列表(未处理)
    * @summary 罚款-管理列表(未处理asin)
    * @param {FinancePenaltyApiPenaltyManageListAsinRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyManageListAsin(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/list/asin', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理列表(未处理)
    * @summary 罚款-管理列表(未处理MSRP)
    * @param {FinancePenaltyApiPenaltyManageListMsrpRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyManageListMsrp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/list/msrp', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理列表(未处理)
    * @summary 罚款-管理列表(未处理责任方)
    * @param {FinancePenaltyApiPenaltyManageListResponsibleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyManageListResponsible(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/list/responsible', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理列表(未处理)
    * @summary 罚款-管理列表(未处理仓库)
    * @param {FinancePenaltyApiPenaltyManageListWarehouseRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyManageListWarehouse(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/list/warehouse', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理分页列表(已处理)
    * @summary 罚款-管理分页列表(已处理asin)
    * @param {FinancePenaltyApiPenaltyManagePageAsinRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyManagePageAsin(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/page/asin', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理分页列表(已处理)
    * @summary 罚款-管理分页列表(已处理MSRP)
    * @param {FinancePenaltyApiPenaltyManagePageMsrpRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyManagePageMsrp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/page/msrp', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理分页列表(已处理)
    * @summary 罚款-管理分页列表(已处理责任方)
    * @param {FinancePenaltyApiPenaltyManagePageResponsibleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyManagePageResponsible(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/page/responsible', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理分页列表(已处理)
    * @summary 罚款-管理分页列表(已处理仓库)
    * @param {FinancePenaltyApiPenaltyManagePageWarehouseRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyManagePageWarehouse(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/page/warehouse', data: requestParameters }));
        });
    }
    /**
    * 罚款-管理列表数量
    * @summary 罚款-管理列表数量
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyMangeSize(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/size' }));
        });
    }
    /**
    * 罚款类型列表
    * @summary 罚款类型列表
    * @param {FinancePenaltyApiPenaltyTypeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    penaltyType(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/penalty_type', data: requestParameters }));
        });
    }
    /**
    * 结算单预览-列表
    * @summary 结算单预览-列表
    * @param {FinancePenaltyApiPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    preview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/settle/preview/list', data: requestParameters }));
        });
    }
    /**
    * 结算单预览-导出
    * @summary 结算单预览-导出
    * @param {FinancePenaltyApiPreviewExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    previewExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/settle/preview/export', data: requestParameters }));
        });
    }
    /**
    * 结算单-撤销
    * @summary 结算单-撤销
    * @param {FinancePenaltyApiRevokeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    revoke(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/penalty/settle/revoke', params: requestParameters }));
        });
    }
    /**
    * 结算单-分页
    * @summary 结算单-分页
    * @param {FinancePenaltyApiSettlePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    settlePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/settle/page', data: requestParameters }));
        });
    }
    /**
    * 金额合计
    * @summary 金额合计
    * @param {FinancePenaltyApiTotalAmountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    totalAmount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/chargebacks/totalAmount', data: requestParameters }));
        });
    }
    /**
    * 补充asin
    * @summary 补充asin
    * @param {FinancePenaltyApiUpdateAsinRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinancePenaltyApi
    */
    updateAsin(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/penalty/penalty/manage/update_asin', data: requestParameters }));
        });
    }
}
exports.SpotterFinancePenaltyApiSdk = SpotterFinancePenaltyApiSdk;
