// @spotter/gmesh-api-sdk/lib/api/finance-expense-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceExpenseApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceExpenseApi - object-oriented interface
* @export
* @class FinanceExpenseApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceExpenseApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 费用调整归档
    * @summary 费用调整归档
    * @param {FinanceExpenseApiArchiveExpenseAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    archiveExpenseAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/archive', data: requestParameters }));
        });
    }
    /**
    * 审核费用调整
    * @summary 审核费用调整
    * @param {FinanceExpenseApiAuditExpenseAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    auditExpenseAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/audit', data: requestParameters }));
        });
    }
    /**
    * 批量审核费用调整
    * @summary 批量审核费用调整
    * @param {FinanceExpenseApiBatchAuditExpenseAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    batchAuditExpenseAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/batch_audit', data: requestParameters }));
        });
    }
    /**
    * 查询费用调整数量
    * @summary 查询费用调整数量
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    countExpenseAdjustment(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/expense/adjustment/count' }));
        });
    }
    /**
    * 创建费用调整
    * @summary 创建费用调整
    * @param {FinanceExpenseApiCreateExpenseAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    createExpenseAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/create', data: requestParameters }));
        });
    }
    /**
    * 创建费用调整记录
    * @summary 创建费用调整记录
    * @param {FinanceExpenseApiCreateExpenseAdjustmentRecordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    createExpenseAdjustmentRecord(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustmentRecord/create', data: requestParameters }));
        });
    }
    /**
    * 创建费用调整备注
    * @summary 创建费用调整备注
    * @param {FinanceExpenseApiCreateExpenseAdjustmentRemarkRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    createExpenseAdjustmentRemark(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/remark/create', data: requestParameters }));
        });
    }
    /**
    * 费用调整详情
    * @summary 费用调整详情
    * @param {FinanceExpenseApiDetailExpenseAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    detailExpenseAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/expense/adjustment/detail', params: requestParameters }));
        });
    }
    /**
    * 编辑费用调整
    * @summary 编辑费用调整
    * @param {FinanceExpenseApiEditExpenseAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    editExpenseAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/edit', data: requestParameters }));
        });
    }
    /**
    * 费用调整导出
    * @summary 费用调整导出
    * @param {FinanceExpenseApiExpenseAdjustmentExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    expenseAdjustmentExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/export', data: requestParameters }));
        });
    }
    /**
    * 分页查询费用调整
    * @summary 分页查询费用调整
    * @param {FinanceExpenseApiPageExpenseAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    pageExpenseAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/page', data: requestParameters }));
        });
    }
    /**
    * 分页查询费用调整处理记录
    * @summary 分页查询费用调整处理记录
    * @param {FinanceExpenseApiPageExpenseAdjustmentRecordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    pageExpenseAdjustmentRecord(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustmentRecord/page', data: requestParameters }));
        });
    }
    /**
    * 费用调整取消归档
    * @summary 费用调整取消归档
    * @param {FinanceExpenseApiUnarchiveExpenseAdjustmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceExpenseApi
    */
    unarchiveExpenseAdjustment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/expense/adjustment/unarchive', data: requestParameters }));
        });
    }
}
exports.SpotterFinanceExpenseApiSdk = SpotterFinanceExpenseApiSdk;
