// @spotter/gmesh-api-sdk/lib/api/finance-other-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceOtherApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceOtherApi - object-oriented interface
* @export
* @class FinanceOtherApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceOtherApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    *
    * @summary 撤销其他费用
    * @param {FinanceOtherApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/delete', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 其他费用分类创建
    * @param {FinanceOtherApiCategoryCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    categoryCreate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/category/create', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 其他费用分类删除
    * @param {FinanceOtherApiCategoryDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    categoryDelete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/category/delete', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 其他费用分类列表
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    categoryList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/other/category' }));
        });
    }
    /**
    *
    * @summary 其他费用分类更新
    * @param {FinanceOtherApiCategoryUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    categoryUpdate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/category/update', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 创建其他费用
    * @param {FinanceOtherApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/create', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 其他费用详情
    * @param {FinanceOtherApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/other/detail', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 其他费用导出表格
    * @param {FinanceOtherApiExportExcelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    exportExcel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/export_excel', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 上一次的款项时间
    * @param {FinanceOtherApiLastSettleDateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof FinanceOtherApi
    */
    lastSettleDate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/other/last_settle_date', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 修改其他费用
    * @param {FinanceOtherApiModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    modify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/update', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 其他费用列表分页
    * @param {FinanceOtherApiQueryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    query(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/query', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 未发货（未扣除，金额为正）信息查询
    * @param {FinanceOtherApiUndeliveredListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    undeliveredList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/undelivered/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 未发货信息（金额为正）分页查询
    * @param {FinanceOtherApiUndeliveredPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    undeliveredPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/undelivered/page', data: requestParameters }));
        });
    }
    /**
    * 上传附件(uploadFile)
    * @summary 上传附件(uploadFile)
    * @param {FinanceOtherApiUploadAttachmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceOtherApi
    */
    uploadAttachment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/other/upload', data: requestParameters }));
        });
    }
}
exports.SpotterFinanceOtherApiSdk = SpotterFinanceOtherApiSdk;
