// @spotter/gmesh-api-sdk/lib/api/finance-settle-check-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceSettleCheckApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceSettleCheckApi - object-oriented interface
* @export
* @class FinanceSettleCheckApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceSettleCheckApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 当前出账日期查询
    * @summary 当前出账日期查询
    * @param {*} [options] Override http request option.
    * @memberof FinanceSettleCheckApi
    */
    accountingDateRange(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/settle/check/accountingDateRange' }));
        });
    }
    /**
    * 结账管理-结账详情列表查询
    * @summary 结账管理-结账详情列表查询
    * @param {FinanceSettleCheckApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceSettleCheckApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/settle/check/detail', data: requestParameters }));
        });
    }
    /**
    * 结账管理-结账概览，月度对账单生成情况查询
    * @summary 结账管理-结账概览，月度对账单生成情况统计查询
    * @param {*} [options] Override http request option.
    * @memberof FinanceSettleCheckApi
    */
    monthReconciliationStatistics(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/settle/check/monthReconciliation/statistics' }));
        });
    }
    /**
    * 结账管理-结账概览分页查询
    * @summary 结账管理-结账概览分页查询
    * @param {FinanceSettleCheckApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceSettleCheckApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/settle/check/page', data: requestParameters }));
        });
    }
    /**
    * 结账日志分页查询
    * @summary 结账日志分页查询
    * @param {FinanceSettleCheckApiSettleLogPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceSettleCheckApi
    */
    settleLogPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/settle/check/settleLog/page', data: requestParameters }));
        });
    }
}
exports.SpotterFinanceSettleCheckApiSdk = SpotterFinanceSettleCheckApiSdk;
