// @spotter/gmesh-api-sdk/lib/api/finance-warehouse-quoted-price-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceWarehouseQuotedPriceApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceWarehouseQuotedPriceApi - object-oriented interface
* @export
* @class FinanceWarehouseQuotedPriceApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceWarehouseQuotedPriceApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 创建海外仓储报价
    * @summary 创建海外仓储报价
    * @param {FinanceWarehouseQuotedPriceApiCreateQuotedPriceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseQuotedPriceApi
    */
    createQuotedPrice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/quotedPrice/create', data: requestParameters }));
        });
    }
    /**
    * 删除海外仓储报价
    * @summary 删除海外仓储报价
    * @param {FinanceWarehouseQuotedPriceApiDeleteQuotedPriceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseQuotedPriceApi
    */
    deleteQuotedPrice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/quotedPrice/delete', data: requestParameters }));
        });
    }
    /**
    * 分页查询海外仓储报价
    * @summary 分页查询海外仓储报价
    * @param {FinanceWarehouseQuotedPriceApiPageQuotedPriceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseQuotedPriceApi
    */
    pageQuotedPrice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/quotedPrice/page', data: requestParameters }));
        });
    }
    /**
    * 海外仓储报价详情
    * @summary 海外仓储报价详情
    * @param {FinanceWarehouseQuotedPriceApiQuotedPriceDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseQuotedPriceApi
    */
    quotedPriceDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/quotedPrice/detail', data: requestParameters }));
        });
    }
    /**
    * 海外仓储报价试算
    * @summary 海外仓储报价试算
    * @param {FinanceWarehouseQuotedPriceApiQuotedPriceTryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseQuotedPriceApi
    */
    quotedPriceTry(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/quotedPrice/try', data: requestParameters }));
        });
    }
    /**
    * 更新海外仓储报价
    * @summary 更新海外仓储报价
    * @param {FinanceWarehouseQuotedPriceApiUpdateQuotedPriceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseQuotedPriceApi
    */
    updateQuotedPrice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/quotedPrice/update', data: requestParameters }));
        });
    }
}
exports.SpotterFinanceWarehouseQuotedPriceApiSdk = SpotterFinanceWarehouseQuotedPriceApiSdk;
