// @spotter/gmesh-api-sdk/lib/api/finance-warehouse-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceWarehouseApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceWarehouseApi - object-oriented interface
* @export
* @class FinanceWarehouseApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceWarehouseApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 生成仓储费结算单
    * @summary 生成结算单
    * @param {FinanceWarehouseApiCreateSettlementRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    createSettlement(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/settlement', data: requestParameters }));
        });
    }
    /**
    * 创建vas附加费类型
    * @summary 创建vas附加费类型
    * @param {FinanceWarehouseApiCreateVasFeeTypeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    createVasFeeType(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas_type/create', data: requestParameters }));
        });
    }
    /**
    * 删除vas附加费类型
    * @summary 删除vas附加费类型
    * @param {FinanceWarehouseApiDeleteVasFeeTypeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    deleteVasFeeType(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas_type/delete', data: requestParameters }));
        });
    }
    /**
    * 用excel批量添加附加服务费
    * @summary 用excel批量添加附加服务费
    * @param {FinanceWarehouseApiExcelSaveBatchVasFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    excelSaveBatchVasFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas/excel_save_batch', data: requestParameters }));
        });
    }
    /**
    * 下载仓储费结算单excel
    * @summary 下载结算单excel
    * @param {FinanceWarehouseApiExportSettlementRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    exportSettlement(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/settlement/export', data: requestParameters }));
        });
    }
    /**
    * 批量导出vas费用
    * @summary 批量导出vas费用(参数同查询，按照查询查询的数据导出)
    * @param {FinanceWarehouseApiExportVasFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    exportVasFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas/export', data: requestParameters }));
        });
    }
    /**
    * 海外仓结算费用详情
    * @summary 海外仓结算费用详情
    * @param {FinanceWarehouseApiGetWarehouseFeeSettlementRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    getWarehouseFeeSettlement(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/settlement/detail', data: requestParameters }));
        });
    }
    /**
    * 库存核对检查的最新时间（
    * @summary 库存核对检查的最新时间（该时间点之前的库存快照都是准确的）
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    inventoryCheckTime(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/warehouse/storage/inventory_check_time' }));
        });
    }
    /**
    * 一次性提货费遗漏查询
    * @summary 一次性提货费遗漏查询
    * @param {FinanceWarehouseApiListMissingFulfillmentFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    listMissingFulfillmentFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/fulfillment/list/missing', data: requestParameters }));
        });
    }
    /**
    * 仓储费遗漏查询
    * @summary 仓储费遗漏查询
    * @param {FinanceWarehouseApiListMissingStorageFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    listMissingStorageFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/storage/list/missing', data: requestParameters }));
        });
    }
    /**
    * 附加服务费遗漏查询
    * @summary 附加服务费遗漏查询
    * @param {FinanceWarehouseApiListMissingVasFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    listMissingVasFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas/list/missing', data: requestParameters }));
        });
    }
    /**
    * 修改附加服务费【包括删除】
    * @summary 修改附加服务费【包括删除】
    * @param {FinanceWarehouseApiModifyVasFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    modifyVasFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas/modify', data: requestParameters }));
        });
    }
    /**
    * 一次性提货费分页【查仓储】
    * @summary 一次性提货费分页【查仓储】
    * @param {FinanceWarehouseApiPageFulfillmentFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    pageFulfillmentFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/fulfillment/page', data: requestParameters }));
        });
    }
    /**
    * 海外仓FF费用明细查询
    * @summary 海外仓FF费用明细查询
    * @param {FinanceWarehouseApiPageFulfillmentFeeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    pageFulfillmentFeeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/fulfillment/detail/page', data: requestParameters }));
        });
    }
    /**
    * 海外仓储费结算单分页
    * @summary 海外仓储费结算单分页
    * @param {FinanceWarehouseApiPageSettlementRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    pageSettlement(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/settlement/page', data: requestParameters }));
        });
    }
    /**
    * 超期仓储费分页【查仓储】
    * @summary 超期仓储费分页【查仓储】
    * @param {FinanceWarehouseApiPageStorageFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    pageStorageFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/storage/page', data: requestParameters }));
        });
    }
    /**
    * 海外仓storage费用明细查询
    * @summary 海外仓storage费用明细查询
    * @param {FinanceWarehouseApiPageStorageFeeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    pageStorageFeeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/storage/detail/page', data: requestParameters }));
        });
    }
    /**
    * 附加服务费分页
    * @summary 附加服务费分页
    * @param {FinanceWarehouseApiPageVasFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    pageVasFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas/page', data: requestParameters }));
        });
    }
    /**
    * 分页查询vas附加费类型
    * @summary 分页查询vas附加费类型
    * @param {FinanceWarehouseApiPageVasFeeTypeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    pageVasFeeType(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas_type/page', data: requestParameters }));
        });
    }
    /**
    * 附加服务费明细分页【金额调整用，非gmesh列表】
    * @summary 附加服务费明细分页【金额调整用，非gmesh列表】
    * @param {FinanceWarehouseApiPageVasFee0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    pageVasFee_1(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas/detail/page', data: requestParameters }));
        });
    }
    /**
    * 撤销仓储费结算单
    * @summary 撤销结算单
    * @param {FinanceWarehouseApiReverseSettlementRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    reverseSettlement(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-finance-web/finance/warehouse/settlement', data: requestParameters }));
        });
    }
    /**
    * 批量添加附加服务费
    * @summary 批量添加附加服务费
    * @param {FinanceWarehouseApiSaveBatchVasFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    saveBatchVasFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas/save_batch', data: requestParameters }));
        });
    }
    /**
    * 更新vas附加费类型
    * @summary 更新vas附加费类型
    * @param {FinanceWarehouseApiUpdateVasFeeTypeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    updateVasFeeType(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas_type/update', data: requestParameters }));
        });
    }
    /**
    * vas附加费类型下发
    * @summary vas附加费类型下发
    * @param {*} [options] Override http request option.
    * @memberof FinanceWarehouseApi
    */
    vasFeeTypes(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vas_type/list' }));
        });
    }
}
exports.SpotterFinanceWarehouseApiSdk = SpotterFinanceWarehouseApiSdk;
