// @spotter/gmesh-api-sdk/lib/api/finance-technology-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceTechnologyApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceTechnologyApi - object-oriented interface
* @export
* @class FinanceTechnologyApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceTechnologyApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 科技服务费导出结算单
    * @summary 科技服务费导出
    * @param {FinanceTechnologyApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceTechnologyApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/technology/export', params: requestParameters }));
        });
    }
    /**
    * 科技服务费创建结算单
    * @summary 科技服务费创建
    * @param {FinanceTechnologyApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceTechnologyApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/technology/create', data: requestParameters }));
        });
    }
    /**
    * 科技服务费预览
    * @summary 科技服务费预览
    * @param {FinanceTechnologyApiPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceTechnologyApi
    */
    preview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/technology/preview', data: requestParameters }));
        });
    }
    /**
    * 科技服务费获取详情
    * @summary 科技服务费获取详情
    * @param {FinanceTechnologyApiQueryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceTechnologyApi
    */
    queryDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/technology/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 科技服务费分页查询
    * @summary 科技服务费分页查询
    * @param {FinanceTechnologyApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceTechnologyApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/technology/queryPage', data: requestParameters }));
        });
    }
    /**
    * 科技服务费获取一二期款编码
    * @summary 科技服务费获取一二期款编码
    * @param {FinanceTechnologyApiQueryPreSettleCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceTechnologyApi
    */
    queryPreSettleCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/technology/queryPreSettleCode', data: requestParameters }));
        });
    }
    /**
    * 科技服务费撤销结算单
    * @summary 科技服务费撤销
    * @param {FinanceTechnologyApiRevokeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceTechnologyApi
    */
    revoke(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/technology/revoke', params: requestParameters }));
        });
    }
}
exports.SpotterFinanceTechnologyApiSdk = SpotterFinanceTechnologyApiSdk;
