// @spotter/gmesh-api-sdk/lib/api/finance-auto-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceAutoApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceAutoApi - object-oriented interface
* @export
* @class FinanceAutoApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceAutoApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 自动结账配置详情
    * @summary 自动结账配置详情
    * @param {FinanceAutoApiAutoSettleConfigDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceAutoApi
    */
    autoSettleConfigDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/auto/config/detail', params: requestParameters }));
        });
    }
    /**
    * 自动结账配置分页查询
    * @summary 自动结账配置分页查询
    * @param {FinanceAutoApiAutoSettleConfigPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceAutoApi
    */
    autoSettleConfigPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/auto/config/page', data: requestParameters }));
        });
    }
    /**
    * 自动结账配置更新
    * @summary 自动结账配置更新
    * @param {FinanceAutoApiAutoSettleConfigUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceAutoApi
    */
    autoSettleConfigUpdate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/auto/config/update', data: requestParameters }));
        });
    }
    /**
    * 异常日志批处理
    * @summary 异常日志处理批处理
    * @param {FinanceAutoApiExceptionBatchHandleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceAutoApi
    */
    exceptionBatchHandle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/auto/exception/batchHandle', data: requestParameters }));
        });
    }
    /**
    * 异常登录人处理数量
    * @summary 异常登录人处理数量
    * @param {*} [options] Override http request option.
    * @memberof FinanceAutoApi
    */
    exceptionCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/auto/exception/count' }));
        });
    }
    /**
    * 异常日志处理
    * @summary 异常日志处理
    * @param {FinanceAutoApiExceptionHandleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceAutoApi
    */
    exceptionHandle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/auto/exception/handle', data: requestParameters }));
        });
    }
    /**
    * 异常日志分页查询
    * @summary 异常日志分页查询
    * @param {FinanceAutoApiExceptionPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceAutoApi
    */
    exceptionPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/auto/exception/page', data: requestParameters }));
        });
    }
    /**
    * 资损核对异常分页查询
    * @summary 资损核对异常分页查询
    * @param {FinanceAutoApiVerifyExceptionPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceAutoApi
    */
    verifyExceptionPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/auto/verify_exception/page', data: requestParameters }));
        });
    }
}
exports.SpotterFinanceAutoApiSdk = SpotterFinanceAutoApiSdk;
