// @spotter/gmesh-api-sdk/lib/api/finance-bbo-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFinanceBboApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FinanceBboApi - object-oriented interface
* @export
* @class FinanceBboApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFinanceBboApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * BBO上次结算单信息
    * @summary BBO上次结算单信息
    * @param {FinanceBboApiBboLastSettleInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceBboApi
    */
    bboLastSettleInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/bbo/lastSettleInfo', params: requestParameters }));
        });
    }
    /**
    * 创建结算单
    * @summary 创建结算单
    * @param {FinanceBboApiCreateSettleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceBboApi
    */
    createSettle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/bbo/create', data: requestParameters }));
        });
    }
    /**
    * 导出结算单
    * @summary 导出结算单
    * @param {FinanceBboApiExportSettleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceBboApi
    */
    exportSettle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/bbo/export', params: requestParameters }));
        });
    }
    /**
    * 生成结算单预览
    * @summary 生成结算单预览
    * @param {FinanceBboApiPreviewSettleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceBboApi
    */
    previewSettle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/bbo/previewSettle', data: requestParameters }));
        });
    }
    /**
    * 获取结算单详情
    * @summary 获取结算单详情
    * @param {FinanceBboApiQueryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceBboApi
    */
    queryDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/bbo/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 分页查询结算单
    * @summary 分页查询结算单
    * @param {FinanceBboApiQuerySettleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceBboApi
    */
    querySettle(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/bbo/querySettle', data: requestParameters }));
        });
    }
    /**
    * 撤销结算单
    * @summary 撤销结算单
    * @param {FinanceBboApiRevokeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FinanceBboApi
    */
    revoke(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/bbo/revoke', params: requestParameters }));
        });
    }
}
exports.SpotterFinanceBboApiSdk = SpotterFinanceBboApiSdk;
