// @spotter/gmesh-api-sdk/lib/api/first-payment-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFirstPaymentApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FirstPaymentApi - object-oriented interface
* @export
* @class FirstPaymentApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFirstPaymentApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 创建一期款
    * @summary 创建一期款
    * @param {FirstPaymentApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPayment/create', data: requestParameters }));
        });
    }
    /**
    * 导出预账单
    * @summary 导出预账单
    * @param {FirstPaymentApiExportAdvanceBillRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    exportAdvanceBill(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/advanceBill/exportAdvanceBill', data: requestParameters }));
        });
    }
    /**
    * 成本结算-导出明细
    * @summary 成本结算-导出明细
    * @param {FirstPaymentApiExportCostDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    exportCostDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/advanceBill/cost/exportDetail', data: requestParameters }));
        });
    }
    /**
    * 根据款项编码导出一期款详情
    * @summary 根据款项编码导出一期款详情
    * @param {FirstPaymentApiExportDetailByPaymentCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    exportDetailByPaymentCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode) !== undefined) {
                params['paymentCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/firstPayment/exportDetail', params: requestParameters }));
        });
    }
    /**
    * 导出预览一期款
    * @summary 导出预览一期款
    * @param {FirstPaymentApiExportPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    exportPreview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPaymentPreview/exportPreview', data: requestParameters }));
        });
    }
    /**
    * 自动生成一期款失败记录
    * @summary 【自动生成一期款】-失败记录
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    listAutoRecord(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPayment/auto/list' }));
        });
    }
    /**
    * 自动生成一期款失败记录
    * @summary 【自动生成一期款】-暂不处理
    * @param {FirstPaymentApiNotTreatRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    notTreat(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPayment/auto/not_treat', params: requestParameters }));
        });
    }
    /**
    * 预览一期款
    * @summary 预览一期款
    * @param {FirstPaymentApiPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    preview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPaymentPreview/preview', data: requestParameters }));
        });
    }
    /**
    * 根据条件查询一期款
    * @summary 根据条件查询一期款
    * @param {FirstPaymentApiQueryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    query(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPayment/query', data: requestParameters }));
        });
    }
    /**
    * 根据主键查询详情
    * @summary 根据主键查询详情
    * @param {FirstPaymentApiQueryByIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    queryById(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/advanceBill/queryById', params: requestParameters }));
        });
    }
    /**
    * 根据款项编码查询一期款详情
    * @summary 根据款项编码查询一期款详情
    * @param {FirstPaymentApiQueryDetailByPaymentCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    queryDetailByPaymentCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode) !== undefined) {
                params['paymentCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/firstPayment/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 查询指定公司最后一次一期款
    * @summary 查询指定公司最后一次一期款
    * @param {FirstPaymentApiQueryLastPaymentCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    queryLastPaymentCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/firstPayment/queryLastPaymentCode', params: requestParameters }));
        });
    }
    /**
    * 预账单列表
    * @summary 预账单列表
    * @param {FirstPaymentApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/advanceBill/queryPage', data: requestParameters }));
        });
    }
    /**
    * 查询预览一期款
    * @summary 查询预览一期款
    * @param {FirstPaymentApiQueryPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    queryPreview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPaymentPreview/queryPreview', data: requestParameters }));
        });
    }
    /**
    * 自动生成一期款失败记录
    * @summary 【自动生成一期款】-重新生成
    * @param {FirstPaymentApiRegenerateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    regenerate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPayment/auto/regenerate', params: requestParameters }));
        });
    }
    /**
    * 撤销一期款
    * @summary 撤销一期款
    * @param {FirstPaymentApiReverseRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FirstPaymentApi
    */
    reverse(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/firstPayment/reverse', data: requestParameters }));
        });
    }
}
exports.SpotterFirstPaymentApiSdk = SpotterFirstPaymentApiSdk;
