// @spotter/gmesh-api-sdk/lib/api/front-inbox-template-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFrontInboxTemplateEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FrontInboxTemplateEndpointApi - object-oriented interface
* @export
* @class FrontInboxTemplateEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFrontInboxTemplateEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 创建模板
    * @summary 创建模板
    * @param {FrontInboxTemplateEndpointApiCreateDevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxTemplateEndpointApi
    */
    createDevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/inboxTemplate/create', data: requestParameters }));
        });
    }
    /**
    * 删除模板
    * @summary 删除模板
    * @param {FrontInboxTemplateEndpointApiDeleteDevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxTemplateEndpointApi
    */
    deleteDevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/inboxTemplate/delete', data: requestParameters }));
        });
    }
    /**
    * 模板详情
    * @summary 模板详情
    * @param {FrontInboxTemplateEndpointApiDetailDevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxTemplateEndpointApi
    */
    detailDevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/inboxTemplate/detail', data: requestParameters }));
        });
    }
    /**
    * 分页查询模板
    * @summary 分页查询模板
    * @param {FrontInboxTemplateEndpointApiQueryPageDevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxTemplateEndpointApi
    */
    queryPageDevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/inboxTemplate/queryPage', data: requestParameters }));
        });
    }
    /**
    * 更新模板
    * @summary 更新模板
    * @param {FrontInboxTemplateEndpointApiUpdateDevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxTemplateEndpointApi
    */
    updateDevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/inboxTemplate/update', data: requestParameters }));
        });
    }
}
exports.SpotterFrontInboxTemplateEndpointApiSdk = SpotterFrontInboxTemplateEndpointApiSdk;
