// @spotter/gmesh-api-sdk/lib/api/front-user-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFrontUserEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FrontUserEndpointApi - object-oriented interface
* @export
* @class FrontUserEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFrontUserEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 通过钉钉直接跳转进来
    * @summary 钉钉登录跳转
    * @param {FrontUserEndpointApiDingCallbackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontUserEndpointApi
    */
    dingCallback(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.authCode) !== undefined) {
                params['authCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.authCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-message-web/develop/user/dingtalk/callback', params: requestParameters }));
        });
    }
    /**
    * 获取所有用户
    * @summary 获取所有用户
    * @param {*} [options] Override http request option.
    * @memberof FrontUserEndpointApi
    */
    getAllUser(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/user/getAllUser' }));
        });
    }
    /**
    * 获取用户信息
    * @summary 获取用户信息
    * @param {FrontUserEndpointApiGetUserInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontUserEndpointApi
    */
    getUserInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.unionId) !== undefined) {
                params['unionId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.unionId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-message-web/develop/user/getUserInfo', params: requestParameters }));
        });
    }
    /**
    * 退出登录
    * @summary 退出登录
    * @param {*} [options] Override http request option.
    * @memberof FrontUserEndpointApi
    */
    logout(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-message-web/develop/user/logout' }));
        });
    }
}
exports.SpotterFrontUserEndpointApiSdk = SpotterFrontUserEndpointApiSdk;
