// @spotter/gmesh-api-sdk/lib/api/front-inbox-msg-type-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFrontInboxMsgTypeEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FrontInboxMsgTypeEndpointApi - object-oriented interface
* @export
* @class FrontInboxMsgTypeEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFrontInboxMsgTypeEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 创建消息类型
    * @summary 创建消息类型
    * @param {FrontInboxMsgTypeEndpointApiCreatedevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxMsgTypeEndpointApi
    */
    createdevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/messageType/createdevelop', data: requestParameters }));
        });
    }
    /**
    * 删除消息类型
    * @summary 删除消息类型
    * @param {FrontInboxMsgTypeEndpointApiDeletedevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxMsgTypeEndpointApi
    */
    deletedevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/messageType/deletedevelop', data: requestParameters }));
        });
    }
    /**
    * 根据名称查询消息类型
    * @summary 根据名称查询消息类型
    * @param {FrontInboxMsgTypeEndpointApiQuerydevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxMsgTypeEndpointApi
    */
    querydevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/messageType/querydevelop', data: requestParameters }));
        });
    }
    /**
    * 更新消息类型
    * @summary 更新消息类型
    * @param {FrontInboxMsgTypeEndpointApiUpdatedevelopRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontInboxMsgTypeEndpointApi
    */
    updatedevelop(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/messageType/updatedevelop', data: requestParameters }));
        });
    }
}
exports.SpotterFrontInboxMsgTypeEndpointApiSdk = SpotterFrontInboxMsgTypeEndpointApiSdk;
