// @spotter/gmesh-api-sdk/lib/api/front-email-template-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFrontEmailTemplateEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FrontEmailTemplateEndpointApi - object-oriented interface
* @export
* @class FrontEmailTemplateEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFrontEmailTemplateEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 删除邮件模板
    * @summary 删除邮件模板
    * @param {FrontEmailTemplateEndpointApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontEmailTemplateEndpointApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/emailTemplate/delete', data: requestParameters }));
        });
    }
    /**
    * 创建邮件模板
    * @summary 创建邮件模板
    * @param {FrontEmailTemplateEndpointApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontEmailTemplateEndpointApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/emailTemplate/create', data: requestParameters }));
        });
    }
    /**
    * 查询详情邮件模板
    * @summary 查询详情邮件模板
    * @param {FrontEmailTemplateEndpointApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontEmailTemplateEndpointApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/emailTemplate/detail', data: requestParameters }));
        });
    }
    /**
    * 分页查询邮件模板
    * @summary 分页查询邮件模板
    * @param {FrontEmailTemplateEndpointApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontEmailTemplateEndpointApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/emailTemplate/queryPage', data: requestParameters }));
        });
    }
    /**
    * 试发邮件模板
    * @summary 试发邮件模板
    * @param {FrontEmailTemplateEndpointApiTrySendRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontEmailTemplateEndpointApi
    */
    trySend(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/emailTemplate/trySend', data: requestParameters }));
        });
    }
    /**
    * 更新邮件模板
    * @summary 更新邮件模板
    * @param {FrontEmailTemplateEndpointApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontEmailTemplateEndpointApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/emailTemplate/update', data: requestParameters }));
        });
    }
}
exports.SpotterFrontEmailTemplateEndpointApiSdk = SpotterFrontEmailTemplateEndpointApiSdk;
