// @spotter/gmesh-api-sdk/lib/api/front-sms-template-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFrontSmsTemplateEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FrontSmsTemplateEndpointApi - object-oriented interface
* @export
* @class FrontSmsTemplateEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFrontSmsTemplateEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 删除短信模板
    * @summary 删除短信模板
    * @param {FrontSmsTemplateEndpointApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontSmsTemplateEndpointApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/smsTemplate/delete', data: requestParameters }));
        });
    }
    /**
    * 创建短信模板
    * @summary 创建短信模板
    * @param {FrontSmsTemplateEndpointApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontSmsTemplateEndpointApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/smsTemplate/create', data: requestParameters }));
        });
    }
    /**
    * 查询详情短信模板
    * @summary 查询详情短信模板
    * @param {FrontSmsTemplateEndpointApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontSmsTemplateEndpointApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/smsTemplate/detail', data: requestParameters }));
        });
    }
    /**
    * 分页查询短信模板
    * @summary 分页查询短信模板
    * @param {FrontSmsTemplateEndpointApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontSmsTemplateEndpointApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/smsTemplate/queryPage', data: requestParameters }));
        });
    }
    /**
    * 试发短信模板
    * @summary 试发短信模板
    * @param {FrontSmsTemplateEndpointApiTrySendRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontSmsTemplateEndpointApi
    */
    trySend(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/smsTemplate/trySend', data: requestParameters }));
        });
    }
    /**
    * 更新短信模板
    * @summary 更新短信模板
    * @param {FrontSmsTemplateEndpointApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FrontSmsTemplateEndpointApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/develop/smsTemplate/update', data: requestParameters }));
        });
    }
}
exports.SpotterFrontSmsTemplateEndpointApiSdk = SpotterFrontSmsTemplateEndpointApiSdk;
