// @spotter/gmesh-api-sdk/lib/api/fulfill-retail-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterFulfillRetailApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* FulfillRetailApi - object-oriented interface
* @export
* @class FulfillRetailApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterFulfillRetailApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 订单取消
    * @summary 订单取消
    * @param {FulfillRetailApiCancelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    cancel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/cancel', data: requestParameters }));
        });
    }
    /**
    * 订单确认
    * @summary 订单确认
    * @param {FulfillRetailApiConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    confirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/confirm', data: requestParameters }));
        });
    }
    /**
    * 订单确认预校验
    * @summary 订单确认预校验
    * @param {FulfillRetailApiConfirmPreCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    confirmPreCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/confirm/preCheck', data: requestParameters }));
        });
    }
    /**
    * 订单详情
    * @summary 订单详情
    * @param {FulfillRetailApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/detail', data: requestParameters }));
        });
    }
    /**
    * 订单导出
    * @summary 订单导出
    * @param {FulfillRetailApiExportExcelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    exportExcel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/exportExcel', data: requestParameters }));
        });
    }
    /**
    * 账号列表
    * @summary 账号列表
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    listAccount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/list/account' }));
        });
    }
    /**
    * 订单单条编辑
    * @summary 订单单条编辑
    * @param {FulfillRetailApiModifyStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    modifyStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/modifyStorage', data: requestParameters }));
        });
    }
    /**
    * 订单分页
    * @summary 订单分页
    * @param {FulfillRetailApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/page', data: requestParameters }));
        });
    }
    /**
    * 订单发货
    * @summary 订单发货
    * @param {FulfillRetailApiShipRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    ship(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/ship', data: requestParameters }));
        });
    }
    /**
    * 订单统计
    * @summary 订单统计
    * @param {FulfillRetailApiStatisticsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof FulfillRetailApi
    */
    statistics(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/statistics', data: requestParameters }));
        });
    }
}
exports.SpotterFulfillRetailApiSdk = SpotterFulfillRetailApiSdk;
