// @spotter/gmesh-api-sdk/lib/api/gmesh-marketing-pd-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshMarketingPdApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshMarketingPdApi - object-oriented interface
* @export
* @class GmeshMarketingPdApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshMarketingPdApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导出PriceDiscount活动列表
    * @summary 导出PriceDiscount活动列表
    * @param {GmeshMarketingPdApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingPdApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/marketing/pd/export', data: requestParameters }));
        });
    }
    /**
    * 获取PriceDiscount活动详情
    * @summary 获取PriceDiscount活动详情
    * @param {GmeshMarketingPdApiGetPriceDiscountDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingPdApi
    */
    getPriceDiscountDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/marketing/pd/detail', data: requestParameters }));
        });
    }
    /**
    * 获取PriceDiscount操作历史列表
    * @summary 获取PriceDiscount操作历史列表
    * @param {GmeshMarketingPdApiGetPriceDiscountOperationRecordPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingPdApi
    */
    getPriceDiscountOperationRecordPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/marketing/pd/historys', data: requestParameters }));
        });
    }
    /**
    * 获取PriceDiscount活动列表
    * @summary 获取PriceDiscount活动列表
    * @param {GmeshMarketingPdApiGetPriceDiscountPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingPdApi
    */
    getPriceDiscountPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/marketing/pd', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshMarketingPdApiSdk = SpotterGmeshMarketingPdApiSdk;
