// @spotter/gmesh-api-sdk/lib/api/gmesh-catalog-offer-template-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshCatalogOfferTemplateApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshCatalogOfferTemplateApi - object-oriented interface
* @export
* @class GmeshCatalogOfferTemplateApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshCatalogOfferTemplateApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 获取模版下载路径
    * @summary 获取模版下载路径
    * @param {GmeshCatalogOfferTemplateApiGetTemplateFilePathRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferTemplateApi
    */
    getTemplateFilePath(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/template/dwonload/filePath', data: requestParameters }));
        });
    }
    /**
    * 获取模版列表
    * @summary 获取模版列表
    * @param {GmeshCatalogOfferTemplateApiGetTemplateListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferTemplateApi
    */
    getTemplateList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/template/list', data: requestParameters }));
        });
    }
    /**
    * 获取模版数量
    * @summary 获取模版数量
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferTemplateApi
    */
    getTemplateTotalCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-items-web/gmesh/catalog/offer/template/count' }));
        });
    }
    /**
    * 拉取最新模版
    * @summary 拉取最新模版
    * @param {GmeshCatalogOfferTemplateApiSyncTemplateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferTemplateApi
    */
    syncTemplate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/template/sync', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshCatalogOfferTemplateApiSdk = SpotterGmeshCatalogOfferTemplateApiSdk;
