// @spotter/gmesh-api-sdk/lib/api/gmesh-ai-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshAiApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshAiApi - object-oriented interface
* @export
* @class GmeshAiApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshAiApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 拉取会话消息历史记录（分页查询）
    * @summary 拉取会话消息历史记录（分页查询）
    * @param {GmeshAiApiFirstCheckAndReturnPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    firstCheckAndReturnPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/ai/msg/page', data: requestParameters }));
        });
    }
    /**
    * 获取会话消息详情
    * @summary 获取会话消息详情
    * @param {GmeshAiApiGetAiChatMsgListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    getAiChatMsgList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.chatId) !== undefined) {
                params['chatId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.chatId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/ai/chat/msg/list', params: requestParameters }));
        });
    }
    /**
    * 根据消息ID获取消息”
    * @summary 根据消息ID获取消息”
    * @param {GmeshAiApiGetMsgByIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    getMsgById(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/ai/msg', params: requestParameters }));
        });
    }
    /**
    * SSE推送“问题”和“答案”
    * @summary SSE推送“问题”和“答案”
    * @param {GmeshAiApiPublishEventStreamRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    publishEventStream(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.deviceId) !== undefined) {
                params['deviceId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.deviceId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.windowId) !== undefined) {
                params['windowId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.windowId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/ai/event/stream', params: requestParameters }));
        });
    }
    /**
    * 客户提交问题
    * @summary 客户提交问题
    * @param {GmeshAiApiQuestionRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    question(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/ai/question', data: requestParameters }));
        });
    }
    /**
    * 质量反馈并返回更新后的消息记录
    * @summary 质量反馈并返回更新后的消息记录
    * @param {GmeshAiApiSetAnswerQualityRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    setAnswerQuality(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/ai/answer/quality', data: requestParameters }));
        });
    }
    /**
    * 查询会话详情
    * @summary 查询会话详情
    * @param {GmeshAiApiTicketAiChatDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    ticketAiChatDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.chatId) !== undefined) {
                params['chatId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.chatId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/ai/chat/detail', params: requestParameters }));
        });
    }
    /**
    * 分页条件查询会话列表
    * @summary 分页条件查询会话列表
    * @param {GmeshAiApiTicketAiChatPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    ticketAiChatPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/ai/chat/page', data: requestParameters }));
        });
    }
    /**
    * Wiki搜索Token初始化
    * @summary Wiki搜索Token初始化
    * @param {GmeshAiApiWikiAccessTokenInitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshAiApi
    */
    wikiAccessTokenInit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.code) !== undefined) {
                params['code'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.code;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/ai/wiki/token', params: requestParameters }));
        });
    }
}
exports.SpotterGmeshAiApiSdk = SpotterGmeshAiApiSdk;
