// @spotter/gmesh-api-sdk/lib/api/gmesh-consulting-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshConsultingApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshConsultingApi - object-oriented interface
* @export
* @class GmeshConsultingApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshConsultingApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 查询咨询工单消息记录
    * @summary 查询咨询工单消息记录
    * @param {GmeshConsultingApiGetTicketConsultingMsgListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshConsultingApi
    */
    getTicketConsultingMsgList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/consulting/msg/list', data: requestParameters }));
        });
    }
    /**
    * 标签列表
    * @summary 标签列表
    * @param {GmeshConsultingApiLabelListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshConsultingApi
    */
    labelList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/consulting/label/list', data: requestParameters }));
        });
    }
    /**
    * 问题类型树
    * @summary 问题类型树
    * @param {GmeshConsultingApiQuestionTypeTreeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshConsultingApi
    */
    questionTypeTree(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.processCategoryId) !== undefined) {
                params['processCategoryId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.processCategoryId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/consulting/question/list', params: requestParameters }));
        });
    }
    /**
    * 添加备注
    * @summary 添加备注
    * @param {GmeshConsultingApiRemarkRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshConsultingApi
    */
    remark(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/consulting/remark', data: requestParameters }));
        });
    }
    /**
    * 数据报告类型
    * @summary 数据报告类型
    * @param {*} [options] Override http request option.
    * @memberof GmeshConsultingApi
    */
    reportList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/consulting/ticket/report/list' }));
        });
    }
    /**
    * 任务批量转移(咨询类工单)
    * @summary 任务批量转移(咨询类工单)
    * @param {GmeshConsultingApiResetAssigneeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshConsultingApi
    */
    resetAssignee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/consulting/assignee', data: requestParameters }));
        });
    }
    /**
    * 设置标签
    * @summary 设置标签
    * @param {GmeshConsultingApiSetLabelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshConsultingApi
    */
    setLabel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/consulting/label/set', data: requestParameters }));
        });
    }
    /**
    * 查询操作记录
    * @summary 查询操作记录
    * @param {GmeshConsultingApiTicketOperationRecordListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshConsultingApi
    */
    ticketOperationRecordList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.instanceId) !== undefined) {
                params['instanceId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.instanceId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/consulting/operation/records', params: requestParameters }));
        });
    }
}
exports.SpotterGmeshConsultingApiSdk = SpotterGmeshConsultingApiSdk;
