// @spotter/gmesh-api-sdk/lib/api/gmesh-process-def-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshProcessDefApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshProcessDefApi - object-oriented interface
* @export
* @class GmeshProcessDefApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshProcessDefApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导入流程定义文件
    * @summary 导入流程定义文件
    * @param {GmeshProcessDefApiImportProcessFileRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshProcessDefApi
    */
    importProcessFile(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/process/def/import', data: requestParameters }));
        });
    }
    /**
    * 保存流程设计器内的xml文件
    * @summary 保存流程设计器内的xml文件
    * @param {GmeshProcessDefApiSaveProcessDefinitionRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshProcessDefApi
    */
    saveProcessDefinition(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/process/def/save', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshProcessDefApiSdk = SpotterGmeshProcessDefApiSdk;
