// @spotter/gmesh-api-sdk/lib/api/gmesh-shipment-amz-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshShipmentAmzApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshShipmentAmzApi - object-oriented interface
* @export
* @class GmeshShipmentAmzApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshShipmentAmzApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 取消shipment
    * @summary 取消shipment
    * @param {GmeshShipmentAmzApiCancelShipmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    cancelShipment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/amz/cancel', data: requestParameters }));
        });
    }
    /**
    * 导出物货运信息
    * @summary 导出货运信息
    * @param {GmeshShipmentAmzApiExportShipmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    exportShipment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/export', data: requestParameters }));
        });
    }
    /**
    * shipment详情
    * @summary shipment详情
    * @param {GmeshShipmentAmzApiGetShipmentDetailWithItemsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    getShipmentDetailWithItems(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/detail', data: requestParameters }));
        });
    }
    /**
    * shipment 获取下载文件路径
    * @summary shipment 获取下载文件路径
    * @param {GmeshShipmentAmzApiGetShipmentFileDownloadPathRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    getShipmentFileDownloadPath(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/download', data: requestParameters }));
        });
    }
    /**
    * shipment列表
    * @summary shipment列表
    * @param {GmeshShipmentAmzApiGetShipmentListWithItemsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    getShipmentListWithItems(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/list', data: requestParameters }));
        });
    }
    /**
    * shipment发货清单数据
    * @summary shipment发货清单数据
    * @param {GmeshShipmentAmzApiGetShipmentShippingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    getShipmentShippingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/shipping/list', data: requestParameters }));
        });
    }
    /**
    * shipment状态
    * @summary shipment状态
    * @param {GmeshShipmentAmzApiGetShipmentStateInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    getShipmentStateInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/progress', data: requestParameters }));
        });
    }
    /**
    * po/df 已发送邮件列表详情
    * @summary shipment 邮件详情
    * @param {GmeshShipmentAmzApiListEmailDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    listEmailDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/email/detail/list', data: requestParameters }));
        });
    }
    /**
    * po/df 已发送邮件列表
    * @summary shipment 邮件列表
    * @param {GmeshShipmentAmzApiListEmailHistoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    listEmailHistory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/email/list', data: requestParameters }));
        });
    }
    /**
    * 手动到货
    * @summary 手动到货
    * @param {GmeshShipmentAmzApiManualArrivalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    manualArrival(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/manualArrival', data: requestParameters }));
        });
    }
    /**
    * 到货异常分页
    * @summary 到货异常分页
    * @param {GmeshShipmentAmzApiPageShipmentArrivalAbnormalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    pageShipmentArrivalAbnormal(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/page/arrivalAbnormal', data: requestParameters }));
        });
    }
    /**
    * po邮件页面重新打包文件
    * @summary 重新打包文件
    * @param {GmeshShipmentAmzApiRepackageFileRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    repackageFile(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/email/file/repackage', data: requestParameters }));
        });
    }
    /**
    * po/df 邮件页面重新发送
    * @summary 重新发送邮件
    * @param {GmeshShipmentAmzApiResendMailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzApi
    */
    resendMail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/email/resend', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshShipmentAmzApiSdk = SpotterGmeshShipmentAmzApiSdk;
