// @spotter/gmesh-api-sdk/lib/api/gmesh-catalog-offer-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshCatalogOfferApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshCatalogOfferApi - object-oriented interface
* @export
* @class GmeshCatalogOfferApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshCatalogOfferApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * Offer字段对比结果
    * @summary Offer字段对比结果
    * @param {GmeshCatalogOfferApiCatalogContrastRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    catalogContrast(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/contrast/field', data: requestParameters }));
        });
    }
    /**
    * 克隆df offer
    * @summary 克隆df offer
    * @param {GmeshCatalogOfferApiCloneOfferRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    cloneOffer(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/clone', data: requestParameters }));
        });
    }
    /**
    * 删除offer
    * @summary 删除offer
    * @param {GmeshCatalogOfferApiDeleteOfferRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    deleteOffer(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-items-web/gmesh/catalog/offer', data: requestParameters }));
        });
    }
    /**
    * 有效Offer列表
    * @summary 有效Offer列表
    * @param {GmeshCatalogOfferApiGetActiveOfferListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    getActiveOfferList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/list/active', data: requestParameters }));
        });
    }
    /**
    * 当前用户（包含所有下级）管理公司所对应的所有offer列表
    * @summary 当前用户管理的Offer分页列表
    * @param {GmeshCatalogOfferApiGetManagementActiveOfferListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    getManagementActiveOfferList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/management/active/list', data: requestParameters }));
        });
    }
    /**
    * offer详情
    * @summary offer详情
    * @param {GmeshCatalogOfferApiGetOfferDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    getOfferDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/detail', data: requestParameters }));
        });
    }
    /**
    * Offer分页列表
    * @summary Offer分页列表
    * @param {GmeshCatalogOfferApiGetOfferListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    getOfferList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/list', data: requestParameters }));
        });
    }
    /**
    * offer信息获取失败详情
    * @summary offer信息获取失败详情
    * @param {GmeshCatalogOfferApiGetOfferValidationFailDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    getOfferValidationFailDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/validation/fail/detail', data: requestParameters }));
        });
    }
    /**
    * 查看Offer流程历史记录详情
    * @summary 查看Offer流程历史记录详情
    * @param {GmeshCatalogOfferApiHistoryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    historyDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/history/detail', data: requestParameters }));
        });
    }
    /**
    * 查看Offer流程历史记录分页列表
    * @summary 查看Offer流程历史记录分页列表
    * @param {GmeshCatalogOfferApiHistoryListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    historyList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/history/list', data: requestParameters }));
        });
    }
    /**
    * 可以clone的offer列表Offer
    * @summary 可以clone的offer列表Offer
    * @param {GmeshCatalogOfferApiPageOfferToCloneRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    pageOfferToClone(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/page/to/clone', data: requestParameters }));
        });
    }
    /**
    * 同步offer信息
    * @summary 同步offer信息
    * @param {GmeshCatalogOfferApiSyncOfferRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferApi
    */
    syncOffer(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/sync', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshCatalogOfferApiSdk = SpotterGmeshCatalogOfferApiSdk;
