// @spotter/gmesh-api-sdk/lib/api/gmesh-marketing-coupon-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshMarketingCouponApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshMarketingCouponApi - object-oriented interface
* @export
* @class GmeshMarketingCouponApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshMarketingCouponApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 优惠券活动导出
    * @summary 优惠券活动导出
    * @param {GmeshMarketingCouponApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingCouponApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/coupon/export', data: requestParameters }));
        });
    }
    /**
    * 获取优惠券详情
    * @summary 获取优惠券详情
    * @param {GmeshMarketingCouponApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingCouponApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/coupon/detail', data: requestParameters }));
        });
    }
    /**
    * 获取各个 tab 的数量
    * @summary 获取各个 tab 的数量
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingCouponApi
    */
    getCouponCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-marketing-web/gmesh/coupon/getCouponCount' }));
        });
    }
    /**
    * 获取Coupon列表
    * @summary 获取Coupon列表
    * @param {GmeshMarketingCouponApiSelectCouponListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingCouponApi
    */
    selectCouponList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/coupon/selectCouponList', data: requestParameters }));
        });
    }
    /**
    * 分页获取Coupon列表
    * @summary 分页获取Coupon列表
    * @param {GmeshMarketingCouponApiSelectCouponPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingCouponApi
    */
    selectCouponPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/coupon/selectCouponPage', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshMarketingCouponApiSdk = SpotterGmeshMarketingCouponApiSdk;
