// @spotter/gmesh-api-sdk/lib/api/gmesh-promotion-code-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshPromotionCodeApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshPromotionCodeApi - object-oriented interface
* @export
* @class GmeshPromotionCodeApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshPromotionCodeApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导出列表
    * @summary 导出列表
    * @param {GmeshPromotionCodeApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshPromotionCodeApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/pc/export', data: requestParameters }));
        });
    }
    /**
    * 详情
    * @summary 详情
    * @param {GmeshPromotionCodeApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshPromotionCodeApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-marketing-web/gmesh/pc/detail', params: requestParameters }));
        });
    }
    /**
    * 获取PromotionCode操作历史列表
    * @summary 获取PromotionCode操作历史列表
    * @param {GmeshPromotionCodeApiHistoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshPromotionCodeApi
    */
    history(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/pc/history', data: requestParameters }));
        });
    }
    /**
    * 分页列表
    * @summary 分页列表
    * @param {GmeshPromotionCodeApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshPromotionCodeApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/pc/list', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshPromotionCodeApiSdk = SpotterGmeshPromotionCodeApiSdk;
