// @spotter/gmesh-api-sdk/lib/api/gmesh-marketing-bd-promotion-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshMarketingBdPromotionApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshMarketingBdPromotionApi - object-oriented interface
* @export
* @class GmeshMarketingBdPromotionApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshMarketingBdPromotionApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导出活动列表
    * @summary 导出活动列表
    * @param {GmeshMarketingBdPromotionApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/export', data: requestParameters }));
        });
    }
    /**
    * 活动创建
    * @summary 活动创建
    * @param {GmeshMarketingBdPromotionApiAddRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    add(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/create', data: requestParameters }));
        });
    }
    /**
    * 活动取消
    * @summary 活动取消
    * @param {GmeshMarketingBdPromotionApiCancelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    cancel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/cancel', data: requestParameters }));
        });
    }
    /**
    * 通过amz活动id获取活动详情
    * @summary 通过amz活动id获取活动详情
    * @param {GmeshMarketingBdPromotionApiDetailByAmazonPromotionIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    detailByAmazonPromotionId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/amz/detail', data: requestParameters }));
        });
    }
    /**
    * 通过活动id获取活动详情
    * @summary 通过活动id获取活动详情
    * @param {GmeshMarketingBdPromotionApiDetailByPromotionIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    detailByPromotionId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/detailById', data: requestParameters }));
        });
    }
    /**
    * 批量同步
    * @summary 批量同步
    * @param {GmeshMarketingBdPromotionApiPromotionDetailSyncRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    promotionDetailSync(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/sync', data: requestParameters }));
        });
    }
    /**
    * 分业获取活动列表
    * @summary 分业获取活动列表
    * @param {GmeshMarketingBdPromotionApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/queryPage', data: requestParameters }));
        });
    }
    /**
    * 活动转移处理人
    * @summary 活动转移处理人
    * @param {GmeshMarketingBdPromotionApiShiftRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    shift(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/shift', data: requestParameters }));
        });
    }
    /**
    * 活动更新
    * @summary 活动更新
    * @param {GmeshMarketingBdPromotionApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdPromotionApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/promotion/update', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshMarketingBdPromotionApiSdk = SpotterGmeshMarketingBdPromotionApiSdk;
