// @spotter/gmesh-api-sdk/lib/api/gmesh-spt-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshSptApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshSptApi - object-oriented interface
* @export
* @class GmeshSptApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshSptApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 确认预计送达时间
    * @summary 确认预计送达时间
    * @param {GmeshSptApiConfirmDeliveryDateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshSptApi
    */
    confirmDeliveryDate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ess-web/gmesh/spt/order/confirm', data: requestParameters }));
        });
    }
    /**
    * 【】修改SPT物流
    * @summary 【】修改SPT物流
    * @param {GmeshSptApiModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshSptApi
    */
    modify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ess-web/gmesh/spt/order/modify', data: requestParameters }));
        });
    }
    /**
    * 查询SPT物流订单详情
    * @summary 查询SPT物流订单详情
    * @param {GmeshSptApiOrderDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshSptApi
    */
    orderDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ess-web/gmesh/spt/order/detail', data: requestParameters }));
        });
    }
    /**
    * SPT物流订单Excel导出
    * @summary SPT物流订单Excel导出
    * @param {GmeshSptApiOrderExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshSptApi
    */
    orderExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ess-web/gmesh/spt/order/export', data: requestParameters }));
        });
    }
    /**
    * 分页条件查询Spt物流
    * @summary 分页条件查询Spt物流
    * @param {GmeshSptApiOrderPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshSptApi
    */
    orderPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ess-web/gmesh/spt/order/page', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshSptApiSdk = SpotterGmeshSptApiSdk;
