// @spotter/gmesh-api-sdk/lib/api/gmesh-marketing-bd-product-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshMarketingBdProductApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshMarketingBdProductApi - object-oriented interface
* @export
* @class GmeshMarketingBdProductApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshMarketingBdProductApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 创建sas
    * @summary 创建sas
    * @param {GmeshMarketingBdProductApiAddSasRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    addSas(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/add/sas', data: requestParameters }));
        });
    }
    /**
    * 产品预期时间检测
    * @summary 产品检测
    * @param {GmeshMarketingBdProductApiCheckProductIsActiveInExpectTimeRangeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    checkProductIsActiveInExpectTimeRange(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/check', data: requestParameters }));
        });
    }
    /**
    * 数据变更记录
    * @summary 数据变更记录
    * @param {GmeshMarketingBdProductApiDataChangeListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    dataChangeList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/data/change/list', data: requestParameters }));
        });
    }
    /**
    * 产品池-Financial报错-分页查询
    * @summary 产品池-Financial报错-分页查询
    * @param {GmeshMarketingBdProductApiFinancialThresholdProductPageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    financialThresholdProductPageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/financial/page', data: requestParameters }));
        });
    }
    /**
    * 根据BD 活动主键 ID 活动活动的产品列表
    * @summary 根据BD 活动主键 ID 活动活动的产品列表
    * @param {GmeshMarketingBdProductApiGetProductByPromotionIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    getProductByPromotionId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/listByPromotionId', data: requestParameters }));
        });
    }
    /**
    * 最近一次上传失败的产品数据
    * @summary 最近一次上传失败的产品数据
    * @param {GmeshMarketingBdProductApiLatestFailUploadProductListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    latestFailUploadProductList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/latest/fail/upload/list', data: requestParameters }));
        });
    }
    /**
    * 产品添加备注
    * @summary 产品添加备注
    * @param {GmeshMarketingBdProductApiProductAddRemarkRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    productAddRemark(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/addRemark', data: requestParameters }));
        });
    }
    /**
    * 产品申请
    * @summary 产品申请
    * @param {GmeshMarketingBdProductApiProductApplyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    productApply(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/apply', data: requestParameters }));
        });
    }
    /**
    * 产品删除
    * @summary 产品删除
    * @param {GmeshMarketingBdProductApiProductDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    productDelete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-marketing-web/gmesh/bd/product/delete', data: requestParameters }));
        });
    }
    /**
    * 手动通过
    * @summary 手动通过
    * @param {GmeshMarketingBdProductApiProductManualApprovedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    productManualApproved(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/manual/approved', data: requestParameters }));
        });
    }
    /**
    * 批量手动反馈
    * @summary 手动反馈
    * @param {GmeshMarketingBdProductApiProductManualFeedbackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    productManualFeedback(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/manual/feedback', data: requestParameters }));
        });
    }
    /**
    * 提交失败
    * @summary 提交失败
    * @param {GmeshMarketingBdProductApiProductManualUploadFailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    productManualUploadFail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/manual/upload/fail', data: requestParameters }));
        });
    }
    /**
    * 转移处理人
    * @summary 转移处理人
    * @param {GmeshMarketingBdProductApiProductShiftRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    productShift(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/shift', data: requestParameters }));
        });
    }
    /**
    * 产品池-需要处理-分页查询
    * @summary 产品池-需要处理-分页查询
    * @param {GmeshMarketingBdProductApiToDoProductPageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    toDoProductPageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/todo/page', data: requestParameters }));
        });
    }
    /**
    * 产品池-提交失败-分页查询
    * @summary 产品池-提交失败-分页查询
    * @param {GmeshMarketingBdProductApiUploadFailProductPageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    uploadFailProductPageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/upload/fail/list', data: requestParameters }));
        });
    }
    /**
    * 产品校验
    * @summary 产品校验
    * @param {GmeshMarketingBdProductApiValidationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingBdProductApi
    */
    validation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/bd/product/validation', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshMarketingBdProductApiSdk = SpotterGmeshMarketingBdProductApiSdk;
