// @spotter/gmesh-api-sdk/lib/api/gmesh-marketing-ld-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshMarketingLdApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshMarketingLdApi - object-oriented interface
* @export
* @class GmeshMarketingLdApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshMarketingLdApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导出LightningDeal活动列表
    * @summary 导出LightningDeal活动列表
    * @param {GmeshMarketingLdApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingLdApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/marketing/ld/export', data: requestParameters }));
        });
    }
    /**
    * 根据ID查询LD详情
    * @summary 根据ID查询LD详情
    * @param {GmeshMarketingLdApiFindLightningDealByIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingLdApi
    */
    findLightningDealById(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-marketing-web/gmesh/marketing/ld', params: requestParameters }));
        });
    }
    /**
    * 获取LightningDeal操作历史列表
    * @summary 获取LightningDeal操作历史列表
    * @param {GmeshMarketingLdApiFindLightningDealOperationRecordPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingLdApi
    */
    findLightningDealOperationRecordPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/marketing/ld/history', data: requestParameters }));
        });
    }
    /**
    * 分页条件查询LightningDeal活动列表
    * @summary 分页条件查询LightningDeal活动列表
    * @param {GmeshMarketingLdApiFindLightningDealPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshMarketingLdApi
    */
    findLightningDealPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-marketing-web/gmesh/marketing/ld', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshMarketingLdApiSdk = SpotterGmeshMarketingLdApiSdk;
