// @spotter/gmesh-api-sdk/lib/api/gmesh-process-instance-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshProcessInstanceApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshProcessInstanceApi - object-oriented interface
* @export
* @class GmeshProcessInstanceApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshProcessInstanceApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 查询流程实例详情
    * @summary 查询流程实例详情
    * @param {GmeshProcessInstanceApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshProcessInstanceApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.instanceId) !== undefined) {
                params['instanceId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.instanceId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.businessKey) !== undefined) {
                params['businessKey'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.businessKey;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/gmesh/process/instance/detail', params: requestParameters }));
        });
    }
    /**
    * 条件查询统计流程实例
    * @summary 条件查询统计流程实例
    * @param {GmeshProcessInstanceApiInstanceCountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshProcessInstanceApi
    */
    instanceCount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/process/instance/count', data: requestParameters }));
        });
    }
    /**
    * 分页条件查询流程实例列表
    * @summary 分页条件查询流程实例列表
    * @param {GmeshProcessInstanceApiPageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshProcessInstanceApi
    */
    pageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/gmesh/process/instance/page', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshProcessInstanceApiSdk = SpotterGmeshProcessInstanceApiSdk;
