// @spotter/gmesh-api-sdk/lib/api/gmesh-catalog-offer-batch-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshCatalogOfferBatchApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshCatalogOfferBatchApi - object-oriented interface
* @export
* @class GmeshCatalogOfferBatchApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshCatalogOfferBatchApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 模版分页-模版下载
    * @summary 模版分页-模版下载
    * @param {GmeshCatalogOfferBatchApiBatchOfferTemplateDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferBatchApi
    */
    batchOfferTemplateDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/batch/template/download', data: requestParameters }));
        });
    }
    /**
    * 模版分页
    * @summary 模版分页
    * @param {GmeshCatalogOfferBatchApiBatchOfferTemplatePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferBatchApi
    */
    batchOfferTemplatePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/batch/template/page', data: requestParameters }));
        });
    }
    /**
    * 模版分页-产品信息分页
    * @summary 模版分页-产品信息分页
    * @param {GmeshCatalogOfferBatchApiBatchOfferTemplateProductPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferBatchApi
    */
    batchOfferTemplateProductPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/batch/template/product/page', data: requestParameters }));
        });
    }
    /**
    * 上传分页
    * @summary 上传分页
    * @param {GmeshCatalogOfferBatchApiBatchOfferUploadPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferBatchApi
    */
    batchOfferUploadPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/batch/upload/page', data: requestParameters }));
        });
    }
    /**
    * 上传分页-产品信息分页
    * @summary 上传分页-产品信息分页
    * @param {GmeshCatalogOfferBatchApiBatchOfferUploadProductPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferBatchApi
    */
    batchOfferUploadProductPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/batch/upload/product/page', data: requestParameters }));
        });
    }
    /**
    * 上传分页-表格下载
    * @summary 上传分页-表格下载
    * @param {GmeshCatalogOfferBatchApiBatchOfferUploadTemplateDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof GmeshCatalogOfferBatchApi
    */
    batchOfferUploadTemplateDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-items-web/gmesh/catalog/offer/batch/upload/template/download', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshCatalogOfferBatchApiSdk = SpotterGmeshCatalogOfferBatchApiSdk;
