// @spotter/gmesh-api-sdk/lib/api/gmesh-shipment-amz-logistics-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterGmeshShipmentAmzLogisticsApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* GmeshShipmentAmzLogisticsApi - object-oriented interface
* @export
* @class GmeshShipmentAmzLogisticsApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterGmeshShipmentAmzLogisticsApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 下载到货证明
    * @summary 下载到货证明
    * @param {GmeshShipmentAmzLogisticsApiDownloadArrivalCertificateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzLogisticsApi
    */
    downloadArrivalCertificate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/logistics/download/zip', data: requestParameters }));
        });
    }
    /**
    * 导出异常到货运单号
    * @summary 导出异常到货运单号
    * @param {GmeshShipmentAmzLogisticsApiExportAbnormalProNoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzLogisticsApi
    */
    exportAbnormalProNo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/logistics/export/abnormal', data: requestParameters }));
        });
    }
    /**
    * VCDF导出运单号
    * @summary VCDF导出运单号
    * @param {GmeshShipmentAmzLogisticsApiExportLogisticsDfListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzLogisticsApi
    */
    exportLogisticsDfList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/logistics/df/export', data: requestParameters }));
        });
    }
    /**
    * VCPO导出运单号
    * @summary VCPO导出运单号
    * @param {GmeshShipmentAmzLogisticsApiExportLogisticsListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzLogisticsApi
    */
    exportLogisticsList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/logistics/export', data: requestParameters }));
        });
    }
    /**
    * shipment logistics 列表
    * @summary shipment logistics 列表
    * @param {GmeshShipmentAmzLogisticsApiGetShipmentLogisticsListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof GmeshShipmentAmzLogisticsApi
    */
    getShipmentLogisticsList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/gmesh/shipment/amz/logistics/list', data: requestParameters }));
        });
    }
}
exports.SpotterGmeshShipmentAmzLogisticsApiSdk = SpotterGmeshShipmentAmzLogisticsApiSdk;
