// @spotter/gmesh-api-sdk/lib/api/imodifi-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterIModifiEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* IModifiEndpointApi - object-oriented interface
* @export
* @class IModifiEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterIModifiEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 支付确认
    * @summary 支付确认
    * @param {IModifiEndpointApiConfirmTradePaymentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    confirmTradePayment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/trade/paid/confirm', data: requestParameters }));
        });
    }
    /**
    * 查询回款明细
    * @summary 查询回款明细
    * @param {IModifiEndpointApiQueryRemittanceDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryRemittanceDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentNo) !== undefined) {
                params['paymentNo'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentNo;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-plutus-web/modifi/remittance/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 分页查询回款列表
    * @summary 分页查询回款列表
    * @param {IModifiEndpointApiQueryRemittancePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryRemittancePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/remittance/queryPage', data: requestParameters }));
        });
    }
    /**
    * 导出回款列表
    * @summary 导出回款列表
    * @param {IModifiEndpointApiQueryRemittancePageExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryRemittancePageExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/remittance/export', data: requestParameters }));
        });
    }
    /**
    * 查询回款列表统计数据
    * @summary 查询回款列表统计数据
    * @param {IModifiEndpointApiQueryRemittanceStatisticsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryRemittanceStatistics(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/remittance/statistics', data: requestParameters }));
        });
    }
    /**
    * 导出交易核对列表
    * @summary 导出交易核对列表
    * @param {IModifiEndpointApiQueryTradeCheckExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryTradeCheckExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/trade/check/export', data: requestParameters }));
        });
    }
    /**
    * 分页查询交易核对列表
    * @summary 分页查询交易核对列表
    * @param {IModifiEndpointApiQueryTradeCheckPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryTradeCheckPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/trade/check/queryPage', data: requestParameters }));
        });
    }
    /**
    * 查询交易明细
    * @summary 查询交易明细
    * @param {IModifiEndpointApiQueryTradeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryTradeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.tradeNo) !== undefined) {
                params['tradeNo'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.tradeNo;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-plutus-web/modifi/trade/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 导出交易列表
    * @summary 导出交易列表
    * @param {IModifiEndpointApiQueryTradeExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryTradeExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/trade/export', data: requestParameters }));
        });
    }
    /**
    * 分页查询交易列表
    * @summary 分页查询交易列表
    * @param {IModifiEndpointApiQueryTradePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryTradePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/trade/queryPage', data: requestParameters }));
        });
    }
    /**
    * 查询交易应付金额合计
    * @summary 查询交易应付金额合计
    * @param {IModifiEndpointApiQueryTradePaidTotalRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof IModifiEndpointApi
    */
    queryTradePaidTotal(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-plutus-web/modifi/trade/queryPaidTotal', data: requestParameters }));
        });
    }
}
exports.SpotterIModifiEndpointApiSdk = SpotterIModifiEndpointApiSdk;
