// @spotter/gmesh-api-sdk/lib/api/instance-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterInstanceApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* InstanceApi - object-oriented interface
* @export
* @class InstanceApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterInstanceApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 完成节点
    * @summary 完成节点
    * @param {InstanceApiCompleteAnyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InstanceApi
    */
    completeAny(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/instance/complete', data: requestParameters }));
        });
    }
    /**
    * 流程图片生成
    * @summary 流程图片生成
    * @param {InstanceApiGetProcessResourceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InstanceApi
    */
    getProcessResource(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.processInstanceId) !== undefined) {
                params['processInstanceId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.processInstanceId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/gmesh-web/instance/img', params: requestParameters }));
        });
    }
    /**
    * 查看流程引擎历史信息
    * @summary 查看流程信息
    * @param {InstanceApiProcessInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InstanceApi
    */
    processInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.processInstanceId) !== undefined) {
                params['processInstanceId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.processInstanceId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/gmesh-web/instance/info', params: requestParameters }));
        });
    }
    /**
    * 开始一个新的流程
    * @summary 开始一个新的流程
    * @param {InstanceApiStartProcessRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InstanceApi
    */
    startProcess(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/instance/start', data: requestParameters }));
        });
    }
}
exports.SpotterInstanceApiSdk = SpotterInstanceApiSdk;
