// @spotter/gmesh-api-sdk/lib/api/invoice-df-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterInvoiceDfApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* InvoiceDfApi - object-oriented interface
* @export
* @class InvoiceDfApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterInvoiceDfApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    *
    * @summary df发票批量导出
    * @param {InvoiceDfApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoiceDfApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/df/export', data: requestParameters }));
        });
    }
    /**
    *
    * @summary df发票批量导出
    * @param {InvoiceDfApiExport0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoiceDfApi
    */
    _export_1(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/df/export', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取一张DF发票的数据
    * @param {InvoiceDfApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoiceDfApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceId) !== undefined) {
                params['invoiceId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/invoice/df/detail', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取一张DF发票的数据
    * @param {InvoiceDfApiDetail0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoiceDfApi
    */
    detail_2(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceId) !== undefined) {
                params['invoiceId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/amz/invoice/df/detail', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取DF发票page列表
    * @param {InvoiceDfApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoiceDfApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/df/page', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取DF发票page列表
    * @param {InvoiceDfApiPage0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoiceDfApi
    */
    page_3(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/df/page', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取DF发票的PDF
    * @param {InvoiceDfApiPdfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoiceDfApi
    */
    pdf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/df/pdf', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取DF发票的PDF
    * @param {InvoiceDfApiPdf0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoiceDfApi
    */
    pdf_4(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/df/pdf', data: requestParameters }));
        });
    }
}
exports.SpotterInvoiceDfApiSdk = SpotterInvoiceDfApiSdk;
