// @spotter/gmesh-api-sdk/lib/api/invoice-po-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterInvoicePoApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* InvoicePoApi - object-oriented interface
* @export
* @class InvoicePoApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterInvoicePoApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    *
    * @summary 绑定ASN
    * @param {InvoicePoApiBindASNRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    bindASN(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/asn/bind', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 绑定ASN
    * @param {InvoicePoApiBindASN0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    bindASN_1(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/asn/bind', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 未绑定ASN发票数
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    countAsn(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/invoice/po/asn/count' }));
        });
    }
    /**
    *
    * @summary 未绑定ASN发票数
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    countAsn_2(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/amz/invoice/po/asn/count' }));
        });
    }
    /**
    *
    * @summary 亚马逊发票管理-红点数量统一查询接口
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    counts(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/invoice/po/counts' }));
        });
    }
    /**
    *
    * @summary 亚马逊发票管理-红点数量统一查询接口
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    counts_3(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/amz/invoice/po/counts' }));
        });
    }
    /**
    *
    * @summary 获取发票详情
    * @param {InvoicePoApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber) !== undefined) {
                params['invoiceNumber'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/invoice/po/detail', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取发票详情
    * @param {InvoicePoApiDetail0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    detail_4(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber) !== undefined) {
                params['invoiceNumber'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/amz/invoice/po/detail', params: requestParameters }));
        });
    }
    /**
    *
    * @summary po发票批量导出
    * @param {InvoicePoApiExportInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    exportInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/export', data: requestParameters }));
        });
    }
    /**
    *
    * @summary po发票批量导出
    * @param {InvoicePoApiExportInvoice0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    exportInvoice_5(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/export', data: requestParameters }));
        });
    }
    /**
    *
    * @summary missing发票列表
    * @param {InvoicePoApiGetFinancePOInvoiceMissingDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getFinancePOInvoiceMissingDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/missingDetail/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary missing发票列表
    * @param {InvoicePoApiGetFinancePOInvoiceMissingDetail0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getFinancePOInvoiceMissingDetail_6(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/missingDetail/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取pdf的URL
    * @param {InvoicePoApiGetPDFUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getPDFUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/pdf', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取pdf的URL
    * @param {InvoicePoApiGetPDFUrl0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getPDFUrl_7(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/pdf', data: requestParameters }));
        });
    }
    /**
    *
    * @summary asn发票查看asn列表
    * @param {InvoicePoApiGetPOInvoiceASNPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getPOInvoiceASNPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/asn/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary asn发票查看asn列表
    * @param {InvoicePoApiGetPOInvoiceASNPage0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getPOInvoiceASNPage_8(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/asn/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 预计付款信息列表
    * @param {InvoicePoApiGetPOInvoiceDueDateGroupPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getPOInvoiceDueDateGroupPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/due/date/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 预计付款信息列表
    * @param {InvoicePoApiGetPOInvoiceDueDateGroupPage0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getPOInvoiceDueDateGroupPage_9(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/due/date/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取发票列表
    * @param {InvoicePoApiGetPOInvoicePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getPOInvoicePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 获取发票列表
    * @param {InvoicePoApiGetPOInvoicePage0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    getPOInvoicePage_10(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 忽略绑定asn
    * @param {InvoicePoApiIgnoreRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    ignore(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/ignore', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 忽略绑定asn
    * @param {InvoicePoApiIgnore0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    ignore_11(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/ignore', data: requestParameters }));
        });
    }
    /**
    *
    * @summary missingDetail备注
    * @param {InvoicePoApiMarkFinancePOInvoiceMissingDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    markFinancePOInvoiceMissingDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/missingDetail/mark', data: requestParameters }));
        });
    }
    /**
    *
    * @summary missingDetail备注
    * @param {InvoicePoApiMarkFinancePOInvoiceMissingDetail0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    markFinancePOInvoiceMissingDetail_12(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/missingDetail/mark', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 同步单张发票
    * @param {InvoicePoApiSyncInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    syncInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/invoice/po/sync', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 同步单张发票
    * @param {InvoicePoApiSyncInvoice0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    syncInvoice_13(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/amz/invoice/po/sync', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 手动同步发票结果
    * @param {InvoicePoApiSyncResultRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    syncResult(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber) !== undefined) {
                params['invoiceNumber'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/invoice/po/sync/result', params: requestParameters }));
        });
    }
    /**
    *
    * @summary 手动同步发票结果
    * @param {InvoicePoApiSyncResult0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof InvoicePoApi
    */
    syncResult_14(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber) !== undefined) {
                params['invoiceNumber'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.invoiceNumber;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/amz/invoice/po/sync/result', params: requestParameters }));
        });
    }
}
exports.SpotterInvoicePoApiSdk = SpotterInvoicePoApiSdk;
