// @spotter/gmesh-api-sdk/lib/api/knowledge-category-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterKnowledgeCategoryApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* KnowledgeCategoryApi - object-oriented interface
* @export
* @class KnowledgeCategoryApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterKnowledgeCategoryApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 创建知识分类
    * @summary 创建知识分类
    * @param {KnowledgeCategoryApiCreateKnowledgeCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeCategoryApi
    */
    createKnowledgeCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/category/create', data: requestParameters }));
        });
    }
    /**
    * 删除分类
    * @summary 删除分类
    * @param {KnowledgeCategoryApiDeleteKnowledgeCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeCategoryApi
    */
    deleteKnowledgeCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-knowledge-web/knowledge/category/delete', params: requestParameters }));
        });
    }
    /**
    * 分类详情
    * @summary 分类详情
    * @param {KnowledgeCategoryApiDetailKnowledgeCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeCategoryApi
    */
    detailKnowledgeCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-knowledge-web/knowledge/category/detail', params: requestParameters }));
        });
    }
    /**
    * 模糊查询分类
    * @summary 模糊查询分类
    * @param {KnowledgeCategoryApiFuzzQueryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeCategoryApi
    */
    fuzzQuery(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.keyWord) !== undefined) {
                params['keyWord'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.keyWord;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.language) !== undefined) {
                params['language'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.language;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-knowledge-web/knowledge/category/fuzzQuery', params: requestParameters }));
        });
    }
    /**
    * 列表查询分类树
    * @summary 列表查询分类树
    * @param {KnowledgeCategoryApiQueryTreeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeCategoryApi
    */
    queryTree(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.keyWord) !== undefined) {
                params['keyWord'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.keyWord;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.language) !== undefined) {
                params['language'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.language;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-knowledge-web/knowledge/category/tree', params: requestParameters }));
        });
    }
    /**
    * 修改分类
    * @summary 修改分类
    * @param {KnowledgeCategoryApiUpdateKnowledgeCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeCategoryApi
    */
    updateKnowledgeCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/category/update', data: requestParameters }));
        });
    }
}
exports.SpotterKnowledgeCategoryApiSdk = SpotterKnowledgeCategoryApiSdk;
