// @spotter/gmesh-api-sdk/lib/api/knowledge-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterKnowledgeApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* KnowledgeApi - object-oriented interface
* @export
* @class KnowledgeApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterKnowledgeApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 编辑知识文档
    * @summary 编辑知识文档，用于修改分组，生效时间等
    * @param {KnowledgeApiBatchUpdateKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    batchUpdateKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/update', data: requestParameters }));
        });
    }
    /**
    * 导入知识文档标题重复校验
    * @summary 导入知识文档标题重复校验
    * @param {KnowledgeApiCheckRepeatKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    checkRepeatKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/import/check/repeat', data: requestParameters }));
        });
    }
    /**
    * 创建知识文档
    * @summary 创建知识文档
    * @param {KnowledgeApiCreateKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    createKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/create', data: requestParameters }));
        });
    }
    /**
    * 删除知识文档
    * @summary 删除知识文档,软删除
    * @param {KnowledgeApiDeleteKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    deleteKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/delete', data: requestParameters }));
        });
    }
    /**
    * 知识文档导出
    * @summary 知识文档导出
    * @param {KnowledgeApiExportKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    exportKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/export', data: requestParameters }));
        });
    }
    /**
    * 全文检索知识文档
    * @summary 全文检索知识文档
    * @param {KnowledgeApiFullTextSearchKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    fullTextSearchKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/full/search', data: requestParameters }));
        });
    }
    /**
    * 导入知识文档
    * @summary 导入知识文档
    * @param {KnowledgeApiImportKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    importKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/import', data: requestParameters }));
        });
    }
    /**
    * 知识文档详情
    * @summary 知识文档详情
    * @param {KnowledgeApiKnowledgeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    knowledgeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-knowledge-web/knowledge/detail', params: requestParameters }));
        });
    }
    /**
    * 分页条件查询知识文档
    * @summary 分页条件查询知识文档
    * @param {KnowledgeApiPageKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    pageKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/page', data: requestParameters }));
        });
    }
    /**
    * 还原知识文档
    * @summary 还原知识文档
    * @param {KnowledgeApiRevertKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    revertKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/revert', data: requestParameters }));
        });
    }
    /**
    * 编辑知识文档详情
    * @summary 编辑知识文档
    * @param {KnowledgeApiUpdateKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof KnowledgeApi
    */
    updateKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/update/detail', data: requestParameters }));
        });
    }
}
exports.SpotterKnowledgeApiSdk = SpotterKnowledgeApiSdk;
