// @spotter/gmesh-api-sdk/lib/api/manage-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterManageApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ManageApi - object-oriented interface
* @export
* @class ManageApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterManageApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 产品活动日历
    * @summary 产品活动日历
    * @param {ManageApiActivityCalendarRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    activityCalendar(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/product/management/activity/calendar', data: requestParameters }));
        });
    }
    /**
    * 产品活动详情
    * @summary 产品活动详情
    * @param {ManageApiActivitySummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    activitySummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku) !== undefined) {
                params['ssku'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/product/management/activity/detail', params: requestParameters }));
        });
    }
    /**
    * 产品管理池--基础数据
    * @summary 产品管理池--基础数据
    * @param {ManageApiBasicDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    basicDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku) !== undefined) {
                params['ssku'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/product/management/basic/detail', params: requestParameters }));
        });
    }
    /**
    * 产品管理池--bbo订单报告
    * @summary 产品管理池--bbo订单报告
    * @param {ManageApiBboOrderDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    bboOrderDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku) !== undefined) {
                params['ssku'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/product/management/order/bbo', params: requestParameters }));
        });
    }
    /**
    * 产品管理池--btr订单报告
    * @summary 产品管理池--btr订单报告
    * @param {ManageApiBtrOrderDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    btrOrderDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku) !== undefined) {
                params['ssku'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/product/management/order/btr', params: requestParameters }));
        });
    }
    /**
    * 产品管理池--基础编码数据
    * @summary 产品管理池--基础编码数据
    * @param {ManageApiCodeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    codeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku) !== undefined) {
                params['ssku'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/product/management/code/detail', params: requestParameters }));
        });
    }
    /**
    * 产品管理池coupon活动数据分页
    * @summary 产品管理池coupon活动数据分页
    * @param {ManageApiCouponActivityPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    couponActivityPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/product/management/activity/coupon/page', data: requestParameters }));
        });
    }
    /**
    * 产品库存提醒记录分页
    * @summary 产品库存提醒记录分页
    * @param {ManageApiInventoryNoticePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    inventoryNoticePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/product/management/inventory/notice/page', data: requestParameters }));
        });
    }
    /**
    * 产品库存详情
    * @summary 产品库存详情
    * @param {ManageApiInventorySummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    inventorySummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku) !== undefined) {
                params['ssku'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/product/management/inventory/detail', params: requestParameters }));
        });
    }
    /**
    * 产品管理池数据分页
    * @summary 产品管理池数据分页
    * @param {ManageApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/product/management/page', data: requestParameters }));
        });
    }
    /**
    * 产品管理池promotion活动数据分页
    * @summary 产品管理池promotion活动数据分页
    * @param {ManageApiPromotionActivityPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ManageApi
    */
    promotionActivityPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/product/management/activity/promotion/page', data: requestParameters }));
        });
    }
}
exports.SpotterManageApiSdk = SpotterManageApiSdk;
