// @spotter/gmesh-api-sdk/lib/api/message-inbox-template-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterMessageInboxTemplateEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* MessageInboxTemplateEndpointApi - object-oriented interface
* @export
* @class MessageInboxTemplateEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterMessageInboxTemplateEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 删除站内信模板
    * @summary 删除站内信模板
    * @param {MessageInboxTemplateEndpointApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxTemplateEndpointApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/template/delete', data: requestParameters }));
        });
    }
    /**
    * 保持草稿箱站内信模板
    * @summary 保持草稿箱站内信模板
    * @param {MessageInboxTemplateEndpointApiCreateTempRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxTemplateEndpointApi
    */
    createTemp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/template/createTemp', data: requestParameters }));
        });
    }
    /**
    * 模板详情
    * @summary 模板详情
    * @param {MessageInboxTemplateEndpointApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxTemplateEndpointApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/template/detail', data: requestParameters }));
        });
    }
    /**
    * 启用禁用站内信模板
    * @summary 启用禁用站内信模板
    * @param {MessageInboxTemplateEndpointApiEnableRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxTemplateEndpointApi
    */
    enable(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/template/enable', data: requestParameters }));
        });
    }
    /**
    * 发布站内信模板
    * @summary 发布站内信模板
    * @param {MessageInboxTemplateEndpointApiPublishRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxTemplateEndpointApi
    */
    publish(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/template/publish', data: requestParameters }));
        });
    }
    /**
    * 分页查询模板
    * @summary 分页查询模板
    * @param {MessageInboxTemplateEndpointApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxTemplateEndpointApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/template/queryPage', data: requestParameters }));
        });
    }
    /**
    * 更新站内信模板
    * @summary 更新站内信模板
    * @param {MessageInboxTemplateEndpointApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxTemplateEndpointApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/template/update', data: requestParameters }));
        });
    }
    /**
    * 修改站内信模板消息类型
    * @summary 修改站内信模板消息类型
    * @param {MessageInboxTemplateEndpointApiUpdateMessageTypeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxTemplateEndpointApi
    */
    updateMessageType(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/template/updateMessageType', data: requestParameters }));
        });
    }
}
exports.SpotterMessageInboxTemplateEndpointApiSdk = SpotterMessageInboxTemplateEndpointApiSdk;
