// @spotter/gmesh-api-sdk/lib/api/message-web-send-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterMessageWebSendEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* MessageWebSendEndpointApi - object-oriented interface
* @export
* @class MessageWebSendEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterMessageWebSendEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 发送邮件
    * @summary 发送邮件
    * @param {MessageWebSendEndpointApiSendEmailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageWebSendEndpointApi
    */
    sendEmail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/message/sendEmail', data: requestParameters }));
        });
    }
    /**
    * 发送飞书
    * @summary 发送飞书
    * @param {MessageWebSendEndpointApiSendFeishuRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageWebSendEndpointApi
    */
    sendFeishu(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/message/sendFeishu', data: requestParameters }));
        });
    }
    /**
    * 发送短信
    * @summary 发送短信
    * @param {MessageWebSendEndpointApiSendSmsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageWebSendEndpointApi
    */
    sendSms(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/message/sendSms', data: requestParameters }));
        });
    }
}
exports.SpotterMessageWebSendEndpointApiSdk = SpotterMessageWebSendEndpointApiSdk;
