// @spotter/gmesh-api-sdk/lib/api/message-receiver-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterMessageReceiverApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* MessageReceiverApi - object-oriented interface
* @export
* @class MessageReceiverApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterMessageReceiverApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 供应商消息接收人详情
    * @summary 供应商消息接收人详情
    * @param {MessageReceiverApiMessageReceiverDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    messageReceiverDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/message/receiver/detail', params: requestParameters }));
        });
    }
    /**
    * 供应商消息接收人详情
    * @summary 供应商消息接收人详情
    * @param {MessageReceiverApiMessageReceiverDetail0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    messageReceiverDetail_1(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/sevc/supplier/message/receiver/detail', params: requestParameters }));
        });
    }
    /**
    * 供应商消息接收人列表
    * @summary 供应商消息接收人列表
    * @param {MessageReceiverApiMessageReceiverListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    messageReceiverList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/message/receiver/list', data: requestParameters }));
        });
    }
    /**
    * 供应商消息接收人列表
    * @summary 供应商消息接收人列表
    * @param {MessageReceiverApiMessageReceiverList0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    messageReceiverList_2(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/sevc/supplier/message/receiver/list', data: requestParameters }));
        });
    }
    /**
    * 分页查询供应商消息接收人列表
    * @summary 分页查询供应商消息接收人列表
    * @param {MessageReceiverApiMessageReceiverPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    messageReceiverPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/message/receiver/page', data: requestParameters }));
        });
    }
    /**
    * 分页查询供应商消息接收人列表
    * @summary 分页查询供应商消息接收人列表
    * @param {MessageReceiverApiMessageReceiverPage0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    messageReceiverPage_3(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/sevc/supplier/message/receiver/page', data: requestParameters }));
        });
    }
    /**
    * 分页查询消息接收人字段变化
    * @summary 分页查询消息接收人字段变化
    * @param {MessageReceiverApiReceiverChangePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    receiverChangePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/message/receiver/change/page', data: requestParameters }));
        });
    }
    /**
    * 分页查询消息接收人字段变化
    * @summary 分页查询消息接收人字段变化
    * @param {MessageReceiverApiReceiverChangePage0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    receiverChangePage_4(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/sevc/supplier/message/receiver/change/page', data: requestParameters }));
        });
    }
    /**
    * 更新供应商消息接收人
    * @summary 更新供应商消息接收人
    * @param {MessageReceiverApiUpdateMessageReceiverRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    updateMessageReceiver(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/message/receiver/create', data: requestParameters }));
        });
    }
    /**
    * 更新供应商消息接收人
    * @summary 更新供应商消息接收人
    * @param {MessageReceiverApiUpdateMessageReceiver0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    updateMessageReceiver_5(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/sevc/supplier/message/receiver/create', data: requestParameters }));
        });
    }
    /**
    * 验证供应商消息接收人
    * @summary 验证供应商消息接收人
    * @param {MessageReceiverApiVerifyMessageReceiverRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    verifyMessageReceiver(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/message/receiver/verify', params: requestParameters }));
        });
    }
    /**
    * 验证供应商消息接收人
    * @summary 验证供应商消息接收人
    * @param {MessageReceiverApiVerifyMessageReceiver0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageReceiverApi
    */
    verifyMessageReceiver_6(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/sevc/supplier/message/receiver/verify', params: requestParameters }));
        });
    }
}
exports.SpotterMessageReceiverApiSdk = SpotterMessageReceiverApiSdk;
