// @spotter/gmesh-api-sdk/lib/api/message-inbox-msg-type-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterMessageInboxMsgTypeEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* MessageInboxMsgTypeEndpointApi - object-oriented interface
* @export
* @class MessageInboxMsgTypeEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterMessageInboxMsgTypeEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 删除消息类型
    * @summary 删除消息类型
    * @param {MessageInboxMsgTypeEndpointApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxMsgTypeEndpointApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageType/delete', data: requestParameters }));
        });
    }
    /**
    * 创建消息类型
    * @summary 创建消息类型
    * @param {MessageInboxMsgTypeEndpointApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxMsgTypeEndpointApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageType/create', data: requestParameters }));
        });
    }
    /**
    * 根据名称查询消息类型
    * @summary 根据名称查询消息类型
    * @param {MessageInboxMsgTypeEndpointApiQueryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxMsgTypeEndpointApi
    */
    query(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageType/query', data: requestParameters }));
        });
    }
    /**
    * 查询消息类型
    * @summary 查询消息类型
    * @param {MessageInboxMsgTypeEndpointApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxMsgTypeEndpointApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageType/queryPage', data: requestParameters }));
        });
    }
    /**
    * 更新消息类型
    * @summary 更新消息类型
    * @param {MessageInboxMsgTypeEndpointApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MessageInboxMsgTypeEndpointApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageType/update', data: requestParameters }));
        });
    }
}
exports.SpotterMessageInboxMsgTypeEndpointApiSdk = SpotterMessageInboxMsgTypeEndpointApiSdk;
