// @spotter/gmesh-api-sdk/lib/api/month-reconciliation-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterMonthReconciliationApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* MonthReconciliationApi - object-oriented interface
* @export
* @class MonthReconciliationApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterMonthReconciliationApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导出月度对账单
    * @summary 导出月度对账单
    * @param {MonthReconciliationApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode) !== undefined) {
                params['reconciliationCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/export', params: requestParameters }));
        });
    }
    /**
    * 批量推送月度对账单到客户
    * @summary 批量推送月度对账单到客户
    * @param {MonthReconciliationApiBatchPushRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    batchPush(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/batchPush', data: requestParameters }));
        });
    }
    /**
    * 月度对账单前置检查
    * @summary 月度对账单前置检查
    * @param {MonthReconciliationApiCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    check(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/check', data: requestParameters }));
        });
    }
    /**
    * 根据是否存在未纳入月度对账单的子款项结算单分组的公司列表
    * @summary 根据是否存在未纳入月度对账单的子款项结算单分组的公司列表
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    companyReconciliationGroup(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/companyGroupList' }));
        });
    }
    /**
    * 月度对账单附件重传
    * @summary 月度对账单附件重传
    * @param {MonthReconciliationApiConfirmAttachReLoadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    confirmAttachReLoad(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/confirmAttachReLoad', data: requestParameters }));
        });
    }
    /**
    * 创建月度对账单
    * @summary 创建月度对账单
    * @param {MonthReconciliationApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/create', data: requestParameters }));
        });
    }
    /**
    * 月度对账单获取每种类型的数量
    * @summary 月度对账单获取每种类型的数量
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    getCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/getCounter' }));
        });
    }
    /**
    * 获取最新的月度对账单
    * @summary 获取最新的月度对账单
    * @param {MonthReconciliationApiGetLastRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    getLast(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/getLast', params: requestParameters }));
        });
    }
    /**
    * 预览月度对账单
    * @summary 预览月度对账单
    * @param {MonthReconciliationApiPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    preview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/preview', data: requestParameters }));
        });
    }
    /**
    * 月度对账单推送事件
    * @summary 月度对账单推送事件
    * @param {MonthReconciliationApiPushEventRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    pushEvent(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/pushEvent', data: requestParameters }));
        });
    }
    /**
    * 获取月度对账单详情
    * @summary 获取月度对账单详情
    * @param {MonthReconciliationApiQueryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    queryDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode) !== undefined) {
                params['reconciliationCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 分页查询月度对账单
    * @summary 分页查询月度对账单
    * @param {MonthReconciliationApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/queryPage', data: requestParameters }));
        });
    }
    /**
    * 撤销月度对账单
    * @summary 撤销月度对账单
    * @param {MonthReconciliationApiRevokeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationApi
    */
    revoke(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode) !== undefined) {
                params['reconciliationCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/revoke', params: requestParameters }));
        });
    }
}
exports.SpotterMonthReconciliationApiSdk = SpotterMonthReconciliationApiSdk;
