// @spotter/gmesh-api-sdk/lib/api/month-reconciliation-payment-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterMonthReconciliationPaymentApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* MonthReconciliationPaymentApi - object-oriented interface
* @export
* @class MonthReconciliationPaymentApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterMonthReconciliationPaymentApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 申请付款
    * @summary 申请付款
    * @param {MonthReconciliationPaymentApiApplyPayRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    applyPay(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payPlan/applyPay', data: requestParameters }));
        });
    }
    /**
    * 余额查询
    * @summary 余额查询
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    balance(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/balance' }));
        });
    }
    /**
    * 付款计划-编辑
    * @summary 付款计划-编辑
    * @param {MonthReconciliationPaymentApiEditPlanRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    editPlan(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payPlan/edit', data: requestParameters }));
        });
    }
    /**
    * 付款计划-导出
    * @summary 付款计划-导出
    * @param {MonthReconciliationPaymentApiExportPlanRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    exportPlan(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payPlan/export', data: requestParameters }));
        });
    }
    /**
    * 付款流水-导出
    * @summary 付款流水-导出
    * @param {MonthReconciliationPaymentApiExportRecordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    exportRecord(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/export', data: requestParameters }));
        });
    }
    /**
    * 付款计划待处理状态计数
    * @summary 付款计划待处理状态计数
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    getPlanCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payPlan/getCount' }));
        });
    }
    /**
    * 付款流水待处理状态计数
    * @summary 付款流水待处理状态计数
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    getRecordCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/getCount' }));
        });
    }
    /**
    * 线上支付
    * @summary 线上支付
    * @param {MonthReconciliationPaymentApiOnlinePayRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    onlinePay(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/onlinePay', params: requestParameters }));
        });
    }
    /**
    * 支付取消
    * @summary 支付取消
    * @param {MonthReconciliationPaymentApiPayCancelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    payCancel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/cancel', params: requestParameters }));
        });
    }
    /**
    * 银行回单上传
    * @summary 银行回单上传
    * @param {MonthReconciliationPaymentApiPayCompleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    payComplete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/complete', data: requestParameters }));
        });
    }
    /**
    * 付款通知
    * @summary 付款通知
    * @param {MonthReconciliationPaymentApiPayNotifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    payNotify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/notify', data: requestParameters }));
        });
    }
    /**
    * 银行回单撤回
    * @summary 银行回单撤回
    * @param {MonthReconciliationPaymentApiPayRevokeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    payRevoke(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/revoke', params: requestParameters }));
        });
    }
    /**
    * 付款计划-列表查询
    * @summary 付款计划-列表查询
    * @param {MonthReconciliationPaymentApiQueryPlanPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    queryPlanPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payPlan/queryPage', data: requestParameters }));
        });
    }
    /**
    * 付款流水-详情查询
    * @summary 付款流水-详情查询
    * @param {MonthReconciliationPaymentApiQueryRecordDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    queryRecordDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.payRecordNo) !== undefined) {
                params['payRecordNo'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.payRecordNo;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 付款流水-列表查询
    * @summary 付款流水-列表查询
    * @param {MonthReconciliationPaymentApiQueryRecordPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof MonthReconciliationPaymentApi
    */
    queryRecordPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payRecord/queryPage', data: requestParameters }));
        });
    }
}
exports.SpotterMonthReconciliationPaymentApiSdk = SpotterMonthReconciliationPaymentApiSdk;
