// @spotter/gmesh-api-sdk/lib/api/operation-inside-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterOperationInsideApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* OperationInsideApi - object-oriented interface
* @export
* @class OperationInsideApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterOperationInsideApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 操作日志查询，根据条件查询
    * @summary 操作日志查询（所有）
    * @param {OperationInsideApiFindListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OperationInsideApi
    */
    findList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-operation-log-web/log/inside/search/list', data: requestParameters }));
        });
    }
    /**
    * 操作日志查询，业务id关联日志查询（分页）
    * @summary 操作日志查询（分页）
    * @param {OperationInsideApiFindPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OperationInsideApi
    */
    findPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-operation-log-web/log/inside/search/page', data: requestParameters }));
        });
    }
}
exports.SpotterOperationInsideApiSdk = SpotterOperationInsideApiSdk;
