// @spotter/gmesh-api-sdk/lib/api/order-btr-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterOrderBtrApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* OrderBtrApi - object-oriented interface
* @export
* @class OrderBtrApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterOrderBtrApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * BTR提交
    * @summary BTR提交
    * @param {OrderBtrApiAddRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    add(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/order/btr/add', data: requestParameters }));
        });
    }
    /**
    * BTR详情
    * @summary BTR详情
    * @param {OrderBtrApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/gmesh-web/order/btr/detail', params: requestParameters }));
        });
    }
    /**
    * 获取Active的Offer列表
    * @summary 获取Active的Offer列表
    * @param {OrderBtrApiListActiveAllRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    listActiveAll(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/order/btr/active/all', data: requestParameters }));
        });
    }
    /**
    * 添加Skus下一步
    * @summary 添加Skus下一步
    * @param {OrderBtrApiNextRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    next(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/order/btr/add/next', data: requestParameters }));
        });
    }
    /**
    * BTR列表
    * @summary BTR列表
    * @param {OrderBtrApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/order/btr/list', data: requestParameters }));
        });
    }
    /**
    * BTR列表-代办
    * @summary BTR列表-代办
    * @param {OrderBtrApiPageToDoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    pageToDo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/order/btr/list/todo', data: requestParameters }));
        });
    }
    /**
    * 审核提交
    * @summary 审核提交
    * @param {OrderBtrApiSubmitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    submit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/order/btr/submit', data: requestParameters }));
        });
    }
    /**
    * 同步订单
    * @summary 同步订单
    * @param {OrderBtrApiSyncOrderRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    syncOrder(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/order/btr/sync', data: requestParameters }));
        });
    }
    /**
    * 更新sSku数量
    * @summary 更新sSku数量
    * @param {OrderBtrApiUpdateSkuRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OrderBtrApi
    */
    updateSku(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/order/btr/sku/update', data: requestParameters }));
        });
    }
}
exports.SpotterOrderBtrApiSdk = SpotterOrderBtrApiSdk;
