// @spotter/gmesh-api-sdk/lib/api/org-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterOrgApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* OrgApi - object-oriented interface
* @export
* @class OrgApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterOrgApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    *
    * @summary 删除组织
    * @param {OrgApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.orgId) !== undefined) {
                params['orgId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.orgId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-guard-web/org', params: requestParameters }));
        });
    }
    /**
    * 批量转出员工/添加成员
    * @summary 批量转出员工/添加成员
    * @param {OrgApiBatchTransferRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    batchTransfer(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/batch-transfer', data: requestParameters }));
        });
    }
    /**
    * 批量转出员工/添加成员
    * @summary 根据业务线获取对应的leader
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    businessLeaderList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/business/leader/list' }));
        });
    }
    /**
    * 添加一个组织
    * @summary 添加一个组织
    * @param {OrgApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org', data: requestParameters }));
        });
    }
    /**
    * 可以通过关键词 搜索, 同时需要对orgId 进行排除
    * @summary 获取组织下的人员列表
    * @param {OrgApiGetUserExcludeOrgIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    getUserExcludeOrgId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/users', data: requestParameters }));
        });
    }
    /**
    * 根据组织架构编码获取用户列表(包括子目录下用户)
    * @summary 根据组织架构编码获取用户列表(包括子目录下用户)
    * @param {OrgApiGetUserListByOrgCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    getUserListByOrgCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/users/all', data: requestParameters }));
        });
    }
    /**
    * 组织列表，通过关键词(成员和部门)进行搜索
    * @summary 关键词搜索用户/组织名
    * @param {OrgApiSearchOrgOrUserByKeyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    searchOrgOrUserByKey(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/list-org', data: requestParameters }));
        });
    }
    /**
    * 组织列表，树形结构，递归
    * @summary 组织列表tree
    * @param {OrgApiTreeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    tree(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/org/tree', params: requestParameters }));
        });
    }
    /**
    * 设置组织管理员
    * @summary 设置组织管理员
    * @param {OrgApiUpdateOrgManageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    updateOrgManage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/update-org-manage', data: requestParameters }));
        });
    }
    /**
    * 更新组织名称
    * @summary 更新组织名称
    * @param {OrgApiUpdateOrgNameRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof OrgApi
    */
    updateOrgName(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/update-org-name', data: requestParameters }));
        });
    }
}
exports.SpotterOrgApiSdk = SpotterOrgApiSdk;
