// @spotter/gmesh-api-sdk/lib/api/otp-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterOtpApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* OtpApi - object-oriented interface
* @export
* @class OtpApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterOtpApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 点击复制OTP账户
    * @summary 点击复制OTP账户
    * @param {OtpApiCopyOtpAccountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OtpApi
    */
    copyOtpAccount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/otp/copy', data: requestParameters }));
        });
    }
    /**
    * 查询svc账号列表
    * @summary 查询svc账号列表
    * @param {*} [options] Override http request option.
    * @memberof OtpApi
    */
    listAccount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/otp/account/list' }));
        });
    }
    /**
    * 分页查询svc邮箱列表
    * @summary 分页查询svc邮箱列表
    * @param {OtpApiQueryAccountPageByParamRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OtpApi
    */
    queryAccountPageByParam(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/otp/account/page', data: requestParameters }));
        });
    }
    /**
    * 新增/编辑svc登陆子账号
    * @summary 新增/编辑svc登陆子账号
    * @param {OtpApiSaveOrUpdateAccountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof OtpApi
    */
    saveOrUpdateAccount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/otp/account/saveOrUpdate', data: requestParameters }));
        });
    }
}
exports.SpotterOtpApiSdk = SpotterOtpApiSdk;
