// @spotter/gmesh-api-sdk/lib/api/payee-account-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterPayeeAccountApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* PayeeAccountApi - object-oriented interface
* @export
* @class PayeeAccountApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterPayeeAccountApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 收款账户-账户名称与公司英文名称检验
    * @summary 收款账户-账户名称与公司英文名称检验
    * @param {PayeeAccountApiAccountNameCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    accountNameCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/account_name_check', data: requestParameters }));
        });
    }
    /**
    * 收款账户-新建账户（银行）
    * @summary 收款账户-新建账户（银行）
    * @param {PayeeAccountApiAddRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    add(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/add', data: requestParameters }));
        });
    }
    /**
    * 收款账户-公司账户基本信息
    * @summary 收款账户-公司账户基本信息
    * @param {PayeeAccountApiCompanyBankBaseInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    companyBankBaseInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/company_bank_base_info', params: requestParameters }));
        });
    }
    /**
    * 收款账户-当前使用账户
    * @summary 收款账户-当前使用账户
    * @param {PayeeAccountApiCurrentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    current(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/current', params: requestParameters }));
        });
    }
    /**
    * 收款账户-账户详情列表
    * @summary 收款账户-账户详情列表
    * @param {PayeeAccountApiDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    detailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/detail/list', params: requestParameters }));
        });
    }
    /**
    * 收款账户-收付款信息绑定异常列表
    * @summary 收款账户-收付款信息绑定异常列表
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    getAbnormal(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/payee/account/payment_acc_abnormal' }));
        });
    }
    /**
    * 收款账户-当前使用账户列表
    * @summary 收款账户-当前使用账户列表
    * @param {PayeeAccountApiGetCurrentListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    getCurrentList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/get_current_list', data: requestParameters }));
        });
    }
    /**
    * 收款账户-最新银行账户
    * @summary 收款账户-最新银行账户
    * @param {PayeeAccountApiNewestBankAccountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    newestBankAccount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/newest_bank_account', params: requestParameters }));
        });
    }
    /**
    * 收款账户-账户分页列表
    * @summary 收款账户-账户分页列表
    * @param {PayeeAccountApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/page', data: requestParameters }));
        });
    }
    /**
    * 收款账户-设为默认账户
    * @summary 收款账户-设为默认账户
    * @param {PayeeAccountApiSetDefaultRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    setDefault(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/set_default', params: requestParameters }));
        });
    }
    /**
    * 收款账户-编辑账户（银行）
    * @summary 收款账户-编辑账户（银行）
    * @param {PayeeAccountApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PayeeAccountApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/update', data: requestParameters }));
        });
    }
}
exports.SpotterPayeeAccountApiSdk = SpotterPayeeAccountApiSdk;
