// @spotter/gmesh-api-sdk/lib/api/payment-info-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterPaymentInfoApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* PaymentInfoApi - object-oriented interface
* @export
* @class PaymentInfoApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterPaymentInfoApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 新增我司付款账户
    * @summary 新增我司付款账户
    * @param {PaymentInfoApiAddRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PaymentInfoApi
    */
    add(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payment/account/add', data: requestParameters }));
        });
    }
    /**
    * 新增我司对客户付款账户
    * @summary 新增我司对客户付款账户
    * @param {PaymentInfoApiAddRelationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PaymentInfoApi
    */
    addRelation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payment/account/customer/add', data: requestParameters }));
        });
    }
    /**
    * 编辑我司付款账户
    * @summary 编辑我司付款账户
    * @param {PaymentInfoApiEditRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PaymentInfoApi
    */
    edit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payment/account/edit', data: requestParameters }));
        });
    }
    /**
    * 编辑我司对客户付款账户
    * @summary 编辑我司对客户付款账户
    * @param {PaymentInfoApiEditRelationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PaymentInfoApi
    */
    editRelation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payment/account/customer/edit', data: requestParameters }));
        });
    }
    /**
    * 获取客户我司付款信息
    * @summary 获取客户我司付款信息
    * @param {PaymentInfoApiGetRelationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PaymentInfoApi
    */
    getRelation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/payment/account/customer/get', params: requestParameters }));
        });
    }
    /**
    * 获取待配置我司对客户列表
    * @summary 获取待配置我司对客户列表
    * @param {*} [options] Override http request option.
    * @memberof PaymentInfoApi
    */
    listCustomer(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/payment/account/customer/list' }));
        });
    }
    /**
    * 获取我司付款账户列表
    * @summary 获取我司付款账户列表
    * @param {*} [options] Override http request option.
    * @memberof PaymentInfoApi
    */
    queryInfoList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/payment/account/queryList' }));
        });
    }
    /**
    * 分页获取我司对客户付款账户列表
    * @summary 分页获取我司对客户付款账户列表
    * @param {PaymentInfoApiQueryRelationPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PaymentInfoApi
    */
    queryRelationPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payment/account/customer/queryPage', data: requestParameters }));
        });
    }
}
exports.SpotterPaymentInfoApiSdk = SpotterPaymentInfoApiSdk;
