// @spotter/gmesh-api-sdk/lib/api/permit-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterPermitApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* PermitApi - object-oriented interface
* @export
* @class PermitApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterPermitApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 添加权限
    * @summary 添加权限
    * @param {PermitApiAddPermitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    addPermit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/add-permit', data: requestParameters }));
        });
    }
    /**
    * 添加权限并绑定角色
    * @summary 添加权限并绑定角色
    * @param {PermitApiAddRoleAndPermitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    addRoleAndPermit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/add-role-and-permit', data: requestParameters }));
        });
    }
    /**
    * 添加权限-单条
    * @summary 添加权限-单条
    * @param {PermitApiAddSoloPermitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    addSoloPermit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/add-permit-solo', data: requestParameters }));
        });
    }
    /**
    * 查看角色绑定权限列表，通过角色Id进行展示
    * @summary 查看角色绑定权限列表
    * @param {PermitApiListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    list(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit', data: requestParameters }));
        });
    }
    /**
    * 查看角色绑定权限列表，通过角色Id进行展示
    * @summary 查看角色绑定权限列表
    * @param {PermitApiListAllRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    listAll(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.type) !== undefined) {
                params['type'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.type;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.appCode) !== undefined) {
                params['appCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.appCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/permit/all', params: requestParameters }));
        });
    }
    /**
    * 添加权限并绑定角色
    * @summary 添加权限并绑定角色
    * @param {PermitApiPermitRoleSettingRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    permitRoleSetting(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/role-permit-setting', data: requestParameters }));
        });
    }
    /**
    * 添加权限并绑定角色给sevc默认角色编辑后批量复制给所有sevc
    * @summary 添加权限并绑定角色给sevc默认角色编辑后批量复制给所有sevc
    * @param {PermitApiPermitRoleSettingForSevcRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    permitRoleSettingForSevc(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/role-permit-setting_for_sevc', data: requestParameters }));
        });
    }
    /**
    * 添加权限并绑定角色(批量接口)
    * @summary 添加权限并绑定角色(批量接口)
    * @param {PermitApiPermitRolesSettingRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    permitRolesSetting(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/roles-permit-setting', data: requestParameters }));
        });
    }
    /**
    * 更新权限
    * @summary 更新权限
    * @param {PermitApiUpdatePermitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    updatePermit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/update', data: requestParameters }));
        });
    }
    /**
    * 更新权限
    * @summary 更新权限
    * @param {PermitApiUpdatePermitsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    updatePermits(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/updates', data: requestParameters }));
        });
    }
    /**
    * 上传前端菜单(权限组)与权限
    * @summary 上传前端菜单(权限组)与权限
    * @param {PermitApiUploadFrontUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof PermitApi
    */
    uploadFrontUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/upload-front-url', data: requestParameters }));
        });
    }
}
exports.SpotterPermitApiSdk = SpotterPermitApiSdk;
