// @spotter/gmesh-api-sdk/lib/api/poa-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterPoaApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* PoaApi - object-oriented interface
* @export
* @class PoaApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterPoaApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 判断当前ticketId是否重复
    * @summary 端创建Poa判断当前ticketId是否重复
    * @param {PoaApiExistRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    exist(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ticketId) !== undefined) {
                params['ticketId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ticketId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-performance-web/poa/gmesh/exist', params: requestParameters }));
        });
    }
    /**
    * 新建Poa正文回复模版(gmesh)
    * @summary 根据id查找模版
    * @param {PoaApiFindTemplateByIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    findTemplateById(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-performance-web/poaTemplate/gmesh/findById', params: requestParameters }));
        });
    }
    /**
    * 根据asin模糊查询产品，查询所有公司(gmesh)
    * @summary 根据asin模糊查询产品，查询所有公司(gmesh)
    * @param {PoaApiGAsinFuzzyPageSearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    gAsinFuzzyPageSearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-performance-web/poa/gmesh/asin/fuzzy/search', data: requestParameters }));
        });
    }
    /**
    * poa详情（gmesh）
    * @summary poa详情（gmesh）
    * @param {PoaApiGDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    gDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-performance-web/poa/gmesh/detail', params: requestParameters }));
        });
    }
    /**
    * gmeshPOA列表
    * @summary gmeshPOA列表
    * @param {PoaApiGPOASearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    gPOASearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-performance-web/poa/gmesh/page/search', data: requestParameters }));
        });
    }
    /**
    * poa客户回复详情（gmesh）
    * @summary poa客户回复详情（gmesh）
    * @param {PoaApiGReplyDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    gReplyDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-performance-web/poa/gmesh/reply/detail', params: requestParameters }));
        });
    }
    /**
    * gmesh发送邮件
    * @summary gmesh发送邮件
    * @param {PoaApiGSendPOAMailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    gSendPOAMail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-performance-web/poa/gmesh/send', data: requestParameters }));
        });
    }
    /**
    * 撤回poa
    * @summary 撤回poa
    * @param {PoaApiRecallRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    recall(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-performance-web/poa/gmesh/recall', data: requestParameters }));
        });
    }
    /**
    * 新建POA(gmesh)
    * @summary 新建POA(gmesh)
    * @param {PoaApiSaveRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    save(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-performance-web/poa/gmesh/save', data: requestParameters }));
        });
    }
    /**
    * 新建Poa正文回复模版(gmesh)
    * @summary 新建Poa正文回复模版(gmesh)
    * @param {PoaApiTemplateSaveRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    templateSave(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-performance-web/poaTemplate/gmesh/save', data: requestParameters }));
        });
    }
    /**
    * 修改Poa正文回复模版(gmesh)
    * @summary 修改Poa正文回复模版(gmesh)
    * @param {PoaApiTemplateUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PoaApi
    */
    templateUpdate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-performance-web/poaTemplate/gmesh/update', data: requestParameters }));
        });
    }
}
exports.SpotterPoaApiSdk = SpotterPoaApiSdk;
