// @spotter/gmesh-api-sdk/lib/api/price-claim-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterPriceClaimApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* PriceClaimApi - object-oriented interface
* @export
* @class PriceClaimApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterPriceClaimApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 价格索赔dispute成功
    * @summary 价格索赔dispute成功
    * @param {PriceClaimApiDisputeSuccessRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceClaimApi
    */
    disputeSuccess(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceClaim/dispute_success', data: requestParameters }));
        });
    }
    /**
    * 价格索赔信息导出
    * @summary 价格索赔信息导出
    * @param {PriceClaimApiPriceClaimExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceClaimApi
    */
    priceClaimExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceClaim/export', data: requestParameters }));
        });
    }
    /**
    * 价格索赔修改责任人
    * @summary 价格索赔修改责任人
    * @param {PriceClaimApiPriceClaimModifyDutyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceClaimApi
    */
    priceClaimModifyDuty(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceClaim/modify_duty', data: requestParameters }));
        });
    }
    /**
    * 价格索赔修改责任人时检查调价编码
    * @summary 价格索赔修改责任人时检查调价编码
    * @param {PriceClaimApiPriceProtectedAdjustNoCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceClaimApi
    */
    priceProtectedAdjustNoCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceClaim/adjust_no_check', data: requestParameters }));
        });
    }
    /**
    * 价格索赔信息确认
    * @summary 价格索赔信息确认
    * @param {PriceClaimApiPriceProtectedConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceClaimApi
    */
    priceProtectedConfirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceClaim/confirm', data: requestParameters }));
        });
    }
    /**
    * 查询价格索赔信息详情
    * @summary 查询价格索赔信息详情
    * @param {PriceClaimApiPriceProtectedDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceClaimApi
    */
    priceProtectedDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/priceClaim/detail', params: requestParameters }));
        });
    }
    /**
    * 分页查询价格索赔信息
    * @summary 分页查询价格索赔信息
    * @param {PriceClaimApiPriceProtectedPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceClaimApi
    */
    priceProtectedPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceClaim/page', data: requestParameters }));
        });
    }
    /**
    * 价格索赔待处理数量
    * @summary 价格索赔待处理数量
    * @param {*} [options] Override http request option.
    * @memberof PriceClaimApi
    */
    priceProtectedPendingNum(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/priceClaim/pending_num' }));
        });
    }
}
exports.SpotterPriceClaimApiSdk = SpotterPriceClaimApiSdk;
