// @spotter/gmesh-api-sdk/lib/api/price-protected-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterPriceProtectedApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* PriceProtectedApi - object-oriented interface
* @export
* @class PriceProtectedApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterPriceProtectedApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 价格保护修改责任人时检查调价编码
    * @summary 价格保护修改责任人时检查调价编码
    * @param {PriceProtectedApiPriceProtectedAdjustNoCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedAdjustNoCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/adjust_no_check', data: requestParameters }));
        });
    }
    /**
    * 价格保护信息确认
    * @summary 价格保护信息确认
    * @param {PriceProtectedApiPriceProtectedConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedConfirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/confirm', data: requestParameters }));
        });
    }
    /**
    * 查询价格保护信息详情与库存分页信息
    * @summary 查询价格保护信息详情与库存分页信息
    * @param {PriceProtectedApiPriceProtectedDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/detail', data: requestParameters }));
        });
    }
    /**
    * 价格保护信息导出
    * @summary 价格保护信息导出
    * @param {PriceProtectedApiPriceProtectedExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/export', data: requestParameters }));
        });
    }
    /**
    * 查询价格保护库存信息（for agreement asin分组归类）
    * @summary 查询价格保护库存信息（for agreement asin分组归类）
    * @param {PriceProtectedApiPriceProtectedInventoryMapForAgreementRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedInventoryMapForAgreement(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId) !== undefined) {
                params['agreementId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/priceProtected/inventory/map', params: requestParameters }));
        });
    }
    /**
    * 价格保护上次结算单信息
    * @summary 价格保护上次结算单信息
    * @param {PriceProtectedApiPriceProtectedLastSettleInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedLastSettleInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/priceProtected/settle/last_settle_info', params: requestParameters }));
        });
    }
    /**
    * 价格保护修改责任人
    * @summary 价格保护修改责任人
    * @param {PriceProtectedApiPriceProtectedModifyDutyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedModifyDuty(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/modify_duty', data: requestParameters }));
        });
    }
    /**
    * 分页查询价格保护信息
    * @summary 分页查询价格保护信息
    * @param {PriceProtectedApiPriceProtectedPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/page', data: requestParameters }));
        });
    }
    /**
    * 价格保护待处理数量
    * @summary 价格保护待处理数量
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedPendingNum(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/priceProtected/pending_num' }));
        });
    }
    /**
    * 价格保护的结算单详情的基础信息
    * @summary 价格保护的结算单详情的基础信息
    * @param {PriceProtectedApiPriceProtectedSettleBaseRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedSettleBase(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/base', data: requestParameters }));
        });
    }
    /**
    * 价格保护的结算单创建
    * @summary 价格保护的结算单创建
    * @param {PriceProtectedApiPriceProtectedSettleCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedSettleCreate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/create', data: requestParameters }));
        });
    }
    /**
    * 价格保护的结算单详情的明细信息分页查询
    * @summary 价格保护的结算单详情的明细信息分页查询
    * @param {PriceProtectedApiPriceProtectedSettleDetailPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedSettleDetailPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/detail_page', data: requestParameters }));
        });
    }
    /**
    * 导出价格保护的结算单
    * @summary 导出价格保护的结算单
    * @param {PriceProtectedApiPriceProtectedSettleExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedSettleExport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/export', data: requestParameters }));
        });
    }
    /**
    * 分页查询价格保护的结算单
    * @summary 分页查询价格保护的结算单
    * @param {PriceProtectedApiPriceProtectedSettlePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedSettlePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/page', data: requestParameters }));
        });
    }
    /**
    * 价格保护的结算单预览（检查遗漏）
    * @summary 价格保护的结算单预览（检查遗漏）
    * @param {PriceProtectedApiPriceProtectedSettlePreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedSettlePreview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/preview', data: requestParameters }));
        });
    }
    /**
    * 价格保护的结算单撤销
    * @summary 价格保护的结算单撤销
    * @param {PriceProtectedApiPriceProtectedSettleRevertRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedSettleRevert(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/revert', data: requestParameters }));
        });
    }
    /**
    * 价格保护的结算单详情的汇总信息分页查询
    * @summary 价格保护的结算单详情的汇总信息分页查询
    * @param {PriceProtectedApiPriceProtectedSettleSummaryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedSettleSummaryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/summary_page', data: requestParameters }));
        });
    }
    /**
    * 分页查询价格保护信息时的金额总和
    * @summary 分页查询价格保护信息时的金额总和
    * @param {PriceProtectedApiPriceProtectedTotalAmountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof PriceProtectedApi
    */
    priceProtectedTotalAmount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/total_amount', data: requestParameters }));
        });
    }
}
exports.SpotterPriceProtectedApiSdk = SpotterPriceProtectedApiSdk;
