// @spotter/gmesh-api-sdk/lib/api/process-record-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterProcessRecordApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ProcessRecordApi - object-oriented interface
* @export
* @class ProcessRecordApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterProcessRecordApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * VC报价审核-添加流程评论
    * @summary VC报价审核-添加流程评论
    * @param {ProcessRecordApiAddCommentsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProcessRecordApi
    */
    addComments(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/process/record/vc/comments/add', data: requestParameters }));
        });
    }
    /**
    * VC报价审核-流程记录
    * @summary VC报价审核-获取指定业务id的所有流程记录
    * @param {ProcessRecordApiAllRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProcessRecordApi
    */
    all(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId) !== undefined) {
                params['agreementId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/process/record/vc/all', params: requestParameters }));
        });
    }
    /**
    * VC报价审核-审核记录
    * @summary VC报价审核-获取指定业务id的审核记录
    * @param {ProcessRecordApiAuditRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProcessRecordApi
    */
    audit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId) !== undefined) {
                params['agreementId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/process/record/vc/audit', params: requestParameters }));
        });
    }
    /**
    * VC报价审核-流程评论
    * @summary VC报价审核-获取指定业务id的流程评论
    * @param {ProcessRecordApiCommentsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProcessRecordApi
    */
    comments(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId) !== undefined) {
                params['agreementId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/process/record/vc/comments', params: requestParameters }));
        });
    }
}
exports.SpotterProcessRecordApiSdk = SpotterProcessRecordApiSdk;
