// @spotter/gmesh-api-sdk/lib/api/product-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterProductApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ProductApi - object-oriented interface
* @export
* @class ProductApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterProductApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 产品详情(gmesh)
    * @summary 产品详情(gmesh)
    * @param {ProductApiGActualDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductApi
    */
    gActualDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.asin) !== undefined) {
                params['asin'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.asin;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/product/gmesh/actual/detail', params: requestParameters }));
        });
    }
    /**
    * 指定公司进行ssku模糊搜索并返回指定分页(gmesh)
    * @summary 指定公司进行ssku模糊搜索并返回指定分页(gmesh)
    * @param {ProductApiGCompanySskuFuzzySearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductApi
    */
    gCompanySskuFuzzySearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/product/gmesh/company/ssku/fuzzy/page/search', data: requestParameters }));
        });
    }
    /**
    * 产品详情(gmesh)
    * @summary 产品详情(gmesh)
    * @param {ProductApiGDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductApi
    */
    gDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku) !== undefined) {
                params['ssku'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.ssku;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/product/gmesh/detail', params: requestParameters }));
        });
    }
    /**
    * 产品概览搜索(gmesh)，仅产品池使用
    * @summary (gmesh)excel表格导出
    * @param {ProductApiGExcelSummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductApi
    */
    gExcelSummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/product/gmesh/excel/summary', data: requestParameters }));
        });
    }
    /**
    * 根据ssku模糊查询有生效VC报价协议的产品，查询所有公司(gmesh)
    * @summary 根据ssku模糊查询有生效VC报价协议的产品，查询所有公司(gmesh)
    * @param {ProductApiGFuzzyVcValidPageSearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductApi
    */
    gFuzzyVcValidPageSearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/product/gmesh/ssku/fuzzy/vc/valid/search', data: requestParameters }));
        });
    }
    /**
    * ssku模糊搜索并返回指定分页(gmesh)
    * @summary ssku模糊搜索并返回指定分页(gmesh)
    * @param {ProductApiGSskuFuzzySearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductApi
    */
    gSskuFuzzySearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/product/gmesh/ssku/fuzzy/page/search', data: requestParameters }));
        });
    }
    /**
    * 修改batchNo
    * @summary 修改batchNo
    * @param {ProductApiNeedBatchNoModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductApi
    */
    needBatchNoModify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/product/modify/needBatchNo', data: requestParameters }));
        });
    }
    /**
    * 产品概览搜索(gmesh)，仅产品池使用
    * @summary 产品概览搜索(gmesh)，仅产品池使用
    * @param {ProductApiSummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductApi
    */
    summary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/product/gmesh/summary', data: requestParameters }));
        });
    }
}
exports.SpotterProductApiSdk = SpotterProductApiSdk;
