// @spotter/gmesh-api-sdk/lib/api/product-line-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterProductLineApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ProductLineApi - object-oriented interface
* @export
* @class ProductLineApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterProductLineApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 查询指定公司的品牌信息
    * @summary 查询指定公司的品牌信息
    * @param {ProductLineApiCompanySearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductLineApi
    */
    companySearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId) !== undefined) {
                params['brandId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/product/line/company/search', params: requestParameters }));
        });
    }
    /**
    * 查询指定产品线信息（含产品线历史销售额）
    * @summary 查询指定产品线信息（含产品线历史销售额）
    * @param {ProductLineApiProductLineSalesSearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ProductLineApi
    */
    productLineSalesSearch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId) !== undefined) {
                params['brandId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.productLineId) !== undefined) {
                params['productLineId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.productLineId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/product/line/company/search/with/sales', params: requestParameters }));
        });
    }
}
exports.SpotterProductLineApiSdk = SpotterProductLineApiSdk;
