// @spotter/gmesh-api-sdk/lib/api/report-settings-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterReportSettingsApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ReportSettingsApi - object-oriented interface
* @export
* @class ReportSettingsApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterReportSettingsApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 添加实时销量asin
    * @summary 添加实时销量asin
    * @param {ReportSettingsApiAddRealtimeSalesAsinRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ReportSettingsApi
    */
    addRealtimeSalesAsin(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/add', data: requestParameters }));
        });
    }
    /**
    * 删除实时销量asin
    * @summary 删除实时销量asin
    * @param {ReportSettingsApiDeleteRealtimeSalesAsinRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ReportSettingsApi
    */
    deleteRealtimeSalesAsin(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/delete', data: requestParameters }));
        });
    }
    /**
    * 获取active offer列表
    * @summary 获取active offer列表
    * @param {ReportSettingsApiGetActiveOfferPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ReportSettingsApi
    */
    getActiveOfferPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/active/offer', data: requestParameters }));
        });
    }
    /**
    * 获取所有已配置的asin总数
    * @summary 获取所有已配置的asin总数
    * @param {*} [options] Override http request option.
    * @memberof ReportSettingsApi
    */
    getCompaniesAllCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/report/settings/total/count' }));
        });
    }
    /**
    * 根据公司id获取所有已配置的asin总数
    * @summary 根据公司id获取所有已配置的asin总数
    * @param {*} [options] Override http request option.
    * @memberof ReportSettingsApi
    */
    getCompanyCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/report/settings/count' }));
        });
    }
    /**
    * 获取active offer列表
    * @summary 查询用户已经配置的产品
    * @param {ReportSettingsApiGetSettingProductPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ReportSettingsApi
    */
    getSettingProductPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/setting/search', data: requestParameters }));
        });
    }
    /**
    * 查询产品配置信息
    * @summary 查询产品配置信息
    * @param {ReportSettingsApiSelectRealtimeSalesListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ReportSettingsApi
    */
    selectRealtimeSalesList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/gmesh/list', data: requestParameters }));
        });
    }
    /**
    * 更新产品配置数量
    * @summary 更新产品配置数量
    * @param {ReportSettingsApiUpdateRealtimeSalesCountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ReportSettingsApi
    */
    updateRealtimeSalesCount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/updateCount', data: requestParameters }));
        });
    }
}
exports.SpotterReportSettingsApiSdk = SpotterReportSettingsApiSdk;
