// @spotter/gmesh-api-sdk/lib/api/role-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterRoleApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* RoleApi - object-oriented interface
* @export
* @class RoleApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterRoleApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 删除角色
    * @summary 删除角色
    * @param {RoleApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof RoleApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/delete', data: requestParameters }));
        });
    }
    /**
    * 添加员工
    * @summary 添加员工
    * @param {RoleApiAddUserRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof RoleApi
    */
    addUser(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/add-users', data: requestParameters }));
        });
    }
    /**
    * 创建角色
    * @summary 创建角色
    * @param {RoleApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof RoleApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role', data: requestParameters }));
        });
    }
    /**
    * 可以通过关键词 搜索, 同时需要对roleCode 进行排除
    * @summary 获取角色下的人员列表
    * @param {RoleApiGetUserExcludeRoleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof RoleApi
    */
    getUserExcludeRole(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/users', data: requestParameters }));
        });
    }
    /**
    * 角色用户列表
    * @summary 角色用户列表
    * @param {RoleApiListUsersRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof RoleApi
    */
    listUsers(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/list-users', data: requestParameters }));
        });
    }
    /**
    * 移除角色中的用户
    * @summary 移除用户
    * @param {RoleApiRemoveUserInRoleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof RoleApi
    */
    removeUserInRole(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/remove-users', data: requestParameters }));
        });
    }
    /**
    * 角色列表
    * @summary 角色列表
    * @param {RoleApiRoleListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof RoleApi
    */
    roleList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/role/list', params: requestParameters }));
        });
    }
}
exports.SpotterRoleApiSdk = SpotterRoleApiSdk;
