// @spotter/gmesh-api-sdk/lib/api/sale-management-line-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSaleManagementLineApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SaleManagementLineApi - object-oriented interface
* @export
* @class SaleManagementLineApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSaleManagementLineApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 新建虚拟业务线
    * @summary 管理虚拟对象-新建虚拟业务线
    * @param {SaleManagementLineApiAddVirtualLineRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    addVirtualLine(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/line/formulate/virtual/add', data: requestParameters }));
        });
    }
    /**
    * 删除虚拟业务线
    * @summary 管理虚拟对象-删除虚拟业务线
    * @param {SaleManagementLineApiDeleteVirtualLineRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    deleteVirtualLine(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-report-web/sale/management/line/formulate/virtual/delete', data: requestParameters }));
        });
    }
    /**
    * 编辑虚拟业务线：编辑操作人，名称
    * @summary 管理虚拟对象-编辑虚拟业务线
    * @param {SaleManagementLineApiEditVirtualLineRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    editVirtualLine(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/line/formulate/virtual/edit', data: requestParameters }));
        });
    }
    /**
    * 制定目标时获取的业务线列表数据，根据用户可查看和可编辑下发
    * @summary 制定目标-业务线列表数据
    * @param {SaleManagementLineApiFormulateLineRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    formulateLine(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/line/formulate/list', data: requestParameters }));
        });
    }
    /**
    * 编辑业务线目标：编辑期望销售额，保存草稿操作，其他人不可见
    * @summary 制定目标-编辑业务线目标(保存草稿)
    * @param {SaleManagementLineApiLineSaveDraftRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    lineSaveDraft(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/line/formulate/save/draft', data: requestParameters }));
        });
    }
    /**
    * 获取业务销售列表
    * @summary 搜索业务线分页数据
    * @param {SaleManagementLineApiLineSearchPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    lineSearchPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/line/search/page', data: requestParameters }));
        });
    }
    /**
    * 目业务线总览列表
    * @summary 业务线总览列表
    * @param {SaleManagementLineApiOverviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    overview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/line/overview', data: requestParameters }));
        });
    }
    /**
    * 发布业务线目标：发布后其他人可见
    * @summary 制定目标-发布业务线目标
    * @param {SaleManagementLineApiPublishLineRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    publishLine(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/line/formulate/publish', data: requestParameters }));
        });
    }
    /**
    * 实际是同步指定当前年度下新加入的业务线到销售目标中，对销售目标的报表查漏补缺
    * @summary 制定目标-更新业务线
    * @param {SaleManagementLineApiSyncUpdateLineRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementLineApi
    */
    syncUpdateLine(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/line/formulate/sync', data: requestParameters }));
        });
    }
}
exports.SpotterSaleManagementLineApiSdk = SpotterSaleManagementLineApiSdk;
