// @spotter/gmesh-api-sdk/lib/api/sale-management-company-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSaleManagementCompanyApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SaleManagementCompanyApi - object-oriented interface
* @export
* @class SaleManagementCompanyApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSaleManagementCompanyApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 新建虚拟公司
    * @summary 管理虚拟对象-新建虚拟公司
    * @param {SaleManagementCompanyApiAddVirtualCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    addVirtualCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/formulate/virtual/add', data: requestParameters }));
        });
    }
    /**
    * 传入指定用户所在公司id，获取公司历史列表，列表中不下发每月的详细数据
    * @summary 公司历史列表(不发月份数据)
    * @param {SaleManagementCompanyApiCompanyHistoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    companyHistory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/history/list', data: requestParameters }));
        });
    }
    /**
    * 传入指定用户所在公司id，获取公司历史列表，列表中下发每月的详细数据
    * @summary 公司历史详细列表(下发月份数据)
    * @param {SaleManagementCompanyApiCompanyHistoryDetailedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    companyHistoryDetailed(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/history/detailed/list', data: requestParameters }));
        });
    }
    /**
    * 编辑公司目标：编辑期望销售额，保存草稿操作，其他人不可见
    * @summary 制定目标-编辑公司目标(保存草稿)
    * @param {SaleManagementCompanyApiCompanySaveDraftRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    companySaveDraft(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/formulate/save/draft', data: requestParameters }));
        });
    }
    /**
    * 获取公司销售列表，列表中不下发每月的详细数据
    * @summary 搜索-公司分页列表(不发月份数据)
    * @param {SaleManagementCompanyApiCompanySearchPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    companySearchPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/search/page', data: requestParameters }));
        });
    }
    /**
    * 获取公司销售列表，列表中下发每月的详细数据
    * @summary 搜索-公司分页详细列表(下发月份数据)
    * @param {SaleManagementCompanyApiCompanySearchPageDetailedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    companySearchPageDetailed(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/search/detailed/page', data: requestParameters }));
        });
    }
    /**
    * 删除虚拟公司
    * @summary 管理虚拟对象-删除虚拟公司
    * @param {SaleManagementCompanyApiDeleteVirtualCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    deleteVirtualCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-report-web/sale/management/company/formulate/virtual/delete', data: requestParameters }));
        });
    }
    /**
    * 编辑虚拟公司：编辑操作人，名称，所属业务线
    * @summary 管理虚拟对象-编辑虚拟公司
    * @param {SaleManagementCompanyApiEditVirtualCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    editVirtualCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/formulate/virtual/edit', data: requestParameters }));
        });
    }
    /**
    * 公司总览列表
    * @summary 公司总览列表
    * @param {SaleManagementCompanyApiOverviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    overview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/overview', data: requestParameters }));
        });
    }
    /**
    * 公司负责人总览列表
    * @summary 公司负责人总览列表
    * @param {SaleManagementCompanyApiPersonOverviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    personOverview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/person/overview', data: requestParameters }));
        });
    }
    /**
    * 发布公司目标：发布后其他人可见
    * @summary 制定目标-发布公司目标
    * @param {SaleManagementCompanyApiPublishCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    publishCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/formulate/publish', data: requestParameters }));
        });
    }
    /**
    * 将公司转移到其他人
    * @summary 制定目标-转移公司
    * @param {SaleManagementCompanyApiShiftCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    shiftCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/formulate/shift', data: requestParameters }));
        });
    }
    /**
    * 实际是同步指定业务线下新加入的公司到销售目标中，对销售目标的报表查漏补缺
    * @summary 制定目标-更新公司
    * @param {SaleManagementCompanyApiSyncUpdateCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementCompanyApi
    */
    syncUpdateCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/company/formulate/sync', data: requestParameters }));
        });
    }
}
exports.SpotterSaleManagementCompanyApiSdk = SpotterSaleManagementCompanyApiSdk;
