// @spotter/gmesh-api-sdk/lib/api/sale-management-target-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSaleManagementTargetApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SaleManagementTargetApi - object-oriented interface
* @export
* @class SaleManagementTargetApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSaleManagementTargetApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 制定目标-新建年度目标
    * @summary 制定目标-新建年度目标
    * @param {SaleManagementTargetApiAddTargetRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    addTarget(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/formulate/add', data: requestParameters }));
        });
    }
    /**
    * 在创建/转移(业务线/公司/品牌)时，可以指定的负责人列表（目前该列表是运营和商务角色的所有人）
    * @summary 可指派的业务线/公司/品牌负责人列表
    * @param {SaleManagementTargetApiAssignablePersonsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    assignablePersons(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/assignable/persons', data: requestParameters }));
        });
    }
    /**
    * 创建虚拟(业务线/公司/品牌)的创建人员列表(目前该列表是运营和商务角色的所有人)
    * @summary 管理虚拟对象-虚拟业务线/公司/品牌创建者列表
    * @param {SaleManagementTargetApiCreatedPersonsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    createdPersons(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/formulate/virtual/created/persons', data: requestParameters }));
        });
    }
    /**
    * 在创建/转移(业务线/公司/品牌)时，已指定的负责人列表（目前该列表是运营和商务角色的所有人）
    * @summary 已指派的业务线/公司/品牌负责人列表
    * @param {SaleManagementTargetApiHistoryAssignablePersonsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    historyAssignablePersons(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/history/assignable/persons', data: requestParameters }));
        });
    }
    /**
    * 年度目标总览
    * @summary 年度目标总览
    * @param {SaleManagementTargetApiOverviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    overview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/overview', data: requestParameters }));
        });
    }
    /**
    * 获取同步指定当前年度下新加入的业务线/公司/品牌到销售目标的操作日志参数定义，通过下发的参数在操作日志服务中拉去指定日志
    * @summary 制定目标-更新日志参数
    * @param {SaleManagementTargetApiSyncUpdateOperationLogParamRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    syncUpdateOperationLogParam(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/formulate/sync/log/param', data: requestParameters }));
        });
    }
    /**
    * 制定目标-年度目标列表
    * @summary 制定目标-年度目标列表
    * @param {SaleManagementTargetApiTargetListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    targetList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/formulate/list', data: requestParameters }));
        });
    }
    /**
    * 制定目标-更新年度目标
    * @summary 制定目标-更新年度目标
    * @param {SaleManagementTargetApiUpdateTargetRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    updateTarget(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/formulate/update', data: requestParameters }));
        });
    }
    /**
    * 制定目标-激活/关闭年度目标
    * @summary 制定目标-激活/关闭年度目标
    * @param {SaleManagementTargetApiUpdateTargetStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementTargetApi
    */
    updateTargetStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/target/formulate/status/update', data: requestParameters }));
        });
    }
}
exports.SpotterSaleManagementTargetApiSdk = SpotterSaleManagementTargetApiSdk;
