// @spotter/gmesh-api-sdk/lib/api/sale-management-brand-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSaleManagementBrandApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SaleManagementBrandApi - object-oriented interface
* @export
* @class SaleManagementBrandApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSaleManagementBrandApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 新建虚拟品牌
    * @summary 管理虚拟对象-新建虚拟品牌
    * @param {SaleManagementBrandApiAddVirtualBrandRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    addVirtualBrand(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/formulate/virtual/add', data: requestParameters }));
        });
    }
    /**
    * 传入指定用户上报的品牌id，获取品牌历史销售列表，列表中不下发每月的详细数据
    * @summary 品牌历史列表(不发月份数据)
    * @param {SaleManagementBrandApiBrandHistoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    brandHistory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/history/list', data: requestParameters }));
        });
    }
    /**
    * 传入指定用户上报的品牌id，获取品牌历史销售列表，列表中下发每月的详细数据
    * @summary 品牌历史详细列表(下发月份数据)
    * @param {SaleManagementBrandApiBrandHistoryDetailedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    brandHistoryDetailed(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/history/detailed/list', data: requestParameters }));
        });
    }
    /**
    * 编辑品牌目标：编辑期望销售额，期望售价等值，保存草稿操作，其他人不可见
    * @summary 制定目标-编辑品牌目标(保存草稿)
    * @param {SaleManagementBrandApiBrandSaveDraftRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    brandSaveDraft(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/formulate/save/draft', data: requestParameters }));
        });
    }
    /**
    * 获取品牌销售列表，列表中不下发每月的详细数据
    * @summary 品牌分页列表(不发月份数据)
    * @param {SaleManagementBrandApiBrandSearchPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    brandSearchPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/search/page', data: requestParameters }));
        });
    }
    /**
    * 获取品牌销售列表，列表中下发每月的详细数据
    * @summary 品牌分页列表(下发月份数据)
    * @param {SaleManagementBrandApiBrandSearchPageDetailedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    brandSearchPageDetailed(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/search/detailed/page', data: requestParameters }));
        });
    }
    /**
    * 删除虚拟品牌
    * @summary 管理虚拟对象-删除虚拟品牌
    * @param {SaleManagementBrandApiDeleteVirtualBrandRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    deleteVirtualBrand(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-report-web/sale/management/brand/formulate/virtual/delete', data: requestParameters }));
        });
    }
    /**
    * 编辑虚拟品牌：编辑操作人，名称，所属公司
    * @summary 管理虚拟对象-编辑虚拟品牌
    * @param {SaleManagementBrandApiEditVirtualBrandRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    editVirtualBrand(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/formulate/virtual/edit', data: requestParameters }));
        });
    }
    /**
    * 品牌总览列表
    * @summary 品牌总览列表
    * @param {SaleManagementBrandApiOverviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    overview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/overview', data: requestParameters }));
        });
    }
    /**
    * 品牌负责人总览列表
    * @summary 品牌负责人总览列表
    * @param {SaleManagementBrandApiPersonOverviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    personOverview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/person/overview', data: requestParameters }));
        });
    }
    /**
    * 发布品牌目标：发布后其他人可见
    * @summary 制定目标-发布品牌目标
    * @param {SaleManagementBrandApiPublishBrandRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    publishBrand(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/formulate/publish', data: requestParameters }));
        });
    }
    /**
    * 将品牌转移到其他人
    * @summary 制定目标-转移品牌
    * @param {SaleManagementBrandApiShiftBrandRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    shiftBrand(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/formulate/shift', data: requestParameters }));
        });
    }
    /**
    * 品牌同步订单数据
    * @summary 品牌同步订单数据
    * @param {SaleManagementBrandApiSyncBrandOrderRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    syncBrandOrder(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/sync/brand/order', data: requestParameters }));
        });
    }
    /**
    * 实际是同步指定公司下新加入的品牌到销售目标中，对销售目标的报表查漏补缺
    * @summary 制定目标-更新品牌
    * @param {SaleManagementBrandApiSyncUpdateBrandRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SaleManagementBrandApi
    */
    syncUpdateBrand(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/sale/management/brand/formulate/sync', data: requestParameters }));
        });
    }
}
exports.SpotterSaleManagementBrandApiSdk = SpotterSaleManagementBrandApiSdk;
