// @spotter/gmesh-api-sdk/lib/api/sas-list-price-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSasListPriceApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SasListPriceApi - object-oriented interface
* @export
* @class SasListPriceApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSasListPriceApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 添加需求
    * @summary 添加需求
    * @param {SasListPriceApiAddRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    add(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/add', data: requestParameters }));
        });
    }
    /**
    * 添加批注
    * @summary 添加批注
    * @param {SasListPriceApiAddAnnotationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    addAnnotation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/annotation/add', data: requestParameters }));
        });
    }
    /**
    * 批注列表
    * @summary 批注列表
    * @param {SasListPriceApiAnnotationListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    annotationList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum) !== undefined) {
                params['sasNum'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/listPrice/annotation/list', params: requestParameters }));
        });
    }
    /**
    * 手动校验
    * @summary 手动校验
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    check(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/check' }));
        });
    }
    /**
    * 手动校验最新记录
    * @summary 手动校验最新记录
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    checkLastRecord(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/listPrice/check/last/record' }));
        });
    }
    /**
    * 详情
    * @summary 详情
    * @param {SasListPriceApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum) !== undefined) {
                params['sasNum'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/listPrice/detail', params: requestParameters }));
        });
    }
    /**
    * 文件确认提交(失败：返回失败的数据)
    * @summary 文件确认提交
    * @param {SasListPriceApiFileConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    fileConfirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/listPrice/file/confirm', params: requestParameters }));
        });
    }
    /**
    * 提交反馈-下载
    * @summary 提交反馈-下载
    * @param {SasListPriceApiFileFeedbackDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    fileFeedbackDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/file/feedback/download', data: requestParameters }));
        });
    }
    /**
    * 下载记录
    * @summary 下载记录
    * @param {SasListPriceApiFileRecordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    fileRecord(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/file/list', data: requestParameters }));
        });
    }
    /**
    * 下载记录中(下载)
    * @summary 下载记录中(下载)
    * @param {SasListPriceApiFileRecordDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    fileRecordDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/listPrice/file/record/download', params: requestParameters }));
        });
    }
    /**
    * 标记失败
    * @summary 标记失败
    * @param {SasListPriceApiManualFailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    manualFail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/manual/fail', data: requestParameters }));
        });
    }
    /**
    * 待提交数量
    * @summary 待提交数量
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    pendingCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/listPrice/list/pending/count' }));
        });
    }
    /**
    * 待提交
    * @summary 待提交
    * @param {SasListPriceApiPendingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    pendingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/list/pending', data: requestParameters }));
        });
    }
    /**
    * 需求池
    * @summary 需求池
    * @param {SasListPriceApiPoolListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    poolList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/list/pool', data: requestParameters }));
        });
    }
    /**
    * 已提交
    * @summary 已提交
    * @param {SasListPriceApiSubmittedListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasListPriceApi
    */
    submittedList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/listPrice/list/submitted', data: requestParameters }));
        });
    }
}
exports.SpotterSasListPriceApiSdk = SpotterSasListPriceApiSdk;
