// @spotter/gmesh-api-sdk/lib/api/sas-cost-price-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSasCostPriceApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SasCostPriceApi - object-oriented interface
* @export
* @class SasCostPriceApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSasCostPriceApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 添加批注
    * @summary 添加批注
    * @param {SasCostPriceApiAddAnnotationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    addAnnotation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/costPrice/annotation/add', data: requestParameters }));
        });
    }
    /**
    * 调价端口列表
    * @summary 调价端口列表
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    adjustPortList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/costPrice/adjustPort/list' }));
        });
    }
    /**
    * 批注列表
    * @summary 批注列表
    * @param {SasCostPriceApiAnnotationListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    annotationList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum) !== undefined) {
                params['sasNum'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/costPrice/annotation/list', params: requestParameters }));
        });
    }
    /**
    * 回退
    * @summary 回退
    * @param {SasCostPriceApiCancelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    cancel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/costPrice/cancel', data: requestParameters }));
        });
    }
    /**
    * 确认提交
    * @summary 确认提交
    * @param {SasCostPriceApiConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    confirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/costPrice/file/confirm', params: requestParameters }));
        });
    }
    /**
    * 详情
    * @summary 详情
    * @param {SasCostPriceApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum) !== undefined) {
                params['sasNum'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/costPrice/detail', params: requestParameters }));
        });
    }
    /**
    * 提交反馈-下载
    * @summary 提交反馈-下载
    * @param {SasCostPriceApiFileFeedbackDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    fileFeedbackDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/costPrice/file/feedback/download', data: requestParameters }));
        });
    }
    /**
    * 文件列表
    * @summary 文件列表
    * @param {SasCostPriceApiFileRecordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    fileRecord(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/costPrice/file/list', data: requestParameters }));
        });
    }
    /**
    * 下载记录中(下载),文件路径获取
    * @summary 下载记录中(下载),文件路径获取
    * @param {SasCostPriceApiFindFilePathRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    findFilePath(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/costPrice/file/record/download', params: requestParameters }));
        });
    }
    /**
    * 待提交
    * @summary 待提交
    * @param {SasCostPriceApiPendingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    pendingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/costPrice/list/pending', data: requestParameters }));
        });
    }
    /**
    * 已提交
    * @summary 已提交
    * @param {SasCostPriceApiSubmittedListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasCostPriceApi
    */
    submittedList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/costPrice/list/submitted', data: requestParameters }));
        });
    }
}
exports.SpotterSasCostPriceApiSdk = SpotterSasCostPriceApiSdk;
