// @spotter/gmesh-api-sdk/lib/api/sas-best-deal-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSasBestDealApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SasBestDealApi - object-oriented interface
* @export
* @class SasBestDealApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSasBestDealApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 批注列表
    * @summary 批注列表
    * @param {SasBestDealApiAnnotationListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    annotationList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum) !== undefined) {
                params['sasNum'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/bestDeal/annotation/list', params: requestParameters }));
        });
    }
    /**
    * 回退
    * @summary 回退
    * @param {SasBestDealApiCancelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    cancel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/cancel', data: requestParameters }));
        });
    }
    /**
    * 创建测试
    * @summary 创建测试
    * @param {SasBestDealApiCreateProductsTestRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    createProductsTest(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/create', data: requestParameters }));
        });
    }
    /**
    * 详情
    * @summary 详情
    * @param {SasBestDealApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum) !== undefined) {
                params['sasNum'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sasNum;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/bestDeal/detail', params: requestParameters }));
        });
    }
    /**
    * 编辑不可提交数据
    * @summary 编辑不可提交数据
    * @param {SasBestDealApiEditSubmitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    editSubmit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/edit/submit', data: requestParameters }));
        });
    }
    /**
    * 文件确认提交(失败：返回失败的数据)
    * @summary 文件确认提交
    * @param {SasBestDealApiFileConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    fileConfirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/bestDeal/file/confirm', params: requestParameters }));
        });
    }
    /**
    * 提交反馈-下载
    * @summary 提交反馈-下载
    * @param {SasBestDealApiFileFeedbackDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    fileFeedbackDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/file/feedback/download', data: requestParameters }));
        });
    }
    /**
    * 下载记录
    * @summary 下载记录
    * @param {SasBestDealApiFileRecordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    fileRecord(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/file/list', data: requestParameters }));
        });
    }
    /**
    * 下载记录中(下载)
    * @summary 下载记录中(下载)
    * @param {SasBestDealApiFileRecordDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    fileRecordDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/bestDeal/file/record/download', params: requestParameters }));
        });
    }
    /**
    * 标记失败
    * @summary 标记失败
    * @param {SasBestDealApiManualFailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    manualFail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/manual/fail', data: requestParameters }));
        });
    }
    /**
    * 补充信息
    * @summary 补充信息
    * @param {SasBestDealApiManualSuccessRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    manualSuccess(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/manual/success', data: requestParameters }));
        });
    }
    /**
    * 待提交数量
    * @summary 待提交数量
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    pendingCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-sas-web/sas/bestDeal/list/pending/count' }));
        });
    }
    /**
    * 待提交
    * @summary 待提交
    * @param {SasBestDealApiPendingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    pendingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/list/pending', data: requestParameters }));
        });
    }
    /**
    * 待提交
    * @summary 产品分页查询
    * @param {SasBestDealApiProductListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    productList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/list/product', data: requestParameters }));
        });
    }
    /**
    * 已提交
    * @summary 已提交
    * @param {SasBestDealApiSubmittedListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    submittedList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/list/submitted', data: requestParameters }));
        });
    }
    /**
    * gls测试
    * @summary gls测试
    * @param {SasBestDealApiTestUpDateProductRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SasBestDealApi
    */
    testUpDateProduct(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sas-web/sas/bestDeal/gls', data: requestParameters }));
        });
    }
}
exports.SpotterSasBestDealApiSdk = SpotterSasBestDealApiSdk;
