// @spotter/gmesh-api-sdk/lib/api/second-payment-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSecondPaymentApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SecondPaymentApi - object-oriented interface
* @export
* @class SecondPaymentApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSecondPaymentApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 创建二期款
    * @summary 创建二期款
    * @param {SecondPaymentApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/secondPayment/create', data: requestParameters }));
        });
    }
    /**
    * 根据款项编码导出二期款详情
    * @summary 根据款项编码导出二期款详情
    * @param {SecondPaymentApiExportDetailByPaymentCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    exportDetailByPaymentCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode) !== undefined) {
                params['paymentCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/secondPayment/exportDetail', params: requestParameters }));
        });
    }
    /**
    * 导出预览二期款
    * @summary 导出预览二期款
    * @param {SecondPaymentApiExportPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    exportPreview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/secondPayment/exportPreview', data: requestParameters }));
        });
    }
    /**
    * 二期款预览创建前检查
    * @summary 二期款预览创建前检查
    * @param {SecondPaymentApiPreCreateCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    preCreateCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.salesEndDateMs) !== undefined) {
                params['salesEndDateMs'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.salesEndDateMs;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/secondPayment/preCreateCheck', params: requestParameters }));
        });
    }
    /**
    * 预览二期款
    * @summary 预览二期款
    * @param {SecondPaymentApiPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    preview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/secondPayment/preview', data: requestParameters }));
        });
    }
    /**
    * 根据款项编码查询二期款详情
    * @summary 根据款项编码查询二期款详情
    * @param {SecondPaymentApiQueryDetailByPaymentCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    queryDetailByPaymentCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode) !== undefined) {
                params['paymentCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/secondPayment/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 查询指定公司最后一次二期款
    * @summary 查询指定公司最后一次二期款
    * @param {SecondPaymentApiQueryLastPaymentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    queryLastPayment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/secondPayment/queryLastPayment', params: requestParameters }));
        });
    }
    /**
    * 根据条件查询二期款
    * @summary 根据条件查询二期款
    * @param {SecondPaymentApiQueryPaymentPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    queryPaymentPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/secondPayment/query', data: requestParameters }));
        });
    }
    /**
    * 查询预览二期款
    * @summary 查询预览二期款
    * @param {SecondPaymentApiQueryPreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    queryPreview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/secondPayment/queryPreview', data: requestParameters }));
        });
    }
    /**
    * 查询指定预账单二期款结算明细
    * @summary 查询指定预账单二期款结算明细
    * @param {SecondPaymentApiQuerySettleDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    querySettleDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/secondPayment/querySettle', params: requestParameters }));
        });
    }
    /**
    * 查询未结算累计池
    * @summary 查询未结算累计池
    * @param {SecondPaymentApiQueryUnsettlementPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    queryUnsettlementPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/secondPaymentUnsettlement/query', data: requestParameters }));
        });
    }
    /**
    * 撤销二期款
    * @summary 撤销二期款
    * @param {SecondPaymentApiReverseRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    reverse(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/secondPayment/reverse', data: requestParameters }));
        });
    }
    /**
    * 本次未结算预览
    * @summary 本次未结算预览
    * @param {SecondPaymentApiUnsettlePreviewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SecondPaymentApi
    */
    unsettlePreview(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/secondPaymentUnsettlement/unsettlePreview', data: requestParameters }));
        });
    }
}
exports.SpotterSecondPaymentApiSdk = SpotterSecondPaymentApiSdk;
