// @spotter/gmesh-api-sdk/lib/api/sevc-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSevcApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SevcApi - object-oriented interface
* @export
* @class SevcApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSevcApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    *
    * @summary 删除组织
    * @param {SevcApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.orgId) !== undefined) {
                params['orgId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.orgId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-guard-web/org', params: requestParameters }));
        });
    }
    /**
    * 删除角色
    * @summary 删除角色
    * @param {SevcApiDelete0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    _delete_1(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/delete', data: requestParameters }));
        });
    }
    /**
    * 导出月度对账单
    * @summary 导出月度对账单
    * @param {SevcApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode) !== undefined) {
                params['reconciliationCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/export', params: requestParameters }));
        });
    }
    /**
    * 科技服务费导出结算单
    * @summary 科技服务费导出
    * @param {SevcApiExport0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    _export_2(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/technology/export', params: requestParameters }));
        });
    }
    /**
    * 新增文件处理任务
    * @summary 新增文件处理任务
    * @param {SevcApiAddRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    add(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/job/add', data: requestParameters }));
        });
    }
    /**
    * 添加app
    * @summary 添加app
    * @param {SevcApiAddAppRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    addApp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/app', data: requestParameters }));
        });
    }
    /**
    * 添加权限
    * @summary 添加权限
    * @param {SevcApiAddPermitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    addPermit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/add-permit', data: requestParameters }));
        });
    }
    /**
    * 添加实时销量asin
    * @summary 添加实时销量asin
    * @param {SevcApiAddRealtimeSalesAsinRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    addRealtimeSalesAsin(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/add', data: requestParameters }));
        });
    }
    /**
    * 添加权限并绑定角色
    * @summary 添加权限并绑定角色
    * @param {SevcApiAddRoleAndPermitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    addRoleAndPermit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/add-role-and-permit', data: requestParameters }));
        });
    }
    /**
    * 添加权限-单条
    * @summary 添加权限-单条
    * @param {SevcApiAddSoloPermitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    addSoloPermit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/add-permit-solo', data: requestParameters }));
        });
    }
    /**
    * 添加员工
    * @summary 添加员工
    * @param {SevcApiAddUserRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    addUser(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/add-users', data: requestParameters }));
        });
    }
    /**
    * FF费价格调整明细信息
    * @summary FF费价格调整明细信息
    * @param {SevcApiAdjustDetailFulfillmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    adjustDetailFulfillment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust_detail/fulfillment', data: requestParameters }));
        });
    }
    /**
    * storage费价格调整明细信息
    * @summary storage费价格调整明细信息
    * @param {SevcApiAdjustDetailStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    adjustDetailStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust_detail/storage', data: requestParameters }));
        });
    }
    /**
    * vas费价格调整明细信息
    * @summary vas费价格调整明细信息
    * @param {SevcApiAdjustDetailVasRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    adjustDetailVas(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/adjust_detail/vas', data: requestParameters }));
        });
    }
    /**
    * 协议类型
    * @summary 枚举-协议类型
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    agreementType(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/enum/agreement/type' }));
        });
    }
    /**
    * 获取所有枚举信息
    * @summary 获取所有枚举信息
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    all(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-operation-log-web/enums/all' }));
        });
    }
    /**
    * 获取所有枚举
    * @summary 获取所有枚举
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    allEnum(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/enum/all' }));
        });
    }
    /**
    * 获取所有枚举
    * @summary 获取所有枚举
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    allEnum_3(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/invoice/enum/all' }));
        });
    }
    /**
    * 获取所有枚举
    * @summary 获取所有枚举
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    allEnum_4(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-monitor-web/enum/monitor/all' }));
        });
    }
    /**
    * 审核海外仓费用对账单
    * @summary 审核海外仓费用对账单
    * @param {SevcApiAuditReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    auditReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/audit', data: requestParameters }));
        });
    }
    /**
    * 自动分配库存
    * @summary 自动分配库存
    * @param {SevcApiAutoAllocateInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    autoAllocateInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/auto/allocate', data: requestParameters }));
        });
    }
    /**
    * 批量转出员工/添加成员
    * @summary 批量转出员工/添加成员
    * @param {SevcApiBatchTransferRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    batchTransfer(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/batch-transfer', data: requestParameters }));
        });
    }
    /**
    * 编辑知识文档
    * @summary 编辑知识文档，用于修改分组，生效时间等
    * @param {SevcApiBatchUpdateKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    batchUpdateKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/update', data: requestParameters }));
        });
    }
    /**
    * 批量出库
    * @summary 批量出库
    * @param {SevcApiBatchUpdateOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    batchUpdateOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/batchUpdate', data: requestParameters }));
        });
    }
    /**
    * 亚马逊BBO订单统计
    * @summary 亚马逊BBO订单统计
    * @param {SevcApiBboStatisticsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    bboStatistics(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/statistics/bbo', data: requestParameters }));
        });
    }
    /**
    * 批量转出员工/添加成员
    * @summary 根据业务线获取对应的leader
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    businessLeaderList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/business/leader/list' }));
        });
    }
    /**
    * 报价测算工具
    * @summary 报价测算工具
    * @param {SevcApiCalFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    calFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/cal/quotationFee', data: requestParameters }));
        });
    }
    /**
    * 仓储费用计算
    * @summary 仓储费用计算
    * @param {SevcApiCalStorageFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    calStorageFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/cal/storage/fee', data: requestParameters }));
        });
    }
    /**
    * 仓储费用计算(批量)
    * @summary 仓储费用计算(批量)
    * @param {SevcApiCalStorageFeeBatchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    calStorageFeeBatch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/cal/storage/fee/batch', data: requestParameters }));
        });
    }
    /**
    * 回调通知
    * @summary 回调通知
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    callback(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/callback' }));
        });
    }
    /**
    * 模拟回调通知
    * @summary 模拟回调通知
    * @param {SevcApiCallbackMockRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    callbackMock(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/callback/callbackMock', data: requestParameters }));
        });
    }
    /**
    * 亚马逊授权回调url
    * @summary 亚马逊授权回调url
    * @param {SevcApiCallback0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    callback_5(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.state) !== undefined) {
                params['state'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.state;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.code) !== undefined) {
                params['code'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.code;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ads-web/amazing/auth/callback', params: requestParameters }));
        });
    }
    /**
    * 订单取消
    * @summary 订单取消
    * @param {SevcApiCancelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    cancel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/cancel', data: requestParameters }));
        });
    }
    /**
    * 修改密码
    * @summary 修改密码
    * @param {SevcApiChangePasswordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    changePassword(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/password', data: requestParameters }));
        });
    }
    /**
    * 用户中心修改密码
    * @summary 用户中心修改密码
    * @param {SevcApiChangePasswordForUserCenterRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    changePasswordForUserCenter(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/update-password', data: requestParameters }));
        });
    }
    /**
    * 用户中心修改密码-检查密码是否正确
    * @summary 用户中心修改密码-检查密码是否正确
    * @param {SevcApiCheckPasswordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    checkPassword(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/check-password', data: requestParameters }));
        });
    }
    /**
    * 导入知识文档标题重复校验
    * @summary 导入知识文档标题重复校验
    * @param {SevcApiCheckRepeatKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    checkRepeatKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/import/check/repeat', data: requestParameters }));
        });
    }
    /**
    * 校验跳转链接加密sign
    * @summary 校验跳转链接加密sign
    * @param {SevcApiCheckSignRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    checkSign(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sign) !== undefined) {
                params['sign'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sign;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/validate/check-sign', params: requestParameters }));
        });
    }
    /**
    * 清理缓存
    * @summary 清理缓存
    * @param {SevcApiClearCacheRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    clearCache(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/cache/clear', data: requestParameters }));
        });
    }
    /**
    * 根据仓库编码获取公司列表
    * @summary 根据仓库编码获取公司列表
    * @param {SevcApiCompanyListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    companyList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/company/list', data: requestParameters }));
        });
    }
    /**
    * 订单确认
    * @summary 订单确认
    * @param {SevcApiConfirmRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    confirm(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/confirm', data: requestParameters }));
        });
    }
    /**
    * 月度对账单附件重传
    * @summary 月度对账单附件重传
    * @param {SevcApiConfirmAttachReLoadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    confirmAttachReLoad(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/confirmAttachReLoad', data: requestParameters }));
        });
    }
    /**
    * 修改bbo接单数量
    * @summary 修改bbo接单数量
    * @param {SevcApiConfirmBboRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    confirmBbo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/bbo/confirm', data: requestParameters }));
        });
    }
    /**
    * 订单确认预校验
    * @summary 订单确认预校验
    * @param {SevcApiConfirmPreCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    confirmPreCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/confirm/preCheck', data: requestParameters }));
        });
    }
    /**
    * 新增银行信息
    * @summary 新增银行信息
    * @param {SevcApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company/bank/create', data: requestParameters }));
        });
    }
    /**
    * 创建入库通知单
    * @summary 创建入库通知单
    * @param {SevcApiCreateInboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    createInbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/create', data: requestParameters }));
        });
    }
    /**
    * 创建知识文档
    * @summary 创建知识文档
    * @param {SevcApiCreateKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    createKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/create', data: requestParameters }));
        });
    }
    /**
    * 创建知识分类
    * @summary 创建知识分类
    * @param {SevcApiCreateKnowledgeCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    createKnowledgeCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/category/create', data: requestParameters }));
        });
    }
    /**
    * 添加一个公司
    * @summary 添加一个公司
    * @param {SevcApiCreate0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    create_6(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company', data: requestParameters }));
        });
    }
    /**
    * 添加一个组织
    * @summary 添加一个组织
    * @param {SevcApiCreate1Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    create_7(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org', data: requestParameters }));
        });
    }
    /**
    * 创建角色
    * @summary 创建角色
    * @param {SevcApiCreate2Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    create_8(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role', data: requestParameters }));
        });
    }
    /**
    * 收款账户-当前使用账户
    * @summary 收款账户-当前使用账户
    * @param {SevcApiCurrentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    current(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/current', params: requestParameters }));
        });
    }
    /**
    * 收款账户-当前使用账户
    * @summary 收款账户-当前使用账户
    * @param {SevcApiCurrent0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    current_9(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-account-web/account/payee/current', params: requestParameters }));
        });
    }
    /**
    * 删除知识文档
    * @summary 删除知识文档,软删除
    * @param {SevcApiDeleteKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    deleteKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/delete', data: requestParameters }));
        });
    }
    /**
    * 删除分类
    * @summary 删除分类
    * @param {SevcApiDeleteKnowledgeCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    deleteKnowledgeCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-knowledge-web/knowledge/category/delete', params: requestParameters }));
        });
    }
    /**
    * 删除实时销量asin
    * @summary 删除实时销量asin
    * @param {SevcApiDeleteRealtimeSalesAsinRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    deleteRealtimeSalesAsin(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/delete', data: requestParameters }));
        });
    }
    /**
    * 结算单-详情
    * @summary 结算单-详情
    * @param {SevcApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/penalty/settle/detail', params: requestParameters }));
        });
    }
    /**
    * 分类详情
    * @summary 分类详情
    * @param {SevcApiDetailKnowledgeCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    detailKnowledgeCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-knowledge-web/knowledge/category/detail', params: requestParameters }));
        });
    }
    /**
    * 海外仓费用对账单详情
    * @summary 海外仓费用对账单详情
    * @param {SevcApiDetailReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    detailReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/detail', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 其他费用详情
    * @param {SevcApiDetail0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    detail_10(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/other/detail', params: requestParameters }));
        });
    }
    /**
    * 查询指定结算单的shortage费用
    * @summary 查询指定结算单的shortage费用
    * @param {SevcApiDetail1Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    detail_11(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/shortage/invoice/payment/detail', params: requestParameters }));
        });
    }
    /**
    * 订单详情
    * @summary 订单详情
    * @param {SevcApiDetail2Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    detail_12(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/detail', data: requestParameters }));
        });
    }
    /**
    * 合同详情
    * @summary 合同详情
    * @param {SevcApiDetail3Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    detail_13(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-business-web/contract/detail', params: requestParameters }));
        });
    }
    /**
    * 亚马逊df统计
    * @summary 亚马逊df统计
    * @param {SevcApiDfStatisticRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    dfStatistic(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/statistics/df', data: requestParameters }));
        });
    }
    /**
    * -1:冻结 1:已激活 2:待激活
    * @summary 更新公司状态
    * @param {SevcApiEnableOrDisableCompanyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    enableOrDisableCompany(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.status) !== undefined) {
                params['status'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.status;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/company/status', params: requestParameters }));
        });
    }
    /**
    * 更新用户状态
    * @summary 更新用户状态
    * @param {SevcApiEnableOrDisableUserRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    enableOrDisableUser(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.status) !== undefined) {
                params['status'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.status;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.isActive) !== undefined) {
                params['isActive'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.isActive;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/status', params: requestParameters }));
        });
    }
    /**
    * 导出预账单
    * @summary 导出预账单
    * @param {SevcApiExportAdvanceBillRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportAdvanceBill(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/advanceBill/exportAdvanceBill', data: requestParameters }));
        });
    }
    /**
    * 根据款项编码导出一期款详情
    * @summary 根据款项编码导出一期款详情
    * @param {SevcApiExportDetailByPaymentCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportDetailByPaymentCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode) !== undefined) {
                params['paymentCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/firstPayment/exportDetail', params: requestParameters }));
        });
    }
    /**
    * 根据款项编码导出二期款详情
    * @summary 根据款项编码导出二期款详情
    * @param {SevcApiExportDetailByPaymentCode0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportDetailByPaymentCode_14(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode) !== undefined) {
                params['paymentCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/secondPayment/exportDetail', params: requestParameters }));
        });
    }
    /**
    * 亚马逊df可售数量导出
    * @summary 亚马逊df可售数量导出
    * @param {SevcApiExportDfInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportDfInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/export/df/inventory', data: requestParameters }));
        });
    }
    /**
    * 亚马逊DF订单列表导出
    * @summary 亚马逊DF订单列表导出
    * @param {SevcApiExportDfOrderRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportDfOrder(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/export/df', data: requestParameters }));
        });
    }
    /**
    * 导出Excel
    * @summary 导出Excel
    * @param {SevcApiExportDfShipmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    exportDfShipment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/shipment/df/amz/export', data: requestParameters }));
        });
    }
    /**
    * 订单导出
    * @summary 订单导出
    * @param {SevcApiExportExcelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportExcel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/exportExcel', data: requestParameters }));
        });
    }
    /**
    * 入库通知单导出
    * @summary 入库通知单导出
    * @param {SevcApiExportInboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    exportInbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/export', data: requestParameters }));
        });
    }
    /**
    * 知识文档导出
    * @summary 知识文档导出
    * @param {SevcApiExportKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/export', data: requestParameters }));
        });
    }
    /**
    * 导出Pdf
    * @summary 导出Pdf
    * @param {SevcApiExportPdfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    exportPdf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ess-web/ess/common/pdf', data: requestParameters }));
        });
    }
    /**
    * 导出海外仓费用对账单excel
    * @summary 导出海外仓费用对账单excel
    * @param {SevcApiExportReconciliationRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportReconciliation(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/reconciliation/export_excel', data: requestParameters }));
        });
    }
    /**
    * 下载仓储费结算单excel
    * @summary 下载结算单excel
    * @param {SevcApiExportSettlementRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    exportSettlement(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/settlement/export', data: requestParameters }));
        });
    }
    /**
    * 下载已签署文件及附属材料
    * @summary 下载已签署文件及附属材料
    * @param {SevcApiFileDownloadUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    fileDownloadUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/org/signFlow/fileDownloadUrl', data: requestParameters }));
        });
    }
    /**
    * 获取文件上传状态
    * @summary 获取文件上传状态
    * @param {SevcApiFileStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    fileStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/files/fileStatus', data: requestParameters }));
        });
    }
    /**
    * 上传本地文件
    * @summary 上传本地文件
    * @param {SevcApiFileUploadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    fileUpload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/files/fileUpload', data: requestParameters }));
        });
    }
    /**
    * 合同分页查询
    * @summary 合同分页查询
    * @param {SevcApiFindPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    findPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/list', data: requestParameters }));
        });
    }
    /**
    * 免仓租期枚举
    * @summary 枚举-免仓租期
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    freeLeaseDays(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/enum/free/lease/days' }));
        });
    }
    /**
    * 全文检索知识文档
    * @summary 全文检索知识文档
    * @param {SevcApiFullTextSearchKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    fullTextSearchKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/full/search', data: requestParameters }));
        });
    }
    /**
    * 模糊查询分类
    * @summary 模糊查询分类
    * @param {SevcApiFuzzQueryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    fuzzQuery(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.keyWord) !== undefined) {
                params['keyWord'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.keyWord;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.language) !== undefined) {
                params['language'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.language;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-knowledge-web/knowledge/category/fuzzQuery', params: requestParameters }));
        });
    }
    /**
    * 获取active offer列表
    * @summary 获取active offer列表
    * @param {SevcApiGetActiveOfferPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getActiveOfferPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/active/offer', data: requestParameters }));
        });
    }
    /**
    * 获取所有APP
    * @summary 获取所有APP
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getAll(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/app' }));
        });
    }
    /**
    * 仓库所在地域下拉列表
    * @summary 仓库所在地域下拉列表
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getAmazonCountryAndStateVO(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/state/all' }));
        });
    }
    /**
    * 亚马逊授权url
    * @summary 亚马逊授权url
    * @param {SevcApiGetAuthUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getAuthUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountName) !== undefined) {
                params['accountName'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountName;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.country) !== undefined) {
                params['country'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.country;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ads-web/amazing/auth/url', params: requestParameters }));
        });
    }
    /**
    * 亚马逊授权url
    * @summary 亚马逊授权url
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getAuthUrlForTest(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ads-web/amazing/auth/url/test' }));
        });
    }
    /**
    * 根据公司id获取所有已配置的asin总数
    * @summary 根据公司id获取所有已配置的asin总数
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getCompanyCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/report/settings/count' }));
        });
    }
    /**
    * 月度对账单获取每种类型的数量
    * @summary 月度对账单获取每种类型的数量
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/getCounter' }));
        });
    }
    /**
    * 查询所有的国家代码枚举值
    * @summary 获取所有的国家代码
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getCountryCode(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/company/country_code' }));
        });
    }
    /**
    * 收款账户-当前使用账户列表
    * @summary 收款账户-当前使用账户列表
    * @param {SevcApiGetCurrentListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getCurrentList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/payee/account/get_current_list', data: requestParameters }));
        });
    }
    /**
    * 收款账户-当前使用账户列表
    * @summary 收款账户-当前使用账户列表
    * @param {SevcApiGetCurrentList0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getCurrentList_15(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-account-web/account/payee/get_current_list', data: requestParameters }));
        });
    }
    /**
    * shipment详情(VCPO和DF)
    * @summary dfShipment详情列表
    * @param {SevcApiGetDetailsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getDetails(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/shipment/df/amz/detailV2', data: requestParameters }));
        });
    }
    /**
    * 获取跨企业授权信息
    * @summary 获取跨企业授权信息
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getExternalAuthInfo(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/org/seal/getExternalAuthInfo' }));
        });
    }
    /**
    * 入库通知单详情
    * @summary 入库通知单详情
    * @param {SevcApiGetInboundNoticeDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getInboundNoticeDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/notice/detail/list', data: requestParameters }));
        });
    }
    /**
    * 入库通知单箱规等其他信息详情
    * @summary 入库通知单箱规等其他信息详情
    * @param {SevcApiGetInboundNoticeOtherDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getInboundNoticeOtherDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/notice/detail/otherInfo', data: requestParameters }));
        });
    }
    /**
    * 入库通知单分页
    * @summary 入库通知单分页
    * @param {SevcApiGetInboundNoticePageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getInboundNoticePageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * 编辑展示，公司信息
    * @summary 公司信息
    * @param {SevcApiGetInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/company', params: requestParameters }));
        });
    }
    /**
    * 根据公司id和部门code获取部门leader
    * @summary 获取部门leader
    * @param {SevcApiGetLeaderByCompanyIdAndOrgIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getLeaderByCompanyIdAndOrgId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.orgId) !== undefined) {
                params['orgId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.orgId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user/find/org/leader', params: requestParameters }));
        });
    }
    /**
    * 查询指标
    * @summary 查询指标
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getMetrics(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/metrics/prometheus' }));
        });
    }
    /**
    * 获取企业认证授权链接
    * @summary 获取企业认证授权链接
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getOrgAuthUrl(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getOrgAuthUrl' }));
        });
    }
    /**
    * 查询企业授权信息
    * @summary 查询企业授权信息
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getOrgAuthorizedInfo(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getOrgAuthorizedInfo' }));
        });
    }
    /**
    * 查询企业认证信息
    * @summary 查询企业认证信息
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getOrgIdentityInfo(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getOrgIdentityInfo' }));
        });
    }
    /**
    * 出库通知单详情
    * @summary 出库通知单详情
    * @param {SevcApiGetOutboundNoticeDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getOutboundNoticeDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/detail/list', data: requestParameters }));
        });
    }
    /**
    * 出库通知单列表分页
    * @summary 出库通知单列表分页
    * @param {SevcApiGetOutboundNoticePageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getOutboundNoticePageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/list', data: requestParameters }));
        });
    }
    /**
    * 从谷歌获取地址预测信息
    * @summary 从谷歌获取地址预测信息
    * @param {SevcApiGetPredictionAddressRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getPredictionAddress(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/getPredictionAddress', data: requestParameters }));
        });
    }
    /**
    * 获取个人认证授权链接
    * @summary 获取个人认证授权链接
    * @param {SevcApiGetPsnAuthUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getPsnAuthUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getPsnAuthUrl', data: requestParameters }));
        });
    }
    /**
    * 查询个人授权信息
    * @summary 查询个人授权信息
    * @param {SevcApiGetPsnAuthorizedInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getPsnAuthorizedInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getPsnAuthorizedInfo', data: requestParameters }));
        });
    }
    /**
    * 查询个人认证信息
    * @summary 查询个人认证信息
    * @param {SevcApiGetPsnIdentityInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getPsnIdentityInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/auth/getPsnIdentityInfo', data: requestParameters }));
        });
    }
    /**
    * 获取active offer列表
    * @summary 查询用户已经配置的产品
    * @param {SevcApiGetSettingProductPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getSettingProductPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/settings/realtime/setting/search', data: requestParameters }));
        });
    }
    /**
    * shipment列表(VCPO和DF)
    * @summary shipment列表
    * @param {SevcApiGetShipmentListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getShipmentList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/shipment/df/amz/list', data: requestParameters }));
        });
    }
    /**
    * 下载ShippingLabel
    * @summary 下载ShippingLabel
    * @param {SevcApiGetShippingLabelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getShippingLabel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/shipment/df/amz/label/url', data: requestParameters }));
        });
    }
    /**
    * 根据userId获取用户所有下级用户信息
    * @summary 根据userId获取用户所有下级用户信息
    * @param {SevcApiGetUnderlingByUserIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getUnderlingByUserId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user/underling', params: requestParameters }));
        });
    }
    /**
    * 根据用户Id 获取某个用户信息
    * @summary 获取用户
    * @param {SevcApiGetUserByIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getUserById(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user', params: requestParameters }));
        });
    }
    /**
    * 可以通过关键词 搜索, 同时需要对orgId 进行排除
    * @summary 获取组织下的人员列表
    * @param {SevcApiGetUserExcludeOrgIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getUserExcludeOrgId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/users', data: requestParameters }));
        });
    }
    /**
    * 可以通过关键词 搜索, 同时需要对roleCode 进行排除
    * @summary 获取角色下的人员列表
    * @param {SevcApiGetUserExcludeRoleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getUserExcludeRole(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/users', data: requestParameters }));
        });
    }
    /**
    * 获取用户列表, 根据多个查询条件进行筛选
    * @summary 用户列表
    * @param {SevcApiGetUserListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getUserList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/list', data: requestParameters }));
        });
    }
    /**
    * 根据组织架构编码获取用户列表(包括子目录下用户)
    * @summary 根据组织架构编码获取用户列表(包括子目录下用户)
    * @param {SevcApiGetUserListByOrgCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    getUserListByOrgCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/users/all', data: requestParameters }));
        });
    }
    /**
    * 用户下拉列表
    * @summary 用户下拉列表
    * @param {SevcApiGetUsersByNameRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getUsersByName(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/users', data: requestParameters }));
        });
    }
    /**
    * 海外仓结算费用详情
    * @summary 海外仓结算费用详情
    * @param {SevcApiGetWarehouseFeeSettlementRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    getWarehouseFeeSettlement(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/settlement/detail', data: requestParameters }));
        });
    }
    /**
    * 出库通知单捡货
    * @summary 出库通知单捡货
    * @param {SevcApiGmeshPickUpOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    gmeshPickUpOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/pickup', data: requestParameters }));
        });
    }
    /**
    * 出库通知单批量捡货
    * @summary 出库通知单批量捡货
    * @param {SevcApiGmeshPickUpOutboundBatchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    gmeshPickUpOutboundBatch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/batch/pickup', data: requestParameters }));
        });
    }
    /**
    * 导入知识文档
    * @summary 导入知识文档
    * @param {SevcApiImportKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    importKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/import', data: requestParameters }));
        });
    }
    /**
    * 查看银行信息
    * @summary 查看银行信息
    * @param {SevcApiInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    info(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/company/bank/info', params: requestParameters }));
        });
    }
    /**
    * 激活设置密码
    * @summary 激活设置密码
    * @param {SevcApiInitPasswordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    initPassword(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/active/password', data: requestParameters }));
        });
    }
    /**
    * 库存变动明细分页
    * @summary 库存变动明细分页
    * @param {SevcApiInventoryChangeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    inventoryChangeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/change/detail', data: requestParameters }));
        });
    }
    /**
    * 获取文件上传状态
    * @summary 获取文件上传状态
    * @param {SevcApiKeywordPositionsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    keywordPositions(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/files/keywordPositions', data: requestParameters }));
        });
    }
    /**
    * 知识文档详情
    * @summary 知识文档详情
    * @param {SevcApiKnowledgeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    knowledgeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-knowledge-web/knowledge/detail', params: requestParameters }));
        });
    }
    /**
    * Seller管理/公司管理/公司列表分页
    * @summary Seller管理/公司管理/公司列表分页
    * @param {SevcApiListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    list(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company/list', data: requestParameters }));
        });
    }
    /**
    * 账号列表
    * @summary 账号列表
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    listAccount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/list/account' }));
        });
    }
    /**
    * 公司列表
    * @summary 公司列表
    * @param {SevcApiListAllRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    listAll(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company/simple/list', data: requestParameters }));
        });
    }
    /**
    * 获取所有一级类目信息
    * @summary 获取所有一级类目信息
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    listAllPrimaryCategory(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/category/first/list/all' }));
        });
    }
    /**
    * 查看角色绑定权限列表，通过角色Id进行展示
    * @summary 查看角色绑定权限列表
    * @param {SevcApiListAll0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    listAll_16(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.type) !== undefined) {
                params['type'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.type;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.appCode) !== undefined) {
                params['appCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.appCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/permit/all', params: requestParameters }));
        });
    }
    /**
    * 带分页搜索用户列表，通过组织Id搜索
    * @summary 通过组织Id查询用户列表
    * @param {SevcApiListByOrgIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    listByOrgId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/list/org', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df待确认订单列表
    * @summary 亚马逊df待确认订单列表
    * @param {SevcApiListDfNewRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    listDfNew(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/list/df/new', data: requestParameters }));
        });
    }
    /**
    * 模糊查询产品信息
    * @summary 模糊查询产品信息
    * @param {SevcApiListFuzzyProductRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    listFuzzyProduct(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/product', data: requestParameters }));
        });
    }
    /**
    * 库存列表
    * @summary 库存列表
    * @param {SevcApiListInboundInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    listInboundInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/inventory/warehouse/list', data: requestParameters }));
        });
    }
    /**
    * 入库通知单详情列表
    * @summary 入库通知单详情列表
    * @param {SevcApiListInboundNoticeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    listInboundNoticeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/notice/detail', data: requestParameters }));
        });
    }
    /**
    * 页面未处理信息提示
    * @summary 未处理信息
    * @param {SevcApiListInfoTipsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    listInfoTips(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/info', data: requestParameters }));
        });
    }
    /**
    * 库存变动明细列表
    * @summary 库存变动明细列表
    * @param {SevcApiListInventoryChangeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    listInventoryChange(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/change/list', data: requestParameters }));
        });
    }
    /**
    * 库存汇总带箱规
    * @summary 库存汇总带箱规
    * @param {SevcApiListInventorySummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    listInventorySummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/list', data: requestParameters }));
        });
    }
    /**
    * 库存产品列表
    * @summary 库存产品列表
    * @param {SevcApiListProductRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    listProduct(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/inventory/warehouse/product/list', data: requestParameters }));
        });
    }
    /**
    * 角色用户列表
    * @summary 角色用户列表
    * @param {SevcApiListUsersRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    listUsers(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/list-users', data: requestParameters }));
        });
    }
    /**
    * 查看角色绑定权限列表，通过角色Id进行展示
    * @summary 查看角色绑定权限列表
    * @param {SevcApiList0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    list_17(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit', data: requestParameters }));
        });
    }
    /**
    * 分区库存列表
    * @summary 分区库存列表
    * @param {SevcApiList1Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    list_18(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/list/each', data: requestParameters }));
        });
    }
    /**
    * SPT物流查询发货备注
    * @summary SPT物流查询发货备注
    * @param {SevcApiList2Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    list_19(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ess-web/spt/remark/list', data: requestParameters }));
        });
    }
    /**
    * 物流商列表
    * @summary 物流商列表
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    list_20(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/carrier/list' }));
        });
    }
    /**
    * 菜单信息
    * @summary 菜单信息
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    menus(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/validate/menus' }));
        });
    }
    /**
    * 亚马逊df可售数量修改
    * @summary 亚马逊df可售数量修改
    * @param {SevcApiModifyDfInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    modifyDfInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/modify/df/inventory', data: requestParameters }));
        });
    }
    /**
    * 订单单条编辑
    * @summary 订单单条编辑
    * @param {SevcApiModifyStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    modifyStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/modifyStorage', data: requestParameters }));
        });
    }
    /**
    * 订单模式-带物流模式
    * @summary 订单模式-带物流模式
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    orderTypeWithWarehouseList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/order/type/with/warehouse' }));
        });
    }
    /**
    * 获取管理机构印章页面链接
    * @summary 获取管理机构印章页面链接
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    orgSealsManageUrl(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/sevc/org/seal/orgSealsManageUrl' }));
        });
    }
    /**
    * 批量下载出库单下的所有文件
    * @summary 批量下载文件
    * @param {SevcApiOutboundFileBatchDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    outboundFileBatchDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/file/warehouse/batch/download', data: requestParameters }));
        });
    }
    /**
    * 上传文件
    * @summary 上传文件
    * @param {SevcApiOutboundFileUploadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    outboundFileUpload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/file/warehouse/upload', data: requestParameters }));
        });
    }
    /**
    * 出库通知单出库
    * @summary 出库通知单出库
    * @param {SevcApiOutboundNoticeDetailListUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    outboundNoticeDetailListUpdate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/detail/update', data: requestParameters }));
        });
    }
    /**
    * 分区库存分页
    * @summary 分区库存分页
    * @param {SevcApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/page/each', data: requestParameters }));
        });
    }
    /**
    * 收货地址列表
    * @summary 收货地址列表
    * @param {SevcApiPageAddressListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    pageAddressList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/list', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df订单分页
    * @summary 亚马逊df订单分页
    * @param {SevcApiPageDfAllRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    pageDfAll(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/page/df', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df可售数量分页
    * @summary 亚马逊df可售数量分页
    * @param {SevcApiPageDfInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    pageDfInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/page/df/inventory', data: requestParameters }));
        });
    }
    /**
    * 亚马逊df可售数量修改分页
    * @summary 亚马逊df可售数量修改分页
    * @param {SevcApiPageDfInventoryModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    pageDfInventoryModify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/amzOrder/page/df/inventory/modify', data: requestParameters }));
        });
    }
    /**
    * 库存汇总分页
    * @summary 库存汇总分页
    * @param {SevcApiPageInventorySummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    pageInventorySummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * 分页条件查询知识文档
    * @summary 分页条件查询知识文档
    * @param {SevcApiPageKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    pageKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/page', data: requestParameters }));
        });
    }
    /**
    * 订单分页
    * @summary 订单分页
    * @param {SevcApiPage0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    page_21(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/page', data: requestParameters }));
        });
    }
    /**
    * 物流商列表分页查询
    * @summary 物流商列表分页查询
    * @param {SevcApiPage1Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    page_22(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/carrier/page', data: requestParameters }));
        });
    }
    /**
    * 付款周期-带金融服务率
    * @summary 付款周期-带金融服务率
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    paymentWithRates(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-business-web/contract/payment/cycle' }));
        });
    }
    /**
    * 添加权限并绑定角色
    * @summary 添加权限并绑定角色
    * @param {SevcApiPermitRoleSettingRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    permitRoleSetting(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/role-permit-setting', data: requestParameters }));
        });
    }
    /**
    * 添加权限并绑定角色给sevc默认角色编辑后批量复制给所有sevc
    * @summary 添加权限并绑定角色给sevc默认角色编辑后批量复制给所有sevc
    * @param {SevcApiPermitRoleSettingForSevcRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    permitRoleSettingForSevc(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/role-permit-setting_for_sevc', data: requestParameters }));
        });
    }
    /**
    * 添加权限并绑定角色(批量接口)
    * @summary 添加权限并绑定角色(批量接口)
    * @param {SevcApiPermitRolesSettingRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    permitRolesSetting(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/roles-permit-setting', data: requestParameters }));
        });
    }
    /**
    * 权限信息
    * @summary 权限信息
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    permits(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/validate/permit' }));
        });
    }
    /**
    * 价格保护的结算单详情的基础信息
    * @summary 价格保护的结算单详情的基础信息
    * @param {SevcApiPriceProtectedSettleBaseRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    priceProtectedSettleBase(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/base', data: requestParameters }));
        });
    }
    /**
    * 价格保护的结算单详情的明细信息分页查询
    * @summary 价格保护的结算单详情的明细信息分页查询
    * @param {SevcApiPriceProtectedSettleDetailPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    priceProtectedSettleDetailPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/detail_page', data: requestParameters }));
        });
    }
    /**
    * 价格保护的结算单详情的汇总信息分页查询
    * @summary 价格保护的结算单详情的汇总信息分页查询
    * @param {SevcApiPriceProtectedSettleSummaryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    priceProtectedSettleSummaryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/priceProtected/settle/summary_page', data: requestParameters }));
        });
    }
    /**
    * 产品关联报价协议列表
    * @summary 产品关联报价协议列表
    * @param {SevcApiProductSummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    productSummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/product/summary', data: requestParameters }));
        });
    }
    /**
    * 月度对账单推送事件
    * @summary 月度对账单推送事件
    * @param {SevcApiPushEventRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    pushEvent(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/pushEvent', data: requestParameters }));
        });
    }
    /**
    * 根据主键查询详情
    * @summary 根据主键查询详情
    * @param {SevcApiQueryByIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryById(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/advanceBill/queryById', params: requestParameters }));
        });
    }
    /**
    * 分页获取coupon明细
    * @summary 分页获取coupon明细
    * @param {SevcApiQueryCouponRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryCoupon(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/coupon/item/queryPage', data: requestParameters }));
        });
    }
    /**
    * 获取结算单详情
    * @summary 获取结算单详情
    * @param {SevcApiQueryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/bbo/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 根据款项编码查询一期款详情
    * @summary 根据款项编码查询一期款详情
    * @param {SevcApiQueryDetailByPaymentCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryDetailByPaymentCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode) !== undefined) {
                params['paymentCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/firstPayment/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 根据款项编码查询二期款详情
    * @summary 根据款项编码查询二期款详情
    * @param {SevcApiQueryDetailByPaymentCode0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryDetailByPaymentCode_23(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode) !== undefined) {
                params['paymentCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.paymentCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/secondPayment/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 获取结算单详情
    * @summary 获取结算单详情
    * @param {SevcApiQueryDetail0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryDetail_24(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/btr/settle/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 获取coupon结算单详情
    * @summary 获取coupon结算单详情
    * @param {SevcApiQueryDetail1Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryDetail_25(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/coupon/settle/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 获取月度对账单详情
    * @summary 获取月度对账单详情
    * @param {SevcApiQueryDetail2Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryDetail_26(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode) !== undefined) {
                params['reconciliationCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.reconciliationCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/month/reconciliation/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 获取结算单详情
    * @summary 获取promotion结算单详情
    * @param {SevcApiQueryDetail3Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryDetail_27(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/promotion/settle/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 科技服务费获取详情
    * @summary 科技服务费获取详情
    * @param {SevcApiQueryDetail4Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryDetail_28(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/finance/technology/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 查询财务操作日志
    * @summary 查询财务操作日志
    * @param {SevcApiQueryLogRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryLog(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/common/operate/query', data: requestParameters }));
        });
    }
    /**
    * 预账单列表
    * @summary 预账单列表
    * @param {SevcApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/advanceBill/queryPage', data: requestParameters }));
        });
    }
    /**
    * 分页查询月度对账单
    * @summary 分页查询月度对账单
    * @param {SevcApiQueryPage0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryPage_29(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/queryPage', data: requestParameters }));
        });
    }
    /**
    * 付款计划-列表查询
    * @summary 付款计划-列表查询
    * @param {SevcApiQueryPlanPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryPlanPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/month/reconciliation/payment/payPlan/queryPage', data: requestParameters }));
        });
    }
    /**
    * 分页获取promotion
    * @summary 分页获取promotion明细
    * @param {SevcApiQueryPromotionRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryPromotion(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/promotion/item/queryPage', data: requestParameters }));
        });
    }
    /**
    * 列表查询分类树
    * @summary 列表查询分类树
    * @param {SevcApiQueryTreeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryTree(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.keyWord) !== undefined) {
                params['keyWord'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.keyWord;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.language) !== undefined) {
                params['language'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.language;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-knowledge-web/knowledge/category/tree', params: requestParameters }));
        });
    }
    /**
    * 查询vas价格配置
    * @summary 查询vas价格配置
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    queryVasPrice(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vasFeePrice/query' }));
        });
    }
    /**
    * 重新发送邀请邮件
    * @summary 重新发送邀请邮件
    * @param {SevcApiReInvitedUserRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    reInvitedUser(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user/email/reinvited', params: requestParameters }));
        });
    }
    /**
    * 添加成员
    * @summary 添加成员
    * @param {SevcApiRegisterRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    register(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/register', data: requestParameters }));
        });
    }
    /**
    * 可以根据Id / code 对app 进行删除
    * @summary 删除应用
    * @param {SevcApiRemoveAppRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    removeApp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-guard-web/app', data: requestParameters }));
        });
    }
    /**
    * 移除角色中的用户
    * @summary 移除用户
    * @param {SevcApiRemoveUserInRoleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    removeUserInRole(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/role/remove-users', data: requestParameters }));
        });
    }
    /**
    * 还原知识文档
    * @summary 还原知识文档
    * @param {SevcApiRevertKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    revertKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/revert', data: requestParameters }));
        });
    }
    /**
    * 角色列表
    * @summary 角色列表
    * @param {SevcApiRoleListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    roleList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/role/list', params: requestParameters }));
        });
    }
    /**
    * 角色信息
    * @summary 角色信息
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    roles(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/validate/roles' }));
        });
    }
    /**
    * 新增地址信息
    * @summary 新增地址信息
    * @param {SevcApiSaveAddressInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    saveAddressInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/save', data: requestParameters }));
        });
    }
    /**
    * 查询类目信息
    * @summary 查询类目信息
    * @param {SevcApiSearchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    search(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/category/search', data: requestParameters }));
        });
    }
    /**
    * 组织列表，通过关键词(成员和部门)进行搜索
    * @summary 关键词搜索用户/组织名
    * @param {SevcApiSearchOrgOrUserByKeyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    searchOrgOrUserByKey(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/list-org', data: requestParameters }));
        });
    }
    /**
    * 发送邮箱验证码
    * @summary 发送邮箱验证码
    * @param {SevcApiSendEmailCaptchaRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    sendEmailCaptcha(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/email/captcha', data: requestParameters }));
        });
    }
    /**
    * 发送短信验证码
    * @summary 发送短信验证码
    * @param {SevcApiSendPhoneCaptchaRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    sendPhoneCaptcha(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/phone/captcha', data: requestParameters }));
        });
    }
    /**
    * 退货结算单详情(退货明细分页)
    * @summary 退货结算单详情(退货明细分页)
    * @param {SevcApiSettlementDetailPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    settlementDetailPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/amazon/return/settlement/detail_page', data: requestParameters }));
        });
    }
    /**
    * 文件处理任务分页
    * @summary 文件处理任务分页
    * @param {SevcApiSevcFileJobPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    sevcFileJobPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/job/page', data: requestParameters }));
        });
    }
    /**
    * 订单发货
    * @summary 订单发货
    * @param {SevcApiShipRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    ship(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/ship', data: requestParameters }));
        });
    }
    /**
    * 基于文件发起签署
    * @summary 基于文件发起签署
    * @param {SevcApiSignByFileRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    signByFile(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/org/signFlow/signByFile', data: requestParameters }));
        });
    }
    /**
    * SPT查询当前公司的仓库
    * @summary SPT查询当前公司的仓库
    * @param {SevcApiSptListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    sptList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/spt/list', data: requestParameters }));
        });
    }
    /**
    * 订单统计
    * @summary 订单统计
    * @param {SevcApiStatisticsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    statistics(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-order-web/fulfill/retail/statistics', data: requestParameters }));
        });
    }
    /**
    * 仓储报价协议详情
    * @summary 仓储报价协议详情
    * @param {SevcApiStorageAgreementDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    storageAgreementDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementCode) !== undefined) {
                params['agreementCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/agreement/storage/detail', params: requestParameters }));
        });
    }
    /**
    * 获取仓库列表,可根据公司id查，也可模糊查询
    * @summary 获取仓库列表,可根据公司id查，也可模糊查询
    * @param {SevcApiStorageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    storageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/storage/list', data: requestParameters }));
        });
    }
    /**
    * 获取仓库列表,可根据公司id查，也可模糊查询
    * @summary 获取仓库列表,可根据公司id查，也可模糊查询，支持VC账号以及vendor查询
    * @param {SevcApiStorageListVRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    storageListV(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/storage/listV2', data: requestParameters }));
        });
    }
    /**
    * 组织列表，树形结构，递归
    * @summary 组织列表tree
    * @param {SevcApiTreeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    tree(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/org/tree', params: requestParameters }));
        });
    }
    /**
    * 银行信息修改
    * @summary 银行信息修改
    * @param {SevcApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/company/bank/update', data: requestParameters }));
        });
    }
    /**
    * 更新地址信息
    * @summary 更新地址信息
    * @param {SevcApiUpdateAddressInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    updateAddressInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/update', data: requestParameters }));
        });
    }
    /**
    * 修改地址状态，启用停用
    * @summary 修改地址状态，启用停用
    * @param {SevcApiUpdateAddressStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    updateAddressStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/address/updateAddressStatus', data: requestParameters }));
        });
    }
    /**
    * 修改应用名
    * @summary 修改app名称
    * @param {SevcApiUpdateAppRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    updateApp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/app', data: requestParameters }));
        });
    }
    /**
    * 编辑知识文档详情
    * @summary 编辑知识文档
    * @param {SevcApiUpdateKnowledgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    updateKnowledge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/update/detail', data: requestParameters }));
        });
    }
    /**
    * 修改分类
    * @summary 修改分类
    * @param {SevcApiUpdateKnowledgeCategoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    updateKnowledgeCategory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-knowledge-web/knowledge/category/update', data: requestParameters }));
        });
    }
    /**
    * 设置组织管理员
    * @summary 设置组织管理员
    * @param {SevcApiUpdateOrgManageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    updateOrgManage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/update-org-manage', data: requestParameters }));
        });
    }
    /**
    * 更新组织名称
    * @summary 更新组织名称
    * @param {SevcApiUpdateOrgNameRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    updateOrgName(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/org/update-org-name', data: requestParameters }));
        });
    }
    /**
    * 更新权限
    * @summary 更新权限
    * @param {SevcApiUpdatePermitRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    updatePermit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/update', data: requestParameters }));
        });
    }
    /**
    * 更新权限
    * @summary 更新权限
    * @param {SevcApiUpdatePermitsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    updatePermits(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/updates', data: requestParameters }));
        });
    }
    /**
    * 更新用户个人信息(登录情况下)
    * @summary 更新用户个人信息(登录情况下)
    * @param {SevcApiUpdateValidateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    updateValidate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/update/validate', data: requestParameters }));
        });
    }
    /**
    * 更新vas价格配置
    * @summary 更新vas价格配置
    * @param {SevcApiUpdateVasPriceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    updateVasPrice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/finance/warehouse/vasFeePrice/update', data: requestParameters }));
        });
    }
    /**
    * 更新公司信息
    * @summary 更新公司信息
    * @param {SevcApiUpdate0Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    update_30(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/company', data: requestParameters }));
        });
    }
    /**
    * 物流商更新
    * @summary 物流商更新
    * @param {SevcApiUpdate1Request} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    update_31(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/open/carrier/update', data: requestParameters }));
        });
    }
    /**
    * 上传前端菜单(权限组)与权限
    * @summary 上传前端菜单(权限组)与权限
    * @param {SevcApiUploadFrontUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    uploadFrontUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/permit/upload-front-url', data: requestParameters }));
        });
    }
    /**
    * 还原明细文件上传结果通知
    * @summary 还原明细文件上传结果通知
    * @param {SevcApiUploadNotifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    uploadNotify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-pay-web/payment/pingpong/uploadNotify', data: requestParameters }));
        });
    }
    /**
    * 校验用户是否存在
    * @summary 校验用户是否存在
    * @param {SevcApiUserExistedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    userExisted(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/existed', data: requestParameters }));
        });
    }
    /**
    * 用户注册链接Key 来获取用户的所有数据
    * @summary 用户注册链接Key 来获取用户的所有数据
    * @param {SevcApiUserRegisterMailCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    userRegisterMailCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sign) !== undefined) {
                params['sign'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sign;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user/register/check', params: requestParameters }));
        });
    }
    /**
    * 验证登录的状态
    * @summary 登录验证
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    validate(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/validate' }));
        });
    }
    /**
    * 校验验证码
    * @summary 校验验证码
    * @param {SevcApiValidateCaptchaRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    validateCaptcha(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/validate/captcha', data: requestParameters }));
        });
    }
    /**
    * 登录验证V2
    * @summary 登录验证V2
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof SevcApi
    */
    validateV(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/validate/v2' }));
        });
    }
    /**
    * VC报价协议详情
    * @summary VC报价协议详情
    * @param {SevcApiVcAgreementDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    vcAgreementDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementCode) !== undefined) {
                params['agreementCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/agreement/vc/detail', params: requestParameters }));
        });
    }
    /**
    * 回调事件
    * @summary 回调事件
    * @param {SevcApiWebhookRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SevcApi
    */
    webhook(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-pay-web/payment/pingpong/webhook', data: requestParameters }));
        });
    }
}
exports.SpotterSevcApiSdk = SpotterSevcApiSdk;
