// @spotter/gmesh-api-sdk/lib/api/shipment-amz-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterShipmentAmzApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ShipmentAmzApi - object-oriented interface
* @export
* @class ShipmentAmzApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterShipmentAmzApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * shipment 缺货备注
    * @summary shipment 缺货备注
    * @param {ShipmentAmzApiAddOutStockMarkRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    addOutStockMark(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/add/outStock/remark', data: requestParameters }));
        });
    }
    /**
    * 取消shipment
    * @summary 取消shipment
    * @param {ShipmentAmzApiCancelShipmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    cancelShipment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/cancel', data: requestParameters }));
        });
    }
    /**
    * 导出Excel
    * @summary 导出Excel
    * @param {ShipmentAmzApiExportDfShipmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof ShipmentAmzApi
    */
    exportDfShipment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/shipment/df/amz/export', data: requestParameters }));
        });
    }
    /**
    * shipment详情(VCPO和DF)
    * @summary dfShipment详情列表
    * @param {ShipmentAmzApiGetDetailsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof ShipmentAmzApi
    */
    getDetails(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/shipment/df/amz/detailV2', data: requestParameters }));
        });
    }
    /**
    * shipment详情
    * @summary shipment详情
    * @param {ShipmentAmzApiGetShipmentDetailWithItemsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    getShipmentDetailWithItems(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/detail', data: requestParameters }));
        });
    }
    /**
    * shipment 获取下载文件路径
    * @summary shipment 获取下载文件路径
    * @param {ShipmentAmzApiGetShipmentFileDownloadPathRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    getShipmentFileDownloadPath(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/download', data: requestParameters }));
        });
    }
    /**
    * shipment列表(VCPO和DF)
    * @summary shipment列表
    * @param {ShipmentAmzApiGetShipmentListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof ShipmentAmzApi
    */
    getShipmentList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/shipment/df/amz/list', data: requestParameters }));
        });
    }
    /**
    * shipment列表
    * @summary shipment列表
    * @param {ShipmentAmzApiGetShipmentListWithItemsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    getShipmentListWithItems(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/list', data: requestParameters }));
        });
    }
    /**
    * shipment发货清单数据
    * @summary shipment发货清单数据
    * @param {ShipmentAmzApiGetShipmentShippingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    getShipmentShippingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/shipping/list', data: requestParameters }));
        });
    }
    /**
    * shipment状态
    * @summary shipment状态
    * @param {ShipmentAmzApiGetShipmentStateInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    getShipmentStateInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/progress', data: requestParameters }));
        });
    }
    /**
    * 下载ShippingLabel
    * @summary 下载ShippingLabel
    * @param {ShipmentAmzApiGetShippingLabelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof ShipmentAmzApi
    */
    getShippingLabel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-shipment-web/shipment/df/amz/label/url', data: requestParameters }));
        });
    }
    /**
    * shipment 获取carrier信息
    * @summary shipment 获取carrier信息
    * @param {ShipmentAmzApiPullCarrierRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    pullCarrier(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/pull/carrierInfo', data: requestParameters }));
        });
    }
    /**
    * shipment 重发邮件
    * @summary shipment 重发邮件
    * @param {ShipmentAmzApiResendEmailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    resendEmail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/resend/email', data: requestParameters }));
        });
    }
    /**
    * shipment 更新FRD
    * @summary shipment 更新FRD
    * @param {ShipmentAmzApiUpdateShipmentFRDRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    updateShipmentFRD(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/update/frd', data: requestParameters }));
        });
    }
    /**
    * bol签单上传
    * @summary bol签单上传
    * @param {ShipmentAmzApiUploadBolSignBillRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShipmentAmzApi
    */
    uploadBolSignBill(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/shipment/amz/upload/bolsignbill', data: requestParameters }));
        });
    }
}
exports.SpotterShipmentAmzApiSdk = SpotterShipmentAmzApiSdk;
