// @spotter/gmesh-api-sdk/lib/api/shortage-invoice-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterShortageInvoiceApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* ShortageInvoiceApi - object-oriented interface
* @export
* @class ShortageInvoiceApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterShortageInvoiceApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 导出shortage结算
    * @summary 导出shortage结算
    * @param {ShortageInvoiceApiExportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    _export(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/payment/export', data: requestParameters }));
        });
    }
    /**
    * shortage异常发票
    * @summary shortage异常发票
    * @param {ShortageInvoiceApiAbnormalInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    abnormalInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/abnormal', data: requestParameters }));
        });
    }
    /**
    * shortage生成结算
    * @summary shortage生成结算
    * @param {ShortageInvoiceApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/payment/create', data: requestParameters }));
        });
    }
    /**
    * 查询指定结算单的shortage费用
    * @summary 查询指定结算单的shortage费用
    * @param {ShortageInvoiceApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode) !== undefined) {
                params['settleCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.settleCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/shortage/invoice/payment/detail', params: requestParameters }));
        });
    }
    /**
    * 导出shortage无VC报价发票
    * @summary 导出shortage无VC报价发票
    * @param {ShortageInvoiceApiExportAbnormalInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    exportAbnormalInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/export/abnormal', data: requestParameters }));
        });
    }
    /**
    * shortage发票列表导出
    * @summary shortage发票列表导出
    * @param {ShortageInvoiceApiExportInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    exportInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/item/export', data: requestParameters }));
        });
    }
    /**
    * 导入shortage无VC报价发票（文件名：shortage）
    * @summary 导入shortage无VC报价发票(文件名：shortage)
    * @param {ShortageInvoiceApiImportAbnormalInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    importAbnormalInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/import/abnormal', data: requestParameters }));
        });
    }
    /**
    * shortage结算列表
    * @summary shortage结算列表
    * @param {ShortageInvoiceApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/payment/page', data: requestParameters }));
        });
    }
    /**
    * shortage发票列表
    * @summary shortage发票列表
    * @param {ShortageInvoiceApiPageInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    pageInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/page', data: requestParameters }));
        });
    }
    /**
    * 查询指定公司最后一次结算的shortage费用
    * @summary 查询指定公司最后一次结算的shortage费用
    * @param {ShortageInvoiceApiQueryLastPaymentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    queryLastPayment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode) !== undefined) {
                params['companyCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-finance-web/shortage/invoice/payment/lastPayment', params: requestParameters }));
        });
    }
    /**
    * 撤销shortage结算
    * @summary 撤销shortage结算
    * @param {ShortageInvoiceApiReverseRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    reverse(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/payment/reverse', data: requestParameters }));
        });
    }
    /**
    * 丢缺金额总计
    * @summary 丢缺金额总计
    * @param {ShortageInvoiceApiShortageTotalAmountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    shortageTotalAmount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/shortageTotalAmount', data: requestParameters }));
        });
    }
    /**
    * shortage生成结算预览
    * @summary shortage生成结算预览
    * @param {ShortageInvoiceApiSummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof ShortageInvoiceApi
    */
    summary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-finance-web/shortage/invoice/payment/summary', data: requestParameters }));
        });
    }
}
exports.SpotterShortageInvoiceApiSdk = SpotterShortageInvoiceApiSdk;
