// @spotter/gmesh-api-sdk/lib/api/sign-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSignApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SignApi - object-oriented interface
* @export
* @class SignApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSignApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 下载已签署文件及附属材料
    * @summary 下载已签署文件及附属材料
    * @param {SignApiFileDownloadUrlRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SignApi
    */
    fileDownloadUrl(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/org/signFlow/fileDownloadUrl', data: requestParameters }));
        });
    }
    /**
    * 获取文件上传状态
    * @summary 获取文件上传状态
    * @param {SignApiFileStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SignApi
    */
    fileStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/files/fileStatus', data: requestParameters }));
        });
    }
    /**
    * 上传本地文件
    * @summary 上传本地文件
    * @param {SignApiFileUploadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SignApi
    */
    fileUpload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/files/fileUpload', data: requestParameters }));
        });
    }
    /**
    * 获取文件上传状态
    * @summary 获取文件上传状态
    * @param {SignApiKeywordPositionsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SignApi
    */
    keywordPositions(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/files/keywordPositions', data: requestParameters }));
        });
    }
    /**
    * 基于文件发起签署
    * @summary 基于文件发起签署
    * @param {SignApiSignByFileRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SignApi
    */
    signByFile(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-sign-web/org/signFlow/signByFile', data: requestParameters }));
        });
    }
}
exports.SpotterSignApiSdk = SpotterSignApiSdk;
