// @spotter/gmesh-api-sdk/lib/api/spotter-invoice-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSpotterInvoiceApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SpotterInvoiceApi - object-oriented interface
* @export
* @class SpotterInvoiceApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSpotterInvoiceApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 手动关联时可选amazon发票
    * @summary 手动关联时可选amazon发票
    * @param {SpotterInvoiceApiAvailableAmazonInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    availableAmazonInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.searchAmazonInvoiceNo) !== undefined) {
                params['searchAmazonInvoiceNo'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.searchAmazonInvoiceNo;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/invoice/availableAmazonInvoice', params: requestParameters }));
        });
    }
    /**
    * 批量下载发票
    * @summary 批量下载发票
    * @param {SpotterInvoiceApiBatchDownloadInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    batchDownloadInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/invoice/batchDownloadInvoice', data: requestParameters }));
        });
    }
    /**
    * 批量导出发票
    * @summary 批量导出发票
    * @param {SpotterInvoiceApiBatchExportInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    batchExportInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/invoice/batchExportInvoice', data: requestParameters }));
        });
    }
    /**
    * onChange查询发票导出进度
    * @summary onChange查询发票导出进度
    * @param {SpotterInvoiceApiExportInvoiceOnChangeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    exportInvoiceOnChange(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/invoice/exportInvoiceOnChange', data: requestParameters }));
        });
    }
    /**
    * 查询发票导出进度
    * @summary 查询发票导出进度
    * @param {SpotterInvoiceApiExportInvoiceProcessRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    exportInvoiceProcess(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/invoice/exportInvoiceProcess', data: requestParameters }));
        });
    }
    /**
    * 发票详情
    * @summary 发票详情
    * @param {SpotterInvoiceApiInvoiceDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    invoiceDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.spotterInvoiceNo) !== undefined) {
                params['spotterInvoiceNo'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.spotterInvoiceNo;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/invoice/detail', params: requestParameters }));
        });
    }
    /**
    * 发票列表
    * @summary 发票列表
    * @param {SpotterInvoiceApiInvoiceListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    invoiceList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/invoice/list', data: requestParameters }));
        });
    }
    /**
    * 展示spotter发票和对应amazon发票（包含贷款单号）
    * @summary 展示spotter发票和对应amazon发票（包含贷款单号）
    * @param {SpotterInvoiceApiInvoiceRelationListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    invoiceRelationList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.spotterInvoiceNumber) !== undefined) {
                params['spotterInvoiceNumber'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.spotterInvoiceNumber;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-invoice-web/invoice/invoiceRelationList', params: requestParameters }));
        });
    }
    /**
    * 手动关联amazon发票
    * @summary 手动关联amazon发票
    * @param {SpotterInvoiceApiManualAssociateInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    manualAssociateInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/invoice/manualAssociateInvoice', data: requestParameters }));
        });
    }
    /**
    * 重试-重新导出发票
    * @summary 重试-重新批量导出发票
    * @param {SpotterInvoiceApiReBatchExportInvoiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SpotterInvoiceApi
    */
    reBatchExportInvoice(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-invoice-web/invoice/reBatchExportInvoice', data: requestParameters }));
        });
    }
}
exports.SpotterSpotterInvoiceApiSdk = SpotterSpotterInvoiceApiSdk;
