// @spotter/gmesh-api-sdk/lib/api/storage-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterStorageApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* StorageApi - object-oriented interface
* @export
* @class StorageApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterStorageApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 新增文件处理任务
    * @summary 新增文件处理任务
    * @param {StorageApiAddRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    add(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/job/add', data: requestParameters }));
        });
    }
    /**
    * 获取所有的仓库信息
    * @summary 获取所有的仓库信息
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    all(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/gmesh-web/storage/all' }));
        });
    }
    /**
    * 仓库所在地域下拉列表
    * @summary 仓库所在地域下拉列表
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    getAmazonCountryAndStateVO(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/storage/state/all' }));
        });
    }
    /**
    * 仓库详情
    * @summary 仓库详情
    * @param {StorageApiGetStorageDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    getStorageDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/storage/detail', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 条件查询仓库信息列表
    * @param {StorageApiGetStorageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    getStorageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/storage/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 分页条件查询仓库信息
    * @param {StorageApiGetStoragePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    getStoragePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/storage/page', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 查询仓库关联VC账号及VendorCode信息
    * @param {StorageApiGetStorageVcAccountListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    getStorageVcAccountList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/storage/vc/list', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 新增仓库
    * @param {StorageApiSaveStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    saveStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/storage/save', data: requestParameters }));
        });
    }
    /**
    * 文件处理任务分页
    * @summary 文件处理任务分页
    * @param {StorageApiSevcFileJobPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    sevcFileJobPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-storage-web/file/job/page', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 仓库绑定VendorCode
    * @param {StorageApiStorageBindVendorCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    storageBindVendorCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/gmesh-web/storage/vc/save', data: requestParameters }));
        });
    }
    /**
    *
    * @summary 根据仓库编码修改仓库配置
    * @param {StorageApiUpdateStorageByCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageApi
    */
    updateStorageByCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PATCH', url: '/gmesh-web/storage/config', data: requestParameters }));
        });
    }
}
exports.SpotterStorageApiSdk = SpotterStorageApiSdk;
