// @spotter/gmesh-api-sdk/lib/api/storage-agreement-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterStorageAgreementApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* StorageAgreementApi - object-oriented interface
* @export
* @class StorageAgreementApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterStorageAgreementApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 批量通过仓储报价补充协议
    * @summary 批量通过仓储报价补充协议
    * @param {StorageAgreementApiAuditBatchAcceptRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    auditBatchAccept(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/batch/accept', data: requestParameters }));
        });
    }
    /**
    * 批量拒绝仓储报价补充协议
    * @summary 批量拒绝仓储报价补充协议
    * @param {StorageAgreementApiAuditBatchRejectedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    auditBatchRejected(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/batch/rejected', data: requestParameters }));
        });
    }
    /**
    * 通过仓储报价补充协议
    * @summary 通过仓储报价补充协议
    * @param {StorageAgreementApiAuditPassRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    auditPass(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/accept', data: requestParameters }));
        });
    }
    /**
    * 不通过仓储报价补充协议
    * @summary 不通过仓储报价补充协议
    * @param {StorageAgreementApiAuditRejectedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    auditRejected(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/rejected', data: requestParameters }));
        });
    }
    /**
    * 批量修改仓储报价补充协议
    * @summary 批量修改仓储报价补充协议
    * @param {StorageAgreementApiBatchDiscountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    batchDiscount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/batch/discount', data: requestParameters }));
        });
    }
    /**
    * 仓储报价补充协议已完结列表
    * @summary 仓储报价补充协议已完结列表
    * @param {StorageAgreementApiFinishedListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    finishedList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/list/finished', data: requestParameters }));
        });
    }
    /**
    * 获取当前登录用户待处理仓储报价协议条数
    * @summary 仓储报价协议审核-待处理报价协议条数
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    gToBeResolveCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/toBeResolve/quotation/count' }));
        });
    }
    /**
    * 仓储报价协议详情
    * @summary 仓储报价协议详情
    * @param {StorageAgreementApiStorageAgreementDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    storageAgreementDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementCode) !== undefined) {
                params['agreementCode'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.agreementCode;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/agreement/storage/detail', params: requestParameters }));
        });
    }
    /**
    * 仓储报价补充协议待审核列表
    * @summary 仓储报价补充协议待审核列表
    * @param {StorageAgreementApiToBeAuditListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    toBeAuditList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/list/toBeAudit', data: requestParameters }));
        });
    }
    /**
    * 仓储报价补充协议待客户确认列表
    * @summary 仓储报价补充协议待客户确认列表
    * @param {StorageAgreementApiToBeCustomerConfirmListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof StorageAgreementApi
    */
    toBeCustomerConfirmList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/agreement/storage/gmesh/list/toBeConfirm', data: requestParameters }));
        });
    }
}
exports.SpotterStorageAgreementApiSdk = SpotterStorageAgreementApiSdk;
