// @spotter/gmesh-api-sdk/lib/api/supplier-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterSupplierApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* SupplierApi - object-oriented interface
* @export
* @class SupplierApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterSupplierApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 账号经理列表接口
    * @summary 账号经理列表接口
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    accountManagerList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/user/accountManager/list' }));
        });
    }
    /**
    * 审核信息详情
    * @summary 审核信息详情
    * @param {SupplierApiAuditDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    auditDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/audit/detail', params: requestParameters }));
        });
    }
    /**
    * 供应商历史审核分页
    * @summary 供应商历史审核分页
    * @param {SupplierApiAuditHistoryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    auditHistoryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/audit/history/page', data: requestParameters }));
        });
    }
    /**
    * 供应商信息审核分页
    * @summary 供应商信息审核分页
    * @param {SupplierApiAuditListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    auditList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/audit/page', data: requestParameters }));
        });
    }
    /**
    * 供应商银行账户信息列表
    * @summary 供应商银行账户信息列表
    * @param {SupplierApiBankListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    bankList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.supplierId) !== undefined) {
                params['supplierId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.supplierId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/bank/list', params: requestParameters }));
        });
    }
    /**
    * 当前供应商关联供应商列表
    * @summary 当前供应商关联供应商列表
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    bindList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/bind/list' }));
        });
    }
    /**
    * 停用供应商
    * @summary 停用供应商
    * @param {SupplierApiBlockRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    block(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/block', data: requestParameters }));
        });
    }
    /**
    * 供应商品牌详情
    * @summary 供应商品牌详情
    * @param {SupplierApiBrandDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    brandDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId) !== undefined) {
                params['brandId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.brandId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/brand/detail', params: requestParameters }));
        });
    }
    /**
    * 供应商品牌列表
    * @summary 供应商品牌列表
    * @param {SupplierApiBrandListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    brandList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/brand/list', data: requestParameters }));
        });
    }
    /**
    * 供应商合同分页
    * @summary 供应商合同分页
    * @param {SupplierApiContractPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    contractPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/contract/page', data: requestParameters }));
        });
    }
    /**
    * 新建供应商
    * @summary 新建供应商
    * @param {SupplierApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/create', data: requestParameters }));
        });
    }
    /**
    * 生命周期变更
    * @summary 生命周期变更
    * @param {SupplierApiCycleUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    cycleUpdate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/cycle/update', data: requestParameters }));
        });
    }
    /**
    * 供应商信息详情
    * @summary 供应商信息详情
    * @param {SupplierApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/detail', params: requestParameters }));
        });
    }
    /**
    * 更新供应商信息
    * @summary 更新供应商信息
    * @param {SupplierApiEditRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    edit(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/edit', data: requestParameters }));
        });
    }
    /**
    * 异常状态修复
    * @summary 异常状态修复
    * @param {SupplierApiExceptionFixRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    exceptionFix(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/exception/fix', data: requestParameters }));
        });
    }
    /**
    * 供应商信息分页
    * @summary 供应商信息分页
    * @param {SupplierApiGetListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    getList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/page', data: requestParameters }));
        });
    }
    /**
    * 供应商关联仓库分页
    * @summary 供应商关联仓库分页
    * @param {SupplierApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * 供应商信息审核通过
    * @summary 供应商信息审核通过
    * @param {SupplierApiPassRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    pass(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/audit/pass', data: requestParameters }));
        });
    }
    /**
    * 供应商信息审核拒绝
    * @summary 供应商信息审核拒绝
    * @param {SupplierApiRejectRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    reject(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/audit/reject', data: requestParameters }));
        });
    }
    /**
    * 供应商简单信息
    * @summary 供应商简单信息
    * @param {SupplierApiSimpleDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    simpleDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/detail/simple', params: requestParameters }));
        });
    }
    /**
    * 统计当前用户待处理数量
    * @summary 统计当前用户待处理数量
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    statics(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/tobeResolve/statics' }));
        });
    }
    /**
    * 仓储费等级变更
    * @summary 仓储费等级变更
    * @param {SupplierApiStorageLevelUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    storageLevelUpdate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/storage/level/update', data: requestParameters }));
        });
    }
    /**
    * 供应商账号分页接口
    * @summary 供应商账号分页接口
    * @param {SupplierApiUserPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof SupplierApi
    */
    userPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/user/page', data: requestParameters }));
        });
    }
}
exports.SpotterSupplierApiSdk = SpotterSupplierApiSdk;
