// @spotter/gmesh-api-sdk/lib/api/task-inbox-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterTaskInboxEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* TaskInboxEndpointApi - object-oriented interface
* @export
* @class TaskInboxEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterTaskInboxEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 删除站内信
    * @summary 删除站内信
    * @param {TaskInboxEndpointApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/delete', data: requestParameters }));
        });
    }
    /**
    * 取消定时发送
    * @summary 取消定时发送
    * @param {TaskInboxEndpointApiCancelSendRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    cancelSend(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/cancelSend', data: requestParameters }));
        });
    }
    /**
    * 普通发送站内信
    * @summary 普通发送站内信
    * @param {TaskInboxEndpointApiCommonSendRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    commonSend(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/commonSend', data: requestParameters }));
        });
    }
    /**
    * 查询站内信详情
    * @summary 查询站内信详情
    * @param {TaskInboxEndpointApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/detail', data: requestParameters }));
        });
    }
    /**
    * 获取发送历史列表
    * @summary 获取发送历史列表
    * @param {TaskInboxEndpointApiGetHistoriesRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    getHistories(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/getHistories', data: requestParameters }));
        });
    }
    /**
    * 获取站内信各个数量
    * @summary 获取站内信各个数量
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    getTaskCount(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/getTaskCount' }));
        });
    }
    /**
    * 获取站内信发送情况
    * @summary 获取站内信发送情况
    * @param {TaskInboxEndpointApiGetTaskSendResultRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    getTaskSendResult(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/getTaskSendResult', data: requestParameters }));
        });
    }
    /**
    * 分页查询站内信
    * @summary 分页查询站内信
    * @param {TaskInboxEndpointApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/queryPage', data: requestParameters }));
        });
    }
    /**
    * 保存为草稿
    * @summary 保存为草稿
    * @param {TaskInboxEndpointApiSaveTempRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    saveTemp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/saveTemp', data: requestParameters }));
        });
    }
    /**
    * 定时发送站内信
    * @summary 定时发送站内信
    * @param {TaskInboxEndpointApiTimingSendRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    timingSend(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/timingSend', data: requestParameters }));
        });
    }
    /**
    * 保存
    * @summary 保存
    * @param {TaskInboxEndpointApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/update', data: requestParameters }));
        });
    }
    /**
    * 撤回消息
    * @summary 撤回消息
    * @param {TaskInboxEndpointApiWithdrawRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskInboxEndpointApi
    */
    withdraw(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/messageInboxTask/withdraw', data: requestParameters }));
        });
    }
}
exports.SpotterTaskInboxEndpointApiSdk = SpotterTaskInboxEndpointApiSdk;
