// @spotter/gmesh-api-sdk/lib/api/task-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterTaskApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* TaskApi - object-oriented interface
* @export
* @class TaskApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterTaskApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 创建任务
    * @summary 创建任务
    * @param {TaskApiCreateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskApi
    */
    create(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/task/create', data: requestParameters }));
        });
    }
    /**
    * 任务管理详情
    * @summary 任务管理详情
    * @param {TaskApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-od-mail-web/od/task/detail', params: requestParameters }));
        });
    }
    /**
    * 完成任务
    * @summary 完成任务
    * @param {TaskApiFinishRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskApi
    */
    finish(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/task/finish', data: requestParameters }));
        });
    }
    /**
    * 任务管理分页
    * @summary 任务管理分页
    * @param {TaskApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/task/page', data: requestParameters }));
        });
    }
    /**
    * 任务asin解析
    * @summary 任务asin解析
    * @param {TaskApiParseAsinRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskApi
    */
    parseAsin(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/task/parse/asin', data: requestParameters }));
        });
    }
    /**
    * 任务撤回
    * @summary 任务撤回
    * @param {TaskApiRevokeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskApi
    */
    revoke(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/task/revoke', data: requestParameters }));
        });
    }
    /**
    * 当前登录用户待办任务统计
    * @summary 当前登录用户待办任务统计
    * @param {*} [options] Override http request option.
    * @memberof TaskApi
    */
    todoStatistics(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/task/todo/statistics' }));
        });
    }
    /**
    * 任务催办
    * @summary 任务催办
    * @param {TaskApiUrgeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TaskApi
    */
    urge(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-od-mail-web/od/task/urge', data: requestParameters }));
        });
    }
}
exports.SpotterTaskApiSdk = SpotterTaskApiSdk;
