// @spotter/gmesh-api-sdk/lib/api/template-inbox-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterTemplateInboxEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* TemplateInboxEndpointApi - object-oriented interface
* @export
* @class TemplateInboxEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterTemplateInboxEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 删除站内信模板
    * @summary 删除站内信模板
    * @param {TemplateInboxEndpointApiDeleteRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TemplateInboxEndpointApi
    */
    _delete(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/templateInbox/delete', data: requestParameters }));
        });
    }
    /**
    * 更改站内信模板状态
    * @summary 更改站内信模板状态
    * @param {TemplateInboxEndpointApiChangeStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TemplateInboxEndpointApi
    */
    changeStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/templateInbox/changeStatus', data: requestParameters }));
        });
    }
    /**
    * 保存草稿箱站内信模板
    * @summary 保存草稿箱站内信模板
    * @param {TemplateInboxEndpointApiCreateTempRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TemplateInboxEndpointApi
    */
    createTemp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/templateInbox/createTemp', data: requestParameters }));
        });
    }
    /**
    * 模板详情
    * @summary 模板详情
    * @param {TemplateInboxEndpointApiDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TemplateInboxEndpointApi
    */
    detail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/templateInbox/detail', data: requestParameters }));
        });
    }
    /**
    * 发布站内信模板
    * @summary 发布站内信模板
    * @param {TemplateInboxEndpointApiPublishRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TemplateInboxEndpointApi
    */
    publish(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/templateInbox/publish', data: requestParameters }));
        });
    }
    /**
    * 分页查询模板
    * @summary 分页查询模板
    * @param {TemplateInboxEndpointApiQueryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TemplateInboxEndpointApi
    */
    queryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/templateInbox/queryPage', data: requestParameters }));
        });
    }
    /**
    * 更新站内信模板
    * @summary 更新站内信模板
    * @param {TemplateInboxEndpointApiUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TemplateInboxEndpointApi
    */
    update(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/templateInbox/update', data: requestParameters }));
        });
    }
    /**
    * 修改站内信模板业务类型
    * @summary 修改站内信模板业务类型
    * @param {TemplateInboxEndpointApiUpdateBusinessTypeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TemplateInboxEndpointApi
    */
    updateBusinessType(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-message-web/gmesh/templateInbox/updateBusinessType', data: requestParameters }));
        });
    }
}
exports.SpotterTemplateInboxEndpointApiSdk = SpotterTemplateInboxEndpointApiSdk;
