// @spotter/gmesh-api-sdk/lib/api/ticket-assignee-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterTicketAssigneeApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* TicketAssigneeApi - object-oriented interface
* @export
* @class TicketAssigneeApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterTicketAssigneeApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 添加或更新工单分流规则配置
    * @summary 添加或更新工单分流规则配置
    * @param {TicketAssigneeApiAddOrUpdateTicketAssigneeSkillRuleRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TicketAssigneeApi
    */
    addOrUpdateTicketAssigneeSkillRule(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/ticket/assignee/skill', data: requestParameters }));
        });
    }
    /**
    * 删除工单分流技能配置
    * @summary 删除工单分流技能配置
    * @param {TicketAssigneeApiDeleteAssigneeSkillRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TicketAssigneeApi
    */
    deleteAssigneeSkill(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'DELETE', url: '/spotter-ticket-web/ticket/assignee/skill/delete', params: requestParameters }));
        });
    }
    /**
    * 分配规则支持的运算符号
    * @summary 分配规则支持的运算符号
    * @param {*} [options] Override http request option.
    * @memberof TicketAssigneeApi
    */
    getOperatorList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/ticket/assignee/rule/operators' }));
        });
    }
    /**
    * 查询兜底规则配置
    * @summary 查询兜底规则配置
    * @param {*} [options] Override http request option.
    * @memberof TicketAssigneeApi
    */
    getTicketAssigneeSecurity(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/ticket/assignee/security/detail' }));
        });
    }
    /**
    * 配置兜底规则
    * @summary 配置兜底规则
    * @param {TicketAssigneeApiSaveTicketAssigneeSecurityRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TicketAssigneeApi
    */
    saveTicketAssigneeSecurity(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/ticket/assignee/security', data: requestParameters }));
        });
    }
    /**
    * 查询工单分流技能配置详情
    * @summary 查询工单分流技能配置详情
    * @param {TicketAssigneeApiTicketAssigneeSkillDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TicketAssigneeApi
    */
    ticketAssigneeSkillDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-ticket-web/ticket/assignee/skill/detail', params: requestParameters }));
        });
    }
    /**
    * 分页条件查询工单分流技能配置
    * @summary 分页条件查询工单分流技能配置
    * @param {TicketAssigneeApiTicketAssigneeSkillPageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TicketAssigneeApi
    */
    ticketAssigneeSkillPageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-ticket-web/ticket/assignee/skill/page', data: requestParameters }));
        });
    }
}
exports.SpotterTicketAssigneeApiSdk = SpotterTicketAssigneeApiSdk;
