// @spotter/gmesh-api-sdk/lib/api/tracking-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterTrackingApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* TrackingApi - object-oriented interface
* @export
* @class TrackingApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterTrackingApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 添加备注
    * @summary 添加备注
    * @param {TrackingApiAddRemarkRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrackingApi
    */
    addRemark(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/gmesh/tracking/remark', data: requestParameters }));
        });
    }
    /**
    * 异常运单关闭操作
    * @summary 异常运单关闭操作
    * @param {TrackingApiClosedByTrackingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrackingApi
    */
    closedByTrackingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/gmesh/tracking/closed', data: requestParameters }));
        });
    }
    /**
    * 轨迹追踪列表导出
    * @summary 轨迹追踪列表导出
    * @param {TrackingApiExportTrackingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrackingApi
    */
    exportTrackingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/gmesh/tracking/export', data: requestParameters }));
        });
    }
    /**
    * 运单号备注批量导入
    * @summary 运单号备注批量导入
    * @param {TrackingApiImportRemarkRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrackingApi
    */
    importRemark(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/gmesh/tracking/import/remark', data: requestParameters }));
        });
    }
    /**
    * 异常运单手动到货操作
    * @summary 异常运单手动到货操作
    * @param {TrackingApiManualArrivedByTrackingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrackingApi
    */
    manualArrivedByTrackingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/gmesh/tracking/manual/arrived', data: requestParameters }));
        });
    }
    /**
    * 物流明细
    * @summary 物流明细
    * @param {TrackingApiRoutingDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrackingApi
    */
    routingDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/gmesh/tracking/routing_detail', data: requestParameters }));
        });
    }
    /**
    * 异常运单分页
    * @summary 异常运单分页
    * @param {TrackingApiTrackingAbnormalListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrackingApi
    */
    trackingAbnormalList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/gmesh/tracking/page/abnormal', data: requestParameters }));
        });
    }
    /**
    * 轨迹追踪列表分页
    * @summary 轨迹追踪列表分页
    * @param {TrackingApiTrackingListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrackingApi
    */
    trackingList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-tms-web/gmesh/tracking/page', data: requestParameters }));
        });
    }
}
exports.SpotterTrackingApiSdk = SpotterTrackingApiSdk;
