// @spotter/gmesh-api-sdk/lib/api/trends-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterTrendsApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* TrendsApi - object-oriented interface
* @export
* @class TrendsApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterTrendsApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 发布评论
    * @summary 发布评论
    * @param {TrendsApiCreateCommentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrendsApi
    */
    createComment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/trends/comment/create', data: requestParameters }));
        });
    }
    /**
    * 新建运营日志
    * @summary 新建运营日志
    * @param {TrendsApiCreateReportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrendsApi
    */
    createReport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/trends/report/create', data: requestParameters }));
        });
    }
    /**
    * 删除评论
    * @summary 删除评论
    * @param {TrendsApiDelCommentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrendsApi
    */
    delComment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/trends/comment/del', params: requestParameters }));
        });
    }
    /**
    * 删除运营日志
    * @summary 删除运营日志
    * @param {TrendsApiDelReportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrendsApi
    */
    delReport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/trends/report/del', params: requestParameters }));
        });
    }
    /**
    * 运营日志详情
    * @summary 运营日志详情
    * @param {TrendsApiReportDetailsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrendsApi
    */
    reportDetails(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-supplier-web/supplier/trends/report/page', params: requestParameters }));
        });
    }
    /**
    * 供应商动态分页
    * @summary 供应商动态分页
    * @param {TrendsApiReportPagingRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrendsApi
    */
    reportPaging(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/trends/page', data: requestParameters }));
        });
    }
    /**
    * 更新运营日志
    * @summary 更新运营日志
    * @param {TrendsApiUpdateReportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof TrendsApi
    */
    updateReport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-supplier-web/supplier/trends/report/update', data: requestParameters }));
        });
    }
}
exports.SpotterTrendsApiSdk = SpotterTrendsApiSdk;
