// @spotter/gmesh-api-sdk/lib/api/upload-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterUploadApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* UploadApi - object-oriented interface
* @export
* @class UploadApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterUploadApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 上传模板
    * @summary 上传模板,文件上传的名称字段是file
    * @param {UploadApiBatchUploadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UploadApi
    */
    batchUpload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/sales/upload', data: requestParameters }));
        });
    }
    /**
    * 确认上传
    * @summary 确认上传
    * @param {UploadApiBatchUploadSureRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UploadApi
    */
    batchUploadSure(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-report-web/report/sales/upload/sure', data: requestParameters }));
        });
    }
    /**
    * 下载模板
    * @summary 下载模板
    * @param {UploadApiDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UploadApi
    */
    download(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year) !== undefined) {
                params['year'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.year;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-report-web/report/sales/download', params: requestParameters }));
        });
    }
}
exports.SpotterUploadApiSdk = SpotterUploadApiSdk;
