// @spotter/gmesh-api-sdk/lib/api/user-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterUserApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* UserApi - object-oriented interface
* @export
* @class UserApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterUserApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 修改密码
    * @summary 修改密码
    * @param {UserApiChangePasswordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    changePassword(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/password', data: requestParameters }));
        });
    }
    /**
    * 用户中心修改密码
    * @summary 用户中心修改密码
    * @param {UserApiChangePasswordForUserCenterRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    changePasswordForUserCenter(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/update-password', data: requestParameters }));
        });
    }
    /**
    * 用户中心修改密码-检查密码是否正确
    * @summary 用户中心修改密码-检查密码是否正确
    * @param {UserApiCheckPasswordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    checkPassword(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/check-password', data: requestParameters }));
        });
    }
    /**
    * 更新用户状态
    * @summary 更新用户状态
    * @param {UserApiEnableOrDisableUserRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    enableOrDisableUser(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.status) !== undefined) {
                params['status'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.status;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.isActive) !== undefined) {
                params['isActive'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.isActive;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/status', params: requestParameters }));
        });
    }
    /**
    * 根据公司id和部门code获取部门leader
    * @summary 获取部门leader
    * @param {UserApiGetLeaderByCompanyIdAndOrgIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    getLeaderByCompanyIdAndOrgId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId) !== undefined) {
                params['companyId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.companyId;
            }
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.orgId) !== undefined) {
                params['orgId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.orgId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user/find/org/leader', params: requestParameters }));
        });
    }
    /**
    * 根据userId获取用户所有下级用户信息
    * @summary 根据userId获取用户所有下级用户信息
    * @param {UserApiGetUnderlingByUserIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    getUnderlingByUserId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user/underling', params: requestParameters }));
        });
    }
    /**
    * 根据用户Id 获取某个用户信息
    * @summary 获取用户
    * @param {UserApiGetUserByIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    getUserById(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user', params: requestParameters }));
        });
    }
    /**
    * 获取用户列表, 根据多个查询条件进行筛选
    * @summary 用户列表
    * @param {UserApiGetUserListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    getUserList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/list', data: requestParameters }));
        });
    }
    /**
    * 查询简易用户分页数据(查询所有乙方公司的用户)
    * @summary 查询简易用户分页数据(查询所有乙方公司的用户)
    * @param {UserApiGetUserWithIdNameListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    getUserWithIdNameList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/simple/page', data: requestParameters }));
        });
    }
    /**
    * 用户下拉列表
    * @summary 用户下拉列表
    * @param {UserApiGetUsersByNameRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    getUsersByName(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/users', data: requestParameters }));
        });
    }
    /**
    * 激活设置密码
    * @summary 激活设置密码
    * @param {UserApiInitPasswordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    initPassword(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/active/password', data: requestParameters }));
        });
    }
    /**
    * 带分页搜索用户列表，通过组织Id搜索
    * @summary 通过组织Id查询用户列表
    * @param {UserApiListByOrgIdRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    listByOrgId(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/list/org', data: requestParameters }));
        });
    }
    /**
    * 重新发送邀请邮件
    * @summary 重新发送邀请邮件
    * @param {UserApiReInvitedUserRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    reInvitedUser(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId) !== undefined) {
                params['userId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.userId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user/email/reinvited', params: requestParameters }));
        });
    }
    /**
    * 添加成员
    * @summary 添加成员
    * @param {UserApiRegisterRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    register(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/register', data: requestParameters }));
        });
    }
    /**
    * 重置密码
    * @summary 重置密码
    * @param {UserApiResetPasswordRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    resetPassword(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PUT', url: '/spotter-guard-web/user/reset-password', data: requestParameters }));
        });
    }
    /**
    * 发送邮箱验证码
    * @summary 发送邮箱验证码
    * @param {UserApiSendEmailCaptchaRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    sendEmailCaptcha(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/email/captcha', data: requestParameters }));
        });
    }
    /**
    * 发送短信验证码
    * @summary 发送短信验证码
    * @param {UserApiSendPhoneCaptchaRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    sendPhoneCaptcha(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/phone/captcha', data: requestParameters }));
        });
    }
    /**
    * 更新用户个人信息(登录情况下)
    * @summary 更新用户个人信息(登录情况下)
    * @param {UserApiUpdateValidateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    updateValidate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/update/validate', data: requestParameters }));
        });
    }
    /**
    * 校验用户是否存在
    * @summary 校验用户是否存在
    * @param {UserApiUserExistedRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    userExisted(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/existed', data: requestParameters }));
        });
    }
    /**
    * 用户注册链接Key 来获取用户的所有数据
    * @summary 用户注册链接Key 来获取用户的所有数据
    * @param {UserApiUserRegisterMailCheckRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    userRegisterMailCheck(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sign) !== undefined) {
                params['sign'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.sign;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-guard-web/user/register/check', params: requestParameters }));
        });
    }
    /**
    * 校验验证码
    * @summary 校验验证码
    * @param {UserApiValidateCaptchaRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof UserApi
    */
    validateCaptcha(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-guard-web/user/validate/captcha', data: requestParameters }));
        });
    }
}
exports.SpotterUserApiSdk = SpotterUserApiSdk;
