// @spotter/gmesh-api-sdk/lib/api/vendor-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterVendorApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* VendorApi - object-oriented interface
* @export
* @class VendorApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterVendorApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 根据账户id查找vendor详情
    * @summary 根据账户id查找vendor详情
    * @param {VendorApiGetMainVendorRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorApi
    */
    getMainVendor(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountId) !== undefined) {
                params['accountId'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.accountId;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-cooperation-web/amazon/account/vendor/getMainVendor', params: requestParameters }));
        });
    }
    /**
    * 查询所有vendor列表
    * @summary 查询所有vendor列表
    * @param {VendorApiListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorApi
    */
    list(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/list', data: requestParameters }));
        });
    }
    /**
    * 根据id更新vendor信息
    * @summary 根据id更新vendor信息
    * @param {VendorApiModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorApi
    */
    modify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/modify', data: requestParameters }));
        });
    }
    /**
    * 根据条件查询vendor（分页）
    * @summary 根据条件查询vendor（分页）
    * @param {VendorApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/page', data: requestParameters }));
        });
    }
    /**
    * 新增一个vendor code
    * @summary 新增一个vendor code
    * @param {VendorApiSaveRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorApi
    */
    save(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/save', data: requestParameters }));
        });
    }
    /**
    * 新增一个DF类型vendor code
    * @summary 新增一个DF类型vendor code
    * @param {VendorApiSaveDirectFulfillmentRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorApi
    */
    saveDirectFulfillment(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/save/df', data: requestParameters }));
        });
    }
    /**
    * 新增一个di类型vendor code
    * @summary 新增一个DI类型vendor code
    * @param {VendorApiSaveDirectImportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorApi
    */
    saveDirectImport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/save/di', data: requestParameters }));
        });
    }
    /**
    * 根据条件查询vendor及其所属类目信息（分页）
    * @summary 根据条件查询vendor及其所属类目信息（分页）
    * @param {VendorApiVendorCategoryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorApi
    */
    vendorCategoryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/page/category', data: requestParameters }));
        });
    }
}
exports.SpotterVendorApiSdk = SpotterVendorApiSdk;
