// @spotter/gmesh-api-sdk/lib/api/vendor-category-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterVendorCategoryApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* VendorCategoryApi - object-oriented interface
* @export
* @class VendorCategoryApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterVendorCategoryApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * vendor类目配置激活
    * @summary vendor类目配置激活
    * @param {VendorCategoryApiActiveRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorCategoryApi
    */
    active(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/category/active', data: requestParameters }));
        });
    }
    /**
    * vendor类目配置冻结
    * @summary vendor类目配置冻结
    * @param {VendorCategoryApiFrozenRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorCategoryApi
    */
    frozen(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/category/frozen', data: requestParameters }));
        });
    }
    /**
    * 根据条件查询类目配置
    * @summary 根据条件查询类目配置
    * @param {VendorCategoryApiListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorCategoryApi
    */
    list(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/category/list', data: requestParameters }));
        });
    }
    /**
    * 根据条件查询类目配置&vendor信息&account信息
    * @summary 根据条件查询类目配置&vendor信息&account信息
    * @param {VendorCategoryApiListDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorCategoryApi
    */
    listDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/category/list/detail', data: requestParameters }));
        });
    }
    /**
    * 更新vendor类目配置
    * @summary 更新vendor类目配置
    * @param {VendorCategoryApiModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorCategoryApi
    */
    modify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/category/modify', data: requestParameters }));
        });
    }
    /**
    * 新增vendor类目配置
    * @summary 新增vendor类目配置
    * @param {VendorCategoryApiSaveRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof VendorCategoryApi
    */
    save(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-cooperation-web/amazon/account/vendor/category/save', data: requestParameters }));
        });
    }
}
exports.SpotterVendorCategoryApiSdk = SpotterVendorCategoryApiSdk;
