// @spotter/gmesh-api-sdk/lib/api/warehouse-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterWarehouseApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* WarehouseApi - object-oriented interface
* @export
* @class WarehouseApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterWarehouseApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 【增值服务】受理/不受理服务单
    * @summary 【增值服务】受理/不受理服务单
    * @param {WarehouseApiAcceptRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    accept(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/accept', data: requestParameters }));
        });
    }
    /**
    * 获取所有的仓库信息
    * @summary 获取所有的仓库信息
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    all(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-warehouse-web/storage/gmesh/all' }));
        });
    }
    /**
    * gmesh查询仓库审批详情
    * @summary gmesh查询仓库审批详情
    * @param {WarehouseApiApproveDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    approveDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/warehouse/approve/detail', data: requestParameters }));
        });
    }
    /**
    * gmesh查询仓库审批列表
    * @summary gmesh查询仓库审批列表
    * @param {WarehouseApiApprovePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    approvePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/warehouse/approve/page', data: requestParameters }));
        });
    }
    /**
    * gmesh审批仓库配置是否通过
    * @summary gmesh审批仓库配置是否通过
    * @param {WarehouseApiApproveStorageConfigRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    approveStorageConfig(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/warehouse/approve', data: requestParameters }));
        });
    }
    /**
    * gmesh审批仓库配置转移审批人
    * @summary gmesh审批仓库配置转移审批人
    * @param {WarehouseApiApproveTransferRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    approveTransfer(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/warehouse/approve/transfer', data: requestParameters }));
        });
    }
    /**
    * 自动分配库存
    * @summary 自动分配库存
    * @param {WarehouseApiAutoAllocateInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    autoAllocateInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/auto/allocate', data: requestParameters }));
        });
    }
    /**
    * 批量出库
    * @summary 批量出库
    * @param {WarehouseApiBatchUpdateOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    batchUpdateOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/batchUpdate', data: requestParameters }));
        });
    }
    /**
    * 公司绑定仓库
    * @summary 公司绑定仓库
    * @param {WarehouseApiBindRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    bind(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/bind', data: requestParameters }));
        });
    }
    /**
    * 【】取消拣货单
    * @summary 【】取消拣货单
    * @param {WarehouseApiCancelPickUpRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    cancelPickUp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/pickUp/warehouse/cancel', data: requestParameters }));
        });
    }
    /**
    * 【gmesh上架】选择库位码
    * @summary 【gmesh上架】选择库位码
    * @param {WarehouseApiChooseShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    chooseShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/choose/shlef/code', data: requestParameters }));
        });
    }
    /**
    * 【弃货处理】关单
    * @summary 【弃货处理】关单
    * @param {WarehouseApiCloseDiscardGoodsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    closeDiscardGoods(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/discard-goods/close', data: requestParameters }));
        });
    }
    /**
    * 【订单拦截】关单
    * @summary 【订单拦截】关单
    * @param {WarehouseApiCloseOrderInterceptionRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    closeOrderInterception(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/order-interception/close', data: requestParameters }));
        });
    }
    /**
    * 【商品盘点】关单
    * @summary 【商品盘点】关单
    * @param {WarehouseApiClosePhysicalInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    closePhysicalInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/physical-inventory/close', data: requestParameters }));
        });
    }
    /**
    * 【更换标签】关单
    * @summary 【更换标签】关单
    * @param {WarehouseApiCloseReplaceLabelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    closeReplaceLabel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/replace-label/gmesh/close', data: requestParameters }));
        });
    }
    /**
    * 【人工服务】关单
    * @summary 【人工服务】关单
    * @param {WarehouseApiCloseStaffServiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    closeStaffService(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/staff-service/close', data: requestParameters }));
        });
    }
    /**
    * 根据仓库编码获取公司列表
    * @summary 根据仓库编码获取公司列表
    * @param {WarehouseApiCompanyListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    companyList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/company/list', data: requestParameters }));
        });
    }
    /**
    * 【gmesh上架】入库单上架完成
    * @summary 【gmesh上架】入库单上架完成
    * @param {WarehouseApiCompletePutOnRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    completePutOn(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/inbound/up/complete', data: requestParameters }));
        });
    }
    /**
    * PDA账户配置仓库
    * @summary PDA账户配置仓库
    * @param {WarehouseApiConfigurePdaAccountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    configurePdaAccount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/pdaAccount/configure', data: requestParameters }));
        });
    }
    /**
    * PDA账户批量配置仓库
    * @summary PDA账户批量配置仓库
    * @param {WarehouseApiConfigurePdaAccountBatchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    configurePdaAccountBatch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/pdaAccount/configure/batch', data: requestParameters }));
        });
    }
    /**
    * 确认未发货数量
    * @summary 确认未发货数量
    * @param {WarehouseApiConfirmUnshippedQuantityRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    confirmUnshippedQuantity(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/confirm/unshippedQuantity', data: requestParameters }));
        });
    }
    /**
    * 创建入库通知单
    * @summary 创建入库通知单
    * @param {WarehouseApiCreateInboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    createInbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/create', data: requestParameters }));
        });
    }
    /**
    * 【】创建拣货单
    * @summary 【】创建拣货单
    * @param {WarehouseApiCreatePickUpRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    createPickUp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/pickUp/warehouse/create', data: requestParameters }));
        });
    }
    /**
    * 箱唛下载
    * @summary 箱唛下载
    * @param {WarehouseApiDownloadShippingMarkRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    downloadShippingMark(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/download/shippingmark', data: requestParameters }));
        });
    }
    /**
    * 入库通知单导出
    * @summary 入库通知单导出
    * @param {WarehouseApiExportInboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    exportInbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/export', data: requestParameters }));
        });
    }
    /**
    * 入库单明细导出【到库位纬度】
    * @summary 入库单明细导出【到库位纬度】
    * @param {WarehouseApiExportInboundDetailWithShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    exportInboundDetailWithShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/detail/export', data: requestParameters }));
        });
    }
    /**
    * 【】历史库存明细导出
    * @summary 【】历史库存明细导出
    * @param {WarehouseApiExportInboundInventorySnapshotRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    exportInboundInventorySnapshot(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inboundInventorySnapshot/warehouse/export', data: requestParameters }));
        });
    }
    /**
    * 【】库存汇总导出
    * @summary 【】库存汇总导出
    * @param {WarehouseApiExportInventoryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    exportInventoryDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/export', data: requestParameters }));
        });
    }
    /**
    * 历史库存汇总导出
    * @summary 历史库存汇总导出
    * @param {WarehouseApiExportInventoryDetailSnapshotRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    exportInventoryDetailSnapshot(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/inventoryDetailSnapshot/export', data: requestParameters }));
        });
    }
    /**
    * 【】拣货单导出
    * @summary 【】拣货单导出
    * @param {WarehouseApiExportPickUpPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    exportPickUpPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/pickUp/warehouse/export', data: requestParameters }));
        });
    }
    /**
    * 【】导出拣货单PDF
    * @summary 【】导出拣货单PDF
    * @param {WarehouseApiExportPickUpPdfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    exportPickUpPdf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/pickUp/warehouse/export/pdf', data: requestParameters }));
        });
    }
    /**
    * 导出仓库信息
    * @summary 导出仓库信息
    * @param {WarehouseApiExportStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    exportStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/warehouse/export', data: requestParameters }));
        });
    }
    /**
    * 出库单导出运单号模版
    * @summary 出库单导出运单号模版
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    exportTransNoTemplate(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/report/transNoTemplate/export' }));
        });
    }
    /**
    * 【弃货处理】完成
    * @summary 【弃货处理】完成
    * @param {WarehouseApiFinishDiscardGoodsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    finishDiscardGoods(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/discard-goods/finish', data: requestParameters }));
        });
    }
    /**
    * 【订单拦截】完成
    * @summary 【订单拦截】完成
    * @param {WarehouseApiFinishOrderInterceptionRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    finishOrderInterception(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/order-interception/finish', data: requestParameters }));
        });
    }
    /**
    * 【商品盘点】完成
    * @summary 【商品盘点】完成
    * @param {WarehouseApiFinishPhysicalInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    finishPhysicalInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/physical-inventory/finish', data: requestParameters }));
        });
    }
    /**
    * 【更换标签】完成
    * @summary 【更换标签】完成
    * @param {WarehouseApiFinishReplaceLabelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    finishReplaceLabel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/replace-label/gmesh/finish', data: requestParameters }));
        });
    }
    /**
    * 【人工服务】完成
    * @summary 【人工服务】完成
    * @param {WarehouseApiFinishStaffServiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    finishStaffService(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/staff-service/finish', data: requestParameters }));
        });
    }
    /**
    * 仓库所在地域下拉列表
    * @summary 仓库所在地域下拉列表
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getAmazonCountryAndStateVO(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/state/all' }));
        });
    }
    /**
    * 【WMS】查询英文文案
    * @summary 【WMS】查询英文文案
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    getInEn(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-warehouse-web/common/warehouse/i18n/en' }));
        });
    }
    /**
    * 【WMS】查询中文文案
    * @summary 【WMS】查询中文文案
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    getInZh(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-warehouse-web/common/warehouse/i18n/zh' }));
        });
    }
    /**
    * 入库通知单详情
    * @summary 入库通知单详情
    * @param {WarehouseApiGetInboundNoticeDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getInboundNoticeDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/notice/detail/list', data: requestParameters }));
        });
    }
    /**
    * 入库通知单箱规等其他信息详情
    * @summary 入库通知单箱规等其他信息详情
    * @param {WarehouseApiGetInboundNoticeOtherDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getInboundNoticeOtherDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/notice/detail/otherInfo', data: requestParameters }));
        });
    }
    /**
    * 入库通知单分页
    * @summary 入库通知单分页
    * @param {WarehouseApiGetInboundNoticePageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getInboundNoticePageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * 出库通知单详情
    * @summary 出库通知单详情
    * @param {WarehouseApiGetOutboundNoticeDetailListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getOutboundNoticeDetailList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/detail/list', data: requestParameters }));
        });
    }
    /**
    * 出库通知单列表分页
    * @summary 出库通知单列表分页
    * @param {WarehouseApiGetOutboundNoticePageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getOutboundNoticePageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/list', data: requestParameters }));
        });
    }
    /**
    * PDA账户详情
    * @summary PDA账户详情
    * @param {WarehouseApiGetPdaAccountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    getPdaAccount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/pdaAccount/detail', data: requestParameters }));
        });
    }
    /**
    * 【】查询拣货单详情
    * @summary 【】查询拣货单详情
    * @param {WarehouseApiGetPickUpPageInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    getPickUpPageInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/pickUp/warehouse/info', data: requestParameters }));
        });
    }
    /**
    * 库位信息详情
    * @summary 库位信息详情
    * @param {WarehouseApiGetShelfDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    getShelfDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/shelf/detail', data: requestParameters }));
        });
    }
    /**
    * 仓库详情
    * @summary 仓库详情
    * @param {WarehouseApiGetStorageDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getStorageDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/detail', data: requestParameters }));
        });
    }
    /**
    * 条件查询仓库信息列表
    * @summary 条件查询仓库信息列表
    * @param {WarehouseApiGetStorageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getStorageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/list', data: requestParameters }));
        });
    }
    /**
    * 分页条件查询仓库信息
    * @summary 分页条件查询仓库信息
    * @param {WarehouseApiGetStoragePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getStoragePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/page', data: requestParameters }));
        });
    }
    /**
    * 查询仓库关联VC账号及VendorCode信息
    * @summary 查询仓库关联VC账号及VendorCode信息
    * @param {WarehouseApiGetStorageVcAccountListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    getStorageVcAccountList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/vc/list', data: requestParameters }));
        });
    }
    /**
    * 【弃货处理】提交弃货处理服务单
    * @summary 【弃货处理】提交弃货处理服务单
    * @param {WarehouseApiGmeshAddDiscardGoodsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshAddDiscardGoods(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/discard-goods/add', data: requestParameters }));
        });
    }
    /**
    * 【订单拦截】提交订单拦截服务单
    * @summary 【订单拦截】提交订单拦截服务单
    * @param {WarehouseApiGmeshAddOrderInterceptionRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshAddOrderInterception(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/order-interception/add', data: requestParameters }));
        });
    }
    /**
    * 添加出库通知单
    * @summary 添加出库通知单
    * @param {WarehouseApiGmeshAddOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshAddOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/notice/add', data: requestParameters }));
        });
    }
    /**
    * 出库通知单批量创建
    * @summary 出库通知单批量创建
    * @param {WarehouseApiGmeshAddOutboundBatchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshAddOutboundBatch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/add/batch', data: requestParameters }));
        });
    }
    /**
    * 【商品盘点】提交商品盘点服务单
    * @summary 【商品盘点】提交商品盘点服务单
    * @param {WarehouseApiGmeshAddPhysicalInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshAddPhysicalInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/physical-inventory/add', data: requestParameters }));
        });
    }
    /**
    * 【更换标签】提交更换标签服务单
    * @summary 【更换标签】提交更换标签服务单
    * @param {WarehouseApiGmeshAddReplaceLabelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshAddReplaceLabel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/replace-label/add', data: requestParameters }));
        });
    }
    /**
    * 【人工服务】提交人工服务服务单
    * @summary 【人工服务】提交人工服务服务单
    * @param {WarehouseApiGmeshAddStaffServiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshAddStaffService(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/staff-service/add', data: requestParameters }));
        });
    }
    /**
    * 【移区】提交移区申请
    * @summary 【移区】提交移区申请
    * @param {WarehouseApiGmeshApplyShiftZoneRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshApplyShiftZone(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/apply', data: requestParameters }));
        });
    }
    /**
    * 出库通知单取消
    * @summary 出库通知单取消
    * @param {WarehouseApiGmeshCancelOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshCancelOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/cancel', data: requestParameters }));
        });
    }
    /**
    * 【移区】移区单取消
    * @summary 【移区】移区单取消
    * @param {WarehouseApiGmeshCancelShiftZoneRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshCancelShiftZone(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/cancel', data: requestParameters }));
        });
    }
    /**
    * 【更换标签】校验ssku的有效性
    * @summary 【更换标签】校验ssku的有效性
    * @param {WarehouseApiGmeshCheckSskuRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshCheckSsku(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/replace-label/check-ssku', data: requestParameters }));
        });
    }
    /**
    * 【移区】移区扫库位校验体积
    * @summary 【移区】移区扫库位校验体积
    * @param {WarehouseApiGmeshCheckVolumeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshCheckVolume(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/check/volume', data: requestParameters }));
        });
    }
    /**
    * 【移区】移区单子单关闭
    * @summary 【移区】移区单子单关闭
    * @param {WarehouseApiGmeshCloseShiftDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshCloseShiftDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/close/detail', data: requestParameters }));
        });
    }
    /**
    * 【移区】移区单主单关闭
    * @summary 【移区】移区单主单关闭
    * @param {WarehouseApiGmeshCloseShiftZoneRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshCloseShiftZone(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/close', data: requestParameters }));
        });
    }
    /**
    * 【弃货处理】列表数据导出
    * @summary 【弃货处理】列表数据导出
    * @param {WarehouseApiGmeshExportDiscardGoodsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportDiscardGoods(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/discard-goods/export', data: requestParameters }));
        });
    }
    /**
    * 入库明细导出
    * @summary 入库明细导出
    * @param {WarehouseApiGmeshExportInboundDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportInboundDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/report/export', data: requestParameters }));
        });
    }
    /**
    * 【订单拦截】列表数据导出
    * @summary 【订单拦截】列表数据导出
    * @param {WarehouseApiGmeshExportOrderInterceptionRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportOrderInterception(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/order-interception/export', data: requestParameters }));
        });
    }
    /**
    * 出库通知单导出
    * @summary 出库通知单导出
    * @param {WarehouseApiGmeshExportOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/export', data: requestParameters }));
        });
    }
    /**
    * 出库明细导出
    * @summary 出库明细导出
    * @param {WarehouseApiGmeshExportOutboundDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    gmeshExportOutboundDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/reprot/export', data: requestParameters }));
        });
    }
    /**
    * 出库明细导出运单号
    * @summary 出库明细导出运单号
    * @param {WarehouseApiGmeshExportOutboundDetailProNoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportOutboundDetailProNo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/report/proNo/export', data: requestParameters }));
        });
    }
    /**
    * 【商品盘点】列表数据导出
    * @summary 【商品盘点】列表数据导出
    * @param {WarehouseApiGmeshExportPhysicalInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportPhysicalInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/physical-inventory/export', data: requestParameters }));
        });
    }
    /**
    * 【更换标签】Gmes列表数据导出
    * @summary 【更换标签】Gmes列表数据导出
    * @param {WarehouseApiGmeshExportReplaceLabelRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportReplaceLabel(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/replace-label/export', data: requestParameters }));
        });
    }
    /**
    * 【人工服务】列表数据导出
    * @summary 【人工服务】列表数据导出
    * @param {WarehouseApiGmeshExportStaffServiceRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportStaffService(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/staff-service/export', data: requestParameters }));
        });
    }
    /**
    * 【物流费用】列表数据导出
    * @summary 【物流费用】列表数据导出
    * @param {WarehouseApiGmeshExportshipmentFeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshExportshipmentFee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/shipment-fee/export', data: requestParameters }));
        });
    }
    /**
    * 【更换标签】下载附件
    * @summary 【更换标签】下载附件
    * @param {WarehouseApiGmeshGetAttachmentsInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetAttachmentsInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/replace-label/download-attachments', data: requestParameters }));
        });
    }
    /**
    * 【弃货处理】查询服务单详情
    * @summary 【弃货处理】查询服务单详情
    * @param {WarehouseApiGmeshGetDiscardGoodsInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetDiscardGoodsInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/discard-goods/info', data: requestParameters }));
        });
    }
    /**
    * 入库通知单详情
    * @summary 入库通知单详情
    * @param {WarehouseApiGmeshGetInboundDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetInboundDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/notice/detail', data: requestParameters }));
        });
    }
    /**
    * 【订单拦截】查询服务单详情
    * @summary 【订单拦截】查询服务单详情
    * @param {WarehouseApiGmeshGetOrderInterceptionInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetOrderInterceptionInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/order-interception/info', data: requestParameters }));
        });
    }
    /**
    * 【商品盘点】查询服务单详情
    * @summary 【商品盘点】查询服务单详情
    * @param {WarehouseApiGmeshGetPhysicalInventoryInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetPhysicalInventoryInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/physical-inventory/info', data: requestParameters }));
        });
    }
    /**
    * 【更换标签】Gmes查询服务单详情
    * @summary 【更换标签】Gmes查询服务单详情
    * @param {WarehouseApiGmeshGetReplaceLabelInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetReplaceLabelInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/replace-label/info', data: requestParameters }));
        });
    }
    /**
    * 【移区】移区单详情查询
    * @summary 【移区】移区单详情查询
    * @param {WarehouseApiGmeshGetShiftZoneDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetShiftZoneDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/detail', data: requestParameters }));
        });
    }
    /**
    * 【移区】ssku长宽高查询【移区操作】
    * @summary 【移区】ssku长宽高查询【移区操作】
    * @param {WarehouseApiGmeshGetSskuSizeInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetSskuSizeInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/ssku/size', data: requestParameters }));
        });
    }
    /**
    * 【人工服务】查询服务单详情
    * @summary 【人工服务】查询服务单详情
    * @param {WarehouseApiGmeshGetStaffServiceInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshGetStaffServiceInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/staff-service/info', data: requestParameters }));
        });
    }
    /**
    * 库存变化明细分页
    * @summary 库存变化明细分页
    * @param {WarehouseApiGmeshInventoryChangePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshInventoryChangePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/change/page', data: requestParameters }));
        });
    }
    /**
    * 【弃货处理】列表查询
    * @summary 【弃货处理】列表查询
    * @param {WarehouseApiGmeshListDiscardGoodsPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListDiscardGoodsPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/discard-goods/page', data: requestParameters }));
        });
    }
    /**
    * 【】分页查询库存明细
    * @summary 【】分页查询库存明细
    * @param {WarehouseApiGmeshListInboundInventoryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListInboundInventoryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/page/each', data: requestParameters }));
        });
    }
    /**
    * 【】查询库存(非明细维度)
    * @summary 【】查询库存(非明细维度)
    * @param {WarehouseApiGmeshListInventoryBySskuAndPcsRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListInventoryBySskuAndPcs(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/ssku-with-pcs/list', data: requestParameters }));
        });
    }
    /**
    * 【订单拦截】列表查询
    * @summary 【订单拦截】列表查询
    * @param {WarehouseApiGmeshListOrderInterceptionPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListOrderInterceptionPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/order-interception/page', data: requestParameters }));
        });
    }
    /**
    * 【商品盘点】列表查询
    * @summary 【商品盘点】列表查询
    * @param {WarehouseApiGmeshListPhysicalInventoryPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListPhysicalInventoryPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/physical-inventory/page', data: requestParameters }));
        });
    }
    /**
    * 【更换标签】Gmes列表查询
    * @summary 【更换标签】Gmes列表查询
    * @param {WarehouseApiGmeshListReplaceLabelPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListReplaceLabelPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/replace-label/page', data: requestParameters }));
        });
    }
    /**
    * 【物流费用】分页查询
    * @summary 【物流费用】分页查询
    * @param {WarehouseApiGmeshListShipmentFeePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListShipmentFeePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/shipment-fee/page', data: requestParameters }));
        });
    }
    /**
    * 【移区】ssku装箱规格和有效期查询【移区申请】
    * @summary 【移区】ssku装箱规格和有效期查询【移区申请】
    * @param {WarehouseApiGmeshListSskuInnerPackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListSskuInnerPack(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/list/ssku/innerPack', data: requestParameters }));
        });
    }
    /**
    * 【移区】ssku库位数量信息查询【移区操作】
    * @summary 【移区】ssku库位数量信息查询【移区操作】
    * @param {WarehouseApiGmeshListSskuShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListSskuShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/list/ssku/shelf', data: requestParameters }));
        });
    }
    /**
    * 【人工服务】列表查询
    * @summary 【人工服务】列表查询
    * @param {WarehouseApiGmeshListStaffServicePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshListStaffServicePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/staff-service/page', data: requestParameters }));
        });
    }
    /**
    * 【】修改装箱数量
    * @summary 【】修改装箱数量
    * @param {WarehouseApiGmeshModifyInnerPackRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshModifyInnerPack(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/modify/pcs', data: requestParameters }));
        });
    }
    /**
    * 出库通知单详情-详情页
    * @summary 出库通知单详情-详情页
    * @param {WarehouseApiGmeshOutboundDetailDetailPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshOutboundDetailDetailPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/detail', data: requestParameters }));
        });
    }
    /**
    * 出库通知单详情--出库前查询
    * @summary 出库通知单详情--出库前查询
    * @param {WarehouseApiGmeshOutboundNoticeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshOutboundNoticeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/notice/detail/list', data: requestParameters }));
        });
    }
    /**
    * 入库通知单分页
    * @summary 入库通知单分页
    * @param {WarehouseApiGmeshPageInboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshPageInbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * 入库明细分页
    * @summary 入库明细分页
    * @param {WarehouseApiGmeshPageInboundDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshPageInboundDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/report/page/detail', data: requestParameters }));
        });
    }
    /**
    * 库存修改记录
    * @summary 库存修改记录
    * @param {WarehouseApiGmeshPageInventoryChangeMessageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshPageInventoryChangeMessage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/change/change/message', data: requestParameters }));
        });
    }
    /**
    * 出库通知单列表分页
    * @summary 出库通知单列表分页
    * @param {WarehouseApiGmeshPageOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshPageOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/notice/list', data: requestParameters }));
        });
    }
    /**
    * 出库明细分页
    * @summary 出库明细分页
    * @param {WarehouseApiGmeshPageOutboundDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshPageOutboundDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/report/detail/page', data: requestParameters }));
        });
    }
    /**
    * 出库单分页查询（拣货单生成）
    * @summary 出库单分页查询（拣货单生成）
    * @param {WarehouseApiGmeshPagePickUpRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshPagePickUp(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/page/forPda', data: requestParameters }));
        });
    }
    /**
    * 【移区】移区单分页查询
    * @summary 【移区】移区单分页查询
    * @param {WarehouseApiGmeshPageShiftZoneRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshPageShiftZone(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/page', data: requestParameters }));
        });
    }
    /**
    * 【商品盘点】盘点表下载
    * @summary 【商品盘点】盘点表下载
    * @param {WarehouseApiGmeshPhysicalInventoryTableDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshPhysicalInventoryTableDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/physical-inventory/table/download', data: requestParameters }));
        });
    }
    /**
    * 出库通知单捡货
    * @summary 出库通知单捡货
    * @param {WarehouseApiGmeshPickUpOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    gmeshPickUpOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/pickup', data: requestParameters }));
        });
    }
    /**
    * 出库通知单批量捡货
    * @summary 出库通知单批量捡货
    * @param {WarehouseApiGmeshPickUpOutboundBatchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    gmeshPickUpOutboundBatch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/batch/pickup', data: requestParameters }));
        });
    }
    /**
    * 【增值服务单】查询出库单的商品列表
    * @summary 【增值服务单】查询出库单的商品列表
    * @param {WarehouseApiGmeshProductListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshProductList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/shipment-fee/product/list', data: requestParameters }));
        });
    }
    /**
    * 【】盘盈亏
    * @summary 【】盘盈亏
    * @param {WarehouseApiGmeshProfitOrLossRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshProfitOrLoss(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/count', data: requestParameters }));
        });
    }
    /**
    * 出库通知单回退
    * @summary 出库通知单回退
    * @param {WarehouseApiGmeshRollbackOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshRollbackOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/rollback', data: requestParameters }));
        });
    }
    /**
    * 【移区】移区扫库位校验箱区，有效期
    * @summary 【移区】移区扫库位校验箱区，有效期
    * @param {WarehouseApiGmeshScanShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshScanShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/scan/shelf', data: requestParameters }));
        });
    }
    /**
    * 【增值服务单】搜索出库通知单
    * @summary 【增值服务单】搜索出库通知单
    * @param {WarehouseApiGmeshSearchOutboundOrderRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshSearchOutboundOrder(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/shipment-fee/outbound/search', data: requestParameters }));
        });
    }
    /**
    * 出库通知单出库
    * @summary 出库通知单出库
    * @param {WarehouseApiGmeshShipOutboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshShipOutbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/ship', data: requestParameters }));
        });
    }
    /**
    * 【移区】移区库位或批次纬度批量提交
    * @summary 【移区】移区库位或批次纬度批量提交
    * @param {WarehouseApiGmeshSubmitShiftZoneRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    gmeshSubmitShiftZone(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/shift/zone/submit/batch', data: requestParameters }));
        });
    }
    /**
    * 入库单收货
    * @summary 入库单收货
    * @param {WarehouseApiInboundRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    inbound(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/inbound', data: requestParameters }));
        });
    }
    /**
    * 库存变动明细分页
    * @summary 库存变动明细分页
    * @param {WarehouseApiInventoryChangeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    inventoryChangeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/change/detail', data: requestParameters }));
        });
    }
    /**
    * 分区库存列表
    * @summary 分区库存列表
    * @param {WarehouseApiListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    list(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/list/each', data: requestParameters }));
        });
    }
    /**
    * 模糊查询产品信息
    * @summary 模糊查询产品信息
    * @param {WarehouseApiListFuzzyProductRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    listFuzzyProduct(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/product', data: requestParameters }));
        });
    }
    /**
    * 库存列表
    * @summary 库存列表
    * @param {WarehouseApiListInboundInventoryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    listInboundInventory(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/inventory/warehouse/list', data: requestParameters }));
        });
    }
    /**
    * 【】分页查询历史库存明细
    * @summary 【】分页查询历史库存明细
    * @param {WarehouseApiListInboundInventorySnapshotPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    listInboundInventorySnapshotPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inboundInventorySnapshot/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * 入库通知单详情列表
    * @summary 入库通知单详情列表
    * @param {WarehouseApiListInboundNoticeDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    listInboundNoticeDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/warehouse/notice/detail', data: requestParameters }));
        });
    }
    /**
    * 库存变动明细列表
    * @summary 库存变动明细列表
    * @param {WarehouseApiListInventoryChangeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    listInventoryChange(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/change/list', data: requestParameters }));
        });
    }
    /**
    * 库存汇总带箱规
    * @summary 库存汇总带箱规
    * @param {WarehouseApiListInventorySummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    listInventorySummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/list', data: requestParameters }));
        });
    }
    /**
    * PDA账户列表
    * @summary PDA账户列表
    * @param {WarehouseApiListPdaAccountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    listPdaAccount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/pdaAccount/list', data: requestParameters }));
        });
    }
    /**
    * 【】分页查询拣货单
    * @summary 【】分页查询拣货单
    * @param {WarehouseApiListPickUpPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    listPickUpPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/pickUp/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * 库存产品列表
    * @summary 库存产品列表
    * @param {WarehouseApiListProductRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    listProduct(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inbound/inventory/warehouse/product/list', data: requestParameters }));
        });
    }
    /**
    * 库位信息列表
    * @summary 库位信息列表
    * @param {WarehouseApiListShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    listShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/shelf/list', data: requestParameters }));
        });
    }
    /**
    * 【移仓】ssku剩余可用数量查询
    * @summary 【移仓】ssku剩余可用数量查询
    * @param {WarehouseApiListSskuAvailableNumRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    listSskuAvailableNum(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/ssku/available/num', data: requestParameters }));
        });
    }
    /**
    * 查含有这个ssku的仓库list
    * @summary 查含有这个ssku的仓库list
    * @param {WarehouseApiListStorageBySskuRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    listStorageBySsku(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/inventory/list', data: requestParameters }));
        });
    }
    /**
    * 编辑入库单
    * @summary 编辑入库单
    * @param {WarehouseApiModifyRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    modify(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/modify', data: requestParameters }));
        });
    }
    /**
    * 编辑入库单,非库存信息编辑
    * @summary 编辑入库单,非库存信息编辑
    * @param {WarehouseApiModifyOtherInfoRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    modifyOtherInfo(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/modifyOtherInfo', data: requestParameters }));
        });
    }
    /**
    * 移仓
    * @summary 移仓
    * @param {WarehouseApiMoveStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    moveStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/move/storage', data: requestParameters }));
        });
    }
    /**
    * 批量下载出库单下的所有文件
    * @summary 批量下载文件
    * @param {WarehouseApiOutboundFileBatchDownloadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    outboundFileBatchDownload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/file/warehouse/batch/download', data: requestParameters }));
        });
    }
    /**
    * 上传文件
    * @summary 上传文件
    * @param {WarehouseApiOutboundFileUploadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    outboundFileUpload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/file/warehouse/upload', data: requestParameters }));
        });
    }
    /**
    * 出库通知单出库
    * @summary 出库通知单出库
    * @param {WarehouseApiOutboundNoticeDetailListUpdateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    outboundNoticeDetailListUpdate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/outbound/warehouse/notice/detail/update', data: requestParameters }));
        });
    }
    /**
    * 分区库存分页
    * @summary 分区库存分页
    * @param {WarehouseApiPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    page(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/page/each', data: requestParameters }));
        });
    }
    /**
    * 【】库存汇总分页查询
    * @summary 【】库存汇总分页查询
    * @param {WarehouseApiPageInventoryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    pageInventoryDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inventory/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * 历史库存汇总分页查询
    * @summary 历史库存汇总分页查询
    * @param {WarehouseApiPageInventoryDetailSnapshotRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    pageInventoryDetailSnapshot(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/inventoryDetailSnapshot/page', data: requestParameters }));
        });
    }
    /**
    * 库存汇总分页
    * @summary 库存汇总分页
    * @param {WarehouseApiPageInventorySummaryRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    pageInventorySummary(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/inventory/warehouse/page', data: requestParameters }));
        });
    }
    /**
    * PDA账户仓库列表分页
    * @summary PDA账户仓库列表分页
    * @param {WarehouseApiPagePdaAccountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    pagePdaAccount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/pdaAccount/page', data: requestParameters }));
        });
    }
    /**
    * 库位信息分页查询
    * @summary 库位信息分页查询
    * @param {WarehouseApiPageShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    pageShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/shelf/page', data: requestParameters }));
        });
    }
    /**
    * 【商品盘点】盘点表上传
    * @summary 【商品盘点】盘点表上传
    * @param {WarehouseApiPhysicalInventoryTableUploadRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    physicalInventoryTableUpload(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/vas/physical-inventory/table/upload', data: requestParameters }));
        });
    }
    /**
    * 打印库位码
    * @summary 打印库位码
    * @param {WarehouseApiPrintShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    printShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/shelf/print', data: requestParameters }));
        });
    }
    /**
    * 批量打印库位码
    * @summary 批量打印库位码
    * @param {WarehouseApiPrintShelfBatchRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    printShelfBatch(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/shelf/print/batch', data: requestParameters }));
        });
    }
    /**
    * 【gmesh上架】上架推荐空货位
    * @summary 【gmesh上架】上架推荐空货位
    * @param {WarehouseApiRecommendShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    recommendShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/inbound/warehouse/recommend/shelf', data: requestParameters }));
        });
    }
    /**
    * 新增库位
    * @summary 新增库位
    * @param {WarehouseApiSaveShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    saveShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/shelf/save', data: requestParameters }));
        });
    }
    /**
    * 新增仓库
    * @summary 新增仓库
    * @param {WarehouseApiSaveStorageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    saveStorage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/save', data: requestParameters }));
        });
    }
    /**
    * SPT查询当前公司的仓库
    * @summary SPT查询当前公司的仓库
    * @param {WarehouseApiSptListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    sptList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/spt/list', data: requestParameters }));
        });
    }
    /**
    * 仓库绑定VendorCode
    * @summary 仓库绑定VendorCode
    * @param {WarehouseApiStorageBindVendorCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    storageBindVendorCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/storage/gmesh/vc/save', data: requestParameters }));
        });
    }
    /**
    * 获取仓库列表,可根据公司id查，也可模糊查询
    * @summary 获取仓库列表,可根据公司id查，也可模糊查询
    * @param {WarehouseApiStorageListRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    storageList(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/storage/list', data: requestParameters }));
        });
    }
    /**
    * 获取仓库列表,可根据公司id查，也可模糊查询
    * @summary 获取仓库列表,可根据公司id查，也可模糊查询，支持VC账号以及vendor查询
    * @param {WarehouseApiStorageListVRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    storageListV(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/company/storage/storage/listV2', data: requestParameters }));
        });
    }
    /**
    * 出库导入运单号
    * @summary 出库单导入运单号
    * @param {WarehouseApiTransNoImportRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    transNoImport(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/outbound/warehouse/report/transNo/import', data: requestParameters }));
        });
    }
    /**
    * 更新库位
    * @summary 更新库位
    * @param {WarehouseApiUpdateShelfRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    updateShelf(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/shelf/update', data: requestParameters }));
        });
    }
    /**
    * 更新库位状态
    * @summary 更新库位状态
    * @param {WarehouseApiUpdateShelfStatusRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseApi
    */
    updateShelfStatus(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-warehouse-web/gmesh/warehouse/shelf/status/update', data: requestParameters }));
        });
    }
    /**
    * 根据仓库编码修改仓库配置
    * @summary 根据仓库编码修改仓库配置
    * @param {WarehouseApiUpdateStorageByCodeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @deprecated
    * @memberof WarehouseApi
    */
    updateStorageByCode(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'PATCH', url: '/spotter-warehouse-web/storage/gmesh/config', data: requestParameters }));
        });
    }
}
exports.SpotterWarehouseApiSdk = SpotterWarehouseApiSdk;
