// @spotter/gmesh-api-sdk/lib/api/warehouse-voucher-endpoint-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * gmesh-web
 * gmesh-web
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: key.chen@spotterio.com.cn
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterWarehouseVoucherEndpointApiSdk = void 0;
// Some imports not used depending on template conditions
// @ts-ignore
const base_api_sdk_1 = __importDefault(require("../base-api-sdk"));
/**
* WarehouseVoucherEndpointApi - object-oriented interface
* @export
* @class WarehouseVoucherEndpointApi
* @extends {BaseSpotterApiSdk}
*/
class SpotterWarehouseVoucherEndpointApiSdk extends base_api_sdk_1.default {
    constructor(...baseApiSdkParams) {
        super(...baseApiSdkParams);
    }
    /**
    * 编辑仓库凭证金额
    * @summary 编辑仓库凭证金额
    * @param {WarehouseVoucherEndpointApiEditVoucherAmountRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    editVoucherAmount(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/warehouse/voucher/editVoucherAmount', data: requestParameters }));
        });
    }
    /**
    * 编辑仓库凭证日期
    * @summary 编辑仓库凭证日期
    * @param {WarehouseVoucherEndpointApiEditVoucherDateRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    editVoucherDate(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/warehouse/voucher/editVoucherDate', data: requestParameters }));
        });
    }
    /**
    * 导出仓库成本凭证
    * @summary 导出仓库成本凭证
    * @param {WarehouseVoucherEndpointApiExportCostRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    exportCost(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/warehouse/voucher/exportCost', data: requestParameters }));
        });
    }
    /**
    * 导出仓库成本金蝶凭证
    * @summary 导出仓库成本金蝶凭证
    * @param {WarehouseVoucherEndpointApiExportCostKingdeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    exportCostKingdee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/warehouse/voucher/kingdee/exportCost', data: requestParameters }));
        });
    }
    /**
    * 导出仓库收入凭证
    * @summary 导出仓库收入凭证
    * @param {WarehouseVoucherEndpointApiExportIncomeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    exportIncome(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/warehouse/voucher/exportIncome', data: requestParameters }));
        });
    }
    /**
    * 导出仓库收入金蝶凭证
    * @summary 导出仓库收入金蝶凭证
    * @param {WarehouseVoucherEndpointApiExportIncomeKingdeeRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    exportIncomeKingdee(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/warehouse/voucher/kingdee/exportIncome', data: requestParameters }));
        });
    }
    /**
    * 获取仓库成本会计科目
    * @summary 获取仓库成本会计科目
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    getCostSubject(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-evidence-web/warehouse/voucher/getCostSubject' }));
        });
    }
    /**
    * 获取仓库列表
    * @summary 获取仓库列表
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    getStorageCodeList(options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-evidence-web/warehouse/voucher/getStorageCodeList' }));
        });
    }
    /**
    * 分页查询仓库成本凭证
    * @summary 分页查询仓库成本凭证
    * @param {WarehouseVoucherEndpointApiQueryCostPageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    queryCostPage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/warehouse/voucher/queryCostPage', data: requestParameters }));
        });
    }
    /**
    * 查询凭证详情
    * @summary 查询凭证详情
    * @param {WarehouseVoucherEndpointApiQueryDetailRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    queryDetail(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const params = {};
            if ((requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id) !== undefined) {
                params['id'] = requestParameters === null || requestParameters === void 0 ? void 0 : requestParameters.id;
            }
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'GET', url: '/spotter-evidence-web/warehouse/voucher/queryDetail', params: requestParameters }));
        });
    }
    /**
    * 分页查询仓库收入凭证
    * @summary 分页查询仓库收入凭证
    * @param {WarehouseVoucherEndpointApiQueryIncomePageRequest} requestParameters Request parameters.
    * @param {*} [options] Override http request option.
    * @memberof WarehouseVoucherEndpointApi
    */
    queryIncomePage(requestParameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(Object.assign(Object.assign({}, options), { method: 'POST', url: '/spotter-evidence-web/warehouse/voucher/queryIncomePage', data: requestParameters }));
        });
    }
}
exports.SpotterWarehouseVoucherEndpointApiSdk = SpotterWarehouseVoucherEndpointApiSdk;
