// @spotter/app-client-toolkit/lib/infra/mainApp/decorator/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotterAppDecorator = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const antd_1 = require("antd");
const i18n_sdk_1 = require("@spotter/i18n-sdk");
const useCompatibleDecorator_1 = require("./useCompatibleDecorator");
const useNetworkDecorator_1 = require("./useNetworkDecorator");
const OFFLINE_HELP_MESSAGE = 'OFFLINE_HELP_MESSAGE';
const BROWSER_TIP_CLOSE_KEY = 'BROWSER_TIP_CLOSE_KEY';
const SpotterAppDecorator = ({ children, getConfig }) => {
    const { t } = (0, i18n_sdk_1.useTranslation)();
    (0, useCompatibleDecorator_1.useCompatibleDecorator)({
        onUnsupportedVersion: () => {
            antd_1.message.warning({
                content: ((0, jsx_runtime_1.jsxs)("span", { className: "leading-16px flex items-center", children: [(0, jsx_runtime_1.jsx)("span", { className: "mr-12px", children: t('common.supported_browser_version_tip') }), (0, jsx_runtime_1.jsx)(antd_1.Button, { style: {
                                padding: 6,
                            }, onClick: () => {
                                antd_1.message.destroy(BROWSER_TIP_CLOSE_KEY);
                            }, type: "link", children: t('common.action_close') })] })),
                duration: 0,
                key: BROWSER_TIP_CLOSE_KEY,
            });
        },
        onUnsupportedBrowser: () => {
            antd_1.message.warning({
                content: ((0, jsx_runtime_1.jsxs)("span", { className: "leading-16px flex items-center", children: [(0, jsx_runtime_1.jsx)("span", { className: "mr-12px", children: t('common.supported_browser_tip') }), (0, jsx_runtime_1.jsx)(antd_1.Button, { style: {
                                padding: 6,
                            }, onClick: () => {
                                antd_1.message.destroy(BROWSER_TIP_CLOSE_KEY);
                            }, type: "link", children: t('common.action_close') })] })),
                duration: 0,
                key: BROWSER_TIP_CLOSE_KEY,
            });
        },
        getConfig
    });
    (0, useNetworkDecorator_1.useNetworkDecorator)({
        onNetworkChange: (isOnline) => {
            if (!isOnline) {
                antd_1.message.error({
                    content: ((0, jsx_runtime_1.jsxs)("span", { className: "leading-16px flex items-center", children: [(0, jsx_runtime_1.jsx)("span", { className: "mr-12px", children: t('infra.network_goes_offline_tips') }), (0, jsx_runtime_1.jsx)(antd_1.Button, { style: {
                                    padding: 6,
                                }, onClick: () => {
                                    antd_1.message.destroy(OFFLINE_HELP_MESSAGE);
                                }, type: "link", children: t('common.action_close') })] })),
                    duration: 0,
                    key: OFFLINE_HELP_MESSAGE,
                });
            }
            else {
                antd_1.message.destroy(OFFLINE_HELP_MESSAGE);
            }
        },
        emitOfflineWhenRouteChange: false,
    });
    return children;
};
exports.SpotterAppDecorator = SpotterAppDecorator;
