// @spotter/app-client-toolkit@1.0.42-nic.8/lib/infra/gui/ErrorBoundary/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useErrorHandler = exports.withErrorBoundary = exports.ErrorBoundary = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const React = __importStar(require("react"));
const changedArray = (a = [], b = []) => a.length !== b.length || a.some((item, index) => !Object.is(item, b[index]));
const initialState = { error: null };
class ErrorBoundary extends React.Component {
    constructor() {
        super(...arguments);
        this.state = initialState;
        this.resetErrorBoundary = (...args) => {
            const { onReset } = this.props;
            onReset?.(...args);
            this.reset();
        };
    }
    static getDerivedStateFromError(error) {
        return { error };
    }
    componentDidUpdate(prevProps, prevState) {
        const { error } = this.state;
        const { resetKeys } = this.props;
        // There's an edge case where if the thing that triggered the error
        // happens to *also* be in the resetKeys array, we'd end up resetting
        // the error boundary immediately. This would likely trigger a second
        // error to be thrown.
        // So we make sure that we don't check the resetKeys on the first call
        // of cDU after the error is set
        if (error !== null &&
            prevState.error !== null &&
            changedArray(prevProps.resetKeys, resetKeys)) {
            const { onResetKeysChange } = this.props;
            onResetKeysChange?.(prevProps.resetKeys, resetKeys);
            this.reset();
        }
    }
    componentDidCatch(error, info) {
        console.error(error);
        const { onError } = this.props;
        onError?.(error, info);
    }
    reset() {
        this.setState(initialState);
    }
    render() {
        const { error } = this.state;
        const { fallbackRender, FallbackComponent, fallback, children } = this.props;
        if (error !== null) {
            const props = {
                error,
                resetErrorBoundary: this.resetErrorBoundary
            };
            if (React.isValidElement(fallback)) {
                return fallback;
            }
            if (typeof fallbackRender === 'function') {
                return fallbackRender(props);
            }
            if (FallbackComponent) {
                return (0, jsx_runtime_1.jsx)(FallbackComponent, { ...props });
            }
            throw new Error('spotter-error-boundary requires either a fallback, fallbackRender, or FallbackComponent prop');
        }
        return children;
    }
}
exports.ErrorBoundary = ErrorBoundary;
function withErrorBoundary(Component, errorBoundaryProps) {
    const Wrapped = (props) => ((0, jsx_runtime_1.jsx)(ErrorBoundary, { ...errorBoundaryProps, children: (0, jsx_runtime_1.jsx)(Component, { ...props }) }));
    // Format for display in DevTools
    const name = Component.displayName || Component.name || 'Unknown';
    Wrapped.displayName = `withErrorBoundary(${name})`;
    return Wrapped;
}
exports.withErrorBoundary = withErrorBoundary;
function useErrorHandler(givenError) {
    const [error, setError] = React.useState(null);
    if (givenError != null)
        throw givenError;
    if (error != null)
        throw error;
    return setError;
}
exports.useErrorHandler = useErrorHandler;
