// @spotter/ui/lib/components/SptBrand/BrandCopyright.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BrandCopyright = void 0;
var _react = _interopRequireDefault(require("react"));
var _BrandFull = require("./BrandFull");
var _antd = require("antd");
require("./style/brand-copyright.less");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var BrandCopyright = exports.BrandCopyright = function BrandCopyright() {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "brand-copyright-container"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "brand-copyright-container-logo"
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "brand-copyright-container-logo-pow"
  }, "Powered By"), /*#__PURE__*/_react.default.createElement(_BrandFull.BrandFull, {
    logoProps: {
      className: 'brand-copyright-container-logo-icon'
    },
    nameProps: {
      className: 'brand-copyright-container-logo-name'
    }
  })), /*#__PURE__*/_react.default.createElement("span", {
    className: "brand-copyright-container-copyright"
  }, "Copyright \xA9 Since 2022 Spotterio.com, Inc. and its affiliates. All Rights Reserved.", /*#__PURE__*/_react.default.createElement(_antd.Divider, {
    type: "vertical"
  }), /*#__PURE__*/_react.default.createElement("a", {
    className: "brand-copyright-container-copyright-link",
    target: "_blank",
    rel: "noopener noreferrer",
    href: "https://beian.miit.gov.cn"
  }, "\u7CA4ICP\u59072022005977\u53F7")));
};
