// @spotter/app-client-toolkit/lib/tool/ossMonitor/ossMonitorSdk.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OSSMonitorSdk = void 0;
/**
 * oss 前端监控SDK
 */
class OSSMonitorSdk {
    /**
     * 构造函数
     * @param bl - 探针对象
     */
    constructor(bl) {
        this.bl = bl;
    }
    /**
     * 检查监控对象是否存在
     * @param bl - 探针对象
     * @returns 监控对象是否存在的布尔值
     */
    static checkBlExists(bl) {
        if (!bl) {
            console.error('未识别到__bl');
            return false;
        }
        return true;
    }
    /**
     * 重新修改部分配置项
     * @param next - 配置项对象
     */
    setConfig(next) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.setConfig(next);
        }
    }
    /**
     * 上报页面的API调用成功率
     * @param info - API信息
     */
    api(info) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.api(info);
        }
    }
    /**
     * 上报页面中的JS错误或使用者想关注的异常
     * @param error - 错误信息
     * @param pos - 错误位置信息
     */
    error(error, pos) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.error(error, pos);
        }
    }
    /**
     * 自定义上报的日志，通常被用于统计业务中特定事件发生的次数
     * @param key - 日志键名
     * @param value - 日志值
     */
    sum(key, value) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.sum(key, value);
        }
    }
    /**
     * 自定义上报的日志，通常被用于计算业务中特定事件发生的平均次数或平均值
     * @param key - 日志键名
     * @param value - 日志值
     */
    avg(key, value) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.avg(key, value);
        }
    }
    /**
     * 上报当前行为队列
     */
    reportBehavior() {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.reportBehavior();
        }
    }
    /**
     * 上报除默认性能指标外的自定义性能指标
     * @param info - 性能信息
     */
    performance(info) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            window.onload = () => {
                setTimeout(() => {
                    this.bl.performance(info);
                }, 1000); //设置一定的延时，确保默认性能数据采集完成。
            };
        }
    }
    /**
     * 重新设置页面的page name
     * @param page - 页面名称
     * @param sendPv - 是否发送PV
     */
    setPage(page, sendPv) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.setPage(page, sendPv);
        }
    }
    /**
     * 设置公共字段
     * @param common - 公共信息
     */
    setCommonInfo(common) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.setCommonInfo(common);
        }
    }
    /**
     * 在当前行为队列末尾添加一条自定义用户行为
     * @param behaviorInfo - 用户行为信息
     */
    addBehavior(behaviorInfo) {
        if (OSSMonitorSdk.checkBlExists(this.bl)) {
            this.bl.addBehavior(behaviorInfo);
        }
    }
}
exports.OSSMonitorSdk = OSSMonitorSdk;
// https://help.aliyun.com/zh/arms/browser-monitoring/developer-reference/api-reference?spm=a2c4g.11186623.0.0.8b601518NlR5i1
