// @spotter/app-client-toolkit/lib/infra/mainApp/network/detector/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useNetworkOnline = exports.useNetworkStatus = void 0;
const React = __importStar(require("react"));
const { useEffect, useState } = React;
const UNSUPPORTED_USER_AGENTS_PATTERN = /Windows.*Chrome|Windows.*Firefox|Linux.*Chrome/;
const defaultConfig = {
    enabled: true,
    url: "https://ipv4.icanhazip.com/",
    timeout: 5000,
    interval: 5000,
};
class SpotterNetworkDetector {
    constructor(pollingConfigs = defaultConfig) {
        this.pollingConfigs = pollingConfigs;
    }
    setIntervalId(intervalId) {
        this.intervalId = intervalId;
    }
    getIntervalId() {
        return this.intervalId;
    }
    static getInstance(pollingConfigs) {
        if (!SpotterNetworkDetector.instance) {
            SpotterNetworkDetector.instance = new SpotterNetworkDetector();
        }
        SpotterNetworkDetector.instance.setPollingConfigs(pollingConfigs);
        return SpotterNetworkDetector.instance;
    }
    setPollingConfigs(pollingConfig) {
        const needsPolling = UNSUPPORTED_USER_AGENTS_PATTERN.test(globalThis.navigator?.userAgent);
        if (needsPolling) {
            if (typeof pollingConfig === "object") {
                this.pollingConfigs = { ...defaultConfig, ...pollingConfig };
            }
            else if (pollingConfig) {
                this.pollingConfigs = defaultConfig;
            }
        }
        else {
            this.pollingConfigs = { ...defaultConfig, enabled: false };
        }
    }
    ping({ url, timeout, }) {
        return new Promise((resolve, reject) => {
            const xhr = new XMLHttpRequest();
            xhr.onerror = reject;
            xhr.ontimeout = reject;
            xhr.onreadystatechange = () => {
                if (xhr.readyState === xhr.HEADERS_RECEIVED) {
                    if (xhr.status) {
                        resolve();
                    }
                    else {
                        reject();
                    }
                }
            };
            xhr.open("HEAD", url);
            xhr.timeout = timeout;
            xhr.send();
        });
    }
}
const useNetworkOnline = (pollingConfigs = true) => {
    const [isOnline, setIsOnline] = useState(true);
    const instance = SpotterNetworkDetector.getInstance(pollingConfigs);
    const goOnline = () => {
        setIsOnline(true);
    };
    const goOffline = () => {
        setIsOnline(false);
    };
    useEffect(() => {
        window.addEventListener("online", goOnline);
        window.addEventListener("offline", goOffline);
        let intervalId = instance.getIntervalId();
        const { enabled, url, timeout, interval } = instance.pollingConfigs;
        if (enabled && url) {
            if (intervalId) {
                clearInterval(intervalId);
            }
            intervalId = window.setInterval(async () => {
                try {
                    await instance.ping({ url, timeout });
                    setIsOnline(true);
                }
                catch (error) {
                    setIsOnline(false);
                }
            }, interval);
            instance.setIntervalId(intervalId);
        }
        return () => {
            window.removeEventListener("online", goOnline);
            window.removeEventListener("offline", goOffline);
            if (enabled) {
                clearInterval(intervalId);
            }
        };
    }, []);
    return { isOnline };
};
exports.useNetworkOnline = useNetworkOnline;
let supported = false;
const useNetworkStatus = (initialEffectiveConnectionType) => {
    supported =
        globalThis.navigator &&
            "connection" in globalThis.navigator &&
            "effectiveType" in globalThis.navigator?.connection;
    const initialNetworkStatus = {
        supported,
        effectiveConnectionType: supported
            // @ts-ignore
            ? (globalThis.navigator?.connection).effectiveType
            : initialEffectiveConnectionType,
    };
    const [networkStatus, updateNetworkStatus] = useState(initialNetworkStatus);
    const setNetworkStatus = (networkStatus) => {
        updateNetworkStatus({
            supported,
            ...networkStatus,
        });
    };
    // @ts-ignore
    useEffect(() => {
        if (supported) {
            // @ts-ignore
            const navigatorConnection = globalThis.navigator?.connection;
            const updateECTStatus = () => {
                updateNetworkStatus({
                    supported,
                    effectiveConnectionType: navigatorConnection.effectiveType,
                });
            };
            navigatorConnection.addEventListener("change", updateECTStatus);
            return () => {
                navigatorConnection.removeEventListener("change", updateECTStatus);
            };
        }
    }, []);
    return { ...networkStatus, setNetworkStatus };
};
exports.useNetworkStatus = useNetworkStatus;
