// @spotter/ui/es/api/api/warehouse-api.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/api/api/warehouse-api.ts
import BaseSpotterApiSdk from "../base-api-sdk";
var SpotterWarehouseApiSdk = class extends BaseSpotterApiSdk {
  constructor(...baseApiSdkParams) {
    super(...baseApiSdkParams);
  }
  /**
   * 获取仓库列表,可根据公司id查，也可模糊查询
   * @summary 获取仓库列表,可根据公司id查，也可模糊查询
   * @param {WarehouseApiStorageListRequest} requestParameters Request parameters.
   * @param {*} [options] Override http request option.
   * @deprecated
   * @memberof WarehouseApi
   */
  storageList(requestParameters, options) {
    return __async(this, null, function* () {
      return this.httpClient.request(__spreadProps(__spreadValues({}, options), {
        method: "POST",
        url: "/spotter-warehouse-web/company/storage/storage/list",
        data: requestParameters
      }));
    });
  }
};
export {
  SpotterWarehouseApiSdk
};
