// @spotter/ui/es/components/SptSelect/index.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptSelect/index.tsx
import { PlusOutlined } from "@ant-design/icons";
import { Modal, Select, Tooltip } from "antd";
import React, { useEffect, useMemo, useState } from "react";
import SptTransfer from "../SptTransfer";
import { SptComponentProvider } from "../Provider";
import "./style/index.less";
var SptSelect = (_a) => {
  var _b = _a, {
    className,
    options,
    style,
    title,
    value,
    loading,
    tooltip,
    onChange,
    request,
    placeholder = "请选择"
  } = _b, resetProps = __objRest(_b, [
    "className",
    "options",
    "style",
    "title",
    "value",
    "loading",
    "tooltip",
    "onChange",
    "request",
    "placeholder"
  ]);
  const [visible, setVisible] = useState(false);
  const toggleVisible = () => setVisible(!visible);
  const [allOptions, setAllOptions] = useState([]);
  const finalOptions = useMemo(() => options != null ? options : allOptions, [options, allOptions]);
  const [checkedValue, setCheckedValue] = useState();
  useEffect(() => {
    request && request().then((o) => {
      setAllOptions(o);
    });
  }, [request]);
  useEffect(() => {
    if (visible) {
      setCheckedValue(value);
    }
  }, [visible]);
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(
    "div",
    {
      className: `spotter-select-container ${className != null ? className : ""}`,
      style: __spreadValues({}, style),
      onClick: (e) => {
        e.stopPropagation();
        e.preventDefault();
      }
    },
    /* @__PURE__ */ React.createElement(
      Select,
      __spreadValues({
        style: { flexGrow: 1 },
        showArrow: false,
        options: finalOptions,
        loading,
        value,
        mode: "multiple",
        showSearch: true,
        maxTagCount: "responsive",
        onChange,
        placeholder
      }, resetProps)
    ),
    /* @__PURE__ */ React.createElement(Tooltip, { placement: "top", title: tooltip, open: !tooltip ? false : void 0 }, /* @__PURE__ */ React.createElement(
      "div",
      {
        className: "spotter-select-tooltip-suffix suffix",
        onClick: () => toggleVisible()
      },
      /* @__PURE__ */ React.createElement(PlusOutlined, null)
    ))
  ), /* @__PURE__ */ React.createElement(
    Modal,
    {
      open: visible,
      width: 600,
      bodyStyle: { height: 500 },
      title,
      onOk: () => {
        onChange && onChange(checkedValue);
        toggleVisible();
      },
      destroyOnClose: true,
      onCancel: () => {
        toggleVisible();
      }
    },
    /* @__PURE__ */ React.createElement(
      SptTransfer,
      {
        defaultValue: value,
        options: finalOptions,
        value: checkedValue,
        onChange: (values) => {
          setCheckedValue(values);
        }
      }
    )
  ));
};
var SptSelect_default = SptSelect;
export {
  SptSelect_default as default
};
