// @spotter/gmesh-api-sdk/lib/http-client.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JSONBig2String = void 0;
const axios_1 = __importDefault(require("axios"));
const json_bigint_1 = __importDefault(require("json-bigint"));
exports.JSONBig2String = (0, json_bigint_1.default)({ storeAsString: true, constructorAction: 'preserve', protoAction: 'preserve' });
const defaultStatusSetMap = {
    authError: new Set([401, 4010000, 4010001, 4010002, 4010003]),
    permissionError: new Set([403]),
    success: new Set([200, 201]),
};
class HttpClient {
    constructor(options) {
        var _a, _b, _c, _d, _e, _f;
        this.options = options;
        this.options.res.isSuccess =
            (_b = (_a = options.res) === null || _a === void 0 ? void 0 : _a.isSuccess) !== null && _b !== void 0 ? _b : ((res) => { var _a; return defaultStatusSetMap.success.has(res.status) && defaultStatusSetMap.success.has((_a = res.data) === null || _a === void 0 ? void 0 : _a.code); });
        this.options.res.isAuthError = (_d = (_c = options.res) === null || _c === void 0 ? void 0 : _c.isAuthError) !== null && _d !== void 0 ? _d : ((res) => { var _a; return defaultStatusSetMap.authError.has(res.status) || defaultStatusSetMap.authError.has((_a = res.data) === null || _a === void 0 ? void 0 : _a.code); });
        this.options.res.isPermissionError = (_f = (_e = options.res) === null || _e === void 0 ? void 0 : _e.isPermissionError) !== null && _f !== void 0 ? _f : ((res) => { var _a; return defaultStatusSetMap.permissionError.has(res.status) || defaultStatusSetMap.permissionError.has((_a = res.data) === null || _a === void 0 ? void 0 : _a.code); });
        this.httpClient = axios_1.default.create(Object.assign(Object.assign({}, this.options.axiosOptions), { baseURL: this.options.context.href, withCredentials: true, transformResponse: [(data) => {
                    try {
                        return exports.JSONBig2String.parse(data || "{}");
                    }
                    catch (error) {
                        console.error(error);
                        return data;
                    }
                }] }));
        this.httpClient.interceptors.request.use((config) => {
            if (!config.data) {
                config.data = {};
            }
            return config;
        }, (error) => {
            console.log('request', error);
            return Promise.reject(error);
        });
        // 响应拦截器
        this.httpClient.interceptors.response.use((response) => {
            var _a, _b, _c, _d, _e, _f, _g;
            if (this.options.res.isSuccess(response)) {
                return Promise.resolve(response);
            }
            if (this.options.res.isAuthError(response)) {
                console.error(response);
                (_b = (_a = this.options.res).onAuthError) === null || _b === void 0 ? void 0 : _b.call(_a, response);
            }
            else if (this.options.res.isPermissionError(response)) {
                console.error(response);
                (_d = (_c = this.options.res).onPermissionError) === null || _d === void 0 ? void 0 : _d.call(_c, response);
            }
            else {
                // @ts-ignore
                if ((_e = response === null || response === void 0 ? void 0 : response.config) === null || _e === void 0 ? void 0 : _e.ignoreError) {
                    return Promise.resolve(response);
                }
                (_g = (_f = this.options.res).onOtherError) === null || _g === void 0 ? void 0 : _g.call(_f, response);
                // content: error.response.data?.message || 'Network Error'
                console.error(JSON.stringify(response.data));
            }
            return Promise.reject(response);
        }, (error) => {
            var _a, _b, _c, _d, _e, _f;
            const response = (error === null || error === void 0 ? void 0 : error.response) || {};
            if (this.options.res.isAuthError(response)) {
                console.log(error);
                (_b = (_a = this.options.res).onAuthError) === null || _b === void 0 ? void 0 : _b.call(_a, response);
            }
            else if (this.options.res.isPermissionError(response)) {
                console.error(response);
                (_d = (_c = this.options.res).onPermissionError) === null || _d === void 0 ? void 0 : _d.call(_c, response);
            }
            else {
                (_f = (_e = this.options.res).onOtherError) === null || _f === void 0 ? void 0 : _f.call(_e, response);
                // content: error.response.data?.message || 'Network Error'
                console.error(JSON.stringify(error.toJSON()));
            }
            return Promise.reject(error);
        });
    }
    request(config) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(config);
        });
    }
    getAxios() {
        return this.httpClient;
    }
}
exports.default = HttpClient;
