// @spotter/app-client-toolkit@1.0.42-nic.8/lib/infra/router/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getUseSpotterRouters = exports.SpotterRouter = exports.GlobalLayoutType = exports.TransparentLayout = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = __importStar(require("react"));
const const_1 = require("../../const");
const tool_1 = require("../../tool");
const react_router_dom_1 = require("react-router-dom");
const react_activation_1 = __importDefault(require("react-activation"));
//  {
//     DEFAULT = 'DEFAULT',
//     // 空白布局
//     BLANK = 'BLANK',
//     // 自定义布局
//     CUSTOM = 'CUSTOM',
// }
const TransparentLayout = ({ indexPath, base, children }) => {
    const location = (0, react_router_dom_1.useLocation)();
    const navigate = (0, react_router_dom_1.useNavigate)();
    (0, react_1.useEffect)(() => {
        if (location.pathname === base) {
            navigate(indexPath);
        }
        /* eslint-disable-next-line react-hooks/exhaustive-deps */
    }, [location.pathname, indexPath]);
    return children ? (0, jsx_runtime_1.jsx)("div", { style: { height: '100%' }, children: children }) : (0, jsx_runtime_1.jsx)(react_router_dom_1.Outlet, {});
};
exports.TransparentLayout = TransparentLayout;
// 命名需要 key 和 value 相同
var GlobalLayoutType;
(function (GlobalLayoutType) {
    // 默认的业务布局
    GlobalLayoutType["DEFAULT"] = "DEFAULT";
    // 未登录场景使用的布局，包括所有未登录场景需要的路由
    GlobalLayoutType["AUTH"] = "AUTH";
})(GlobalLayoutType = exports.GlobalLayoutType || (exports.GlobalLayoutType = {}));
class SpotterRouter {
    constructor(options) {
        this.options = options;
        this.layoutMap = {
            [GlobalLayoutType.DEFAULT]: {
                getLayoutRoute: () => ({
                    element: this.options.guard?.element ? (0, jsx_runtime_1.jsx)(this.options.guard.element, {}) : (0, jsx_runtime_1.jsx)(react_router_dom_1.Outlet, {}),
                    children: [
                        {
                            element: this.options.layout?.DEFAULT ? ((0, jsx_runtime_1.jsx)(this.options.layout.DEFAULT, {})) : ((0, jsx_runtime_1.jsx)(react_router_dom_1.Outlet, {})),
                            children: [
                                ...this.layoutMap[GlobalLayoutType.DEFAULT].pathRouteMap.values(),
                                ...(this.options.guard?.downgradeRoutes ?? [])
                            ]
                        }
                    ]
                }),
                // 完整路径定义到 router 对象的映射
                pathRouteMap: new Map()
            },
            [GlobalLayoutType.AUTH]: {
                getLayoutRoute: () => ({
                    element: this.options.layout?.AUTH ? (0, jsx_runtime_1.jsx)(this.options.layout.AUTH, {}) : (0, jsx_runtime_1.jsx)(react_router_dom_1.Outlet, {}),
                    children: [
                        ...this.layoutMap[GlobalLayoutType.AUTH].pathRouteMap.values()
                    ]
                }),
                pathRouteMap: new Map()
            }
        };
    }
    generateRedirectLayout(route, parentPath = '') {
        const routeAbsolutePath = (0, tool_1.pathJoin)(parentPath, route.path);
        if (route.redirect) {
            const redirectAbsolutePath = route.redirect.startsWith('/')
                ? route.redirect
                : (0, tool_1.pathJoin)(routeAbsolutePath, route.redirect);
            route.element = ((0, jsx_runtime_1.jsx)(exports.TransparentLayout, { base: routeAbsolutePath, indexPath: redirectAbsolutePath, children: route.element }));
        }
        if (route.children?.length) {
            route.children = route.children.map((child) => this.generateRedirectLayout(child, routeAbsolutePath));
        }
        if (route.keepAlive) {
            const cacheName = route.key ?? routeAbsolutePath;
            route.element = ((0, jsx_runtime_1.jsx)(react_activation_1.default, { cacheKey: cacheName, name: cacheName, ...route.keepAliveProps, children: route.pageLayout ? (react_1.default.cloneElement(route.pageLayout, {
                    children: route.element
                })) : ((0, jsx_runtime_1.jsx)("div", { className: 'keep-alive-scroll-cache', children: route.element })) }));
        }
        return route;
    }
    onChange(handler) {
        const_1.eventBus.on(SpotterRouter.SPOTTER_ROUTE_CHANGE, handler);
    }
    offChange(handler) {
        const_1.eventBus.off(SpotterRouter.SPOTTER_ROUTE_CHANGE, handler);
    }
    /**
     * 当前路由会直接添加在指定的 Layout 路由下面
     * 举例来说，一级菜单需要添加在 DefaultLayout 下面
     */
    addLayoutRoute(route, layout = GlobalLayoutType.DEFAULT) {
        const { pathRouteMap } = this.layoutMap[layout];
        const routePath = route?.path ?? '';
        // 如果已经存在了路径定义，那么在控制台输出警告，然后用新的路由定义覆盖旧的定义
        pathRouteMap.has(routePath) &&
            console.warn('⚠️存在重复的路由定义，新定义会覆盖旧定义：\n', '已有定义：', pathRouteMap.get(routePath), '新定义:', route);
        pathRouteMap.set(routePath, this.generateRedirectLayout(route));
        const_1.eventBus.emit(SpotterRouter.SPOTTER_ROUTE_CHANGE, this.getRoutes());
    }
    getRoutes() {
        return [
            ...Object.values(GlobalLayoutType).map((layout) => this.layoutMap[layout].getLayoutRoute())
        ];
    }
}
exports.SpotterRouter = SpotterRouter;
SpotterRouter.SPOTTER_ROUTE_CHANGE = 'SPOTTER_ROUTER_CHANGE';
const getUseSpotterRouters = (getSpotterRouter, initialRoutes = []) => () => {
    const [routes, setRoutes] = (0, react_1.useState)([...initialRoutes, ...getSpotterRouter().getRoutes()]);
    (0, react_1.useEffect)(() => {
        const handler = (newRoutes) => {
            setRoutes([...initialRoutes, ...newRoutes]);
        };
        getSpotterRouter().onChange(handler);
        return () => {
            getSpotterRouter().offChange(handler);
        };
    }, []);
    return routes;
};
exports.getUseSpotterRouters = getUseSpotterRouters;
