// @spotter/ui/es/components/SptDrawer/index.js
var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptDrawer/index.tsx
import { Drawer, Spin, Tabs, Typography } from "antd";
import React, {
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState
} from "react";
import { useSearchParams } from "react-router-dom";
import classNames from "classnames";
import SptIcon from "../SptIcon";
import { SptComponentProvider } from "../Provider";
import "./style/index.less";
import qs from "qs";
import updateSearchParams from "../../utils/updateSearchParams";
var { Paragraph } = Typography;
var DRAWER_PERSIST = "drawerPersist";
var inBoundSelectorWhiteList = [".ant-table", ".spotter-layout-header"];
var motion = {
  visible: true
};
var SpotterDrawerBase = (_a) => {
  var _b = _a, {
    children,
    title,
    subTitle,
    topSection,
    tabs,
    activeTabKey,
    defaultTabKey,
    destroyInactiveTabPane,
    action,
    extra,
    loading = false,
    open = false,
    onClose,
    className,
    rootClassName
  } = _b, drawerProps = __objRest(_b, [
    "children",
    "title",
    "subTitle",
    "topSection",
    "tabs",
    "activeTabKey",
    "defaultTabKey",
    "destroyInactiveTabPane",
    "action",
    "extra",
    "loading",
    "open",
    "onClose",
    "className",
    "rootClassName"
  ]);
  var _a2;
  const [activeKey, setActiveKey] = useState((_a2 = activeTabKey != null ? activeTabKey : defaultTabKey) != null ? _a2 : tabs == null ? void 0 : tabs[0]);
  const clickNotInDrawer = useRef(null);
  useEffect(() => {
    const outboundClickHandler = (e) => {
      var _a3;
      if (open) {
        if (clickNotInDrawer.current !== null && !clickNotInDrawer.current) {
          clickNotInDrawer.current = null;
          return;
        }
        if (!((_a3 = document.querySelector("#main-application")) == null ? void 0 : _a3.contains(e.target))) {
          return;
        }
        const inBound = inBoundSelectorWhiteList.some((selector) => {
          const nodeList = document.querySelectorAll(selector);
          return [...nodeList].some((node) => node.contains(e.target));
        });
        if (!inBound) {
          onClose();
        }
      }
    };
    window.addEventListener("click", outboundClickHandler);
    return () => {
      window.removeEventListener("click", outboundClickHandler);
    };
  }, [onClose]);
  useEffect(() => {
    setActiveKey(activeTabKey);
  }, [activeTabKey]);
  return /* @__PURE__ */ React.createElement(
    Drawer,
    __spreadValues({
      width: "calc(75vw - 56px)",
      className: classNames("spotter-drawer-container", className),
      rootClassName: classNames("spotter-drawer", rootClassName),
      open,
      motion,
      closeIcon: null,
      keyboard: false,
      mask: false,
      onClose,
      destroyOnClose: true,
      getContainer: () => {
        var _a3;
        return (_a3 = document.querySelector(".keep-alive-scroll-cache")) != null ? _a3 : document.querySelector("#main-application");
      }
    }, drawerProps),
    /* @__PURE__ */ React.createElement("div", { className: "spotter-drawer-close-icon", onClick: onClose }, /* @__PURE__ */ React.createElement(SptIcon, { size: 24, color: "#949EB9", type: "sptfont-drawerClose" })),
    /* @__PURE__ */ React.createElement(
      "div",
      {
        onClick: () => {
          clickNotInDrawer.current = false;
        },
        className: "spotter-drawer-ref-wrapper"
      },
      /* @__PURE__ */ React.createElement(Spin, { spinning: loading }, topSection, (title || subTitle || action || extra || (tabs == null ? void 0 : tabs.length)) && /* @__PURE__ */ React.createElement(
        "div",
        {
          className: classNames("spotter-drawer-container-title", {
            // 当不存在 children 时
            "with-tab-children": !children
          })
        },
        (title || subTitle || action) && /* @__PURE__ */ React.createElement("div", { className: "spotter-drawer-container-title-row" }, /* @__PURE__ */ React.createElement("div", { className: "spotter-drawer-container-title-row-inner" }, /* @__PURE__ */ React.createElement(
          Paragraph,
          {
            className: "spotter-drawer-container-title-text",
            ellipsis: {
              tooltip: title,
              rows: 3
            }
          },
          title != null ? title : "-"
        ), subTitle ? /* @__PURE__ */ React.createElement("div", { className: "spotter-drawer-container-title-subtext" }, subTitle != null ? subTitle : "-") : null), /* @__PURE__ */ React.createElement("div", { className: "spotter-drawer-container-title-action" }, action)),
        extra,
        (tabs == null ? void 0 : tabs.length) ? /* @__PURE__ */ React.createElement("div", { className: "spotter-drawer-tabs" }, /* @__PURE__ */ React.createElement(
          Tabs,
          {
            size: "small",
            items: tabs,
            activeKey,
            destroyInactiveTabPane,
            onChange: (key) => {
              setActiveKey(key);
            }
          }
        )) : null
      ), /* @__PURE__ */ React.createElement("div", { className: "spotter-drawer-container-content" }, children))
    )
  );
};
function useDrawerOpen(initialData = {}) {
  var _a, _b, _c;
  const [searchParams] = useSearchParams();
  const initialPersist = useRef((_a = initialData.persist) != null ? _a : true);
  const [data, setData] = useState(__spreadValues({
    open: initialPersist.current ? searchParams.has(DRAWER_PERSIST) : (_b = initialData.open) != null ? _b : false
  }, initialPersist.current ? qs.parse(searchParams.toString()) : (_c = initialData.payload) != null ? _c : {}));
  const updateDrawer = useCallback((newOpen, newData) => {
    setData(__spreadValues({ open: newOpen }, newData));
    const newSearchParams = __spreadValues(__spreadValues({}, newData), newOpen ? { [DRAWER_PERSIST]: newOpen.toString() } : {});
    updateSearchParams(newSearchParams);
  }, []);
  return useMemo(
    () => ({
      open: data.open,
      payload: data,
      updateDrawer
    }),
    [data]
  );
}
var getDrawerUrl = (path, payload) => {
  const searchParams = new URLSearchParams();
  for (const key of Object.keys(payload)) {
    searchParams.set(key, payload[key]);
  }
  searchParams.set(DRAWER_PERSIST, "true");
  return `${path}?${searchParams.toString()}`;
};
var SptDrawer = (props) => {
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(SpotterDrawerBase, __spreadValues({}, props)));
};
var SptDrawer_default = SptDrawer;
export {
  DRAWER_PERSIST,
  SptDrawer_default as default,
  getDrawerUrl,
  useDrawerOpen
};
