// @spotter/ui/es/components/SptEntitySelect/components/SelectBase/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/components/SptEntitySelect/components/SelectBase/index.tsx
import SptSearchSelect from "../../../SptSearchSelect";
import {
  ProFormSelect
} from "@ant-design/pro-components";
import React, { useCallback } from "react";
function runPropFunction(prop, item) {
  return typeof prop === "function" ? prop(item) : item[prop];
}
var runFormat = (value, format) => {
  return format ? format(value) : value;
};
var isContainsString = (node, searchString) => {
  if (typeof node === "string") {
    const upperSearchString = searchString.toUpperCase();
    return node.toUpperCase().includes(upperSearchString);
  }
  if (Array.isArray(node)) {
    return node.some((child) => isContainsString(child, searchString));
  }
  if ((node == null ? void 0 : node.props) && node.props.children) {
    return isContainsString(node.props.children, searchString);
  }
  return false;
};
var SelectBase = (_a) => {
  var _b = _a, {
    request: customRequest,
    viewMode = "normal",
    params,
    fieldProps
  } = _b, props = __objRest(_b, [
    "request",
    "viewMode",
    "params",
    "fieldProps"
  ]);
  const customFormat = useCallback(
    (value) => {
      var _a2;
      return (_a2 = runFormat(value, props.format)) == null ? void 0 : _a2.map((item) => ({
        label: runPropFunction(props.labelName, item),
        value: runPropFunction(props.valueName, item)
      }));
    },
    [props.format, customRequest]
  );
  const customProps = __spreadProps(__spreadValues({
    allowClear: true,
    showSearch: true,
    params,
    fieldProps: __spreadProps(__spreadValues({}, fieldProps), {
      // 搜索时，过滤掉前后空字符串
      filterOption: (searchValue, option) => {
        searchValue = searchValue.trim();
        if (!searchValue.trim()) {
          return true;
        }
        return isContainsString(option == null ? void 0 : option.label, searchValue);
      }
    })
  }, props), {
    request: (params2) => __async(void 0, null, function* () {
      try {
        const res = yield customRequest(params2);
        return customFormat(res);
      } catch (error) {
        console.error(error);
        return [];
      }
    })
  });
  const renderedDom = viewMode === "light" ? /* @__PURE__ */ React.createElement(SptSearchSelect, __spreadValues({}, customProps)) : /* @__PURE__ */ React.createElement(ProFormSelect, __spreadValues({}, customProps));
  return renderedDom;
};
var SelectBase_default = SelectBase;
export {
  SelectBase_default as default,
  runFormat,
  runPropFunction
};
