// @spotter/app-client-toolkit@1.0.42-nic.8/lib/infra/tabs/CachedTabs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TabEventClient = exports.refisterCachedTab = exports.cachedTab = exports.matchRoutePath = exports.TAB_EVENT_TYPE = void 0;
const react_router_dom_1 = require("react-router-dom");
const const_1 = require("../../const");
const EVENT_NAME = Symbol('Cache Tab Update Event');
var TAB_EVENT_TYPE;
(function (TAB_EVENT_TYPE) {
    TAB_EVENT_TYPE["CLOSE"] = "close";
    TAB_EVENT_TYPE["CLOSE_OTHER"] = "close-other";
    TAB_EVENT_TYPE["STICKY"] = "sticky";
    TAB_EVENT_TYPE["CANCEL_STICKY"] = "cancel-sticky";
    TAB_EVENT_TYPE["REFRESH"] = "refresh";
    TAB_EVENT_TYPE["UPDATE"] = "update";
})(TAB_EVENT_TYPE = exports.TAB_EVENT_TYPE || (exports.TAB_EVENT_TYPE = {}));
const matchRoutePath = (pattern, path) => {
    return (Array.isArray(pattern) ? pattern : [pattern]).some((p) => {
        return (0, react_router_dom_1.matchPath)(p, path);
    });
};
exports.matchRoutePath = matchRoutePath;
class CachedTab {
    constructor() {
        this.cache = new Map();
        this.routes = new Map();
        this.pathToKeyMap = new Map();
        this.onShowFuncCache = new Map();
        this.onHideFuncCache = new Map();
        this.onCloseFuncCache = new Map();
        this.eventBus = const_1.eventBus;
        this.addBlocker = (key, blocker) => {
            const tab = this.get(key);
            if (tab) {
                if (!tab.blockers) {
                    tab.blockers = [];
                }
                tab.blockers.push(blocker);
            }
        };
        this.removeBlocker = (key, blocker) => {
            const tab = this.get(key);
            if (tab?.blockers) {
                tab.blockers = tab.blockers.filter((b) => b !== blocker);
            }
        };
        this.addOnShowFunc = (key, func) => {
            const cache = this.onShowFuncCache.get(key);
            // 检查func是否已经在缓存中
            if (cache?.includes(func)) {
                return;
            }
            if (cache) {
                cache.push(func);
            }
            else {
                this.onShowFuncCache.set(key, [func]);
            }
        };
        this.addOnHideFunc = (key, func) => {
            const cache = this.onHideFuncCache.get(key);
            // 检查func是否已经在缓存中
            if (cache?.includes(func)) {
                return;
            }
            if (cache) {
                cache.push(func);
            }
            else {
                this.onHideFuncCache.set(key, [func]);
            }
        };
        this.addOnCloseFunc = (key, func) => {
            const cache = this.onCloseFuncCache.get(key);
            // 检查func是否已经在缓存中
            if (cache?.includes(func)) {
                return;
            }
            if (cache) {
                cache.push(func);
            }
            else {
                this.onCloseFuncCache.set(key, [func]);
            }
        };
        this.removeOnShowFunc = (key, func) => {
            const cache = this.onShowFuncCache.get(key);
            if (cache) {
                this.onShowFuncCache.set(key, cache.filter((f) => f !== func));
            }
        };
        this.removeOnHideFunc = (key, func) => {
            const cache = this.onHideFuncCache.get(key);
            if (cache) {
                this.onHideFuncCache.set(key, cache.filter((f) => f !== func));
            }
        };
        this.removeOnCloseFunc = (key, func) => {
            const cache = this.onCloseFuncCache.get(key);
            if (cache) {
                this.onCloseFuncCache.set(key, cache.filter((f) => f !== func));
            }
        };
        /**
         * onShow 事件, 用于 Tab 显示时触发
         * @param key
         */
        this.onTabShow = (key) => {
            const cache = this.onShowFuncCache.get(key);
            if (cache) {
                cache.forEach((func) => func?.());
            }
        };
        this.onTabHide = (key) => {
            const cache = this.onHideFuncCache.get(key);
            if (cache) {
                cache.forEach((func) => func?.());
            }
        };
    }
    emit(data) {
        const_1.eventBus.emit(EVENT_NAME, data);
    }
    on(callback, key = EVENT_NAME) {
        const_1.eventBus.on(key, callback);
    }
    off(callback, key = EVENT_NAME) {
        const_1.eventBus.off(key, callback);
    }
    get(key) {
        return this.cache.get(key);
    }
    set(key, tab) {
        this.cache.set(key, tab);
    }
    setRoutes(routes) {
        routes.forEach((route) => {
            const key = route.key || route.path;
            this.routes.set(key, route);
            const patterns = Array.isArray(route.path) ? route.path : [route.path];
            patterns.forEach(pattern => {
                this.pathToKeyMap.set(pattern, key);
            });
        });
    }
    delete(key) {
        const _this = this;
        const tab = this.cache.get(key);
        return new Promise((resolve, reject) => {
            try {
                if (tab?.blockers && tab.blockers.length > 0) {
                    Promise.all(tab.blockers.map((blocker) => blocker())).then((results) => {
                        if (results.every((r) => r)) {
                            _this.cache.delete(key);
                            resolve(true);
                        }
                        else {
                            resolve(false);
                        }
                    });
                }
                else {
                    resolve(true);
                }
            }
            catch (error) {
                reject(error);
            }
        });
    }
    clear() {
        this.cache.clear();
    }
    getAll() {
        return this.cache;
    }
    has(key) {
        return this.cache.has(key);
    }
    getKeyByPath(path) {
        if (this.pathToKeyMap.has(path)) {
            return this.pathToKeyMap.get(path);
        }
        for (const [pattern, key] of this.pathToKeyMap) {
            if ((0, react_router_dom_1.matchPath)(pattern, path)) {
                return key;
            }
        }
        return undefined;
    }
    /**
     * 关闭 Tab
     * @param pathname 待关闭的 Tab 的 pathname, <Location.pathname>
     */
    async close(key, autoRedirect = false) {
        const deleteTab = this.get(key);
        const isDelete = await this.delete(key);
        if (isDelete) {
            this.emit({
                type: TAB_EVENT_TYPE.CLOSE,
                data: {
                    key,
                    tab: deleteTab,
                    autoRedirect,
                }
            });
        }
    }
    /**
     * 通过 pathname 关闭 Tab
     * @param pathname
     */
    closeByPathname(pathname, autoRedirect = false) {
        const key = this.getKeyByPath(pathname);
        if (key) {
            this.close(key, autoRedirect);
        }
    }
    /**
     * 关闭当前 Tab
     */
    closeCurrent(autoRedirect = false) {
        const pathname = window.location.pathname;
        this.closeByPathname(pathname, autoRedirect);
    }
    closeOther(key) {
        const tab = this.get(key);
        this.emit({
            type: TAB_EVENT_TYPE.CLOSE_OTHER,
            data: {
                key,
                tab,
            }
        });
    }
    /**
     * 刷新标签页面内容
     * @param pathname 待刷新的 Tab 的 pathname, <Location.pathname>
     */
    refresh(key, force = false) {
        const tab = this.get(key);
        this.emit({
            type: TAB_EVENT_TYPE.REFRESH,
            data: {
                key,
                tab,
                force,
            }
        });
    }
    /**
     * 更新标签内容，比如标签名称，标签跳转地址，标签是否可关闭等
     * @param tab
     */
    update(key, tab) {
        this.emit({ type: TAB_EVENT_TYPE.UPDATE, data: {
                key,
                tab,
            } });
    }
    /**
     * 固定标签页
     * @param pathname
     */
    sticky(key) {
        const tab = this.get(key);
        this.emit({
            type: TAB_EVENT_TYPE.STICKY,
            data: {
                key,
                tab,
            }
        });
    }
    unSticky(key) {
        const tab = this.get(key);
        this.emit({
            type: TAB_EVENT_TYPE.CANCEL_STICKY,
            data: {
                key,
                tab,
            }
        });
    }
}
exports.cachedTab = new CachedTab();
const refisterCachedTab = () => {
    window.TabEvent = exports.cachedTab;
};
exports.refisterCachedTab = refisterCachedTab;
exports.TabEventClient = (window.TabEvent ?? exports.cachedTab);
