// @spotter/ui/es/components/SptSearchCascader/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptSearchCascader/index.tsx
import { ProFormCascader } from "@ant-design/pro-components";
import React, { useEffect, useState } from "react";
import { nanoid } from "nanoid";
import { SptComponentProvider } from "../Provider";
import classNames from "classnames";
import "./style/index.less";
var SptSearchCascader = (_a) => {
  var _b = _a, {
    label,
    className,
    fieldProps
  } = _b, proFormSelectProps = __objRest(_b, [
    "label",
    "className",
    "fieldProps"
  ]);
  const baseClassName = "spt-search-cascader-with-inner-label";
  const [select, setSelect] = useState([]);
  const [searchText, setSearchText] = useState("");
  const id = `${baseClassName}-${nanoid(8).toLocaleLowerCase()}`;
  const onChangeGuard = (_value, _label) => {
    setSelect(_value);
    (fieldProps == null ? void 0 : fieldProps.onChange) && fieldProps.onChange(_value, _label);
  };
  const onSearchGuard = (value) => {
    value !== searchText && setSearchText(value);
    (fieldProps == null ? void 0 : fieldProps.onSearch) && fieldProps.onSearch(value);
  };
  const onBlurGuard = (e) => {
    setSearchText(searchText === "" ? "-" : "");
    (fieldProps == null ? void 0 : fieldProps.onBlur) && fieldProps.onBlur(e);
  };
  useEffect(() => {
    var _a2, _b2, _c;
    (_a2 = document.querySelector(`.${id} .ant-select-selector`)) == null ? void 0 : _a2.setAttribute("data-spt-search-cascader-inner-label", label);
    (_b2 = document.querySelector(`.${id} .ant-select-selector .ant-select-selection-search`)) == null ? void 0 : _b2.setAttribute("data-spt-search-cascader-inner-label", label);
    if ((fieldProps == null ? void 0 : fieldProps.multiple) && select.length === 0) {
      (_c = document.querySelector(`.${id} .ant-select-selector .ant-select-selection-placeholder`)) == null ? void 0 : _c.setAttribute("data-spt-search-select-inner-label", label);
    }
  });
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(
    ProFormCascader,
    __spreadValues({
      noStyle: true,
      colon: false,
      label,
      fieldProps: __spreadProps(__spreadValues({
        className: classNames(
          baseClassName,
          className,
          id,
          (fieldProps == null ? void 0 : fieldProps.multiple) ? "multiple" : ""
        )
      }, fieldProps), {
        onChange: onChangeGuard,
        onSearch: onSearchGuard,
        onBlur: onBlurGuard
      })
    }, proFormSelectProps)
  ));
};
var SptSearchCascader_default = SptSearchCascader;
export {
  SptSearchCascader_default as default
};
