// @spotter/app-client-toolkit/lib/infra/mainApp/logger/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.boostrapLogSystem = exports.Logger = void 0;
__exportStar(require("./consoleProxy"), exports);
class Logger {
    constructor(prefix) {
        this.prefix = '';
        this.log = console.log.bind(console, this.prefix);
        this.info = console.log.bind(console, this.prefix);
        this.warn = console.warn.bind(console, this.prefix);
        this.error = console.error.bind(console, this.prefix);
        this.prefix = `[${prefix}]`;
    }
}
exports.Logger = Logger;
function boostrapLogSystem(options, sptApp) {
    sptApp.log = {};
    sptApp.log.logger = options instanceof Logger ? options : new Logger(options.prefix);
}
exports.boostrapLogSystem = boostrapLogSystem;
