// @spotter/ui/lib/components/SptPageContainer/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SptPageContainerProvider = exports.SptPageContainerContext = void 0;
var _antd = require("antd");
var _reactRouterDom = require("react-router-dom");
var _react = _interopRequireWildcard(require("react"));
var _appClientToolkit = require("@spotter/app-client-toolkit");
var _classnames = _interopRequireDefault(require("classnames"));
var _SptTabWithBadge = _interopRequireWildcard(require("../SptTabWithBadge"));
require("./style/index.less");
var _SptFooterToolbar = _interopRequireDefault(require("../SptFooterToolbar"));
var _Provider = require("../Provider");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure " + obj); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var SptPageContainerContext = exports.SptPageContainerContext = /*#__PURE__*/_react.default.createContext(null);
var SptPageContainerProvider = exports.SptPageContainerProvider = SptPageContainerContext.Provider;
var Text = _antd.Typography.Text;
var SptPageContainerBase = function SptPageContainerBase(_ref) {
  var _innerTabs$0$key, _innerTabs$;
  var title = _ref.title,
    subTitle = _ref.subTitle,
    _ref$tabs = _ref.tabs,
    tabs = _ref$tabs === void 0 ? [] : _ref$tabs,
    _ref$tabBadgeType = _ref.tabBadgeType,
    tabBadgeType = _ref$tabBadgeType === void 0 ? _SptTabWithBadge.SptTabBadgeType.Corner : _ref$tabBadgeType,
    _ref$basePath = _ref.basePath,
    basePath = _ref$basePath === void 0 ? '' : _ref$basePath,
    topSection = _ref.topSection,
    _ref$children = _ref.children,
    children = _ref$children === void 0 ? /*#__PURE__*/_react.default.createElement(_reactRouterDom.Outlet, null) : _ref$children,
    extra = _ref.extra,
    action = _ref.action,
    footerToolbar = _ref.footerToolbar,
    _ref$loading = _ref.loading,
    loading = _ref$loading === void 0 ? false : _ref$loading,
    _ref$tabNavigationMod = _ref.tabNavigationMode,
    tabNavigationMode = _ref$tabNavigationMod === void 0 ? true : _ref$tabNavigationMod,
    _ref$autoDefaultTabKe = _ref.autoDefaultTabKey,
    autoDefaultTabKey = _ref$autoDefaultTabKe === void 0 ? true : _ref$autoDefaultTabKe,
    _ref$fitContainer = _ref.fitContainer,
    fitContainer = _ref$fitContainer === void 0 ? false : _ref$fitContainer,
    tableAction = _ref.tableAction,
    onTabChange = _ref.onTabChange;
  var innerTabs = (0, _react.useMemo)(function () {
    return tabs === null || tabs === void 0 ? void 0 : tabs.map(function (p) {
      return {
        // ...p,
        label: p.tab,
        key: tabNavigationMode ? (0, _appClientToolkit.pathJoin)(basePath, p.tabKey) : p.tabKey,
        count: p === null || p === void 0 ? void 0 : p.count
      };
    });
  }, [basePath, tabs, tabNavigationMode]);
  var location = (0, _reactRouterDom.useLocation)();
  var navigate = (0, _reactRouterDom.useNavigate)();
  var _useState = (0, _react.useState)(tabNavigationMode ? location.pathname : (_innerTabs$0$key = (_innerTabs$ = innerTabs[0]) === null || _innerTabs$ === void 0 ? void 0 : _innerTabs$.key) !== null && _innerTabs$0$key !== void 0 ? _innerTabs$0$key : undefined),
    _useState2 = _slicedToArray(_useState, 2),
    activeKey = _useState2[0],
    setActiveKey = _useState2[1];
  (0, _react.useEffect)(function () {
    console.log('刷新。。。。');
    if (tabNavigationMode && location.pathname !== (0, _appClientToolkit.pathJoin)('/', basePath)) {
      setActiveKey(location.pathname);
    } else {
      autoDefaultTabKey && innerTabs.length && setActiveKey(innerTabs[0].key);
    }
  }, [location.pathname, tabNavigationMode, innerTabs]);
  (0, _react.useEffect)(function () {
    onTabChange === null || onTabChange === void 0 || onTabChange(activeKey);
  }, [activeKey]);
  var ref = (0, _react.useRef)(null);
  var tableProvideAction = (0, _react.useMemo)(function () {
    return {
      tableAction: tableAction
    };
  }, [activeKey, innerTabs, tabNavigationMode, basePath]);
  return /*#__PURE__*/_react.default.createElement(SptPageContainerProvider, {
    value: tableProvideAction
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)({
      'with-tabs': tabs === null || tabs === void 0 ? void 0 : tabs.length,
      'fit-container': fitContainer,
      'tab-navigation-mode': tabNavigationMode,
      'spotter-page-container-fit-pagination': footerToolbar
    }, 'spotter-page-container'),
    ref: ref
  }, topSection, (title || subTitle || action || extra) && /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-page-container-title"
  }, (title || subTitle || action) && /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-page-container-title-inner"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-page-container-title-inner-wrap"
  }, /*#__PURE__*/_react.default.createElement(Text, {
    style: {
      verticalAlign: 'middle',
      overflow: 'hidden'
    },
    className: "spotter-page-container-title-main",
    ellipsis: {
      tooltip: title
    }
  }, title), subTitle ? /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-page-container-title-sub"
  }, subTitle) : null), /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-page-container-title-action"
  }, action)), extra), innerTabs && innerTabs !== null && innerTabs !== void 0 && innerTabs.length ? /*#__PURE__*/_react.default.createElement(_SptTabWithBadge.default, {
    className: "spotter-page-container-tabs",
    activeKey: activeKey,
    onChange: function onChange(key) {
      setActiveKey(key);
      tabNavigationMode && navigate(key);
    },
    items: innerTabs.map(function (item, i) {
      return {
        label: tabs[i].tab,
        key: tabNavigationMode ? item.key : tabs[i].tabKey,
        count: item === null || item === void 0 ? void 0 : item.count
      };
    }),
    badgeType: tabBadgeType,
    tabBarGutter: 24
  }) : null, /*#__PURE__*/_react.default.createElement(_antd.Spin, {
    style: {
      height: ''
    },
    spinning: loading
  }, children), footerToolbar ? /*#__PURE__*/_react.default.createElement(_SptFooterToolbar.default, null, " ", footerToolbar, " ") : null));
};
var SptPageContainer = function SptPageContainer(_ref2) {
  var props = Object.assign({}, (_objectDestructuringEmpty(_ref2), _ref2));
  return /*#__PURE__*/_react.default.createElement(_Provider.SptComponentProvider, null, /*#__PURE__*/_react.default.createElement(SptPageContainerBase, props));
};
var _default = exports.default = SptPageContainer;
