// @spotter/ui/lib/components/SptDrawer/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getDrawerUrl = exports.default = exports.DRAWER_PERSIST = void 0;
exports.useDrawerOpen = useDrawerOpen;
var _antd = require("antd");
var _react = _interopRequireWildcard(require("react"));
var _reactRouterDom = require("react-router-dom");
var _classnames = _interopRequireDefault(require("classnames"));
var _SptIcon = _interopRequireDefault(require("../SptIcon"));
var _Provider = require("../Provider");
require("./style/index.less");
var _qs = _interopRequireDefault(require("qs"));
var _updateSearchParams = _interopRequireDefault(require("../../utils/updateSearchParams"));
var _excluded = ["children", "title", "subTitle", "topSection", "tabs", "activeTabKey", "defaultTabKey", "destroyInactiveTabPane", "action", "extra", "loading", "open", "onClose", "className", "rootClassName"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }
function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var Paragraph = _antd.Typography.Paragraph;
var DRAWER_PERSIST = exports.DRAWER_PERSIST = 'drawerPersist';
// 出界点击的选择器白名单，匹配到的选择器都不会在点击事件发生时关闭抽屉
var inBoundSelectorWhiteList = ['.ant-table', '.spotter-layout-header'];
var motion = {
  visible: true
};
var SpotterDrawerBase = function SpotterDrawerBase(_ref) {
  var _ref2;
  var children = _ref.children,
    title = _ref.title,
    subTitle = _ref.subTitle,
    topSection = _ref.topSection,
    tabs = _ref.tabs,
    activeTabKey = _ref.activeTabKey,
    defaultTabKey = _ref.defaultTabKey,
    destroyInactiveTabPane = _ref.destroyInactiveTabPane,
    action = _ref.action,
    extra = _ref.extra,
    _ref$loading = _ref.loading,
    loading = _ref$loading === void 0 ? false : _ref$loading,
    _ref$open = _ref.open,
    open = _ref$open === void 0 ? false : _ref$open,
    onClose = _ref.onClose,
    className = _ref.className,
    rootClassName = _ref.rootClassName,
    drawerProps = _objectWithoutProperties(_ref, _excluded);
  var _useState = (0, _react.useState)((_ref2 = activeTabKey !== null && activeTabKey !== void 0 ? activeTabKey : defaultTabKey) !== null && _ref2 !== void 0 ? _ref2 : tabs === null || tabs === void 0 ? void 0 : tabs[0]),
    _useState2 = _slicedToArray(_useState, 2),
    activeKey = _useState2[0],
    setActiveKey = _useState2[1];
  var clickNotInDrawer = (0, _react.useRef)(null);
  (0, _react.useEffect)(function () {
    var outboundClickHandler = function outboundClickHandler(e) {
      if (open) {
        var _document$querySelect;
        // 如果点击区域在 drawer 内部，则直接 skip
        if (clickNotInDrawer.current !== null && !clickNotInDrawer.current) {
          clickNotInDrawer.current = null;
          return;
        }
        // 如果点击区域不在应用 dom 下面，说明是动态插入的浮层 dom，则直接 skip
        if (!((_document$querySelect = document.querySelector('#main-application')) !== null && _document$querySelect !== void 0 && _document$querySelect.contains(e.target))) {
          return;
        }

        // 如果点击 drawer 外部的区域在白名单里，则 skip，否则执行关闭 drawer
        var inBound = inBoundSelectorWhiteList.some(function (selector) {
          var nodeList = document.querySelectorAll(selector);
          return _toConsumableArray(nodeList).some(function (node) {
            return node.contains(e.target);
          });
        });
        if (!inBound) {
          onClose();
        }
      }
    };
    window.addEventListener('click', outboundClickHandler);
    return function () {
      window.removeEventListener('click', outboundClickHandler);
    };
  }, [onClose]);
  (0, _react.useEffect)(function () {
    setActiveKey(activeTabKey);
  }, [activeTabKey]);
  return /*#__PURE__*/_react.default.createElement(_antd.Drawer, _extends({
    width: "calc(75vw - 56px)",
    className: (0, _classnames.default)('spotter-drawer-container', className),
    rootClassName: (0, _classnames.default)('spotter-drawer', rootClassName),
    open: open,
    motion: motion,
    closeIcon: null
    // 只在 focus 到抽屉本体时有效，疑似 bug
    ,
    keyboard: false,
    mask: false,
    onClose: onClose,
    destroyOnClose: true
    // getContainer="#main-application"
    ,
    getContainer: function getContainer() {
      var _document$querySelect2;
      return (_document$querySelect2 = document.querySelector('.keep-alive-scroll-cache')) !== null && _document$querySelect2 !== void 0 ? _document$querySelect2 : document.querySelector('#main-application');
    }
  }, drawerProps), /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-drawer-close-icon",
    onClick: onClose
  }, /*#__PURE__*/_react.default.createElement(_SptIcon.default, {
    size: 24,
    color: "#949EB9",
    type: "sptfont-drawerClose"
  })), /*#__PURE__*/_react.default.createElement("div", {
    onClick: function onClick() {
      clickNotInDrawer.current = false;
    },
    className: "spotter-drawer-ref-wrapper"
  }, /*#__PURE__*/_react.default.createElement(_antd.Spin, {
    spinning: loading
  }, topSection, (title || subTitle || action || extra || (tabs === null || tabs === void 0 ? void 0 : tabs.length)) && /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)('spotter-drawer-container-title', {
      // 当不存在 children 时
      'with-tab-children': !children
    })
  }, (title || subTitle || action) && /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-drawer-container-title-row"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-drawer-container-title-row-inner"
  }, /*#__PURE__*/_react.default.createElement(Paragraph, {
    className: "spotter-drawer-container-title-text",
    ellipsis: {
      tooltip: title,
      rows: 3
    }
  }, title !== null && title !== void 0 ? title : '-'), subTitle ? /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-drawer-container-title-subtext"
  }, subTitle !== null && subTitle !== void 0 ? subTitle : '-') : null), /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-drawer-container-title-action"
  }, action)), extra, tabs !== null && tabs !== void 0 && tabs.length ? /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-drawer-tabs"
  }, /*#__PURE__*/_react.default.createElement(_antd.Tabs, {
    size: "small",
    items: tabs,
    activeKey: activeKey,
    destroyInactiveTabPane: destroyInactiveTabPane,
    onChange: function onChange(key) {
      setActiveKey(key);
    }
  })) : null), /*#__PURE__*/_react.default.createElement("div", {
    className: "spotter-drawer-container-content"
  }, children))));
};
function useDrawerOpen() {
  var _initialData$persist, _initialData$open, _initialData$payload;
  var initialData = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _useSearchParams = (0, _reactRouterDom.useSearchParams)(),
    _useSearchParams2 = _slicedToArray(_useSearchParams, 1),
    searchParams = _useSearchParams2[0];
  var initialPersist = (0, _react.useRef)((_initialData$persist = initialData.persist) !== null && _initialData$persist !== void 0 ? _initialData$persist : true);
  var _useState3 = (0, _react.useState)(_objectSpread({
      open: initialPersist.current ? searchParams.has(DRAWER_PERSIST) : (_initialData$open = initialData.open) !== null && _initialData$open !== void 0 ? _initialData$open : false
    }, initialPersist.current ? _qs.default.parse(searchParams.toString()) : (_initialData$payload = initialData.payload) !== null && _initialData$payload !== void 0 ? _initialData$payload : {})),
    _useState4 = _slicedToArray(_useState3, 2),
    data = _useState4[0],
    setData = _useState4[1];
  var updateDrawer = (0, _react.useCallback)(function (newOpen, newData) {
    // 在传入 persistKey 的前提下进入 open 状态时，同步 persistKey 到 searchParams
    setData(_objectSpread({
      open: newOpen
    }, newData));
    var newSearchParams = _objectSpread(_objectSpread({}, newData), newOpen ? _defineProperty({}, DRAWER_PERSIST, newOpen.toString()) : {});
    (0, _updateSearchParams.default)(newSearchParams);
  }, []);
  return (0, _react.useMemo)(function () {
    return {
      open: data.open,
      payload: data,
      updateDrawer: updateDrawer
    };
  }, [data]);
}
var getDrawerUrl = exports.getDrawerUrl = function getDrawerUrl(path, payload) {
  var searchParams = new URLSearchParams();
  for (var _i = 0, _Object$keys = Object.keys(payload); _i < _Object$keys.length; _i++) {
    var key = _Object$keys[_i];
    searchParams.set(key, payload[key]);
  }
  // 自动添加弹窗持久化参数，跳转后自动打开
  searchParams.set(DRAWER_PERSIST, 'true');
  return "".concat(path, "?").concat(searchParams.toString());
};

/**
 * 详情抽屉组件
 * @param props
 * @returns
 */
var SptDrawer = function SptDrawer(props) {
  return /*#__PURE__*/_react.default.createElement(_Provider.SptComponentProvider, null, /*#__PURE__*/_react.default.createElement(SpotterDrawerBase, props));
};
var _default = exports.default = SptDrawer;
