// @spotter/ui/lib/components/SptUserAvatar/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _antd = require("antd");
var _react = _interopRequireDefault(require("react"));
var _icons = require("@ant-design/icons");
var _Provider = require("../Provider");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure " + obj); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
var UserAvatarBase = function UserAvatarBase(_ref) {
  var _name$length;
  var _ref$maxNameLength = _ref.maxNameLength,
    maxNameLength = _ref$maxNameLength === void 0 ? 1 : _ref$maxNameLength,
    _ref$size = _ref.size,
    size = _ref$size === void 0 ? 32 : _ref$size,
    fontSize = _ref.fontSize,
    name = _ref.name,
    loading = _ref.loading,
    style = _ref.style;
  return /*#__PURE__*/_react.default.createElement(_antd.Avatar, {
    className: "bg-primary-500",
    style: _objectSpread({
      display: 'flex',
      alignItems: 'center',
      backgroundColor: 'var(--color-primary)',
      width: typeof size === 'number' ? size : size.width,
      height: typeof size === 'number' ? size : size.height,
      fontSize: fontSize !== null && fontSize !== void 0 ? fontSize : Math.max(14, (typeof size === 'number' ? size : size.height) / 3)
    }, style),
    gap: 16
  }, loading ? /*#__PURE__*/_react.default.createElement(_antd.Spin, {
    indicator: /*#__PURE__*/_react.default.createElement(_icons.LoadingOutlined, null),
    spinning: true,
    style: {
      color: '#fff',
      display: 'flex',
      alignItems: 'center'
    },
    size: "small"
  }) : name === null || name === void 0 ? void 0 : name.slice(0, Math.min((_name$length = name === null || name === void 0 ? void 0 : name.length) !== null && _name$length !== void 0 ? _name$length : Number.MAX_SAFE_INTEGER, maxNameLength)));
};
var SptUserAvatar = function SptUserAvatar(_ref2) {
  var props = Object.assign({}, (_objectDestructuringEmpty(_ref2), _ref2));
  return /*#__PURE__*/_react.default.createElement(_Provider.SptComponentProvider, null, /*#__PURE__*/_react.default.createElement(UserAvatarBase, props));
};
var _default = exports.default = SptUserAvatar;
