// @spotter/ui/lib/components/Downgrade/NotFound.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NotFound = void 0;
var _react = _interopRequireDefault(require("react"));
var _antd = require("antd");
var _reactRouterDom = require("react-router-dom");
var _ = _interopRequireDefault(require("./img/404.svg"));
var _i18nSdk = require("@spotter/i18n-sdk");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/**
 * 应用无法匹配到浏览器当前路由时降级页面
 * @param param1
 * @returns
 */
var NotFound = exports.NotFound = function NotFound() {
  var navigate = (0, _reactRouterDom.useNavigate)();
  var _useSptTranslation = (0, _i18nSdk.useSptTranslation)(),
    t = _useSptTranslation.t;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "h-full p-24px"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "h-full bg-white"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-auto pt-100px text-center"
  }, /*#__PURE__*/_react.default.createElement("img", {
    className: "mx-auto",
    src: _.default,
    alt: "Not Found"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12px text-16px font-bold leading-24px"
  }, t('infra.downgrade_not_found_title')), /*#__PURE__*/_react.default.createElement("div", {
    className: "leading-24px text-gray-400"
  }, t('infra.downgrade_not_found_help')), /*#__PURE__*/_react.default.createElement(_antd.Button, {
    className: "mt-24px",
    type: "primary",
    onClick: function onClick() {
      navigate('/');
    }
  }, t('infra.back_home')))));
};
