// @spotter/ui/lib/components/Downgrade/ServerError.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ServerError = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactRouterDom = require("react-router-dom");
var _antd = require("antd");
var _ = _interopRequireDefault(require("./img/500.svg"));
var _i18nSdk = require("@spotter/i18n-sdk");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var ServerError = exports.ServerError = function ServerError() {
  var navigate = (0, _reactRouterDom.useNavigate)();
  var _useSptTranslation = (0, _i18nSdk.useSptTranslation)(),
    t = _useSptTranslation.t;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "h-full p-24px"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "h-full bg-white"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "mx-auto pt-100px text-center"
  }, /*#__PURE__*/_react.default.createElement("img", {
    className: "mx-auto",
    src: _.default,
    alt: "Server Error"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "mb-12px text-16px font-bold leading-24px"
  }, t('infra.downgrade_server_error_title')), /*#__PURE__*/_react.default.createElement("div", {
    className: "leading-24px text-gray-400"
  }, t('infra.downgrade_server_error_help')), /*#__PURE__*/_react.default.createElement(_antd.Button, {
    className: "mt-24px",
    type: "primary",
    onClick: function onClick() {
      navigate('/');
    }
  }, t('infra.back_home')))));
};
