// @spotter/ui/es/components/SptTable/Form/Field.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};

// src/components/SptTable/Form/Field.tsx
import omit from "omit.js";
import {
  ProFormField,
  ProProvider,
  isDeepEqualReact
} from "@ant-design/pro-components";
import React, { memo, useContext } from "react";
import { runFunction } from "../../../utils/runFunction";
import SptSearchSelect from "../../SptSearchSelect";
import SptSearchText from "../../SptSearchText";
import { SptSearchDatePicker } from "../../SptSearchDatePicker";
import { SptSearchDateRangePicker } from "../../SptSearchDateRangePicker";
import { SptSearchTreeSelect } from "../../SptSearchTreeSelect";
import { omitUndefined } from "../../../utils/omitUndefined";
import SptEntitySelect from "../../SptEntitySelect";
import SptSearchCascader from "../../SptSearchCascader";
import { SptSearchDateTimeRangePicker } from "../../SptSearchDateTimeRangePicker";
import { Form } from "antd";
var blackList = [
  "company",
  "companyCode",
  "storage",
  "svcCode",
  "svcAccount",
  "vendorCode",
  "sptDateDay",
  "sptDateMonth"
];
var SptField = memo(
  ({ item, helpers }) => {
    var _a, _b, _c, _d, _e, _f;
    const context = useContext(ProProvider);
    const { type, action, originItem, formRef } = helpers;
    const formItemProps = (_a = item == null ? void 0 : item.getFormItemProps) == null ? void 0 : _a.call(item);
    const formFieldProps = __spreadProps(__spreadValues({}, omit(item, [
      "dataIndex",
      "width",
      "render",
      "renderFormItem",
      "renderText",
      "title",
      "valueType",
      "label"
    ])), {
      label: (formItemProps == null ? void 0 : formItemProps.label) || item.label,
      name: ((_b = item.formItemProps) == null ? void 0 : _b.name) || item.name || item.key || item.dataIndex,
      width: item.width,
      initialValue: item.initialValue || ((_c = item.formItemProps) == null ? void 0 : _c.initialValue) || (formItemProps == null ? void 0 : formItemProps.initialValue),
      render: (item == null ? void 0 : item.render) ? (dom, entity, renderIndex) => {
        var _a2, _b2, _c2, _d2;
        return (_d2 = item == null ? void 0 : item.render) == null ? void 0 : _d2.call(item, dom, entity, renderIndex, action == null ? void 0 : action.current, __spreadProps(__spreadValues({
          type
        }, item), {
          key: (_a2 = item.key) == null ? void 0 : _a2.toString(),
          formItemProps: (_b2 = item.getFormItemProps) == null ? void 0 : _b2.call(item),
          fieldProps: (_c2 = item.getFieldProps) == null ? void 0 : _c2.call(item)
        }));
      } : void 0,
      valueEnum: runFunction(item.valueEnum),
      fieldProps: (_d = item == null ? void 0 : item.getFieldProps) == null ? void 0 : _d.call(item),
      formItemProps
    });
    const defaultRender = () => {
      return /* @__PURE__ */ React.createElement(ProFormField, __spreadProps(__spreadValues({}, formFieldProps), { ignoreFormItem: true }));
    };
    const renderFormItem = (item == null ? void 0 : item.renderFormItem) ? (_, config) => {
      var _a2, _b2, _c2, _d2;
      const renderConfig = omitUndefined(__spreadProps(__spreadValues({}, config), { onChange: void 0 }));
      return (_d2 = item == null ? void 0 : item.renderFormItem) == null ? void 0 : _d2.call(
        item,
        __spreadProps(__spreadValues({
          type
        }, item), {
          key: (_a2 = item.key) == null ? void 0 : _a2.toString(),
          formItemProps: (_b2 = item.getFormItemProps) == null ? void 0 : _b2.call(item),
          fieldProps: (_c2 = item.getFieldProps) == null ? void 0 : _c2.call(item),
          originProps: originItem
        }),
        __spreadProps(__spreadValues({}, renderConfig), {
          defaultRender,
          type
        }),
        formRef.current
      );
    } : void 0;
    const renderedDom = () => {
      var _a2, _b2, _c2, _d2, _e2, _f2, _g, _h;
      if (item == null ? void 0 : item.renderFormItem) {
        const dom = renderFormItem == null ? void 0 : renderFormItem(null, {});
        if (!dom || item.ignoreFormItem)
          return dom;
      }
      const text = (_a2 = item == null ? void 0 : item.fieldProps) == null ? void 0 : _a2["value"];
      const dataValue = (_d2 = (_c2 = (_b2 = item == null ? void 0 : item.fieldProps) == null ? void 0 : _b2.value) != null ? _c2 : text) != null ? _d2 : "";
      const valueTypeMap = context.valueTypeMap || {};
      const customValueTypeConfig = valueTypeMap && valueTypeMap[item.valueType];
      if (customValueTypeConfig && !(blackList == null ? void 0 : blackList.includes(item.valueType))) {
        const props = __spreadValues(__spreadValues(__spreadValues({
          label: item.label,
          tooltip: (_e2 = item.proFieldProps) == null ? void 0 : _e2.tooltip,
          transform: item.transform,
          request: item.request,
          params: item.params
        }, item.fieldProps), item.proFieldProps), formFieldProps);
        return (_f2 = customValueTypeConfig == null ? void 0 : customValueTypeConfig.renderFormItem) == null ? void 0 : _f2.call(
          customValueTypeConfig,
          dataValue,
          __spreadProps(__spreadValues({
            text: dataValue
          }, props), {
            valueEnum: runFunction(item.valueEnum),
            viewMode: "light",
            // 直接给自定义渲染一个viewMode light的参数
            mode: "edit"
          }),
          /* @__PURE__ */ React.createElement(React.Fragment, null, dataValue)
        );
      }
      if (renderFormItem) {
        const newDom = renderFormItem == null ? void 0 : renderFormItem(text, __spreadValues({
          mode: "edit"
        }, (_g = formFieldProps == null ? void 0 : formFieldProps.fieldProps) != null ? _g : {}));
        if (React.isValidElement(newDom))
          return React.cloneElement(newDom, __spreadValues(__spreadValues({
            viewMode: "light"
          }, formFieldProps), newDom.props || {}));
        return newDom;
      }
      if (item.valueType === "select" || item.valueType === "text" && (formFieldProps.valueEnum || item.request)) {
        return /* @__PURE__ */ React.createElement(
          SptSearchSelect,
          __spreadValues(__spreadProps(__spreadValues(__spreadValues({
            noStyle: true,
            label: item.label,
            tooltip: (_h = item.proFieldProps) == null ? void 0 : _h.tooltip,
            transform: item.transform,
            request: item.request,
            params: item.params
          }, item.fieldProps), item.proFieldProps), {
            fieldProps: __spreadValues({}, item.fieldProps)
          }), formFieldProps)
        );
      }
      if (item.valueType === "date") {
        return /* @__PURE__ */ React.createElement(SptSearchDatePicker, __spreadValues({ noStyle: true }, formFieldProps));
      }
      if (["dateRange", "sptDateDay"].includes(item.valueType)) {
        return /* @__PURE__ */ React.createElement(SptSearchDateRangePicker, __spreadValues({ noStyle: true }, formFieldProps));
      }
      if (item.valueType === "dateTimeRange") {
        return /* @__PURE__ */ React.createElement(SptSearchDateTimeRangePicker, __spreadValues({ noStyle: true }, formFieldProps));
      }
      if (["dateMonthRange", "sptDateMonth"].includes(item.valueType)) {
        const _i = formFieldProps, { fieldProps } = _i, otherFormFieldProps = __objRest(_i, ["fieldProps"]);
        return /* @__PURE__ */ React.createElement(
          SptSearchDateRangePicker,
          __spreadValues({
            noStyle: true,
            dataFormat: "YYYY-MM",
            fieldProps: __spreadValues({
              picker: "month",
              format: "YYYY-MM"
            }, fieldProps)
          }, otherFormFieldProps)
        );
      }
      if (item.valueType === "treeSelect") {
        return /* @__PURE__ */ React.createElement(SptSearchTreeSelect, __spreadValues({ noStyle: true }, formFieldProps));
      }
      if (item.valueType === "company") {
        return /* @__PURE__ */ React.createElement(
          SptEntitySelect.Company,
          __spreadValues(__spreadValues({
            viewMode: "light",
            noStyle: true
          }, formFieldProps.fieldProps), formFieldProps)
        );
      }
      if (item.valueType === "svcCode" || item.valueType === "svcAccount") {
        return /* @__PURE__ */ React.createElement(
          SptEntitySelect.SvcAccount,
          __spreadValues(__spreadValues({
            viewMode: "light",
            noStyle: true
          }, formFieldProps.fieldProps), formFieldProps)
        );
      }
      if (item.valueType === "store" || item.valueType === "storage") {
        return /* @__PURE__ */ React.createElement(
          SptEntitySelect.Store,
          __spreadValues(__spreadProps(__spreadValues({
            viewMode: "light"
          }, formFieldProps.fieldProps), {
            noStyle: true
          }), formFieldProps)
        );
      }
      if (item.valueType === "cascader") {
        return /* @__PURE__ */ React.createElement(SptSearchCascader, __spreadValues({ noStyle: true }, formFieldProps));
      }
      if (item.valueType === "companyCode") {
        return /* @__PURE__ */ React.createElement(
          SptEntitySelect.Company,
          __spreadValues(__spreadValues({
            noStyle: true,
            viewMode: "light",
            valueName: "businessCode"
          }, formFieldProps.fieldProps), formFieldProps)
        );
      }
      if (item.valueType === "vendorCode") {
        return /* @__PURE__ */ React.createElement(
          SptEntitySelect.VendorCode,
          __spreadValues(__spreadValues({
            noStyle: true,
            viewMode: "light",
            valueName: "vendorCode"
          }, formFieldProps.fieldProps), formFieldProps)
        );
      }
      return /* @__PURE__ */ React.createElement(SptSearchText, __spreadValues({ noStyle: true }, formFieldProps));
    };
    return /* @__PURE__ */ React.createElement(
      Form.Item,
      __spreadValues({
        name: formFieldProps.name,
        style: { marginBottom: 0 },
        key: ((_e = item.proFieldProps) == null ? void 0 : _e.proFieldKey) || ((_f = item.name) == null ? void 0 : _f.toString())
      }, omit(formFieldProps.formItemProps, ["label", "initialValue"])),
      renderedDom()
    );
  },
  (prevProps, nextProps) => {
    return isDeepEqualReact(nextProps, prevProps, ["onChange", "onBlur"]);
  }
);
SptField.displayName = "SptField";
var Field_default = SptField;
export {
  Field_default as default
};
