// @spotter/ui/es/components/SptTable/Form/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/components/SptTable/Form/index.tsx
import { runFunction } from "../../../utils/runFunction";
import {
  omitUndefined
} from "@ant-design/pro-components";
import { Form, Table } from "antd";
import classNames from "classnames";
import omit from "omit.js";
import React, { memo, useEffect, useMemo } from "react";
import SptField from "./Field";
import Submitter from "./Submitter";
import transformKey from "./transformKey";
var FieldColumns = ({ columns, action, type, formRef }) => {
  return /* @__PURE__ */ React.createElement(React.Fragment, null, columns == null ? void 0 : columns.map((item) => /* @__PURE__ */ React.createElement(
    SptField,
    {
      key: item.proFieldProps.proFieldKey,
      item,
      helpers: {
        action,
        type,
        originItem: item,
        formRef
      }
    }
  )));
};
var FiledsColumnRender = memo(FieldColumns);
function Filter({
  action,
  formRef,
  beforeSearchSubmit = (searchParams) => searchParams,
  manualRequest,
  onSubmit,
  onReset,
  onFormSearchSubmit,
  columns,
  type,
  loading,
  form,
  searchKeyMap,
  search: searchConfig
}) {
  const baseClassName = "spt-table-search";
  const [innerForm] = Form.useForm();
  const submit = (params, firstLoad) => {
    var _a;
    const pagination = (_a = action == null ? void 0 : action.current) == null ? void 0 : _a.pageInfo;
    const pageInfo = pagination ? omitUndefined({
      current: pagination.current,
      pageSize: pagination.pageSize
    }) : {};
    const submitParams = __spreadValues(__spreadProps(__spreadValues({}, params), {
      _timestamp: Date.now()
    }), pageInfo);
    const omitParams = omit(beforeSearchSubmit(submitParams), Object.keys(pageInfo));
    onFormSearchSubmit(omitParams);
    if (onSubmit && !firstLoad) {
      onSubmit == null ? void 0 : onSubmit(params);
    }
  };
  const reset = (value) => {
    var _a, _b, _c;
    const pagination = (_a = action == null ? void 0 : action.current) == null ? void 0 : _a.pageInfo;
    const pageInfo = pagination ? omitUndefined({
      current: pagination.current,
      pageSize: pagination.pageSize
    }) : {};
    const omitParams = omit(
      beforeSearchSubmit(__spreadValues(__spreadValues({}, value), pageInfo)),
      Object.keys(pageInfo)
    );
    onFormSearchSubmit(omitParams);
    (_c = (_b = action == null ? void 0 : action.current) == null ? void 0 : _b.setPageInfo) == null ? void 0 : _c.call(_b, {
      current: 1
    });
    onReset == null ? void 0 : onReset();
  };
  const columnsList = useMemo(() => {
    var _a, _b;
    return (_b = (_a = columns == null ? void 0 : columns.filter((item) => {
      if (item === Table.EXPAND_COLUMN || item === Table.SELECTION_COLUMN) {
        return false;
      }
      if (item.hideInSearch || item.search === false) {
        return false;
      }
      if (type === "form" && item.hideInForm) {
        return false;
      }
      if (item.valueType && typeof item.valueType === "string" && ["index", "indexBorder", "option"].includes(item == null ? void 0 : item.valueType)) {
        return false;
      }
      return true;
    })) == null ? void 0 : _a.sort((a, b) => {
      if (b.order || a.order) {
        return (b.order || 0) - (a.order || 0);
      }
      return (b.index || 0) - (a.index || 0);
    })) == null ? void 0 : _b.map((item) => {
      var _a2;
      const finalValueType = !item.valueType || ["textarea", "jsonCode", "code"].includes(item == null ? void 0 : item.valueType) && type === "table" ? "text" : item == null ? void 0 : item.valueType;
      const columnKey = (item == null ? void 0 : item.key) || ((_a2 = item == null ? void 0 : item.dataIndex) == null ? void 0 : _a2.toString());
      const title = runFunction(item.title, item, "form");
      return __spreadProps(__spreadValues(__spreadProps(__spreadValues({}, item), {
        title,
        label: title,
        width: void 0
      }), (item == null ? void 0 : item.search) ? item.search : {}), {
        valueType: finalValueType,
        proFieldProps: __spreadProps(__spreadValues({}, item.proFieldProps), {
          proFieldKey: columnKey ? `table-field-${columnKey}` : void 0
        }),
        getFieldProps: item.fieldProps ? () => runFunction(item.fieldProps, formRef == null ? void 0 : formRef.current, item) : void 0,
        getFormItemProps: item.formItemProps ? () => runFunction(item.formItemProps, formRef == null ? void 0 : formRef.current, item) : void 0
      });
    });
  }, [columns, type]);
  useEffect(() => {
    var _a, _b, _c, _d;
    if (!(formRef == null ? void 0 : formRef.current)) {
      formRef.current = innerForm;
    }
    const finalValues = transformKey((_a = innerForm == null ? void 0 : innerForm.getFieldsValue()) != null ? _a : {}, searchKeyMap);
    const pageInfo = (_b = action == null ? void 0 : action.current) == null ? void 0 : _b.pageInfo;
    const { current = pageInfo == null ? void 0 : pageInfo.current, pageSize = pageInfo == null ? void 0 : pageInfo.pageSize } = finalValues;
    (_d = (_c = action == null ? void 0 : action.current) == null ? void 0 : _c.setPageInfo) == null ? void 0 : _d.call(_c, __spreadProps(__spreadValues({}, pageInfo), {
      current: parseInt(current, 10),
      pageSize: parseInt(pageSize, 10)
    }));
    if (manualRequest)
      return;
    submit(finalValues, true);
  }, []);
  return /* @__PURE__ */ React.createElement("div", { className: classNames(baseClassName) }, /* @__PURE__ */ React.createElement(
    Form,
    __spreadProps(__spreadValues(__spreadValues({
      layout: "horizontal",
      autoComplete: "off"
    }, form || {}), omit(__spreadValues({}, searchConfig), ["filterType"])), {
      onKeyPress: (event) => {
        var _a;
        if (event.key === "Enter" && (form == null ? void 0 : form.isKeyPressSubmit) !== false) {
          (_a = formRef == null ? void 0 : formRef.current) == null ? void 0 : _a.submit();
        }
      },
      form: innerForm,
      className: `${baseClassName}-form`,
      initialValues: form == null ? void 0 : form.initialValues
    }),
    /* @__PURE__ */ React.createElement("div", { className: `${baseClassName}-cols` }, /* @__PURE__ */ React.createElement(
      FiledsColumnRender,
      {
        columns: columnsList,
        action,
        type,
        formRef
      }
    ), /* @__PURE__ */ React.createElement("div", { className: `${baseClassName}-submitter-wrap` }, /* @__PURE__ */ React.createElement(
      Submitter,
      {
        submit: () => __async(this, null, function* () {
          if (loading)
            return;
          yield innerForm.validateFields();
          const values = innerForm.getFieldsValue();
          submit(transformKey(values, searchKeyMap), false);
        }),
        reset: () => __async(this, null, function* () {
          if (loading)
            return;
          innerForm.resetFields();
          const values = innerForm.getFieldsValue();
          reset(transformKey(values, searchKeyMap));
        }),
        submitButtonProps: {
          loading
        }
      }
    )))
  ));
}
var Form_default = memo(Filter);
export {
  Form_default as default
};
