// @spotter/app-client-toolkit/lib/hooks/cosUpload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useCosUpload = void 0;
const react_1 = require("react");
/**
 * COS上传的hooks
 * @param {useCosUploadProps} { getCredential, useFileContentType }
 * @return {*}
 */
const useCosUpload = ({ getCredential, useFileContentType }) => {
    (0, react_1.useEffect)(() => {
        if (typeof window.CosAuth !== 'function') {
            const myScript = document.createElement('script');
            // 设置标签的src属性
            myScript.src = 'https://romerx-cos.spotterio.com/crucial_js/cos-auth.min.js';
            // 明确设置为同步加载
            myScript.async = false;
            // 追加到body标签的最下面
            document.body.append(myScript);
            return () => {
                // 组件即将卸载时，移除标签
                myScript.remove();
            };
        }
        return () => { };
    }, []);
    return async (file, fileName) => {
        const { data } = await getCredential();
        const Authorization = window.CosAuth({
            SecretId: data.secretId,
            SecretKey: data.secretKey,
            Method: 'PUT',
            Pathname: `${data.fileUrlBase.split('.com')[1]}${fileName}`
        });
        return fetch(`${data.fileUrlBase}${fileName}`, {
            method: 'PUT',
            headers: {
                Accept: 'application/octet-stream, text/plain, */*',
                'Content-Type': useFileContentType ? file.type : 'application/octet-stream',
                Authorization,
                'x-cos-security-token': data.sessionToken
            },
            body: file
        });
    };
};
exports.useCosUpload = useCosUpload;
