// @spotter/app-client-toolkit/lib/hooks/tools.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSwitch = exports.useDependency = exports.useToggledValue = exports.useSafeContext = exports.useDomSelector = exports.useStack = void 0;
const react_1 = require("react");
function useStack(initStack) {
    const [stack, setStack] = (0, react_1.useState)(initStack);
    const push = (item) => {
        setStack([...stack, ...(Array.isArray(item) ? item : [item])]);
    };
    const pop = (len) => {
        setStack(stack.slice(stack.length - len - 1));
    };
    return { stack, push, pop };
}
exports.useStack = useStack;
function useDomSelector(selector) {
    const domElement = (0, react_1.useRef)();
    (0, react_1.useEffect)(() => {
        !domElement.current &&
            (domElement.current = document.querySelector(selector));
    }, [selector]);
    return domElement;
}
exports.useDomSelector = useDomSelector;
function useSafeContext(c) {
    const context = (0, react_1.useContext)(c);
    if (!context) {
        throw new Error('Component used a uninitialized DocDirectoryTreeContext, please provide a context value.');
    }
    return context;
}
exports.useSafeContext = useSafeContext;
function useToggledValue(init) {
    const [value, setValue] = (0, react_1.useState)(init);
    function toggle() {
        setValue((origin) => !origin);
    }
    return [value, toggle];
}
exports.useToggledValue = useToggledValue;
function useDependency() {
    const [dependency, updateDependency] = (0, react_1.useState)(false);
    const update = () => {
        updateDependency((origin) => !origin);
    };
    return [dependency, update];
}
exports.useDependency = useDependency;
function useSwitch() {
    const [visible, setVisible] = (0, react_1.useState)(false);
    const open = () => setVisible(true);
    const close = () => setVisible(false);
    return [visible, open, close];
}
exports.useSwitch = useSwitch;
