// @spotter/i18n-sdk@3.1.21/lib/api-sdk/utils/http-client.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const axios_1 = __importDefault(require("axios"));
const json_bigint_1 = __importDefault(require("json-bigint"));
const JSONBig2String = (0, json_bigint_1.default)({ storeAsString: true, constructorAction: 'preserve', protoAction: 'preserve' });
const defaultStatusSetMap = {
    authError: new Set([401, 403]),
    success: new Set([200, 201]),
};
class HttpClient {
    constructor(options) {
        var _a, _b, _c, _d;
        this.options = options;
        this.options.res.isSuccess =
            (_b = (_a = options.res) === null || _a === void 0 ? void 0 : _a.isSuccess) !== null && _b !== void 0 ? _b : ((res) => defaultStatusSetMap.success.has(res.status));
        this.options.res.isAuthError =
            (_d = (_c = options.res) === null || _c === void 0 ? void 0 : _c.isAuthError) !== null && _d !== void 0 ? _d : ((res) => defaultStatusSetMap.authError.has(res.status));
        this.httpClient = axios_1.default.create(Object.assign(Object.assign({}, this.options.axiosOptions), { baseURL: this.options.context.href, withCredentials: true, transformResponse: [(data) => JSONBig2String.parse(data)] }));
        this.httpClient.interceptors.request.use((config) => {
            if (!config.data) {
                config.data = {};
            }
            return config;
        }, (error) => {
            console.log('request', error);
            return Promise.reject(error);
        });
        // 响应拦截器
        this.httpClient.interceptors.response.use((response) => {
            if (this.options.res.isSuccess(response)) {
                return Promise.resolve(response);
            }
            if (this.options.res.isAuthError(response)) {
                console.log(response);
                this.options.res.onAuthError(response);
            }
            else {
                this.options.res.onOtherError(response);
                // content: error.response.data?.message || 'Network Error'
                console.error(JSON.stringify(response.data));
            }
            return Promise.reject(response);
        }, (error) => {
            if (this.options.res.isAuthError(error)) {
                console.log(error);
                this.options.res.onAuthError(error);
            }
            else {
                this.options.res.onOtherError(error);
                // content: error.response.data?.message || 'Network Error'
                console.error(JSON.stringify(error.response));
            }
            return Promise.reject(error);
        });
    }
    request(config) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.httpClient.request(config);
        });
    }
    getAxios() {
        return this.httpClient;
    }
}
exports.default = HttpClient;
