// @spotter/ui/lib/components/SptLink/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classnames = _interopRequireDefault(require("classnames"));
var _reactRouterDom = require("react-router-dom");
var _antd = require("antd");
var _react = _interopRequireDefault(require("react"));
require("./style/index.less");
var _Provider = require("../Provider");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
/**
 *
 * default: 更好的性能表现，使用轻量的路由切换， 基于 react-router 的 history 模式.
 * new: 新建页签，等同于 target: '_blank', target 详见 https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target
 * replace: 跳转并刷新当前页面，等同于 target: '_self'
 */

var ExpensiveLink = _antd.Typography.Link;
var SpotterLinkBase = function SpotterLinkBase(_ref) {
  var children = _ref.children,
    className = _ref.className,
    type = _ref.type,
    to = _ref.to,
    noStyle = _ref.noStyle;
  if (!type && to.toString().startsWith('http')) {
    type = 'new';
  }
  var _c = (0, _classnames.default)('spotter-link', className);
  var style = noStyle ? {
    color: 'inherit'
  } : {
    display: 'inline-flex',
    flexDirection: 'row',
    alignItems: 'center'
  };
  switch (type) {
    case 'new':
      {
        return /*#__PURE__*/_react.default.createElement(ExpensiveLink, {
          className: _c,
          style: style,
          href: to,
          target: "_blank"
        }, children);
      }
    case 'replace':
      {
        return /*#__PURE__*/_react.default.createElement(ExpensiveLink, {
          className: _c,
          style: style,
          href: to,
          target: "_self"
        }, children);
      }
    default:
      {
        return /*#__PURE__*/_react.default.createElement(_reactRouterDom.Link, {
          className: _c,
          style: style,
          to: to
        }, children);
      }
  }
};

/**
 * 对 React Router、Typography.Link 进行二次封装，满足应用内和应用外跳转需求
 * @param param0
 * @returns
 */
var SptLink = function SptLink(props) {
  return /*#__PURE__*/_react.default.createElement(_Provider.SptComponentProvider, null, /*#__PURE__*/_react.default.createElement(SpotterLinkBase, props));
};
var _default = exports.default = SptLink;
