// @spotter/app-client-toolkit@1.0.42-nic.8/lib/tool/siteTenant.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.transformUnit = exports.checkMultiChannel = exports.getSiteTenantWeightUnit = exports.getSiteTenantLinearUnit = exports.getSiteTenantTimeZone = exports.getSiteTenantCurrencyCode = exports.getSiteTenantCurrencySymbol = exports.getCurrentPlatformRegionCode = exports.getCurrentPlatformCode = exports.getSiteTenantCode = void 0;
/** 渠道站点相关方法 */
const const_1 = require("../const");
const localCache_1 = require("./localCache");
const big_js_1 = __importDefault(require("big.js"));
/**
 * 获取当前渠道站点
 */
function getSiteTenantCode() {
    return localCache_1.localCache.get(const_1.SPT_SITE_TENANT_CODE) || const_1.DEFAULT_SITE_TENANT_CODE;
}
exports.getSiteTenantCode = getSiteTenantCode;
/**
 * 获取当前渠道
 */
function getCurrentPlatformCode() {
    return localCache_1.localCache.get(const_1.SPT_PLATFORM_CODE) || const_1.DEFAULT_PLATFORM_CODE;
}
exports.getCurrentPlatformCode = getCurrentPlatformCode;
/**
 * 获取当前站点
 */
function getCurrentPlatformRegionCode() {
    return localCache_1.localCache.get(const_1.SPT_PLATFORM_REGION_CODE) || const_1.DEFAULT_PLATFORM_REGION_CODE;
}
exports.getCurrentPlatformRegionCode = getCurrentPlatformRegionCode;
/**
 * 获取当前渠道站点货币符号
 */
function getSiteTenantCurrencySymbol() {
    return localCache_1.localCache.get(const_1.SPT_SITE_TENANT_CURRENCY_SYMBOL) || const_1.DEFAULT_SITE_TENANT_CURRENCY_SYMBOL;
}
exports.getSiteTenantCurrencySymbol = getSiteTenantCurrencySymbol;
/**
 * 获取当前渠道站点货币编码
 */
function getSiteTenantCurrencyCode() {
    return localCache_1.localCache.get(const_1.SPT_SITE_TENANT_CURRENCY_CODE) || const_1.USD;
}
exports.getSiteTenantCurrencyCode = getSiteTenantCurrencyCode;
/**
 * 获取当前渠道站点时区
 */
function getSiteTenantTimeZone() {
    return localCache_1.localCache.get(const_1.SPT_SITE_TENANT_TIMEZONE) || const_1.DEFAULT_TIMEZONE;
}
exports.getSiteTenantTimeZone = getSiteTenantTimeZone;
/**
 * 获取当前渠道站点长度单位
 */
function getSiteTenantLinearUnit() {
    return localCache_1.localCache.get(const_1.SPT_SITE_TENANT_LINEAR_UNIT_CODE) || const_1.DEFAULT_LINEAR_UNIT;
}
exports.getSiteTenantLinearUnit = getSiteTenantLinearUnit;
/**
 * 获取当前渠道站点重量单位
 */
function getSiteTenantWeightUnit() {
    return localCache_1.localCache.get(const_1.SPT_SITE_TENANT_WEIGHT_UNIT_CODE) || const_1.DEFAULT_WEIGHT_UNIT;
}
exports.getSiteTenantWeightUnit = getSiteTenantWeightUnit;
// 判断多渠道，根据传入的渠道配置,返回布尔值,默认false,提供默认布尔值可以选择自定义填入
const checkMultiChannel = (checkObject, defaultBoolean) => {
    // 获取当前站点
    const [platformRegion, platform] = getSiteTenantCode()?.split('_');
    // 返回当前站点的配置值
    return checkObject?.[platform]?.[platformRegion] || defaultBoolean || false;
};
exports.checkMultiChannel = checkMultiChannel;
/**
 * @description: 根据配置文件转化单位
 * @param { string|number } originalValue 原始值,string类型或者number类型的数字
 * @param { string } originalUnit 原始单位
 * @param { string } expectUnit 期望的单位
 * @param { number } decimal 保留小数位数, 默认为2
 * @returns { number|string } 转换后的数值，以string为主
*/
const transformUnit = ({ originalUnit, originalValue, expectUnit, decimal = 2 }) => {
    if (originalUnit === expectUnit)
        return originalValue;
    try {
        // 从localStorage中获取单位转换配置
        const transformConfig = (localCache_1.localCache.get(const_1.SPT_UNIT_TRANSFORM_CONFIG) || []);
        if (!transformConfig.length)
            return originalValue;
        // 查找对应的单位转化进制数据
        const targetConfig = transformConfig.find((item) => item.from === originalUnit && item.to === expectUnit);
        if (!targetConfig)
            return originalValue;
        return new big_js_1.default(originalValue).mul(targetConfig.value).toFixed(decimal);
    }
    catch (error) {
        console.error(error);
        return originalValue;
    }
};
exports.transformUnit = transformUnit;
