// @spotter/app-client-toolkit@1.0.42-nic.8/lib/tool/sso.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SSOSDKClient = exports.PermissionErrorCode = exports.AuthErrorCode = void 0;
const getAppBootData_1 = require("./getAppBootData");
exports.AuthErrorCode = new Set([
    401,
    4010000,
    4010001,
    4010002,
    4010003,
    4010004,
    -1,
    -2,
    -3,
    -4,
    -5,
    -6,
    -7,
]);
exports.PermissionErrorCode = new Set([403, 403001]);
/**
 * 存在前后端使用
 */
class SSOSDKClient {
    constructor({ appCode, }) {
        this.SSO_TICKET = 'spotter_sso_ticket';
        this.appCode = '';
        this.removeQueryParam = (url, param) => {
            const urlObj = new URL(url);
            const searchParams = new URLSearchParams(urlObj.search);
            searchParams.delete(param);
            urlObj.search = searchParams.toString();
            return urlObj.toString();
        };
        this.appCode = appCode;
    }
    /**
     * 前端登出方法
     */
    logout() {
        const iamApiUrl = (0, getAppBootData_1.getAppBootData)().iamApiUrl;
        window.location.replace(`${iamApiUrl}/sso/logout?callback=${encodeURIComponent(this.removeQueryParam(window.location.href, this.SSO_TICKET))}&appCode=${this.appCode}`);
    }
    /**
     * 登出跳转sso提示页面
     * @param i18key 提示
     */
    logoutWithInfo(i18key) {
        const iamApiUrl = (0, getAppBootData_1.getAppBootData)().iamApiUrl;
        window.location.replace(`${iamApiUrl}/sso/auth/error?errorMessage=${i18key}&callback=${encodeURIComponent(this.removeQueryParam(window.location.href, this.SSO_TICKET))}&appCode=${this.appCode}`);
    }
}
exports.SSOSDKClient = SSOSDKClient;
