// @spotter/app-client-toolkit@1.0.42-nic.8/lib/tool/time.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.checkAllExportTime = exports.checkExportTime = exports.formatDateWithWeek = exports.formatDate = exports.formatTime = exports.tzTimestampRangeTransformer = exports.toTzTimestamp = exports.tzTimeFormatter = exports.tzTime = void 0;
const utc_1 = __importDefault(require("dayjs/plugin/utc"));
const timezone_1 = __importDefault(require("dayjs/plugin/timezone"));
const duration_1 = __importDefault(require("dayjs/plugin/duration"));
const dayjs_1 = __importDefault(require("dayjs"));
const const_1 = require("../const");
const siteTenant_1 = require("./siteTenant");
dayjs_1.default.extend(utc_1.default);
// 用于计算ms转换min，hour，day
dayjs_1.default.extend(duration_1.default);
dayjs_1.default.extend(timezone_1.default);
/**
 * 时区时间
 * @param localTime required 本地时间字符串 或 unix 时间戳
 * @param tz optional 转换目标时区，默认值为平台默认时区 DEFAULT_TIMEZONE，
 * @param keepLocalTime optional 是否保持本地时间，默认值为 false, 如果为 true，则 tz 不生效
 */
function tzTime(localTime, tz = (0, siteTenant_1.getSiteTenantTimeZone)(), keepLocalTime = false) {
    return (0, dayjs_1.default)(localTime).tz(tz, keepLocalTime);
}
exports.tzTime = tzTime;
/**
 * 本地时间转其他时区格式化工具
 * @param localTime required 本地时间字符串 或 unix 时间戳
 * @param format optional 时间格式，默认值为 YYYY-MM-DD HH:mm:ss
 * @param tz optional 转换目标时区，默认值为平台默认时区 DEFAULT_TIMEZONE，
 */
function tzTimeFormatter(localTime, format = const_1.TIME_SECOND_FORMAT, tz = (0, siteTenant_1.getSiteTenantTimeZone)()) {
    return localTime ? tzTime(localTime, tz, false).format(format) : const_1.DEFAULT_EMPTY_VALUE;
}
exports.tzTimeFormatter = tzTimeFormatter;
/**
 * 任意时区时间转换时间戳
 * @description
 * @param sourceTime required 源时间
 * @param operator 操作
 * @param unit 单位
 * @param tz 时区，默认为平台默认时间
 * @param keepLocalTime 是否保持本地时间，默认值 true
 */
const toTzTimestamp = (sourceTime, operator, unit, tz = (0, siteTenant_1.getSiteTenantTimeZone)(), keepLocalTime = true) => (sourceTime ? tzTime(sourceTime, tz, keepLocalTime)[operator](unit).valueOf() : undefined);
exports.toTzTimestamp = toTzTimestamp;
/**
 * 指定时区对应的 timestamp range 对象的转换工具
 * @param fields required 最终生成对象的字段，数量会影响最终返回的数据结构
 * @param unit optional 单位, 默认值为 'd'
 * @param tz 时区，默认为平台默认时间
 */
const tzTimestampRangeTransformer = (fields, unit = 'd', tz = (0, siteTenant_1.getSiteTenantTimeZone)()) => (timeRange) => {
    if (!timeRange)
        return {};
    const startTime = (0, exports.toTzTimestamp)(timeRange[0], 'startOf', unit, tz);
    const endTime = (0, exports.toTzTimestamp)(timeRange[1], 'endOf', unit, tz);
    if (fields.length > 1) {
        return {
            [fields[0]]: startTime,
            [fields[1]]: endTime
        };
    }
    return { [fields[0]]: [startTime, endTime] };
};
exports.tzTimestampRangeTransformer = tzTimestampRangeTransformer;
/**
 * 日期的格式化，后续改成formatDate
 * @param {*} val
 */
const formatTime = (val) => tzTimeFormatter(val, const_1.TIME_DAY_FORMAT);
exports.formatTime = formatTime;
/**
 * 日期的格式化
 * @param {*} val
 */
const formatDate = (val) => tzTimeFormatter(val, const_1.TIME_DAY_FORMAT);
exports.formatDate = formatDate;
/**
 * 日期带周的格式化
 * @param {*} val
 */
const formatDateWithWeek = (val) => tzTimeFormatter(val, const_1.TIME_DAY_WITH_WEEK_FORMAT);
exports.formatDateWithWeek = formatDateWithWeek;
/**
 * 导出时间的校验
 * @param {IdentifierType} [start]
 * @param {IdentifierType} [end]
 * @param {{ ignoreEmpty?: boolean; max?: number }} [opt]
 */
const checkExportTime = (start, end, opt) => (opt?.ignoreEmpty || (!!start && !!end)) &&
    (0, dayjs_1.default)(end).diff((0, dayjs_1.default)(start), 'day') <= (opt?.max ?? 31);
exports.checkExportTime = checkExportTime;
/**
 * 多个导出时间的校验
 * @param {Parameters<typeof checkExportTime>[]} arr
 * @param {number} [max=31]
 */
const checkAllExportTime = (arr, max = 31) => arr.some((timeRange) => (0, exports.checkExportTime)(timeRange?.[0], timeRange?.[1], { max }));
exports.checkAllExportTime = checkAllExportTime;
