// @spotter/ui/es/components/SptGridForm/index.js
var __defProp = Object.defineProperty;
var __defProps = Object.defineProperties;
var __getOwnPropDescs = Object.getOwnPropertyDescriptors;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __spreadProps = (a, b) => __defProps(a, __getOwnPropDescs(b));

// src/components/SptGridForm/index.tsx
import {
  ProForm,
  ProFormDatePicker,
  ProFormDateTimePicker,
  ProFormDigit,
  ProFormSelect,
  ProFormText,
  ProFormTimePicker,
  ProFormUploadButton,
  ProFormUploadDragger
} from "@ant-design/pro-components";
import {
  Col,
  Row,
  Spin
} from "antd";
import React, { createElement } from "react";
import classNames from "classnames";
import { formatUsAmount } from "@spotter/app-client-toolkit";
import "./style/index.less";
import { SptComponentProvider } from "../Provider";
var getFormItemComponent = (type) => {
  switch (type) {
    case "number": {
      return ProFormDigit;
    }
    case "enum": {
      return ProFormSelect;
    }
    case "date": {
      return ProFormDatePicker;
    }
    case "time": {
      return ProFormTimePicker;
    }
    case "date-time": {
      return ProFormDateTimePicker;
    }
    case "file-dragger": {
      return ProFormUploadDragger;
    }
    case "file": {
      return ProFormUploadButton;
    }
    default: {
      return ProFormText;
    }
  }
};
var SptGridForm = ({
  fields,
  formRef,
  col,
  formItems,
  layout = "vertical",
  labelCol,
  loading = false,
  rowJustify = "start"
}) => {
  for (const b of formItems) {
    if (b.type === "number" && b.readonly && !(b == null ? void 0 : b.render)) {
      b.render = (val) => /* @__PURE__ */ React.createElement(React.Fragment, null, formatUsAmount(val));
    }
  }
  return /* @__PURE__ */ React.createElement(SptComponentProvider, null, /* @__PURE__ */ React.createElement(Spin, { spinning: loading }, /* @__PURE__ */ React.createElement(
    ProForm,
    {
      className: "spotter-grid-form",
      fields,
      submitter: {
        render: false
      },
      layout,
      formRef,
      labelCol
    },
    formItems.reduce((p, c, i) => {
      if (i % col === 0) {
        p.push([c]);
      } else {
        p[p.length - 1].push(c);
      }
      return p;
    }, []).map((cols, index) => (
      // eslint-disable-next-line react/no-array-index-key
      /* @__PURE__ */ React.createElement(Row, { key: index, gutter: [36, 0], justify: rowJustify }, cols.map((item) => {
        var _a, _b, _c, _d;
        return /* @__PURE__ */ React.createElement(
          Col,
          __spreadProps(__spreadValues({}, item.colProps), {
            key: item.name,
            span: (_b = (_a = item.colProps) == null ? void 0 : _a.span) != null ? _b : 24 / col
          }),
          createElement(
            getFormItemComponent(item.type),
            __spreadProps(__spreadValues({}, item), {
              filedConfig: __spreadProps(__spreadValues({}, item.filedConfig), {
                className: classNames(
                  (_d = (_c = item.filedConfig) == null ? void 0 : _c.className) != null ? _d : "",
                  {
                    "spotter-form-item-readonly": item.readonly
                  }
                )
              }),
              key: item.name,
              name: item.name,
              label: /* @__PURE__ */ React.createElement("span", { className: "spotter-form-item-label" }, item.label)
            })
          )
        );
      }))
    ))
  )));
};
var SptGridForm_default = SptGridForm;
export {
  SptGridForm_default as default
};
