// @spotter/app-client-toolkit@1.0.42-nic.8/lib/const/site.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PLATFORM_REGION_ENUM = exports.PLATFORM_ENUM = void 0;
// 平台code枚举
var PLATFORM_ENUM;
(function (PLATFORM_ENUM) {
    // 亚马逊站点
    PLATFORM_ENUM["AMZ"] = "AMZ";
    // 供应链使用
    PLATFORM_ENUM["SPT"] = "SPT";
    // shopify租户站点
    PLATFORM_ENUM["SPF"] = "SPF";
    // target租户站点
    PLATFORM_ENUM["TGT"] = "TGT";
})(PLATFORM_ENUM = exports.PLATFORM_ENUM || (exports.PLATFORM_ENUM = {}));
// 站点code枚举
var PLATFORM_REGION_ENUM;
(function (PLATFORM_REGION_ENUM) {
    // 美国
    PLATFORM_REGION_ENUM["US"] = "US";
    // 加拿大
    PLATFORM_REGION_ENUM["CA"] = "CA";
    // 英国
    PLATFORM_REGION_ENUM["UK"] = "GB";
    // 英国
    PLATFORM_REGION_ENUM["GB"] = "GB";
    // 德国
    PLATFORM_REGION_ENUM["DE"] = "DE";
    // 意大利
    PLATFORM_REGION_ENUM["IT"] = "IT";
    // 西班牙
    PLATFORM_REGION_ENUM["ES"] = "ES";
    // 法国
    PLATFORM_REGION_ENUM["FR"] = "FR";
})(PLATFORM_REGION_ENUM = exports.PLATFORM_REGION_ENUM || (exports.PLATFORM_REGION_ENUM = {}));
