// @spotter/ui/lib/components/SptPopoverList/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _appClientToolkit = require("@spotter/app-client-toolkit");
var _antd = require("antd");
var _Provider = require("../Provider");
require("./style/index.less");
var _excluded = ["list", "placement", "trigger", "renderItem"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var PopoverListBase = function PopoverListBase(_ref) {
  var _list$length;
  var list = _ref.list,
    _ref$placement = _ref.placement,
    placement = _ref$placement === void 0 ? 'right' : _ref$placement,
    _ref$trigger = _ref.trigger,
    trigger = _ref$trigger === void 0 ? 'hover' : _ref$trigger,
    renderItem = _ref.renderItem,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "popover-list-container",
    id: "popover-list-container"
  }, list && (list === null || list === void 0 ? void 0 : list.length) > 0 ? (list === null || list === void 0 ? void 0 : list.length) > 1 ? /*#__PURE__*/_react.default.createElement(_antd.Popover, _extends({
    placement: placement,
    content: list === null || list === void 0 ? void 0 : list.map(function (b, index) {
      return /*#__PURE__*/_react.default.createElement("div", {
        className: "popover-list-item",
        key: index
      }, renderItem ? renderItem(b) : b);
    }),
    trigger: trigger,
    overlayClassName: "popover-list-overlay"
  }, props), /*#__PURE__*/_react.default.createElement("div", {
    className: "popover-list-text"
  }, /*#__PURE__*/_react.default.createElement("span", null, list === null || list === void 0 ? void 0 : list[0]), /*#__PURE__*/_react.default.createElement("div", {
    className: "popover-list-tab-num"
  }, "+", ((_list$length = list === null || list === void 0 ? void 0 : list.length) !== null && _list$length !== void 0 ? _list$length : 0) - 1))) : renderItem ? renderItem(list[0]) : list[0] : _appClientToolkit.DEFAULT_EMPTY_VALUE);
};

/**
 * 额外数据气泡展示
 * @param props
 * @returns
 */
var SptPopoverList = function SptPopoverList(props) {
  return /*#__PURE__*/_react.default.createElement(_Provider.SptComponentProvider, null, /*#__PURE__*/_react.default.createElement(PopoverListBase, props));
};
var _default = exports.default = SptPopoverList;
